{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "entitled-poison",
   "metadata": {},
   "source": [
    "# Choose Genomes\n",
    "\n",
    "Choose some genomes for the Prophage Prediction Comparison Website.\n",
    "\n",
    "We want to prepare about 100 genomes that meet these criteria:\n",
    "1. Diverse genomes (not proteobacteria, firmicutes, etc)\n",
    "2. Complete genomes (<5 contigs)\n",
    "3. At least one, but no more than 6 prophages per genome\n",
    "\n",
    "Then we will reannotate them with PhiSpy to get good annotations that people can manually curate."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "objective-trinity",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A lot of this is not used, but we import it so we have it later!\n",
    "import os\n",
    "import sys\n",
    "\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "from PhiSpyAnalysis import printmd\n",
    "from PhiSpyAnalysis import read_phages, read_gtdb, read_checkv, read_base_pp, read_categories, read_metadata, read_transposons\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "invalid-ultimate",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Please note that this was run with git commit 1a2a28b that has 567,404 genomes parsed.\n",
      "Initially there were 3,265,453 kept phages, but after filtering we kept 75,444 prophages from 19,529 genomes"
     ]
    }
   ],
   "source": [
    "phages = read_phages(maxcontigs=5)\n",
    "phages = phages[phages['Kept'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "wooden-concrete",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_lineage</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>coding_density</th>\n",
       "      <th>...</th>\n",
       "      <th>trna_aa_count</th>\n",
       "      <th>trna_count</th>\n",
       "      <th>trna_selenocysteine_count</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GB_GCA_000006155.2</td>\n",
       "      <td>1916</td>\n",
       "      <td>93.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1171</td>\n",
       "      <td>g__Bacillus (UID902)</td>\n",
       "      <td>324</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4305660</td>\n",
       "      <td>80.178992</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Bacillaceae_G</td>\n",
       "      <td>Bacillus_A</td>\n",
       "      <td>Bacillus_A anthracis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GB_GCA_000007385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.82</td>\n",
       "      <td>0.00</td>\n",
       "      <td>481</td>\n",
       "      <td>c__Gammaproteobacteria (UID4202)</td>\n",
       "      <td>276</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4190634</td>\n",
       "      <td>84.805944</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Xanthomonas</td>\n",
       "      <td>Xanthomonas oryzae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GB_GCA_000008605.1</td>\n",
       "      <td>67</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>235</td>\n",
       "      <td>f__Spirochaetaceae (UID2512)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1048744</td>\n",
       "      <td>92.155875</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Spirochaetota</td>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>Treponematales</td>\n",
       "      <td>Treponemataceae</td>\n",
       "      <td>Treponema</td>\n",
       "      <td>Treponema pallidum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GB_GCA_000010565.1</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.63</td>\n",
       "      <td>295</td>\n",
       "      <td>p__Firmicutes (UID1022)</td>\n",
       "      <td>158</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2608397</td>\n",
       "      <td>86.217312</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>51</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_B</td>\n",
       "      <td>Desulfotomaculia</td>\n",
       "      <td>Desulfotomaculales</td>\n",
       "      <td>Pelotomaculaceae</td>\n",
       "      <td>Pelotomaculum</td>\n",
       "      <td>Pelotomaculum thermopropionicum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GB_GCA_000013845.2</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>332</td>\n",
       "      <td>o__Clostridiales (UID1375)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2428396</td>\n",
       "      <td>82.037966</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Clostridiaceae</td>\n",
       "      <td>Clostridium_P</td>\n",
       "      <td>Clostridium_P perfringens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191522</th>\n",
       "      <td>RS_GCF_902166935.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.62</td>\n",
       "      <td>0.09</td>\n",
       "      <td>1312</td>\n",
       "      <td>g__Klebsiella (UID5140)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4894244</td>\n",
       "      <td>87.293148</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>87</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Klebsiella</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191523</th>\n",
       "      <td>RS_GCF_902166945.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.94</td>\n",
       "      <td>0.72</td>\n",
       "      <td>1162</td>\n",
       "      <td>f__Enterobacteriaceae (UID5121)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5162721</td>\n",
       "      <td>86.540042</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Klebsiella</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191524</th>\n",
       "      <td>RS_GCF_902167295.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5737447</td>\n",
       "      <td>90.122088</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagales</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Chitinophaga</td>\n",
       "      <td>Chitinophaga pinensis_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191525</th>\n",
       "      <td>RS_GCF_902167305.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5738096</td>\n",
       "      <td>90.147873</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagales</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Chitinophaga</td>\n",
       "      <td>Chitinophaga pinensis_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191526</th>\n",
       "      <td>RS_GCF_902167325.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5671443</td>\n",
       "      <td>90.098258</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagales</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Chitinophaga</td>\n",
       "      <td>Chitinophaga pinensis_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191527 rows × 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 accession  ambiguous_bases  checkm_completeness  \\\n",
       "0       GB_GCA_000006155.2             1916                93.12   \n",
       "1       GB_GCA_000007385.1                0                99.82   \n",
       "2       GB_GCA_000008605.1               67               100.00   \n",
       "3       GB_GCA_000010565.1                0               100.00   \n",
       "4       GB_GCA_000013845.2                0               100.00   \n",
       "...                    ...              ...                  ...   \n",
       "191522  RS_GCF_902166935.1                0                99.62   \n",
       "191523  RS_GCF_902166945.1                0                99.94   \n",
       "191524  RS_GCF_902167295.1                0                99.51   \n",
       "191525  RS_GCF_902167305.1                0                99.51   \n",
       "191526  RS_GCF_902167325.1                0                99.51   \n",
       "\n",
       "        checkm_contamination  checkm_marker_count  \\\n",
       "0                       0.00                 1171   \n",
       "1                       0.00                  481   \n",
       "2                       0.00                  235   \n",
       "3                       0.63                  295   \n",
       "4                       0.00                  332   \n",
       "...                      ...                  ...   \n",
       "191522                  0.09                 1312   \n",
       "191523                  0.72                 1162   \n",
       "191524                  0.00                  303   \n",
       "191525                  0.00                  303   \n",
       "191526                  0.00                  303   \n",
       "\n",
       "                   checkm_marker_lineage  checkm_marker_set_count  \\\n",
       "0                   g__Bacillus (UID902)                      324   \n",
       "1       c__Gammaproteobacteria (UID4202)                      276   \n",
       "2           f__Spirochaetaceae (UID2512)                      124   \n",
       "3                p__Firmicutes (UID1022)                      158   \n",
       "4             o__Clostridiales (UID1375)                      124   \n",
       "...                                  ...                      ...   \n",
       "191522           g__Klebsiella (UID5140)                      336   \n",
       "191523   f__Enterobacteriaceae (UID5121)                      336   \n",
       "191524        p__Bacteroidetes (UID2591)                      203   \n",
       "191525        p__Bacteroidetes (UID2591)                      203   \n",
       "191526        p__Bacteroidetes (UID2591)                      203   \n",
       "\n",
       "        checkm_strain_heterogeneity  coding_bases  coding_density  ...  \\\n",
       "0                               0.0       4305660       80.178992  ...   \n",
       "1                               0.0       4190634       84.805944  ...   \n",
       "2                               0.0       1048744       92.155875  ...   \n",
       "3                               0.0       2608397       86.217312  ...   \n",
       "4                               0.0       2428396       82.037966  ...   \n",
       "...                             ...           ...             ...  ...   \n",
       "191522                          0.0       4894244       87.293148  ...   \n",
       "191523                          0.0       5162721       86.540042  ...   \n",
       "191524                          0.0       5737447       90.122088  ...   \n",
       "191525                          0.0       5738096       90.147873  ...   \n",
       "191526                          0.0       5671443       90.098258  ...   \n",
       "\n",
       "        trna_aa_count  trna_count  trna_selenocysteine_count    domain  \\\n",
       "0                  16          31                          0  Bacteria   \n",
       "1                  20          53                          0  Bacteria   \n",
       "2                  20          45                          0  Bacteria   \n",
       "3                  20          51                          1  Bacteria   \n",
       "4                  20          95                          1  Bacteria   \n",
       "...               ...         ...                        ...       ...   \n",
       "191522             20          87                          1  Bacteria   \n",
       "191523             20          86                          1  Bacteria   \n",
       "191524             20          58                          0  Bacteria   \n",
       "191525             20          58                          0  Bacteria   \n",
       "191526             20          59                          0  Bacteria   \n",
       "\n",
       "                phylum                class               order  \\\n",
       "0           Firmicutes              Bacilli          Bacillales   \n",
       "1       Proteobacteria  Gammaproteobacteria     Xanthomonadales   \n",
       "2        Spirochaetota         Spirochaetia      Treponematales   \n",
       "3         Firmicutes_B     Desulfotomaculia  Desulfotomaculales   \n",
       "4         Firmicutes_A           Clostridia       Clostridiales   \n",
       "...                ...                  ...                 ...   \n",
       "191522  Proteobacteria  Gammaproteobacteria    Enterobacterales   \n",
       "191523  Proteobacteria  Gammaproteobacteria    Enterobacterales   \n",
       "191524    Bacteroidota          Bacteroidia     Chitinophagales   \n",
       "191525    Bacteroidota          Bacteroidia     Chitinophagales   \n",
       "191526    Bacteroidota          Bacteroidia     Chitinophagales   \n",
       "\n",
       "                    family          genus                          species  \n",
       "0            Bacillaceae_G     Bacillus_A             Bacillus_A anthracis  \n",
       "1         Xanthomonadaceae    Xanthomonas               Xanthomonas oryzae  \n",
       "2          Treponemataceae      Treponema               Treponema pallidum  \n",
       "3         Pelotomaculaceae  Pelotomaculum  Pelotomaculum thermopropionicum  \n",
       "4           Clostridiaceae  Clostridium_P        Clostridium_P perfringens  \n",
       "...                    ...            ...                              ...  \n",
       "191522  Enterobacteriaceae     Klebsiella            Klebsiella pneumoniae  \n",
       "191523  Enterobacteriaceae     Klebsiella            Klebsiella pneumoniae  \n",
       "191524    Chitinophagaceae   Chitinophaga          Chitinophaga pinensis_A  \n",
       "191525    Chitinophagaceae   Chitinophaga          Chitinophaga pinensis_A  \n",
       "191526    Chitinophagaceae   Chitinophaga          Chitinophaga pinensis_A  \n",
       "\n",
       "[191527 rows x 117 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtdb = c\n",
    "gtdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "suspected-keyboard",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000007385.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Xanthomonas</td>\n",
       "      <td>Xanthomonas oryzae</td>\n",
       "      <td>ASM738v1</td>\n",
       "      <td>4941439.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>129141.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000008605.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Spirochaetota</td>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>Treponematales</td>\n",
       "      <td>Treponemataceae</td>\n",
       "      <td>Treponema</td>\n",
       "      <td>Treponema pallidum</td>\n",
       "      <td>ASM860v1</td>\n",
       "      <td>1138011.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>48873.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000010565.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_B</td>\n",
       "      <td>Desulfotomaculia</td>\n",
       "      <td>Desulfotomaculales</td>\n",
       "      <td>Pelotomaculaceae</td>\n",
       "      <td>Pelotomaculum</td>\n",
       "      <td>Pelotomaculum thermopropionicum</td>\n",
       "      <td>ASM1056v1</td>\n",
       "      <td>3025375.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>107997.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000013845.2</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Clostridiaceae</td>\n",
       "      <td>Clostridium_P</td>\n",
       "      <td>Clostridium_P perfringens</td>\n",
       "      <td>ASM1384v2</td>\n",
       "      <td>2960088.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>56955.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000014305.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Streptococcaceae</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>Streptococcus suis</td>\n",
       "      <td>ASM1430v1</td>\n",
       "      <td>2096309.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>84555.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12522</th>\n",
       "      <td>GCA_901544215.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Streptococcaceae</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>Streptococcus dysgalactiae</td>\n",
       "      <td>42880_F01</td>\n",
       "      <td>2193653.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>130605.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12523</th>\n",
       "      <td>GCA_901544385.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Streptococcaceae</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>Streptococcus gordonii</td>\n",
       "      <td>42912_F01</td>\n",
       "      <td>2185555.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>36608.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12524</th>\n",
       "      <td>GCA_901553725.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Enterococcaceae</td>\n",
       "      <td>Enterococcus</td>\n",
       "      <td>Enterococcus faecalis</td>\n",
       "      <td>42197_D01</td>\n",
       "      <td>2864420.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>146449.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12525</th>\n",
       "      <td>GCA_901553735.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Streptococcaceae</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>Streptococcus porcinus</td>\n",
       "      <td>41965_D01</td>\n",
       "      <td>1954699.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>150445.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12526</th>\n",
       "      <td>GCA_901553755.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Streptococcaceae</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>Streptococcus porcinus</td>\n",
       "      <td>41851_D01</td>\n",
       "      <td>1949104.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>124214.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12527 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      assembly_accession    domain          phylum                class  \\\n",
       "0        GCA_000007385.1  Bacteria  Proteobacteria  Gammaproteobacteria   \n",
       "1        GCA_000008605.1  Bacteria   Spirochaetota         Spirochaetia   \n",
       "2        GCA_000010565.1  Bacteria    Firmicutes_B     Desulfotomaculia   \n",
       "3        GCA_000013845.2  Bacteria    Firmicutes_A           Clostridia   \n",
       "4        GCA_000014305.1  Bacteria      Firmicutes              Bacilli   \n",
       "...                  ...       ...             ...                  ...   \n",
       "12522    GCA_901544215.1  Bacteria      Firmicutes              Bacilli   \n",
       "12523    GCA_901544385.1  Bacteria      Firmicutes              Bacilli   \n",
       "12524    GCA_901553725.1  Bacteria      Firmicutes              Bacilli   \n",
       "12525    GCA_901553735.1  Bacteria      Firmicutes              Bacilli   \n",
       "12526    GCA_901553755.1  Bacteria      Firmicutes              Bacilli   \n",
       "\n",
       "                    order            family          genus  \\\n",
       "0         Xanthomonadales  Xanthomonadaceae    Xanthomonas   \n",
       "1          Treponematales   Treponemataceae      Treponema   \n",
       "2      Desulfotomaculales  Pelotomaculaceae  Pelotomaculum   \n",
       "3           Clostridiales    Clostridiaceae  Clostridium_P   \n",
       "4         Lactobacillales  Streptococcaceae  Streptococcus   \n",
       "...                   ...               ...            ...   \n",
       "12522     Lactobacillales  Streptococcaceae  Streptococcus   \n",
       "12523     Lactobacillales  Streptococcaceae  Streptococcus   \n",
       "12524     Lactobacillales   Enterococcaceae   Enterococcus   \n",
       "12525     Lactobacillales  Streptococcaceae  Streptococcus   \n",
       "12526     Lactobacillales  Streptococcaceae  Streptococcus   \n",
       "\n",
       "                               species assembly_name  Genome length  Contigs  \\\n",
       "0                   Xanthomonas oryzae      ASM738v1      4941439.0      1.0   \n",
       "1                   Treponema pallidum      ASM860v1      1138011.0      1.0   \n",
       "2      Pelotomaculum thermopropionicum     ASM1056v1      3025375.0      1.0   \n",
       "3            Clostridium_P perfringens     ASM1384v2      2960088.0      4.0   \n",
       "4                   Streptococcus suis     ASM1430v1      2096309.0      1.0   \n",
       "...                                ...           ...            ...      ...   \n",
       "12522       Streptococcus dysgalactiae     42880_F01      2193653.0      1.0   \n",
       "12523           Streptococcus gordonii     42912_F01      2185555.0      1.0   \n",
       "12524            Enterococcus faecalis     42197_D01      2864420.0      1.0   \n",
       "12525           Streptococcus porcinus     41965_D01      1954699.0      1.0   \n",
       "12526           Streptococcus porcinus     41851_D01      1949104.0      1.0   \n",
       "\n",
       "       Phage Contigs  Total Predicted Prophages  Kept  No phage genes  \\\n",
       "0                1.0                       29.0   4.0            12.0   \n",
       "1                1.0                        5.0   1.0             2.0   \n",
       "2                1.0                       25.0   2.0             9.0   \n",
       "3                4.0                       12.0   2.0             4.0   \n",
       "4                1.0                       19.0   4.0             9.0   \n",
       "...              ...                        ...   ...             ...   \n",
       "12522            1.0                       18.0   5.0             1.0   \n",
       "12523            1.0                       17.0   2.0             8.0   \n",
       "12524            1.0                       38.0   5.0             0.0   \n",
       "12525            1.0                       14.0   4.0             3.0   \n",
       "12526            1.0                       17.0   4.0             5.0   \n",
       "\n",
       "       Not enough genes  bp prophage  \n",
       "0                  13.0     129141.0  \n",
       "1                   2.0      48873.0  \n",
       "2                  14.0     107997.0  \n",
       "3                   6.0      56955.0  \n",
       "4                   6.0      84555.0  \n",
       "...                 ...          ...  \n",
       "12522              12.0     130605.0  \n",
       "12523               7.0      36608.0  \n",
       "12524              33.0     146449.0  \n",
       "12525               7.0     150445.0  \n",
       "12526               8.0     124214.0  \n",
       "\n",
       "[12527 rows x 17 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cvtax = pd.merge(gtdb[['assembly_accession', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'species']], phages, on='assembly_accession', how='inner')\n",
    "cvtax"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "general-mongolia",
   "metadata": {},
   "source": [
    "### Choose some based on not being abundant\n",
    "\n",
    "Note: see below for a better way to do this"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "suburban-requirement",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000008605.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Spirochaetota</td>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>Treponematales</td>\n",
       "      <td>Treponemataceae</td>\n",
       "      <td>Treponema</td>\n",
       "      <td>Treponema pallidum</td>\n",
       "      <td>ASM860v1</td>\n",
       "      <td>1138011.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>48873.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000010565.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_B</td>\n",
       "      <td>Desulfotomaculia</td>\n",
       "      <td>Desulfotomaculales</td>\n",
       "      <td>Pelotomaculaceae</td>\n",
       "      <td>Pelotomaculum</td>\n",
       "      <td>Pelotomaculum thermopropionicum</td>\n",
       "      <td>ASM1056v1</td>\n",
       "      <td>3025375.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>107997.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000013845.2</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Clostridiaceae</td>\n",
       "      <td>Clostridium_P</td>\n",
       "      <td>Clostridium_P perfringens</td>\n",
       "      <td>ASM1384v2</td>\n",
       "      <td>2960088.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>56955.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_000018565.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Chloroflexota</td>\n",
       "      <td>Chloroflexia</td>\n",
       "      <td>Chloroflexales</td>\n",
       "      <td>Herpetosiphonaceae</td>\n",
       "      <td>Herpetosiphon</td>\n",
       "      <td>Herpetosiphon aurantiacus</td>\n",
       "      <td>ASM1856v1</td>\n",
       "      <td>6785430.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>175634.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_000091165.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Methylomirabilota</td>\n",
       "      <td>Methylomirabilia</td>\n",
       "      <td>Methylomirabilales</td>\n",
       "      <td>Methylomirabilaceae</td>\n",
       "      <td>Methylomirabilis</td>\n",
       "      <td>Methylomirabilis oxyfera_B</td>\n",
       "      <td>ASM9116v1</td>\n",
       "      <td>2752854.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>80192.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12385</th>\n",
       "      <td>GCA_900638335.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Campylobacterota</td>\n",
       "      <td>Campylobacteria</td>\n",
       "      <td>Campylobacterales</td>\n",
       "      <td>Campylobacteraceae</td>\n",
       "      <td>Campylobacter_D</td>\n",
       "      <td>Campylobacter_D lari_C</td>\n",
       "      <td>57141_E01</td>\n",
       "      <td>1542878.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>5696.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12387</th>\n",
       "      <td>GCA_900638355.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Campylobacterota</td>\n",
       "      <td>Campylobacteria</td>\n",
       "      <td>Campylobacterales</td>\n",
       "      <td>Helicobacteraceae</td>\n",
       "      <td>Helicobacter_D</td>\n",
       "      <td>Helicobacter_D pullorum</td>\n",
       "      <td>57428_B01</td>\n",
       "      <td>1733410.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>144914.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12395</th>\n",
       "      <td>GCA_900638585.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Tissierellales</td>\n",
       "      <td>Peptoniphilaceae</td>\n",
       "      <td>Peptoniphilus_C</td>\n",
       "      <td>Peptoniphilus_C ivorii</td>\n",
       "      <td>57845_G01</td>\n",
       "      <td>1587771.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>34016.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12402</th>\n",
       "      <td>GCA_900638675.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Campylobacterota</td>\n",
       "      <td>Campylobacteria</td>\n",
       "      <td>Campylobacterales</td>\n",
       "      <td>Campylobacteraceae</td>\n",
       "      <td>Campylobacter_D</td>\n",
       "      <td>Campylobacter_D insulaenigrae</td>\n",
       "      <td>58225_E01</td>\n",
       "      <td>1500187.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>49721.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12504</th>\n",
       "      <td>GCA_901482605.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Clostridiaceae</td>\n",
       "      <td>Hathewaya</td>\n",
       "      <td>Hathewaya histolytica</td>\n",
       "      <td>45532_E01</td>\n",
       "      <td>2740791.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>177991.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1308 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      assembly_accession    domain             phylum             class  \\\n",
       "1        GCA_000008605.1  Bacteria      Spirochaetota      Spirochaetia   \n",
       "2        GCA_000010565.1  Bacteria       Firmicutes_B  Desulfotomaculia   \n",
       "3        GCA_000013845.2  Bacteria       Firmicutes_A        Clostridia   \n",
       "8        GCA_000018565.1  Bacteria      Chloroflexota      Chloroflexia   \n",
       "11       GCA_000091165.1  Bacteria  Methylomirabilota  Methylomirabilia   \n",
       "...                  ...       ...                ...               ...   \n",
       "12385    GCA_900638335.1  Bacteria   Campylobacterota   Campylobacteria   \n",
       "12387    GCA_900638355.1  Bacteria   Campylobacterota   Campylobacteria   \n",
       "12395    GCA_900638585.1  Bacteria       Firmicutes_A        Clostridia   \n",
       "12402    GCA_900638675.1  Bacteria   Campylobacterota   Campylobacteria   \n",
       "12504    GCA_901482605.1  Bacteria       Firmicutes_A        Clostridia   \n",
       "\n",
       "                    order               family             genus  \\\n",
       "1          Treponematales      Treponemataceae         Treponema   \n",
       "2      Desulfotomaculales     Pelotomaculaceae     Pelotomaculum   \n",
       "3           Clostridiales       Clostridiaceae     Clostridium_P   \n",
       "8          Chloroflexales   Herpetosiphonaceae     Herpetosiphon   \n",
       "11     Methylomirabilales  Methylomirabilaceae  Methylomirabilis   \n",
       "...                   ...                  ...               ...   \n",
       "12385   Campylobacterales   Campylobacteraceae   Campylobacter_D   \n",
       "12387   Campylobacterales    Helicobacteraceae    Helicobacter_D   \n",
       "12395      Tissierellales     Peptoniphilaceae   Peptoniphilus_C   \n",
       "12402   Campylobacterales   Campylobacteraceae   Campylobacter_D   \n",
       "12504       Clostridiales       Clostridiaceae         Hathewaya   \n",
       "\n",
       "                               species assembly_name  Genome length  Contigs  \\\n",
       "1                   Treponema pallidum      ASM860v1      1138011.0      1.0   \n",
       "2      Pelotomaculum thermopropionicum     ASM1056v1      3025375.0      1.0   \n",
       "3            Clostridium_P perfringens     ASM1384v2      2960088.0      4.0   \n",
       "8            Herpetosiphon aurantiacus     ASM1856v1      6785430.0      3.0   \n",
       "11          Methylomirabilis oxyfera_B     ASM9116v1      2752854.0      1.0   \n",
       "...                                ...           ...            ...      ...   \n",
       "12385           Campylobacter_D lari_C     57141_E01      1542878.0      1.0   \n",
       "12387          Helicobacter_D pullorum     57428_B01      1733410.0      1.0   \n",
       "12395           Peptoniphilus_C ivorii     57845_G01      1587771.0      1.0   \n",
       "12402    Campylobacter_D insulaenigrae     58225_E01      1500187.0      1.0   \n",
       "12504            Hathewaya histolytica     45532_E01      2740791.0      1.0   \n",
       "\n",
       "       Phage Contigs  Total Predicted Prophages  Kept  No phage genes  \\\n",
       "1                1.0                        5.0   1.0             2.0   \n",
       "2                1.0                       25.0   2.0             9.0   \n",
       "3                4.0                       12.0   2.0             4.0   \n",
       "8                3.0                       32.0   5.0             4.0   \n",
       "11               1.0                       24.0   3.0            11.0   \n",
       "...              ...                        ...   ...             ...   \n",
       "12385            1.0                       13.0   1.0             5.0   \n",
       "12387            1.0                       11.0   3.0             2.0   \n",
       "12395            1.0                       13.0   1.0             4.0   \n",
       "12402            1.0                        9.0   1.0             2.0   \n",
       "12504            1.0                       16.0   5.0             4.0   \n",
       "\n",
       "       Not enough genes  bp prophage  \n",
       "1                   2.0      48873.0  \n",
       "2                  14.0     107997.0  \n",
       "3                   6.0      56955.0  \n",
       "8                  23.0     175634.0  \n",
       "11                 10.0      80192.0  \n",
       "...                 ...          ...  \n",
       "12385               7.0       5696.0  \n",
       "12387               6.0     144914.0  \n",
       "12395               8.0      34016.0  \n",
       "12402               6.0      49721.0  \n",
       "12504               7.0     177991.0  \n",
       "\n",
       "[1308 rows x 17 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "notwanted = set({'Proteobacteria', 'Firmicutes', 'Actinobacteriota', 'Bacteroidota'})\n",
    "cvtax[~cvtax['phylum'].isin(notwanted)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "earlier-richmond",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phylum</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Acidobacteriota</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Aquificota</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Armatimonadota</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Bdellovibrionota</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Calditrichota</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Chloroflexota</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Chrysiogenetota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Cloacimonadota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Coprothermobacterota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Deferribacterota</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Deinococcota</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Dictyoglomota</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Elusimicrobiota</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Fibrobacterota</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Firmicutes_B</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Firmicutes_C</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Firmicutes_D</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Firmicutes_E</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Firmicutes_F</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Firmicutes_G</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Fusobacteriota</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Gemmatimonadota</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Margulisbacteria</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Methylomirabilota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Myxococcota</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Nitrospinota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Nitrospirota</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Omnitrophota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Patescibacteria</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Planctomycetota</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Sumerlaeota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Synergistota</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Thermodesulfobiota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Thermosulfidibacterota</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Thermotogota</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Verrucomicrobiota</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Verrucomicrobiota_A</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Zixibacteria</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    phylum  counts\n",
       "0          Acidobacteriota      13\n",
       "2               Aquificota       5\n",
       "3           Armatimonadota       2\n",
       "5         Bdellovibrionota       9\n",
       "6            Calditrichota       2\n",
       "8            Chloroflexota      42\n",
       "9          Chrysiogenetota       1\n",
       "10          Cloacimonadota       1\n",
       "11    Coprothermobacterota       1\n",
       "13        Deferribacterota       2\n",
       "14            Deinococcota      14\n",
       "16           Dictyoglomota       2\n",
       "17         Elusimicrobiota       2\n",
       "18          Fibrobacterota       4\n",
       "21            Firmicutes_B      35\n",
       "22            Firmicutes_C      32\n",
       "23            Firmicutes_D       1\n",
       "24            Firmicutes_E       5\n",
       "25            Firmicutes_F       5\n",
       "26            Firmicutes_G       1\n",
       "27          Fusobacteriota      35\n",
       "28         Gemmatimonadota       3\n",
       "29        Margulisbacteria       5\n",
       "30       Methylomirabilota       1\n",
       "31             Myxococcota      32\n",
       "32            Nitrospinota       1\n",
       "33            Nitrospirota       9\n",
       "34            Omnitrophota       1\n",
       "35         Patescibacteria      24\n",
       "36         Planctomycetota      19\n",
       "39             Sumerlaeota       1\n",
       "40            Synergistota      13\n",
       "41      Thermodesulfobiota       1\n",
       "42  Thermosulfidibacterota       1\n",
       "43            Thermotogota      13\n",
       "44       Verrucomicrobiota      31\n",
       "45     Verrucomicrobiota_A       4\n",
       "46            Zixibacteria       1"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tt = pd.DataFrame(cvtax.groupby('phylum')['phylum'].agg('count')).rename({'phylum':'counts'}, axis=1).reset_index()\n",
    "tt[tt['counts']<50]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "proprietary-hungarian",
   "metadata": {},
   "source": [
    "Choose about 100 organisms to be evenly distributed. Here we take the first four per phylum."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "decent-partner",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phylum</th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>domain</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Acidobacteriota</td>\n",
       "      <td>GCA_000022565.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Acidobacteriae</td>\n",
       "      <td>Acidobacteriales</td>\n",
       "      <td>Acidobacteriaceae</td>\n",
       "      <td>Acidobacterium</td>\n",
       "      <td>Acidobacterium capsulatum</td>\n",
       "      <td>ASM2256v1</td>\n",
       "      <td>4127356.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>34236.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Acidobacteriota</td>\n",
       "      <td>GCA_001664505.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Mor1</td>\n",
       "      <td>Mor1</td>\n",
       "      <td>Mor1</td>\n",
       "      <td>Mor1</td>\n",
       "      <td>Mor1 sp001664505</td>\n",
       "      <td>ASM166450v1</td>\n",
       "      <td>5989545.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>35778.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Acidobacteriota</td>\n",
       "      <td>GCA_000014005.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Acidobacteriae</td>\n",
       "      <td>Acidobacteriales</td>\n",
       "      <td>Koribacteraceae</td>\n",
       "      <td>Koribacter</td>\n",
       "      <td>Koribacter versatilis</td>\n",
       "      <td>ASM1400v1</td>\n",
       "      <td>5650368.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>65668.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Acidobacteriota</td>\n",
       "      <td>GCA_000014905.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Acidobacteriae</td>\n",
       "      <td>Bryobacterales</td>\n",
       "      <td>Bryobacteraceae</td>\n",
       "      <td>Solibacter</td>\n",
       "      <td>Solibacter usitatus</td>\n",
       "      <td>ASM1490v1</td>\n",
       "      <td>9965640.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>202405.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Aquificota</td>\n",
       "      <td>GCA_000021565.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Aquificae</td>\n",
       "      <td>Hydrogenothermales</td>\n",
       "      <td>Hydrogenothermaceae</td>\n",
       "      <td>Persephonella</td>\n",
       "      <td>Persephonella marina</td>\n",
       "      <td>ASM2156v1</td>\n",
       "      <td>1983966.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>76158.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Verrucomicrobiota_A</td>\n",
       "      <td>GCA_000237205.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Chlamydiia</td>\n",
       "      <td>Parachlamydiales</td>\n",
       "      <td>Simkaniaceae</td>\n",
       "      <td>Simkania</td>\n",
       "      <td>Simkania negevensis</td>\n",
       "      <td>ASM23720v1</td>\n",
       "      <td>2628375.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>48385.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>Verrucomicrobiota_A</td>\n",
       "      <td>GCA_000011565.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Chlamydiia</td>\n",
       "      <td>Parachlamydiales</td>\n",
       "      <td>Parachlamydiaceae</td>\n",
       "      <td>Protochlamydia</td>\n",
       "      <td>Protochlamydia amoebophila</td>\n",
       "      <td>ASM1156v1</td>\n",
       "      <td>2414465.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>60933.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>Verrucomicrobiota_A</td>\n",
       "      <td>GCA_001499655.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Chlamydiia</td>\n",
       "      <td>Parachlamydiales</td>\n",
       "      <td>Parachlamydiaceae</td>\n",
       "      <td>Protochlamydia</td>\n",
       "      <td>Protochlamydia naegleriophila</td>\n",
       "      <td>PNK1</td>\n",
       "      <td>3030375.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>82376.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>Verrucomicrobiota_A</td>\n",
       "      <td>GCA_000253035.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Chlamydiia</td>\n",
       "      <td>Parachlamydiales</td>\n",
       "      <td>Parachlamydiaceae</td>\n",
       "      <td>Parachlamydia</td>\n",
       "      <td>Parachlamydia acanthamoebae</td>\n",
       "      <td>ASM25303v1</td>\n",
       "      <td>3072383.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>14095.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>Zixibacteria</td>\n",
       "      <td>GCA_000447245.1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>MSB-5A5</td>\n",
       "      <td>MSB-5A5</td>\n",
       "      <td>RBG-1</td>\n",
       "      <td>RBG-1</td>\n",
       "      <td>RBG-1 sp000447245</td>\n",
       "      <td>RBG_1</td>\n",
       "      <td>2122767.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24195.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>105 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  phylum assembly_accession    domain           class  \\\n",
       "0        Acidobacteriota    GCA_000022565.1  Bacteria  Acidobacteriae   \n",
       "1        Acidobacteriota    GCA_001664505.1  Bacteria            Mor1   \n",
       "2        Acidobacteriota    GCA_000014005.1  Bacteria  Acidobacteriae   \n",
       "3        Acidobacteriota    GCA_000014905.1  Bacteria  Acidobacteriae   \n",
       "4             Aquificota    GCA_000021565.1  Bacteria       Aquificae   \n",
       "..                   ...                ...       ...             ...   \n",
       "100  Verrucomicrobiota_A    GCA_000237205.1  Bacteria      Chlamydiia   \n",
       "101  Verrucomicrobiota_A    GCA_000011565.1  Bacteria      Chlamydiia   \n",
       "102  Verrucomicrobiota_A    GCA_001499655.1  Bacteria      Chlamydiia   \n",
       "103  Verrucomicrobiota_A    GCA_000253035.1  Bacteria      Chlamydiia   \n",
       "104         Zixibacteria    GCA_000447245.1  Bacteria         MSB-5A5   \n",
       "\n",
       "                  order               family           genus  \\\n",
       "0      Acidobacteriales    Acidobacteriaceae  Acidobacterium   \n",
       "1                  Mor1                 Mor1            Mor1   \n",
       "2      Acidobacteriales      Koribacteraceae      Koribacter   \n",
       "3        Bryobacterales      Bryobacteraceae      Solibacter   \n",
       "4    Hydrogenothermales  Hydrogenothermaceae   Persephonella   \n",
       "..                  ...                  ...             ...   \n",
       "100    Parachlamydiales         Simkaniaceae        Simkania   \n",
       "101    Parachlamydiales    Parachlamydiaceae  Protochlamydia   \n",
       "102    Parachlamydiales    Parachlamydiaceae  Protochlamydia   \n",
       "103    Parachlamydiales    Parachlamydiaceae   Parachlamydia   \n",
       "104             MSB-5A5                RBG-1           RBG-1   \n",
       "\n",
       "                           species assembly_name  Genome length  Contigs  \\\n",
       "0        Acidobacterium capsulatum     ASM2256v1      4127356.0      1.0   \n",
       "1                 Mor1 sp001664505   ASM166450v1      5989545.0      1.0   \n",
       "2            Koribacter versatilis     ASM1400v1      5650368.0      1.0   \n",
       "3              Solibacter usitatus     ASM1490v1      9965640.0      1.0   \n",
       "4             Persephonella marina     ASM2156v1      1983966.0      2.0   \n",
       "..                             ...           ...            ...      ...   \n",
       "100            Simkania negevensis    ASM23720v1      2628375.0      2.0   \n",
       "101     Protochlamydia amoebophila     ASM1156v1      2414465.0      1.0   \n",
       "102  Protochlamydia naegleriophila          PNK1      3030375.0      2.0   \n",
       "103    Parachlamydia acanthamoebae    ASM25303v1      3072383.0      1.0   \n",
       "104              RBG-1 sp000447245         RBG_1      2122767.0      1.0   \n",
       "\n",
       "     Phage Contigs  Total Predicted Prophages  Kept  No phage genes  \\\n",
       "0              1.0                       18.0   1.0             6.0   \n",
       "1              1.0                        7.0   2.0             4.0   \n",
       "2              1.0                       16.0   2.0             1.0   \n",
       "3              1.0                       60.0   8.0            11.0   \n",
       "4              2.0                        7.0   2.0             2.0   \n",
       "..             ...                        ...   ...             ...   \n",
       "100            2.0                       18.0   3.0            10.0   \n",
       "101            1.0                        9.0   1.0             4.0   \n",
       "102            2.0                       20.0   2.0             8.0   \n",
       "103            1.0                       14.0   1.0             5.0   \n",
       "104            1.0                        9.0   1.0             6.0   \n",
       "\n",
       "     Not enough genes  bp prophage  \n",
       "0                11.0      34236.0  \n",
       "1                 1.0      35778.0  \n",
       "2                13.0      65668.0  \n",
       "3                41.0     202405.0  \n",
       "4                 3.0      76158.0  \n",
       "..                ...          ...  \n",
       "100               5.0      48385.0  \n",
       "101               4.0      60933.0  \n",
       "102              10.0      82376.0  \n",
       "103               8.0      14095.0  \n",
       "104               2.0      24195.0  \n",
       "\n",
       "[105 rows x 17 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cvtax[cvtax['phylum'].isin(tt[tt['counts']<50]['phylum'])].groupby('phylum').nth([0,1,2,3]).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "incomplete-cleaner",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GCA_000022565.1',\n",
       " 'GCA_001664505.1',\n",
       " 'GCA_000014005.1',\n",
       " 'GCA_000014905.1',\n",
       " 'GCA_000021565.1',\n",
       " 'GCA_003518825.1',\n",
       " 'GCA_000164905.1',\n",
       " 'GCA_000010785.1',\n",
       " 'GCA_000724625.1',\n",
       " 'GCA_002162075.1',\n",
       " 'GCA_000348725.1',\n",
       " 'GCA_000196175.1',\n",
       " 'GCA_004011035.1',\n",
       " 'GCA_000317895.1',\n",
       " 'GCA_000241815.1',\n",
       " 'GCA_001886815.1',\n",
       " 'GCA_000009025.1',\n",
       " 'GCA_001761445.1',\n",
       " 'GCA_000018565.1',\n",
       " 'GCA_000011905.1',\n",
       " 'GCA_000177635.2',\n",
       " 'GCA_000146065.1',\n",
       " 'GCA_000020945.1',\n",
       " 'GCA_000025725.1',\n",
       " 'GCA_004087915.1',\n",
       " 'GCA_007165645.1',\n",
       " 'GCA_007182855.1',\n",
       " 'GCA_000008565.1',\n",
       " 'GCA_003568585.1',\n",
       " 'GCA_000020965.1',\n",
       " 'GCA_000021645.1',\n",
       " 'GCA_000020145.1',\n",
       " 'GCA_001027545.1',\n",
       " 'GCA_002300445.1',\n",
       " 'GCA_002300465.1',\n",
       " 'GCA_002846255.1',\n",
       " 'GCA_000024665.1',\n",
       " 'GCA_000013105.1',\n",
       " 'GCA_000012865.1',\n",
       " 'GCA_000010565.1',\n",
       " 'GCA_000214705.1',\n",
       " 'GCA_000209975.1',\n",
       " 'GCA_007164725.1',\n",
       " 'GCA_003537535.1',\n",
       " 'GCA_002082765.1',\n",
       " 'GCA_000020005.1',\n",
       " 'GCA_000009905.1',\n",
       " 'GCA_000219855.1',\n",
       " 'GCA_000237975.1',\n",
       " 'GCA_004634385.1',\n",
       " 'GCA_000144695.1',\n",
       " 'GCA_000166415.1',\n",
       " 'GCA_000020485.1',\n",
       " 'GCA_000328625.1',\n",
       " 'GCA_001544015.1',\n",
       " 'GCA_002573475.1',\n",
       " 'GCA_001296125.1',\n",
       " 'GCA_000023905.1',\n",
       " 'GCA_000024405.1',\n",
       " 'GCA_000695095.2',\n",
       " 'GCA_000522985.1',\n",
       " 'GCA_000010305.1',\n",
       " 'GCA_001771465.1',\n",
       " 'GCA_001771575.1',\n",
       " 'GCA_001771495.1',\n",
       " 'GCA_001771425.1',\n",
       " 'GCA_000091165.1',\n",
       " 'GCA_000012685.1',\n",
       " 'GCA_001263205.1',\n",
       " 'GCA_001931535.1',\n",
       " 'GCA_000020805.1',\n",
       " 'GCA_000341545.2',\n",
       " 'GCA_000299235.1',\n",
       " 'GCA_000020985.1',\n",
       " 'GCA_000284315.1',\n",
       " 'GCA_000196815.1',\n",
       " 'GCA_004102945.1',\n",
       " 'GCA_000505065.1',\n",
       " 'GCA_000503875.1',\n",
       " 'GCA_000503915.1',\n",
       " 'GCA_000990655.1',\n",
       " 'GCA_001999965.1',\n",
       " 'GCA_001610875.1',\n",
       " 'GCA_000092105.1',\n",
       " 'GCA_002007645.1',\n",
       " 'GCA_003290465.1',\n",
       " 'GCA_000210715.1',\n",
       " 'GCA_000025885.1',\n",
       " 'GCA_000161995.1',\n",
       " 'GCA_000024905.1',\n",
       " 'GCA_000212395.1',\n",
       " 'GCA_001547735.1',\n",
       " 'GCA_000016905.1',\n",
       " 'GCA_000018605.1',\n",
       " 'GCA_000023325.1',\n",
       " 'GCA_000021285.1',\n",
       " 'GCA_000020225.1',\n",
       " 'GCA_000025905.1',\n",
       " 'GCA_000019965.1',\n",
       " 'GCA_000242935.3',\n",
       " 'GCA_000237205.1',\n",
       " 'GCA_000011565.1',\n",
       " 'GCA_001499655.1',\n",
       " 'GCA_000253035.1',\n",
       " 'GCA_000447245.1']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(cvtax[cvtax['phylum'].isin(tt[tt['counts']<50]['phylum'])].groupby('phylum').nth([0,1,2,3]).reset_index()['assembly_accession'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "proved-projector",
   "metadata": {},
   "source": [
    "# The problem with this approach!\n",
    "\n",
    "The problem is that these are all _unusual_ bacteria and so the phages are _unusual_. Not so great for beginners to learn on. Lets revist and choose a few other bacteria that people might be able to work with."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "recent-thumbnail",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phylum</th>\n",
       "      <th>counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Actinobacteriota</td>\n",
       "      <td>1929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Campylobacterota</td>\n",
       "      <td>307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Desulfobacterota</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>2817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>6096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Spirochaetota</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              phylum  counts\n",
       "1   Actinobacteriota    1929\n",
       "4       Bacteroidota     377\n",
       "7   Campylobacterota     307\n",
       "12     Cyanobacteria      92\n",
       "15  Desulfobacterota      65\n",
       "19        Firmicutes    2817\n",
       "20      Firmicutes_A     398\n",
       "37    Proteobacteria    6096\n",
       "38     Spirochaetota      72"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tt[tt['counts']>50]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "traditional-electric",
   "metadata": {},
   "source": [
    "Let's repeat the above approach with just these organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "lasting-investigator",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GCA_000389905.1',\n",
       " 'GCA_000277085.1',\n",
       " 'GCA_000277105.1',\n",
       " 'GCA_000389925.1',\n",
       " 'GCA_000418455.1',\n",
       " 'GCA_000270345.1',\n",
       " 'GCA_000210075.1',\n",
       " 'GCA_000380305.1',\n",
       " 'GCA_000738065.1',\n",
       " 'GCA_001274835.1',\n",
       " 'GCA_001051075.1',\n",
       " 'GCA_001444325.1',\n",
       " 'GCA_001506325.1',\n",
       " 'GCA_001808165.1',\n",
       " 'GCA_000495505.1',\n",
       " 'GCA_002082995.2',\n",
       " 'GCA_002312965.1',\n",
       " 'GCA_000274865.1',\n",
       " 'GCA_000011485.1',\n",
       " 'GCA_002813895.1',\n",
       " 'GCA_003846445.1',\n",
       " 'GCA_000007925.1',\n",
       " 'GCA_000011465.1',\n",
       " 'GCA_000012505.1',\n",
       " 'GCA_001595385.3',\n",
       " 'GCA_000007985.2',\n",
       " 'GCA_000177215.2',\n",
       " 'GCA_900095395.1',\n",
       " 'GCA_900079715.1',\n",
       " 'GCA_900079685.1',\n",
       " 'GCA_000014325.1',\n",
       " 'GCA_000017965.1',\n",
       " 'GCA_000161535.1',\n",
       " 'GCA_000160995.1',\n",
       " 'GCA_000014305.1',\n",
       " 'GCA_000143435.1',\n",
       " 'GCA_000209915.1',\n",
       " 'GCA_000209895.1',\n",
       " 'GCA_000209835.1',\n",
       " 'GCA_000209855.1',\n",
       " 'GCA_000209815.1',\n",
       " 'GCA_000013845.2',\n",
       " 'GCA_000023545.1',\n",
       " 'GCA_000023565.1',\n",
       " 'GCA_000146265.4',\n",
       " 'GCA_000152565.1',\n",
       " 'GCA_000015425.1',\n",
       " 'GCA_000007385.1',\n",
       " 'GCA_002269215.1',\n",
       " 'GCA_000007685.1',\n",
       " 'GCA_000808095.1',\n",
       " 'GCA_000413015.1',\n",
       " 'GCA_000008185.1',\n",
       " 'GCA_000008605.1']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(cvtax[cvtax['phylum'].isin(tt[tt['counts']>50]['phylum'])].groupby('phylum').nth([0,1,2,3,4,5]).reset_index()['assembly_accession'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "polyphonic-heather",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
