{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "84b3492d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A lot of this is not used, but we import it so we have it later!\n",
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from PhiSpyAnalysis import theils_u, DateConverter, printmd\n",
    "from PhiSpyAnalysis import read_phages, read_gtdb, read_checkv, read_base_pp, read_categories\n",
    "from PhiSpyAnalysis import read_metadata, read_gbk_metadata, read_phage_locations\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "from sklearn import decomposition\n",
    "from sklearn.preprocessing import OrdinalEncoder\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "\n",
    "\n",
    "import subprocess\n",
    "import gzip\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c07e3328",
   "metadata": {},
   "outputs": [],
   "source": [
    "locs = read_phage_locations(use_small_data=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "65d0f075",
   "metadata": {},
   "outputs": [],
   "source": [
    "locs['density'] = locs['# CDS']/locs['Length']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "89b95760",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENOMEID</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Start</th>\n",
       "      <th>Stop</th>\n",
       "      <th>Length</th>\n",
       "      <th># CDS</th>\n",
       "      <th>Decision</th>\n",
       "      <th>density</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>VTFL01000006.1</td>\n",
       "      <td>337</td>\n",
       "      <td>34176</td>\n",
       "      <td>33840</td>\n",
       "      <td>27</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "      <td>0.000798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_013484905.1</td>\n",
       "      <td>DABJIP010000020.1</td>\n",
       "      <td>36363</td>\n",
       "      <td>53031</td>\n",
       "      <td>16669</td>\n",
       "      <td>22</td>\n",
       "      <td>Kept</td>\n",
       "      <td>0.001320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000485715.1</td>\n",
       "      <td>KI522405.1</td>\n",
       "      <td>1</td>\n",
       "      <td>9505</td>\n",
       "      <td>9505</td>\n",
       "      <td>10</td>\n",
       "      <td>Kept</td>\n",
       "      <td>0.001052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_010929115.1</td>\n",
       "      <td>AAPAQY010000001.1</td>\n",
       "      <td>113322</td>\n",
       "      <td>114212</td>\n",
       "      <td>891</td>\n",
       "      <td>1</td>\n",
       "      <td>Dropped. Not enough genes</td>\n",
       "      <td>0.001122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_005232675.1</td>\n",
       "      <td>AACJPY010000001.1</td>\n",
       "      <td>791108</td>\n",
       "      <td>812149</td>\n",
       "      <td>21042</td>\n",
       "      <td>25</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "      <td>0.001188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>GCA_004844065.1</td>\n",
       "      <td>AABJPG010000023.1</td>\n",
       "      <td>96</td>\n",
       "      <td>1160</td>\n",
       "      <td>1065</td>\n",
       "      <td>1</td>\n",
       "      <td>Dropped. Not enough genes</td>\n",
       "      <td>0.000939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>GCA_003580095.1</td>\n",
       "      <td>QVSP01000007.1</td>\n",
       "      <td>21665</td>\n",
       "      <td>22153</td>\n",
       "      <td>489</td>\n",
       "      <td>1</td>\n",
       "      <td>Dropped. Not enough genes</td>\n",
       "      <td>0.002045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>GCA_017448945.1</td>\n",
       "      <td>JAFSKM010000017.1</td>\n",
       "      <td>111759</td>\n",
       "      <td>126778</td>\n",
       "      <td>15020</td>\n",
       "      <td>24</td>\n",
       "      <td>Kept</td>\n",
       "      <td>0.001598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>GCA_012404705.1</td>\n",
       "      <td>AASKON010000022.1</td>\n",
       "      <td>59264</td>\n",
       "      <td>59644</td>\n",
       "      <td>381</td>\n",
       "      <td>1</td>\n",
       "      <td>Dropped. Not enough genes</td>\n",
       "      <td>0.002625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000</th>\n",
       "      <td>GCA_013173585.1</td>\n",
       "      <td>JAAUCA010000001.1</td>\n",
       "      <td>17668</td>\n",
       "      <td>18525</td>\n",
       "      <td>858</td>\n",
       "      <td>1</td>\n",
       "      <td>Dropped. Not enough genes</td>\n",
       "      <td>0.001166</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1001 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             GENOMEID             Contig   Start    Stop  Length  # CDS  \\\n",
       "0     GCA_008244535.1     VTFL01000006.1     337   34176   33840     27   \n",
       "1     GCA_013484905.1  DABJIP010000020.1   36363   53031   16669     22   \n",
       "2     GCA_000485715.1         KI522405.1       1    9505    9505     10   \n",
       "3     GCA_010929115.1  AAPAQY010000001.1  113322  114212     891      1   \n",
       "4     GCA_005232675.1  AACJPY010000001.1  791108  812149   21042     25   \n",
       "...               ...                ...     ...     ...     ...    ...   \n",
       "996   GCA_004844065.1  AABJPG010000023.1      96    1160    1065      1   \n",
       "997   GCA_003580095.1     QVSP01000007.1   21665   22153     489      1   \n",
       "998   GCA_017448945.1  JAFSKM010000017.1  111759  126778   15020     24   \n",
       "999   GCA_012404705.1  AASKON010000022.1   59264   59644     381      1   \n",
       "1000  GCA_013173585.1  JAAUCA010000001.1   17668   18525     858      1   \n",
       "\n",
       "                                              Decision   density  \n",
       "0     Dropped. No genes were identified as phage genes  0.000798  \n",
       "1                                                 Kept  0.001320  \n",
       "2                                                 Kept  0.001052  \n",
       "3                            Dropped. Not enough genes  0.001122  \n",
       "4     Dropped. No genes were identified as phage genes  0.001188  \n",
       "...                                                ...       ...  \n",
       "996                          Dropped. Not enough genes  0.000939  \n",
       "997                          Dropped. Not enough genes  0.002045  \n",
       "998                                               Kept  0.001598  \n",
       "999                          Dropped. Not enough genes  0.002625  \n",
       "1000                         Dropped. Not enough genes  0.001166  \n",
       "\n",
       "[1001 rows x 8 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "locs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e3005891",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
