{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "internal-anchor",
   "metadata": {},
   "source": [
    "## Compare the assemblies in GenBank, GTDB, and RAST\n",
    "\n",
    "Each has a different set. What are the unions and intersections?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "acute-prospect",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'Bio'",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mModuleNotFoundError\u001B[0m                       Traceback (most recent call last)",
      "Input \u001B[1;32mIn [1]\u001B[0m, in \u001B[0;36m<cell line: 13>\u001B[1;34m()\u001B[0m\n\u001B[0;32m     10\u001B[0m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;21;01mmath\u001B[39;00m\n\u001B[0;32m     11\u001B[0m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;21;01mre\u001B[39;00m\n\u001B[1;32m---> 13\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mPhiSpyAnalysis\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m theils_u, DateConverter, printmd\n\u001B[0;32m     14\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mPhiSpyAnalysis\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m read_phages, read_gtdb, read_checkv, read_base_pp, read_categories, read_metadata, read_gbk_metadata\n\u001B[0;32m     16\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mscipy\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mstats\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m pearsonr, f_oneway\n",
      "File \u001B[1;32mC:\\Workspace\\edwa0468\\GitHubs\\PhispyAnalysis\\PhiSpyAnalysis\\__init__.py:8\u001B[0m, in \u001B[0;36m<module>\u001B[1;34m\u001B[0m\n\u001B[0;32m      6\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mread_data_frames\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m read_insertion_lengths, read_phage_locations\n\u001B[0;32m      7\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mdisplay\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m printmd\n\u001B[1;32m----> 8\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mgenbank\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m is_gzip_file, parse_genbank\n\u001B[0;32m     10\u001B[0m __all__ \u001B[38;5;241m=\u001B[39m [\n\u001B[0;32m     11\u001B[0m     \u001B[38;5;124m'\u001B[39m\u001B[38;5;124mtheils_u\u001B[39m\u001B[38;5;124m'\u001B[39m,\n\u001B[0;32m     12\u001B[0m     \u001B[38;5;124m'\u001B[39m\u001B[38;5;124mCantConvertDate\u001B[39m\u001B[38;5;124m'\u001B[39m, \u001B[38;5;124m'\u001B[39m\u001B[38;5;124mDateConverter\u001B[39m\u001B[38;5;124m'\u001B[39m,\n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m     17\u001B[0m \n\u001B[0;32m     18\u001B[0m ]\n",
      "File \u001B[1;32mC:\\Workspace\\edwa0468\\GitHubs\\PhispyAnalysis\\PhiSpyAnalysis\\genbank.py:7\u001B[0m, in \u001B[0;36m<module>\u001B[1;34m\u001B[0m\n\u001B[0;32m      5\u001B[0m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;21;01msys\u001B[39;00m\n\u001B[0;32m      6\u001B[0m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;21;01mgzip\u001B[39;00m\n\u001B[1;32m----> 7\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mBio\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m SeqIO\n\u001B[0;32m      8\u001B[0m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;21;01mbinascii\u001B[39;00m\n\u001B[0;32m     10\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21mis_gzip_file\u001B[39m(f):\n",
      "\u001B[1;31mModuleNotFoundError\u001B[0m: No module named 'Bio'"
     ]
    }
   ],
   "source": [
    "# A lot of this is not used, but we import it so we have it later!\n",
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from PhiSpyAnalysis import theils_u, DateConverter, printmd\n",
    "from PhiSpyAnalysis import read_phages, read_gtdb, read_checkv, read_base_pp, read_categories, read_metadata, read_gbk_metadata\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "\n",
    "\n",
    "import subprocess\n",
    "import gzip\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "digital-cancellation",
   "metadata": {},
   "outputs": [],
   "source": [
    "rv = re.compile('^\\\\w+')\n",
    "def remove_ver(x):\n",
    "    return rv.search(str(x)).group()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "statewide-terror",
   "metadata": {},
   "source": [
    "# GTDB\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "compact-shell",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000012145.1</td>\n",
       "      <td>GCA_000012145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000013845.2</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000016465.1</td>\n",
       "      <td>GCA_000016465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000024525.1</td>\n",
       "      <td>GCA_000024525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000152525.1</td>\n",
       "      <td>GCA_000152525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311475</th>\n",
       "      <td>GCA_910574395.1</td>\n",
       "      <td>GCA_910574395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311476</th>\n",
       "      <td>GCA_910589955.1</td>\n",
       "      <td>GCA_910589955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311477</th>\n",
       "      <td>GCA_910590315.1</td>\n",
       "      <td>GCA_910590315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311478</th>\n",
       "      <td>GCA_910592735.1</td>\n",
       "      <td>GCA_910592735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311479</th>\n",
       "      <td>GCA_910593725.1</td>\n",
       "      <td>GCA_910593725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311480 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000012145.1  GCA_000012145\n",
       "1         GCA_000013845.2  GCA_000013845\n",
       "2         GCA_000016465.1  GCA_000016465\n",
       "3         GCA_000024525.1  GCA_000024525\n",
       "4         GCA_000152525.1  GCA_000152525\n",
       "...                   ...            ...\n",
       "311475    GCA_910574395.1  GCA_910574395\n",
       "311476    GCA_910589955.1  GCA_910589955\n",
       "311477    GCA_910590315.1  GCA_910590315\n",
       "311478    GCA_910592735.1  GCA_910592735\n",
       "311479    GCA_910593725.1  GCA_910593725\n",
       "\n",
       "[311480 rows x 2 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GTDB\n",
    "gtdb = read_gtdb()\n",
    "gtdb['assembly_nover'] = gtdb['assembly_accession'].apply(remove_ver)\n",
    "gtdb[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "pediatric-seven",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>accession</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_lineage</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>...</th>\n",
       "      <th>trna_count</th>\n",
       "      <th>trna_selenocysteine_count</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000012145.1</td>\n",
       "      <td>GB_GCA_000012145.1</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.47</td>\n",
       "      <td>324</td>\n",
       "      <td>o__Rickettsiales (UID3809)</td>\n",
       "      <td>211</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1344326</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rickettsiales</td>\n",
       "      <td>Rickettsiaceae</td>\n",
       "      <td>Rickettsia</td>\n",
       "      <td>Rickettsia felis</td>\n",
       "      <td>GCA_000012145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000013845.2</td>\n",
       "      <td>GB_GCA_000013845.2</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>332</td>\n",
       "      <td>o__Clostridiales (UID1375)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2428396</td>\n",
       "      <td>...</td>\n",
       "      <td>95</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Clostridiaceae</td>\n",
       "      <td>Clostridium_P</td>\n",
       "      <td>Clostridium_P perfringens</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000016465.1</td>\n",
       "      <td>GB_GCA_000016465.1</td>\n",
       "      <td>14</td>\n",
       "      <td>99.55</td>\n",
       "      <td>0.23</td>\n",
       "      <td>767</td>\n",
       "      <td>f__Pasteurellaceae (UID4932)</td>\n",
       "      <td>440</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1584718</td>\n",
       "      <td>...</td>\n",
       "      <td>58</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Pasteurellaceae</td>\n",
       "      <td>Haemophilus</td>\n",
       "      <td>Haemophilus influenzae</td>\n",
       "      <td>GCA_000016465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000024525.1</td>\n",
       "      <td>GB_GCA_000024525.1</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.89</td>\n",
       "      <td>454</td>\n",
       "      <td>o__Cytophagales (UID2936)</td>\n",
       "      <td>336</td>\n",
       "      <td>25.0</td>\n",
       "      <td>7481814</td>\n",
       "      <td>...</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Cytophagales</td>\n",
       "      <td>Spirosomaceae</td>\n",
       "      <td>Spirosoma</td>\n",
       "      <td>Spirosoma linguale</td>\n",
       "      <td>GCA_000024525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000152525.1</td>\n",
       "      <td>GB_GCA_000152525.1</td>\n",
       "      <td>0</td>\n",
       "      <td>96.45</td>\n",
       "      <td>0.11</td>\n",
       "      <td>813</td>\n",
       "      <td>o__Pseudomonadales (UID4488)</td>\n",
       "      <td>308</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5429460</td>\n",
       "      <td>...</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>Pseudomonadaceae</td>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>Pseudomonas aeruginosa</td>\n",
       "      <td>GCA_000152525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311475</th>\n",
       "      <td>GCA_910574395.1</td>\n",
       "      <td>RS_GCF_910574395.1</td>\n",
       "      <td>1</td>\n",
       "      <td>99.22</td>\n",
       "      <td>0.78</td>\n",
       "      <td>449</td>\n",
       "      <td>g__Lactobacillus (UID380)</td>\n",
       "      <td>129</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1773962</td>\n",
       "      <td>...</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>Lactobacillaceae</td>\n",
       "      <td>Lactobacillus</td>\n",
       "      <td>Lactobacillus sp910589675</td>\n",
       "      <td>GCA_910574395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311476</th>\n",
       "      <td>GCA_910589955.1</td>\n",
       "      <td>RS_GCF_910589955.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.81</td>\n",
       "      <td>0.66</td>\n",
       "      <td>773</td>\n",
       "      <td>g__Staphylococcus (UID294)</td>\n",
       "      <td>178</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2102381</td>\n",
       "      <td>...</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Staphylococcales</td>\n",
       "      <td>Staphylococcaceae</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>GCA_910589955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311477</th>\n",
       "      <td>GCA_910590315.1</td>\n",
       "      <td>RS_GCF_910590315.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.40</td>\n",
       "      <td>0.00</td>\n",
       "      <td>773</td>\n",
       "      <td>g__Staphylococcus (UID294)</td>\n",
       "      <td>178</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2058841</td>\n",
       "      <td>...</td>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Staphylococcales</td>\n",
       "      <td>Staphylococcaceae</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>Staphylococcus epidermidis</td>\n",
       "      <td>GCA_910590315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311478</th>\n",
       "      <td>GCA_910592735.1</td>\n",
       "      <td>RS_GCF_910592735.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.97</td>\n",
       "      <td>0.67</td>\n",
       "      <td>1173</td>\n",
       "      <td>f__Enterobacteriaceae (UID5124)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4666456</td>\n",
       "      <td>...</td>\n",
       "      <td>81</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Escherichia</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>GCA_910592735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311479</th>\n",
       "      <td>GCA_910593725.1</td>\n",
       "      <td>RS_GCF_910593725.1</td>\n",
       "      <td>0</td>\n",
       "      <td>64.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>570</td>\n",
       "      <td>o__Burkholderiales (UID4001)</td>\n",
       "      <td>250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>767513</td>\n",
       "      <td>...</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Mycetohabitans</td>\n",
       "      <td>Mycetohabitans tarda</td>\n",
       "      <td>GCA_910593725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311480 rows Ã— 119 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession           accession  ambiguous_bases  \\\n",
       "0         GCA_000012145.1  GB_GCA_000012145.1                0   \n",
       "1         GCA_000013845.2  GB_GCA_000013845.2                0   \n",
       "2         GCA_000016465.1  GB_GCA_000016465.1               14   \n",
       "3         GCA_000024525.1  GB_GCA_000024525.1                0   \n",
       "4         GCA_000152525.1  GB_GCA_000152525.1                0   \n",
       "...                   ...                 ...              ...   \n",
       "311475    GCA_910574395.1  RS_GCF_910574395.1                1   \n",
       "311476    GCA_910589955.1  RS_GCF_910589955.1                0   \n",
       "311477    GCA_910590315.1  RS_GCF_910590315.1                0   \n",
       "311478    GCA_910592735.1  RS_GCF_910592735.1                0   \n",
       "311479    GCA_910593725.1  RS_GCF_910593725.1                0   \n",
       "\n",
       "        checkm_completeness  checkm_contamination  checkm_marker_count  \\\n",
       "0                    100.00                  0.47                  324   \n",
       "1                    100.00                  0.00                  332   \n",
       "2                     99.55                  0.23                  767   \n",
       "3                    100.00                  0.89                  454   \n",
       "4                     96.45                  0.11                  813   \n",
       "...                     ...                   ...                  ...   \n",
       "311475                99.22                  0.78                  449   \n",
       "311476                99.81                  0.66                  773   \n",
       "311477                99.40                  0.00                  773   \n",
       "311478                99.97                  0.67                 1173   \n",
       "311479                64.08                  0.00                  570   \n",
       "\n",
       "                  checkm_marker_lineage  checkm_marker_set_count  \\\n",
       "0            o__Rickettsiales (UID3809)                      211   \n",
       "1            o__Clostridiales (UID1375)                      124   \n",
       "2          f__Pasteurellaceae (UID4932)                      440   \n",
       "3             o__Cytophagales (UID2936)                      336   \n",
       "4          o__Pseudomonadales (UID4488)                      308   \n",
       "...                                 ...                      ...   \n",
       "311475        g__Lactobacillus (UID380)                      129   \n",
       "311476       g__Staphylococcus (UID294)                      178   \n",
       "311477       g__Staphylococcus (UID294)                      178   \n",
       "311478  f__Enterobacteriaceae (UID5124)                      336   \n",
       "311479     o__Burkholderiales (UID4001)                      250   \n",
       "\n",
       "        checkm_strain_heterogeneity  coding_bases  ...  trna_count  \\\n",
       "0                               0.0       1344326  ...          33   \n",
       "1                               0.0       2428396  ...          95   \n",
       "2                               0.0       1584718  ...          58   \n",
       "3                              25.0       7481814  ...          47   \n",
       "4                               0.0       5429460  ...          37   \n",
       "...                             ...           ...  ...         ...   \n",
       "311475                          0.0       1773962  ...          56   \n",
       "311476                          0.0       2102381  ...          54   \n",
       "311477                          0.0       2058841  ...          59   \n",
       "311478                          0.0       4666456  ...          81   \n",
       "311479                          0.0        767513  ...          41   \n",
       "\n",
       "        trna_selenocysteine_count    domain          phylum  \\\n",
       "0                               0  Bacteria  Proteobacteria   \n",
       "1                               1  Bacteria    Firmicutes_A   \n",
       "2                               1  Bacteria  Proteobacteria   \n",
       "3                               0  Bacteria    Bacteroidota   \n",
       "4                               1  Bacteria  Proteobacteria   \n",
       "...                           ...       ...             ...   \n",
       "311475                          0  Bacteria      Firmicutes   \n",
       "311476                          0  Bacteria      Firmicutes   \n",
       "311477                          0  Bacteria      Firmicutes   \n",
       "311478                          1  Bacteria  Proteobacteria   \n",
       "311479                          0  Bacteria  Proteobacteria   \n",
       "\n",
       "                      class             order              family  \\\n",
       "0       Alphaproteobacteria     Rickettsiales      Rickettsiaceae   \n",
       "1                Clostridia     Clostridiales      Clostridiaceae   \n",
       "2       Gammaproteobacteria  Enterobacterales     Pasteurellaceae   \n",
       "3               Bacteroidia      Cytophagales       Spirosomaceae   \n",
       "4       Gammaproteobacteria   Pseudomonadales    Pseudomonadaceae   \n",
       "...                     ...               ...                 ...   \n",
       "311475              Bacilli   Lactobacillales    Lactobacillaceae   \n",
       "311476              Bacilli  Staphylococcales   Staphylococcaceae   \n",
       "311477              Bacilli  Staphylococcales   Staphylococcaceae   \n",
       "311478  Gammaproteobacteria  Enterobacterales  Enterobacteriaceae   \n",
       "311479  Gammaproteobacteria   Burkholderiales    Burkholderiaceae   \n",
       "\n",
       "                 genus                     species assembly_nover  \n",
       "0           Rickettsia            Rickettsia felis  GCA_000012145  \n",
       "1        Clostridium_P   Clostridium_P perfringens  GCA_000013845  \n",
       "2          Haemophilus      Haemophilus influenzae  GCA_000016465  \n",
       "3            Spirosoma          Spirosoma linguale  GCA_000024525  \n",
       "4          Pseudomonas      Pseudomonas aeruginosa  GCA_000152525  \n",
       "...                ...                         ...            ...  \n",
       "311475   Lactobacillus   Lactobacillus sp910589675  GCA_910574395  \n",
       "311476  Staphylococcus  Staphylococcus epidermidis  GCA_910589955  \n",
       "311477  Staphylococcus  Staphylococcus epidermidis  GCA_910590315  \n",
       "311478     Escherichia            Escherichia coli  GCA_910592735  \n",
       "311479  Mycetohabitans        Mycetohabitans tarda  GCA_910593725  \n",
       "\n",
       "[311480 rows x 119 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "textile-software",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>GCA_000003135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003215.1</td>\n",
       "      <td>GCA_000003215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>GCA_000003645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>GCA_000003925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>GCA_000003955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320171</th>\n",
       "      <td>GCF_900167595.1</td>\n",
       "      <td>GCF_900167595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320172</th>\n",
       "      <td>GCF_900167605.1</td>\n",
       "      <td>GCF_900167605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320173</th>\n",
       "      <td>GCF_900167615.1</td>\n",
       "      <td>GCF_900167615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320174</th>\n",
       "      <td>GCF_900167625.1</td>\n",
       "      <td>GCF_900167625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320175</th>\n",
       "      <td>GCF_900167635.1</td>\n",
       "      <td>GCF_900167635</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320176 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000003135.1  GCA_000003135\n",
       "1         GCA_000003215.1  GCA_000003215\n",
       "2         GCA_000003645.1  GCA_000003645\n",
       "3         GCA_000003925.1  GCA_000003925\n",
       "4         GCA_000003955.1  GCA_000003955\n",
       "...                   ...            ...\n",
       "320171    GCF_900167595.1  GCF_900167595\n",
       "320172    GCF_900167605.1  GCF_900167605\n",
       "320173    GCF_900167615.1  GCF_900167615\n",
       "320174    GCF_900167625.1  GCF_900167625\n",
       "320175    GCF_900167635.1  GCF_900167635\n",
       "\n",
       "[320176 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# RAST\n",
    "# the full data set. Don't try this at home!\n",
    "# metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "rast = read_metadata()\n",
    "rast['assembly_nover'] = rast['assembly_accession'].apply(remove_ver)\n",
    "rast[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "sustained-broadcast",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>...</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "      <th>isolation_date</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>5.484803e+05</td>\n",
       "      <td>Bifidobacterium longum subsp. longum ATCC 55813</td>\n",
       "      <td>Bifidobacterium longum subsp. longum ATCC 55813</td>\n",
       "      <td>548480</td>\n",
       "      <td>WGS</td>\n",
       "      <td>ATCC 55813</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Anaerobic</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bifidobacterium longum subsp. longum ATCC 5581...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003215.1</td>\n",
       "      <td>4.556313e+05</td>\n",
       "      <td>Clostridium difficile QCD-66c26</td>\n",
       "      <td>Clostridium difficile QCD-66c26</td>\n",
       "      <td>455631</td>\n",
       "      <td>WGS</td>\n",
       "      <td>QCD-66c26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Anaerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Antibiotic-associated Diarrhea;Pseudomembraneo...</td>\n",
       "      <td>Clostridium difficile QCD-66c26. was isolated ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>5.269733e+05</td>\n",
       "      <td>Bacillus cereus m1293</td>\n",
       "      <td>Bacillus cereus m1293</td>\n",
       "      <td>526973</td>\n",
       "      <td>WGS</td>\n",
       "      <td>m1293</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>Bacillus cereus m1293. This strain, also know...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>5.269973e+05</td>\n",
       "      <td>Bacillus mycoides DSM 2048</td>\n",
       "      <td>Bacillus mycoides DSM 2048</td>\n",
       "      <td>526997</td>\n",
       "      <td>WGS</td>\n",
       "      <td>DSM 2048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacillus mycoides DSM 2048. Bacillus mycoides...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>5.269943e+05</td>\n",
       "      <td>Bacillus cereus AH1273</td>\n",
       "      <td>Bacillus cereus AH1273</td>\n",
       "      <td>526994</td>\n",
       "      <td>WGS</td>\n",
       "      <td>AH1273</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>Bacillus cereus AH1273. This strain was isola...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320171</th>\n",
       "      <td>GCF_900167595.1</td>\n",
       "      <td>1.496160e+03</td>\n",
       "      <td>Clostridioides difficile strain VRECD0186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1496</td>\n",
       "      <td>WGS</td>\n",
       "      <td>VRECD0186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>These data are part of a pre-publication relea...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015.488706</td>\n",
       "      <td>GCF_900167595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320172</th>\n",
       "      <td>GCF_900167605.1</td>\n",
       "      <td>1.147123e+06</td>\n",
       "      <td>Caloramator quimbayensis strain USBA 833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1147123</td>\n",
       "      <td>WGS</td>\n",
       "      <td>USBA 833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Caloramator quimbayensis USBA 833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_900167605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320173</th>\n",
       "      <td>GCF_900167615.1</td>\n",
       "      <td>1.496159e+03</td>\n",
       "      <td>Clostridioides difficile strain CD19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1496</td>\n",
       "      <td>WGS</td>\n",
       "      <td>CD19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>http://www.sanger.ac.uk/resources/downloads/ba...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014.489391</td>\n",
       "      <td>GCF_900167615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320174</th>\n",
       "      <td>GCF_900167625.1</td>\n",
       "      <td>1.496159e+03</td>\n",
       "      <td>Clostridioides difficile strain VRECD0076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1496</td>\n",
       "      <td>WGS</td>\n",
       "      <td>VRECD0076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>These data are part of a pre-publication relea...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014.489391</td>\n",
       "      <td>GCF_900167625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320175</th>\n",
       "      <td>GCF_900167635.1</td>\n",
       "      <td>3.591109e+05</td>\n",
       "      <td>Pseudomonas extremaustralis strain USBA 515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>359110</td>\n",
       "      <td>WGS</td>\n",
       "      <td>USBA 515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pseudomonas extremaustralis USBA 515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_900167635</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320176 rows Ã— 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession     genome_id  \\\n",
       "0         GCA_000003135.1  5.484803e+05   \n",
       "1         GCA_000003215.1  4.556313e+05   \n",
       "2         GCA_000003645.1  5.269733e+05   \n",
       "3         GCA_000003925.1  5.269973e+05   \n",
       "4         GCA_000003955.1  5.269943e+05   \n",
       "...                   ...           ...   \n",
       "320171    GCF_900167595.1  1.496160e+03   \n",
       "320172    GCF_900167605.1  1.147123e+06   \n",
       "320173    GCF_900167615.1  1.496159e+03   \n",
       "320174    GCF_900167625.1  1.496159e+03   \n",
       "320175    GCF_900167635.1  3.591109e+05   \n",
       "\n",
       "                                            genome_name  \\\n",
       "0       Bifidobacterium longum subsp. longum ATCC 55813   \n",
       "1                       Clostridium difficile QCD-66c26   \n",
       "2                                 Bacillus cereus m1293   \n",
       "3                            Bacillus mycoides DSM 2048   \n",
       "4                                Bacillus cereus AH1273   \n",
       "...                                                 ...   \n",
       "320171        Clostridioides difficile strain VRECD0186   \n",
       "320172         Caloramator quimbayensis strain USBA 833   \n",
       "320173             Clostridioides difficile strain CD19   \n",
       "320174        Clostridioides difficile strain VRECD0076   \n",
       "320175      Pseudomonas extremaustralis strain USBA 515   \n",
       "\n",
       "                                          organism_name  taxon_id  \\\n",
       "0       Bifidobacterium longum subsp. longum ATCC 55813    548480   \n",
       "1                       Clostridium difficile QCD-66c26    455631   \n",
       "2                                 Bacillus cereus m1293    526973   \n",
       "3                            Bacillus mycoides DSM 2048    526997   \n",
       "4                                Bacillus cereus AH1273    526994   \n",
       "...                                                 ...       ...   \n",
       "320171                                              NaN      1496   \n",
       "320172                                              NaN   1147123   \n",
       "320173                                              NaN      1496   \n",
       "320174                                              NaN      1496   \n",
       "320175                                              NaN    359110   \n",
       "\n",
       "       genome_status      strain serovar biovar pathovar  ...  \\\n",
       "0                WGS  ATCC 55813     NaN    NaN      NaN  ...   \n",
       "1                WGS   QCD-66c26     NaN    NaN      NaN  ...   \n",
       "2                WGS       m1293     NaN    NaN      NaN  ...   \n",
       "3                WGS    DSM 2048     NaN    NaN      NaN  ...   \n",
       "4                WGS      AH1273     NaN    NaN      NaN  ...   \n",
       "...              ...         ...     ...    ...      ...  ...   \n",
       "320171           WGS   VRECD0186     NaN    NaN      NaN  ...   \n",
       "320172           WGS    USBA 833     NaN    NaN      NaN  ...   \n",
       "320173           WGS        CD19     NaN    NaN      NaN  ...   \n",
       "320174           WGS   VRECD0076     NaN    NaN      NaN  ...   \n",
       "320175           WGS    USBA 515     NaN    NaN      NaN  ...   \n",
       "\n",
       "       temperature_range optimal_temperature salinity oxygen_requirement  \\\n",
       "0             Mesophilic                   -      NaN          Anaerobic   \n",
       "1             Mesophilic                  37      NaN          Anaerobic   \n",
       "2             Mesophilic                   -      NaN            Aerobic   \n",
       "3                    NaN                   -      NaN                NaN   \n",
       "4             Mesophilic                   -      NaN            Aerobic   \n",
       "...                  ...                 ...      ...                ...   \n",
       "320171               NaN                 NaN      NaN                NaN   \n",
       "320172               NaN                 NaN      NaN                NaN   \n",
       "320173               NaN                 NaN      NaN                NaN   \n",
       "320174               NaN                 NaN      NaN                NaN   \n",
       "320175               NaN                 NaN      NaN                NaN   \n",
       "\n",
       "                habitat                                            disease  \\\n",
       "0       Host-associated                                                NaN   \n",
       "1              Multiple  Antibiotic-associated Diarrhea;Pseudomembraneo...   \n",
       "2              Multiple                                     Food poisoning   \n",
       "3                   NaN                                                NaN   \n",
       "4              Multiple                                     Food poisoning   \n",
       "...                 ...                                                ...   \n",
       "320171              NaN                                                NaN   \n",
       "320172              NaN                                                NaN   \n",
       "320173              NaN                                                NaN   \n",
       "320174              NaN                                                NaN   \n",
       "320175              NaN                                                NaN   \n",
       "\n",
       "                                                 comments additional_metadata  \\\n",
       "0       Bifidobacterium longum subsp. longum ATCC 5581...                 NaN   \n",
       "1       Clostridium difficile QCD-66c26. was isolated ...                 NaN   \n",
       "2        Bacillus cereus m1293. This strain, also know...                 NaN   \n",
       "3        Bacillus mycoides DSM 2048. Bacillus mycoides...                 NaN   \n",
       "4        Bacillus cereus AH1273. This strain was isola...                 NaN   \n",
       "...                                                   ...                 ...   \n",
       "320171  These data are part of a pre-publication relea...                 NaN   \n",
       "320172                  Caloramator quimbayensis USBA 833                 NaN   \n",
       "320173  http://www.sanger.ac.uk/resources/downloads/ba...                 NaN   \n",
       "320174  These data are part of a pre-publication relea...                 NaN   \n",
       "320175               Pseudomonas extremaustralis USBA 515                 NaN   \n",
       "\n",
       "       isolation_date assembly_nover  \n",
       "0                 NaN  GCA_000003135  \n",
       "1                 NaN  GCA_000003215  \n",
       "2                 NaN  GCA_000003645  \n",
       "3                 NaN  GCA_000003925  \n",
       "4                 NaN  GCA_000003955  \n",
       "...               ...            ...  \n",
       "320171    2015.488706  GCF_900167595  \n",
       "320172            NaN  GCF_900167605  \n",
       "320173    2014.489391  GCF_900167615  \n",
       "320174    2014.489391  GCF_900167625  \n",
       "320175            NaN  GCF_900167635  \n",
       "\n",
       "[320176 rows x 68 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rast"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "considerable-brief",
   "metadata": {},
   "source": [
    "# GenBank\n",
    "\n",
    "This assembly summary comes from [GenBank ftp site](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/) and you want the [assembly_summary.txt](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/assembly_summary.txt) file explicitly from bacteria (but don't try and open the bacteria list in your browser!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "curious-compatibility",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3338: DtypeWarning: Columns (21) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if await self.run_code(code, result, async_=asy):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_019958295.1</td>\n",
       "      <td>GCA_019958295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>GCA_900128725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>GCA_008244535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_008015785.1</td>\n",
       "      <td>GCA_008015785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001735525.1</td>\n",
       "      <td>GCA_001735525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195588</th>\n",
       "      <td>GCA_914768085.1</td>\n",
       "      <td>GCA_914768085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195589</th>\n",
       "      <td>GCA_910595835.1</td>\n",
       "      <td>GCA_910595835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195590</th>\n",
       "      <td>GCA_910595855.1</td>\n",
       "      <td>GCA_910595855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195591</th>\n",
       "      <td>GCA_910595815.1</td>\n",
       "      <td>GCA_910595815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195592</th>\n",
       "      <td>GCA_023573945.1</td>\n",
       "      <td>GCA_023573945</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1195593 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        assembly_accession assembly_nover\n",
       "0          GCA_019958295.1  GCA_019958295\n",
       "1          GCA_900128725.1  GCA_900128725\n",
       "2          GCA_008244535.1  GCA_008244535\n",
       "3          GCA_008015785.1  GCA_008015785\n",
       "4          GCA_001735525.1  GCA_001735525\n",
       "...                    ...            ...\n",
       "1195588    GCA_914768085.1  GCA_914768085\n",
       "1195589    GCA_910595835.1  GCA_910595835\n",
       "1195590    GCA_910595855.1  GCA_910595855\n",
       "1195591    GCA_910595815.1  GCA_910595815\n",
       "1195592    GCA_023573945.1  GCA_023573945\n",
       "\n",
       "[1195593 rows x 2 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GenBank\n",
    "gbk = read_gbk_metadata()\n",
    "gbk['assembly_nover'] = gbk['assembly_accession'].apply(remove_ver)\n",
    "gbk[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "toxic-applicant",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>...</th>\n",
       "      <th>seq_rel_date</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>submitter</th>\n",
       "      <th>gbrs_paired_asm</th>\n",
       "      <th>paired_asm_comp</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>asm_not_live_date</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_019958295.1</td>\n",
       "      <td>GCA_019958295.1</td>\n",
       "      <td>PRJNA755163</td>\n",
       "      <td>SAMN20395209</td>\n",
       "      <td>JAIOWO000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>Buchnera aphidicola</td>\n",
       "      <td>strain=Tabriz.4</td>\n",
       "      <td>...</td>\n",
       "      <td>2021/09/13</td>\n",
       "      <td>ASM1995829v1</td>\n",
       "      <td>Tabriz University, Tabriz, Iran</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...</td>\n",
       "      <td>fragmented assembly</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_019958295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>PRJEB15400</td>\n",
       "      <td>SAMEA4556317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>Buchnera aphidicola</td>\n",
       "      <td>strain=BCifornacula</td>\n",
       "      <td>...</td>\n",
       "      <td>2016/11/25</td>\n",
       "      <td>BCifornacula_v1.0</td>\n",
       "      <td>Centre de Biologie pour la Gestion des Populat...</td>\n",
       "      <td>GCF_900128725.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_900128725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>PRJNA562146</td>\n",
       "      <td>SAMN12634886</td>\n",
       "      <td>VTFL00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "      <td>strain=PYS_80_B</td>\n",
       "      <td>...</td>\n",
       "      <td>2019/09/02</td>\n",
       "      <td>ASM824453v1</td>\n",
       "      <td>Agharkar Research Institute</td>\n",
       "      <td>GCF_008244535.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_008244535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_008015785.1</td>\n",
       "      <td>GCA_008015785.1</td>\n",
       "      <td>PRJNA428383</td>\n",
       "      <td>SAMN10026444</td>\n",
       "      <td>SSGG00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>Methylophilus methylotrophus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2019/08/19</td>\n",
       "      <td>ASM801578v1</td>\n",
       "      <td>Colorado School of Mines</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...</td>\n",
       "      <td>derived from metagenome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_008015785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001735525.1</td>\n",
       "      <td>GCA_001735525.1</td>\n",
       "      <td>PRJNA328548</td>\n",
       "      <td>SAMN05384437</td>\n",
       "      <td>MCBT00000000.1</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>Shewanella colwelliana</td>\n",
       "      <td>strain=CSB03KR</td>\n",
       "      <td>...</td>\n",
       "      <td>2016/09/19</td>\n",
       "      <td>ASM173552v1</td>\n",
       "      <td>Chonnam National University</td>\n",
       "      <td>GCF_001735525.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_001735525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195588</th>\n",
       "      <td>GCA_914768085.1</td>\n",
       "      <td>GCA_914768085.1</td>\n",
       "      <td>PRJEB47378</td>\n",
       "      <td>SAMEA9570072</td>\n",
       "      <td>CAJZCJ000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2946699</td>\n",
       "      <td>2946699</td>\n",
       "      <td>Snodgrassella communis</td>\n",
       "      <td>strain=R-54236</td>\n",
       "      <td>...</td>\n",
       "      <td>2022/02/17</td>\n",
       "      <td>R-54236</td>\n",
       "      <td>Sciensano</td>\n",
       "      <td>GCF_914768085.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_914768085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195589</th>\n",
       "      <td>GCA_910595835.1</td>\n",
       "      <td>GCA_910595835.1</td>\n",
       "      <td>PRJEB42974</td>\n",
       "      <td>SAMEA8541143</td>\n",
       "      <td>CAJVCI000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2946838</td>\n",
       "      <td>2946838</td>\n",
       "      <td>Candidatus Thioglobus vadi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2021/07/07</td>\n",
       "      <td>SUP05-3</td>\n",
       "      <td>MAX PLANCK INSTITUTE FOR MARINE MICROBIOLOGY</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...</td>\n",
       "      <td>derived from environmental source</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_910595835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195590</th>\n",
       "      <td>GCA_910595855.1</td>\n",
       "      <td>GCA_910595855.1</td>\n",
       "      <td>PRJEB42974</td>\n",
       "      <td>SAMEA8541145</td>\n",
       "      <td>CAJVCJ000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2946839</td>\n",
       "      <td>2946839</td>\n",
       "      <td>Candidatus Thioglobus vulcanius</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2021/07/07</td>\n",
       "      <td>SUP05-5</td>\n",
       "      <td>MAX PLANCK INSTITUTE FOR MARINE MICROBIOLOGY</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...</td>\n",
       "      <td>derived from environmental source</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_910595855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195591</th>\n",
       "      <td>GCA_910595815.1</td>\n",
       "      <td>GCA_910595815.1</td>\n",
       "      <td>PRJEB42974</td>\n",
       "      <td>SAMEA8541147</td>\n",
       "      <td>CAJVCM000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2946840</td>\n",
       "      <td>2946840</td>\n",
       "      <td>Candidatus Thioglobus plumae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2021/07/07</td>\n",
       "      <td>SUP05-6</td>\n",
       "      <td>MAX PLANCK INSTITUTE FOR MARINE MICROBIOLOGY</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...</td>\n",
       "      <td>derived from environmental source</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_910595815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195592</th>\n",
       "      <td>GCA_023573945.1</td>\n",
       "      <td>GCA_023573945.1</td>\n",
       "      <td>PRJNA841762</td>\n",
       "      <td>SAMN28614168</td>\n",
       "      <td>JAMKEZ000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2946863</td>\n",
       "      <td>2946863</td>\n",
       "      <td>Apilactobacillus sp. F1</td>\n",
       "      <td>strain=F1</td>\n",
       "      <td>...</td>\n",
       "      <td>2022/05/29</td>\n",
       "      <td>ASM2357394v1</td>\n",
       "      <td>NIBGE</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>GCA_023573945</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1195593 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        assembly_accession # assembly_accession   bioproject     biosample  \\\n",
       "0          GCA_019958295.1      GCA_019958295.1  PRJNA755163  SAMN20395209   \n",
       "1          GCA_900128725.1      GCA_900128725.1   PRJEB15400  SAMEA4556317   \n",
       "2          GCA_008244535.1      GCA_008244535.1  PRJNA562146  SAMN12634886   \n",
       "3          GCA_008015785.1      GCA_008015785.1  PRJNA428383  SAMN10026444   \n",
       "4          GCA_001735525.1      GCA_001735525.1  PRJNA328548  SAMN05384437   \n",
       "...                    ...                  ...          ...           ...   \n",
       "1195588    GCA_914768085.1      GCA_914768085.1   PRJEB47378  SAMEA9570072   \n",
       "1195589    GCA_910595835.1      GCA_910595835.1   PRJEB42974  SAMEA8541143   \n",
       "1195590    GCA_910595855.1      GCA_910595855.1   PRJEB42974  SAMEA8541145   \n",
       "1195591    GCA_910595815.1      GCA_910595815.1   PRJEB42974  SAMEA8541147   \n",
       "1195592    GCA_023573945.1      GCA_023573945.1  PRJNA841762  SAMN28614168   \n",
       "\n",
       "                wgs_master        refseq_category    taxid  species_taxid  \\\n",
       "0        JAIOWO000000000.1                     na        9              9   \n",
       "1                      NaN                     na        9              9   \n",
       "2           VTFL00000000.1                     na       14             14   \n",
       "3           SSGG00000000.1                     na       17             17   \n",
       "4           MCBT00000000.1  representative genome       23             23   \n",
       "...                    ...                    ...      ...            ...   \n",
       "1195588  CAJZCJ000000000.1                     na  2946699        2946699   \n",
       "1195589  CAJVCI000000000.1                     na  2946838        2946838   \n",
       "1195590  CAJVCJ000000000.1                     na  2946839        2946839   \n",
       "1195591  CAJVCM000000000.1                     na  2946840        2946840   \n",
       "1195592  JAMKEZ000000000.1                     na  2946863        2946863   \n",
       "\n",
       "                           organism_name   infraspecific_name  ...  \\\n",
       "0                    Buchnera aphidicola      strain=Tabriz.4  ...   \n",
       "1                    Buchnera aphidicola  strain=BCifornacula  ...   \n",
       "2              Dictyoglomus thermophilum      strain=PYS_80_B  ...   \n",
       "3           Methylophilus methylotrophus                  NaN  ...   \n",
       "4                 Shewanella colwelliana       strain=CSB03KR  ...   \n",
       "...                                  ...                  ...  ...   \n",
       "1195588           Snodgrassella communis       strain=R-54236  ...   \n",
       "1195589       Candidatus Thioglobus vadi                  NaN  ...   \n",
       "1195590  Candidatus Thioglobus vulcanius                  NaN  ...   \n",
       "1195591     Candidatus Thioglobus plumae                  NaN  ...   \n",
       "1195592          Apilactobacillus sp. F1            strain=F1  ...   \n",
       "\n",
       "        seq_rel_date           asm_name  \\\n",
       "0         2021/09/13       ASM1995829v1   \n",
       "1         2016/11/25  BCifornacula_v1.0   \n",
       "2         2019/09/02        ASM824453v1   \n",
       "3         2019/08/19        ASM801578v1   \n",
       "4         2016/09/19        ASM173552v1   \n",
       "...              ...                ...   \n",
       "1195588   2022/02/17            R-54236   \n",
       "1195589   2021/07/07            SUP05-3   \n",
       "1195590   2021/07/07            SUP05-5   \n",
       "1195591   2021/07/07            SUP05-6   \n",
       "1195592   2022/05/29       ASM2357394v1   \n",
       "\n",
       "                                                 submitter  gbrs_paired_asm  \\\n",
       "0                          Tabriz University, Tabriz, Iran               na   \n",
       "1        Centre de Biologie pour la Gestion des Populat...  GCF_900128725.1   \n",
       "2                              Agharkar Research Institute  GCF_008244535.1   \n",
       "3                                 Colorado School of Mines               na   \n",
       "4                              Chonnam National University  GCF_001735525.1   \n",
       "...                                                    ...              ...   \n",
       "1195588                                          Sciensano  GCF_914768085.1   \n",
       "1195589       MAX PLANCK INSTITUTE FOR MARINE MICROBIOLOGY               na   \n",
       "1195590       MAX PLANCK INSTITUTE FOR MARINE MICROBIOLOGY               na   \n",
       "1195591       MAX PLANCK INSTITUTE FOR MARINE MICROBIOLOGY               na   \n",
       "1195592                                              NIBGE               na   \n",
       "\n",
       "        paired_asm_comp                                           ftp_path  \\\n",
       "0                    na  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...   \n",
       "1             identical  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...   \n",
       "2             identical  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...   \n",
       "3                    na  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...   \n",
       "4             identical  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...   \n",
       "...                 ...                                                ...   \n",
       "1195588       identical  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...   \n",
       "1195589              na  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...   \n",
       "1195590              na  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...   \n",
       "1195591              na  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/9...   \n",
       "1195592              na  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/0...   \n",
       "\n",
       "                      excluded_from_refseq    relation_to_type_material  \\\n",
       "0                      fragmented assembly                          NaN   \n",
       "1                                      NaN                          NaN   \n",
       "2                                      NaN                          NaN   \n",
       "3                  derived from metagenome                          NaN   \n",
       "4                                      NaN                          NaN   \n",
       "...                                    ...                          ...   \n",
       "1195588                                NaN                          NaN   \n",
       "1195589  derived from environmental source  assembly from type material   \n",
       "1195590  derived from environmental source  assembly from type material   \n",
       "1195591  derived from environmental source  assembly from type material   \n",
       "1195592                                NaN                          NaN   \n",
       "\n",
       "        asm_not_live_date assembly_nover  \n",
       "0                      na  GCA_019958295  \n",
       "1                      na  GCA_900128725  \n",
       "2                      na  GCA_008244535  \n",
       "3                      na  GCA_008015785  \n",
       "4                      na  GCA_001735525  \n",
       "...                   ...            ...  \n",
       "1195588                na  GCA_914768085  \n",
       "1195589                na  GCA_910595835  \n",
       "1195590                na  GCA_910595855  \n",
       "1195591                na  GCA_910595815  \n",
       "1195592                na  GCA_023573945  \n",
       "\n",
       "[1195593 rows x 25 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbk"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab832325",
   "metadata": {},
   "source": [
    "# Phages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "endless-entrance",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Please note that this was run with git commit 93794ca that has 1,195,494 genomes parsed.\n",
      "Initially there were 5,005,011 kept phages, but after filtering we kept 5,005,011 prophages from 945,685 genomes"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>GENOMEID</th>\n",
       "      <th>Total_bp</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Contigs_Checked</th>\n",
       "      <th>Total_Predicted_Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No_phage_genes</th>\n",
       "      <th>Not_enough_genes</th>\n",
       "      <th>bp_of_kept_prophages</th>\n",
       "      <th>bp_no_phage_genes</th>\n",
       "      <th>bp_not_enough_genes</th>\n",
       "      <th>Note</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>1963348</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>79796</td>\n",
       "      <td>104676</td>\n",
       "      <td>13431</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_008244535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_008015785.1</td>\n",
       "      <td>GCA_008015785.1</td>\n",
       "      <td>1188656</td>\n",
       "      <td>94</td>\n",
       "      <td>94</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>183294</td>\n",
       "      <td>26642</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_008015785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001735525.1</td>\n",
       "      <td>GCA_001735525.1</td>\n",
       "      <td>4642543</td>\n",
       "      <td>52</td>\n",
       "      <td>33</td>\n",
       "      <td>30</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "      <td>83559</td>\n",
       "      <td>58207</td>\n",
       "      <td>33920</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_001735525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_019654935.1</td>\n",
       "      <td>GCA_019654935.1</td>\n",
       "      <td>4631520</td>\n",
       "      <td>171</td>\n",
       "      <td>47</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>59552</td>\n",
       "      <td>10210</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_019654935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_019655395.1</td>\n",
       "      <td>GCA_019655395.1</td>\n",
       "      <td>4473905</td>\n",
       "      <td>472</td>\n",
       "      <td>218</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>17577</td>\n",
       "      <td>24469</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_019655395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195489</th>\n",
       "      <td>GCA_020319415.1</td>\n",
       "      <td>GCA_020319415.1</td>\n",
       "      <td>5559579</td>\n",
       "      <td>310</td>\n",
       "      <td>143</td>\n",
       "      <td>76</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>63</td>\n",
       "      <td>178853</td>\n",
       "      <td>9738</td>\n",
       "      <td>127058</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_020319415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195490</th>\n",
       "      <td>GCA_020319515.1</td>\n",
       "      <td>GCA_020319515.1</td>\n",
       "      <td>5497958</td>\n",
       "      <td>201</td>\n",
       "      <td>116</td>\n",
       "      <td>69</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>215550</td>\n",
       "      <td>5011</td>\n",
       "      <td>105851</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_020319515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195491</th>\n",
       "      <td>GCA_020319535.1</td>\n",
       "      <td>GCA_020319535.1</td>\n",
       "      <td>5294704</td>\n",
       "      <td>195</td>\n",
       "      <td>95</td>\n",
       "      <td>68</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>59</td>\n",
       "      <td>114216</td>\n",
       "      <td>0</td>\n",
       "      <td>133228</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_020319535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195492</th>\n",
       "      <td>GCA_020319715.1</td>\n",
       "      <td>GCA_020319715.1</td>\n",
       "      <td>4928082</td>\n",
       "      <td>165</td>\n",
       "      <td>87</td>\n",
       "      <td>44</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>40</td>\n",
       "      <td>116732</td>\n",
       "      <td>0</td>\n",
       "      <td>73535</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_020319715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195493</th>\n",
       "      <td>GCA_020319735.1</td>\n",
       "      <td>GCA_020319735.1</td>\n",
       "      <td>5653408</td>\n",
       "      <td>409</td>\n",
       "      <td>163</td>\n",
       "      <td>79</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>67</td>\n",
       "      <td>185500</td>\n",
       "      <td>0</td>\n",
       "      <td>101065</td>\n",
       "      <td>Complete</td>\n",
       "      <td>GCA_020319735</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>945685 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        assembly_accession         GENOMEID  Total_bp  Contigs  \\\n",
       "2          GCA_008244535.1  GCA_008244535.1   1963348       23   \n",
       "3          GCA_008015785.1  GCA_008015785.1   1188656       94   \n",
       "4          GCA_001735525.1  GCA_001735525.1   4642543       52   \n",
       "5          GCA_019654935.1  GCA_019654935.1   4631520      171   \n",
       "6          GCA_019655395.1  GCA_019655395.1   4473905      472   \n",
       "...                    ...              ...       ...      ...   \n",
       "1195489    GCA_020319415.1  GCA_020319415.1   5559579      310   \n",
       "1195490    GCA_020319515.1  GCA_020319515.1   5497958      201   \n",
       "1195491    GCA_020319535.1  GCA_020319535.1   5294704      195   \n",
       "1195492    GCA_020319715.1  GCA_020319715.1   4928082      165   \n",
       "1195493    GCA_020319735.1  GCA_020319735.1   5653408      409   \n",
       "\n",
       "         Contigs_Checked  Total_Predicted_Prophages  Kept  No_phage_genes  \\\n",
       "2                     14                         17     4               5   \n",
       "3                     94                         35     0              22   \n",
       "4                     33                         30     3               4   \n",
       "5                     47                         13     0               2   \n",
       "6                    218                         11     0               3   \n",
       "...                  ...                        ...   ...             ...   \n",
       "1195489              143                         76    11               2   \n",
       "1195490              116                         69    13               1   \n",
       "1195491               95                         68     9               0   \n",
       "1195492               87                         44     4               0   \n",
       "1195493              163                         79    12               0   \n",
       "\n",
       "         Not_enough_genes  bp_of_kept_prophages  bp_no_phage_genes  \\\n",
       "2                       8                 79796             104676   \n",
       "3                      13                     0             183294   \n",
       "4                      23                 83559              58207   \n",
       "5                      11                     0              59552   \n",
       "6                       8                     0              17577   \n",
       "...                   ...                   ...                ...   \n",
       "1195489                63                178853               9738   \n",
       "1195490                55                215550               5011   \n",
       "1195491                59                114216                  0   \n",
       "1195492                40                116732                  0   \n",
       "1195493                67                185500                  0   \n",
       "\n",
       "         bp_not_enough_genes      Note assembly_nover  \n",
       "2                      13431  Complete  GCA_008244535  \n",
       "3                      26642  Complete  GCA_008015785  \n",
       "4                      33920  Complete  GCA_001735525  \n",
       "5                      10210  Complete  GCA_019654935  \n",
       "6                      24469  Complete  GCA_019655395  \n",
       "...                      ...       ...            ...  \n",
       "1195489               127058  Complete  GCA_020319415  \n",
       "1195490               105851  Complete  GCA_020319515  \n",
       "1195491               133228  Complete  GCA_020319535  \n",
       "1195492                73535  Complete  GCA_020319715  \n",
       "1195493               101065  Complete  GCA_020319735  \n",
       "\n",
       "[945685 rows x 14 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagesdf = read_phages(maxcontigs=-1) # this disables contig length filtering\n",
    "phagesdf['assembly_nover'] = phagesdf['assembly_accession'].apply(remove_ver)\n",
    "phagesdf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "medieval-canyon",
   "metadata": {},
   "source": [
    "# What is in common between the groups?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "changing-president",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1,195,593 genomes in GenBank\n",
      "There are 320,176 genomes in PATRIC\n",
      "There are 311,468 genomes in GTDB\n",
      "We have analyzed phages in 945,685 genomes\n",
      "\n",
      "Between GBK and RAST there are 242,239 genomes in common\n",
      "Between GBK and GTDB there are 308,578 genomes in common\n",
      "Between GTDB and RAST there are 205,940 genomes in common\n",
      "\n",
      "There are 249,908 phages in Genbank that have not been analyzed\n",
      "There are 115,918 phages in GTDB that have not been analyzed\n",
      "There are 176,920 phages in Genbank that have not been analyzed\n",
      "\n",
      "There are 83,931 phages in all three that have not been analyzed\n"
     ]
    }
   ],
   "source": [
    "gbkaa=set(gbk['assembly_accession'])\n",
    "rastaa=set(rast['assembly_accession'])\n",
    "gtdbaa=set(gtdb['assembly_accession'])\n",
    "phagesaa=set(phagesdf['assembly_accession'])\n",
    "\n",
    "\n",
    "gbr = gbkaa.intersection(rastaa)\n",
    "gbg = gbkaa.intersection(gtdbaa)\n",
    "gtr = gtdbaa.intersection(rastaa)\n",
    "\n",
    "print(f\"There are {len(gbkaa):,} genomes in GenBank\")\n",
    "print(f\"There are {len(rastaa):,} genomes in PATRIC\")\n",
    "print(f\"There are {len(gtdbaa):,} genomes in GTDB\")\n",
    "print(f\"We have analyzed phages in {len(phagesaa):,} genomes\")\n",
    "print()\n",
    "print(f\"Between GBK and RAST there are {len(gbr):,} genomes in common\")\n",
    "print(f\"Between GBK and GTDB there are {len(gbg):,} genomes in common\")\n",
    "print(f\"Between GTDB and RAST there are {len(gtr):,} genomes in common\")\n",
    "print()\n",
    "\n",
    "gbnotdone = gbkaa - phagesaa\n",
    "gtnotdone = gtdbaa - phagesaa\n",
    "ranotdone = rastaa - phagesaa\n",
    "print(f\"There are {len(gbnotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print(f\"There are {len(gtnotdone):,} phages in GTDB that have not been analyzed\")\n",
    "print(f\"There are {len(ranotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print()\n",
    "\n",
    "allmissing = gbnotdone.intersection(gtnotdone).intersection(ranotdone)\n",
    "print(f\"There are {len(allmissing):,} phages in all three that have not been analyzed\")\n",
    "\n",
    "if False:\n",
    "    with open(\"../data/unprocessed_phages.txt\", 'w') as out:\n",
    "        for o in allmissing:\n",
    "            gbk[gbk['assembly_accession'] == 'GCA_002129805.1'][['assembly_accession', 'ftp_path']].to_csv(out, sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "surrounded-fireplace",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IF WE IGNORE GENOME VERSIONS\n",
      "There are 1,195,593 genomes in GenBank\n",
      "There are 320,033 genomes in PATRIC\n",
      "There are 311,468 genomes in GTDB\n",
      "We have analyzed phages in 945,685 genomes\n",
      "\n",
      "Between GenBank and PATRIC there are 247,087 genomes in common\n",
      "Between GenBank and GTDB there are 309,616 genomes in common\n",
      "Between GTDB and PATRIC there are 209,749 genomes in common\n",
      "\n",
      "There are 249,908 phages in Genbank that have not been analyzed\n",
      "There are 115,149 phages in GTDB that have not been analyzed\n",
      "There are 173,347 phages in PATRIC that have not been analyzed\n",
      "\n",
      "There are 85,249 phages in all three that have not been analyzed\n"
     ]
    }
   ],
   "source": [
    "print(\"IF WE IGNORE GENOME VERSIONS\")\n",
    "gbkaa=set(gbk['assembly_nover'])\n",
    "rastaa=set(rast['assembly_nover'])\n",
    "gtdbaa=set(gtdb['assembly_nover'])\n",
    "phagesaa=set(phagesdf['assembly_nover'])\n",
    "\n",
    "\n",
    "gbr = gbkaa.intersection(rastaa)\n",
    "gbg = gbkaa.intersection(gtdbaa)\n",
    "gtr = gtdbaa.intersection(rastaa)\n",
    "\n",
    "print(f\"There are {len(gbkaa):,} genomes in GenBank\")\n",
    "print(f\"There are {len(rastaa):,} genomes in PATRIC\")\n",
    "print(f\"There are {len(gtdbaa):,} genomes in GTDB\")\n",
    "print(f\"We have analyzed phages in {len(phagesaa):,} genomes\")\n",
    "print()\n",
    "print(f\"Between GenBank and PATRIC there are {len(gbr):,} genomes in common\")\n",
    "print(f\"Between GenBank and GTDB there are {len(gbg):,} genomes in common\")\n",
    "print(f\"Between GTDB and PATRIC there are {len(gtr):,} genomes in common\")\n",
    "print()\n",
    "\n",
    "gbnotdone = gbkaa - phagesaa\n",
    "gtnotdone = gtdbaa - phagesaa\n",
    "ranotdone = rastaa - phagesaa\n",
    "print(f\"There are {len(gbnotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print(f\"There are {len(gtnotdone):,} phages in GTDB that have not been analyzed\")\n",
    "print(f\"There are {len(ranotdone):,} phages in PATRIC that have not been analyzed\")\n",
    "print()\n",
    "\n",
    "allmissing = gbnotdone.intersection(gtnotdone).intersection(ranotdone)\n",
    "print(f\"There are {len(allmissing):,} phages in all three that have not been analyzed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "incorrect-following",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}