{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "internal-anchor",
   "metadata": {},
   "source": [
    "## Compare the assemblies in GenBank, GTDB, and RAST\n",
    "\n",
    "Each has a different set. What are the unions and intersections?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "acute-prospect",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A lot of this is not used, but we import it so we have it later!\n",
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from PhiSpyAnalysis import theils_u, DateConverter, printmd\n",
    "from PhiSpyAnalysis import read_phages, read_gtdb, read_checkv, read_base_pp, read_categories, read_metadata, read_gbk_metadata\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "\n",
    "\n",
    "import subprocess\n",
    "import gzip\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "digital-cancellation",
   "metadata": {},
   "outputs": [],
   "source": [
    "rv = re.compile('^\\\\w+')\n",
    "def remove_ver(x):\n",
    "    return rv.search(str(x)).group()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "statewide-terror",
   "metadata": {},
   "source": [
    "# GTDB\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "compact-shell",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000006155.2</td>\n",
       "      <td>GCA_000006155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000007385.1</td>\n",
       "      <td>GCA_000007385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000008605.1</td>\n",
       "      <td>GCA_000008605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000010565.1</td>\n",
       "      <td>GCA_000010565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000013845.2</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191522</th>\n",
       "      <td>GCA_902166935.1</td>\n",
       "      <td>GCA_902166935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191523</th>\n",
       "      <td>GCA_902166945.1</td>\n",
       "      <td>GCA_902166945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191524</th>\n",
       "      <td>GCA_902167295.1</td>\n",
       "      <td>GCA_902167295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191525</th>\n",
       "      <td>GCA_902167305.1</td>\n",
       "      <td>GCA_902167305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191526</th>\n",
       "      <td>GCA_902167325.1</td>\n",
       "      <td>GCA_902167325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191527 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000006155.2  GCA_000006155\n",
       "1         GCA_000007385.1  GCA_000007385\n",
       "2         GCA_000008605.1  GCA_000008605\n",
       "3         GCA_000010565.1  GCA_000010565\n",
       "4         GCA_000013845.2  GCA_000013845\n",
       "...                   ...            ...\n",
       "191522    GCA_902166935.1  GCA_902166935\n",
       "191523    GCA_902166945.1  GCA_902166945\n",
       "191524    GCA_902167295.1  GCA_902167295\n",
       "191525    GCA_902167305.1  GCA_902167305\n",
       "191526    GCA_902167325.1  GCA_902167325\n",
       "\n",
       "[191527 rows x 2 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GTDB\n",
    "gtdb = read_gtdb()\n",
    "gtdb['assembly_nover'] = gtdb['assembly_accession'].apply(remove_ver)\n",
    "gtdb[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "pediatric-seven",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['accession',\n",
       " 'ambiguous_bases',\n",
       " 'checkm_completeness',\n",
       " 'checkm_contamination',\n",
       " 'checkm_marker_count',\n",
       " 'checkm_marker_lineage',\n",
       " 'checkm_marker_set_count',\n",
       " 'checkm_strain_heterogeneity',\n",
       " 'coding_bases',\n",
       " 'coding_density',\n",
       " 'contig_count',\n",
       " 'gc_count',\n",
       " 'gc_percentage',\n",
       " 'genome_size',\n",
       " 'gtdb_genome_representative',\n",
       " 'gtdb_representative',\n",
       " 'gtdb_taxonomy',\n",
       " 'gtdb_type_designation',\n",
       " 'gtdb_type_designation_sources',\n",
       " 'gtdb_type_species_of_genus',\n",
       " 'l50_contigs',\n",
       " 'l50_scaffolds',\n",
       " 'longest_contig',\n",
       " 'longest_scaffold',\n",
       " 'lsu_23s_contig_len',\n",
       " 'lsu_23s_count',\n",
       " 'lsu_23s_length',\n",
       " 'lsu_23s_query_id',\n",
       " 'lsu_5s_contig_len',\n",
       " 'lsu_5s_count',\n",
       " 'lsu_5s_length',\n",
       " 'lsu_5s_query_id',\n",
       " 'lsu_silva_23s_blast_align_len',\n",
       " 'lsu_silva_23s_blast_bitscore',\n",
       " 'lsu_silva_23s_blast_evalue',\n",
       " 'lsu_silva_23s_blast_perc_identity',\n",
       " 'lsu_silva_23s_blast_subject_id',\n",
       " 'lsu_silva_23s_taxonomy',\n",
       " 'mean_contig_length',\n",
       " 'mean_scaffold_length',\n",
       " 'mimag_high_quality',\n",
       " 'mimag_low_quality',\n",
       " 'mimag_medium_quality',\n",
       " 'n50_contigs',\n",
       " 'n50_scaffolds',\n",
       " 'ncbi_assembly_level',\n",
       " 'ncbi_assembly_name',\n",
       " 'ncbi_assembly_type',\n",
       " 'ncbi_bioproject',\n",
       " 'ncbi_biosample',\n",
       " 'ncbi_contig_count',\n",
       " 'ncbi_contig_n50',\n",
       " 'ncbi_country',\n",
       " 'ncbi_date',\n",
       " 'assembly_accession',\n",
       " 'ncbi_genome_category',\n",
       " 'ncbi_genome_representation',\n",
       " 'ncbi_isolate',\n",
       " 'ncbi_isolation_source',\n",
       " 'ncbi_lat_lon',\n",
       " 'ncbi_molecule_count',\n",
       " 'ncbi_ncrna_count',\n",
       " 'ncbi_organism_name',\n",
       " 'ncbi_protein_count',\n",
       " 'ncbi_refseq_category',\n",
       " 'ncbi_rrna_count',\n",
       " 'ncbi_scaffold_count',\n",
       " 'ncbi_scaffold_l50',\n",
       " 'ncbi_scaffold_n50',\n",
       " 'ncbi_scaffold_n75',\n",
       " 'ncbi_scaffold_n90',\n",
       " 'ncbi_seq_rel_date',\n",
       " 'ncbi_spanned_gaps',\n",
       " 'ncbi_species_taxid',\n",
       " 'ncbi_ssu_count',\n",
       " 'ncbi_strain_identifiers',\n",
       " 'ncbi_submitter',\n",
       " 'ncbi_taxid',\n",
       " 'ncbi_taxonomy',\n",
       " 'ncbi_taxonomy_unfiltered',\n",
       " 'ncbi_total_gap_length',\n",
       " 'ncbi_total_length',\n",
       " 'ncbi_translation_table',\n",
       " 'ncbi_trna_count',\n",
       " 'ncbi_type_material_designation',\n",
       " 'ncbi_ungapped_length',\n",
       " 'ncbi_unspanned_gaps',\n",
       " 'ncbi_wgs_master',\n",
       " 'protein_count',\n",
       " 'scaffold_count',\n",
       " 'ssu_contig_len',\n",
       " 'ssu_count',\n",
       " 'ssu_gg_blast_align_len',\n",
       " 'ssu_gg_blast_bitscore',\n",
       " 'ssu_gg_blast_evalue',\n",
       " 'ssu_gg_blast_perc_identity',\n",
       " 'ssu_gg_blast_subject_id',\n",
       " 'ssu_gg_taxonomy',\n",
       " 'ssu_length',\n",
       " 'ssu_query_id',\n",
       " 'ssu_silva_blast_align_len',\n",
       " 'ssu_silva_blast_bitscore',\n",
       " 'ssu_silva_blast_evalue',\n",
       " 'ssu_silva_blast_perc_identity',\n",
       " 'ssu_silva_blast_subject_id',\n",
       " 'ssu_silva_taxonomy',\n",
       " 'total_gap_length',\n",
       " 'trna_aa_count',\n",
       " 'trna_count',\n",
       " 'trna_selenocysteine_count',\n",
       " 'assembly_nover']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(gtdb.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "textile-software",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>GCA_000003135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003215.1</td>\n",
       "      <td>GCA_000003215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>GCA_000003645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>GCA_000003925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>GCA_000003955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320171</th>\n",
       "      <td>GCF_900167595.1</td>\n",
       "      <td>GCF_900167595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320172</th>\n",
       "      <td>GCF_900167605.1</td>\n",
       "      <td>GCF_900167605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320173</th>\n",
       "      <td>GCF_900167615.1</td>\n",
       "      <td>GCF_900167615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320174</th>\n",
       "      <td>GCF_900167625.1</td>\n",
       "      <td>GCF_900167625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320175</th>\n",
       "      <td>GCF_900167635.1</td>\n",
       "      <td>GCF_900167635</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320176 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000003135.1  GCA_000003135\n",
       "1         GCA_000003215.1  GCA_000003215\n",
       "2         GCA_000003645.1  GCA_000003645\n",
       "3         GCA_000003925.1  GCA_000003925\n",
       "4         GCA_000003955.1  GCA_000003955\n",
       "...                   ...            ...\n",
       "320171    GCF_900167595.1  GCF_900167595\n",
       "320172    GCF_900167605.1  GCF_900167605\n",
       "320173    GCF_900167615.1  GCF_900167615\n",
       "320174    GCF_900167625.1  GCF_900167625\n",
       "320175    GCF_900167635.1  GCF_900167635\n",
       "\n",
       "[320176 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# RAST\n",
    "# the full data set. Don't try this at home!\n",
    "# metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "rast = read_metadata()\n",
    "rast['assembly_nover'] = rast['assembly_accession'].apply(remove_ver)\n",
    "rast[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "considerable-brief",
   "metadata": {},
   "source": [
    "# GenBank\n",
    "\n",
    "This assembly summary comes from [GenBank ftp site](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/) and you want the [assembly_summary.txt](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/assembly_summary.txt) file explicitly from bacteria (but don't try and open the bacteria list in your browser!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "curious-compatibility",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3357: DtypeWarning: Columns (21) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>GCA_900128725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>GCA_008244535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_011046815.1</td>\n",
       "      <td>GCA_011046815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_011054035.1</td>\n",
       "      <td>GCA_011054035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_011331185.1</td>\n",
       "      <td>GCA_011331185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949928</th>\n",
       "      <td>GCA_017746755.1</td>\n",
       "      <td>GCA_017746755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949929</th>\n",
       "      <td>GCA_017746715.1</td>\n",
       "      <td>GCA_017746715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949930</th>\n",
       "      <td>GCA_013266695.1</td>\n",
       "      <td>GCA_013266695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949931</th>\n",
       "      <td>GCA_015831295.1</td>\n",
       "      <td>GCA_015831295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949932</th>\n",
       "      <td>GCA_015686645.1</td>\n",
       "      <td>GCA_015686645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>949933 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_900128725.1  GCA_900128725\n",
       "1         GCA_008244535.1  GCA_008244535\n",
       "2         GCA_011046815.1  GCA_011046815\n",
       "3         GCA_011054035.1  GCA_011054035\n",
       "4         GCA_011331185.1  GCA_011331185\n",
       "...                   ...            ...\n",
       "949928    GCA_017746755.1  GCA_017746755\n",
       "949929    GCA_017746715.1  GCA_017746715\n",
       "949930    GCA_013266695.1  GCA_013266695\n",
       "949931    GCA_015831295.1  GCA_015831295\n",
       "949932    GCA_015686645.1  GCA_015686645\n",
       "\n",
       "[949933 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GenBank\n",
    "gbk = read_gbk_metadata()\n",
    "gbk['assembly_nover'] = gbk['assembly_accession'].apply(remove_ver)\n",
    "gbk[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "endless-entrance",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Please note that this was run with git commit 86420e1 that has 567,404 genomes parsed.\n",
      "Initially there were 3,265,453 kept phages,but after filtering we kept 3,265,453 prophages from 567,399 genomes"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000043285.1</td>\n",
       "      <td>ASM4328v1</td>\n",
       "      <td>705557.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000043285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000046685.1</td>\n",
       "      <td>ASM4668v1</td>\n",
       "      <td>1581384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>97011.0</td>\n",
       "      <td>GCA_000046685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000046705.1</td>\n",
       "      <td>ASM4670v1</td>\n",
       "      <td>1931047.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>278490.0</td>\n",
       "      <td>GCA_000046705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000046845.1</td>\n",
       "      <td>ASM4684v1</td>\n",
       "      <td>3598621.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>83112.0</td>\n",
       "      <td>GCA_000046845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000047365.1</td>\n",
       "      <td>ASM4736v1</td>\n",
       "      <td>4840898.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>213427.0</td>\n",
       "      <td>GCA_000047365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567399</th>\n",
       "      <td>GCA_905187425.1</td>\n",
       "      <td>Xanthomonas_sp._CPBF_424_-_hybrid_assembly</td>\n",
       "      <td>4900930.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>214278.0</td>\n",
       "      <td>GCA_905187425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567400</th>\n",
       "      <td>GCA_905188235.1</td>\n",
       "      <td>ASM90518823v1</td>\n",
       "      <td>4963609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>178993.0</td>\n",
       "      <td>GCA_905188235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567401</th>\n",
       "      <td>GCA_905219375.1</td>\n",
       "      <td>QI0054</td>\n",
       "      <td>3034314.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>178527.0</td>\n",
       "      <td>GCA_905219375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567402</th>\n",
       "      <td>GCA_905219385.1</td>\n",
       "      <td>QI0055</td>\n",
       "      <td>3034113.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>178526.0</td>\n",
       "      <td>GCA_905219385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567403</th>\n",
       "      <td>GCA_905220645.1</td>\n",
       "      <td>NMYAN.1</td>\n",
       "      <td>3400020.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>65454.0</td>\n",
       "      <td>GCA_905220645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>567399 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession                               assembly_name  \\\n",
       "0         GCA_000043285.1                                   ASM4328v1   \n",
       "1         GCA_000046685.1                                   ASM4668v1   \n",
       "2         GCA_000046705.1                                   ASM4670v1   \n",
       "3         GCA_000046845.1                                   ASM4684v1   \n",
       "4         GCA_000047365.1                                   ASM4736v1   \n",
       "...                   ...                                         ...   \n",
       "567399    GCA_905187425.1  Xanthomonas_sp._CPBF_424_-_hybrid_assembly   \n",
       "567400    GCA_905188235.1                               ASM90518823v1   \n",
       "567401    GCA_905219375.1                                      QI0054   \n",
       "567402    GCA_905219385.1                                      QI0055   \n",
       "567403    GCA_905220645.1                                     NMYAN.1   \n",
       "\n",
       "        Genome length  Contigs  Phage Contigs  Total Predicted Prophages  \\\n",
       "0            705557.0      1.0            1.0                        2.0   \n",
       "1           1581384.0      1.0            1.0                       18.0   \n",
       "2           1931047.0      1.0            1.0                       36.0   \n",
       "3           3598621.0      1.0            1.0                       13.0   \n",
       "4           4840898.0      3.0            3.0                       22.0   \n",
       "...               ...      ...            ...                        ...   \n",
       "567399      4900930.0      1.0            1.0                       46.0   \n",
       "567400      4963609.0      1.0            1.0                       29.0   \n",
       "567401      3034314.0      1.0            1.0                       12.0   \n",
       "567402      3034113.0      1.0            1.0                       12.0   \n",
       "567403      3400020.0     55.0           33.0                       31.0   \n",
       "\n",
       "        Kept  No phage genes  Not enough genes  bp prophage assembly_nover  \n",
       "0        0.0             2.0               0.0          0.0  GCA_000043285  \n",
       "1        3.0             0.0              15.0      97011.0  GCA_000046685  \n",
       "2        7.0             0.0              29.0     278490.0  GCA_000046705  \n",
       "3        3.0             4.0               6.0      83112.0  GCA_000046845  \n",
       "4        3.0             3.0              16.0     213427.0  GCA_000047365  \n",
       "...      ...             ...               ...          ...            ...  \n",
       "567399   7.0             9.0              30.0     214278.0  GCA_905187425  \n",
       "567400   4.0             4.0              21.0     178993.0  GCA_905188235  \n",
       "567401   4.0             3.0               5.0     178527.0  GCA_905219375  \n",
       "567402   4.0             3.0               5.0     178526.0  GCA_905219385  \n",
       "567403   4.0             5.0              22.0      65454.0  GCA_905220645  \n",
       "\n",
       "[567399 rows x 11 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagesdf = read_phages(maxcontigs=-1) # this disables contig length filtering\n",
    "phagesdf['assembly_nover'] = phagesdf['assembly_accession'].apply(remove_ver)\n",
    "phagesdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "downtown-subject",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Total Predicted Prophages    21562059.0\n",
       "Not enough genes             15047785.0\n",
       "No phage genes                3248821.0\n",
       "Kept                          3265453.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagesdf[['Total Predicted Prophages', 'Not enough genes', 'No phage genes', 'Kept']].agg('sum')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "medieval-canyon",
   "metadata": {},
   "source": [
    "# What is in common between the groups?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "changing-president",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Between GBK and RAST there are 244,870 genomes in common\n",
      "Between GBK and GTDB there are 189,704 genomes in common\n",
      "Between GTDB and RAST there are 140,463 genomes in common\n",
      "\n",
      "There are 339,231 phages in Genbank that have not been analyzed\n",
      "There are 65,392 phages in GTDB that have not been analyzed\n",
      "There are 186,485 phages in Genbank that have not been analyzed\n",
      "\n",
      "There are 52,219 phages in all three that have not been analyzed\n"
     ]
    }
   ],
   "source": [
    "gbkaa=set(gbk['assembly_accession'])\n",
    "rastaa=set(rast['assembly_accession'])\n",
    "gtdbaa=set(gtdb['assembly_accession'])\n",
    "phagesaa=set(phagesdf['assembly_accession'])\n",
    "\n",
    "\n",
    "gbr = gbkaa.intersection(rastaa)\n",
    "gbg = gbkaa.intersection(gtdbaa)\n",
    "gtr = gtdbaa.intersection(rastaa)\n",
    "\n",
    "print(f\"Between GBK and RAST there are {len(gbr):,} genomes in common\")\n",
    "print(f\"Between GBK and GTDB there are {len(gbg):,} genomes in common\")\n",
    "print(f\"Between GTDB and RAST there are {len(gtr):,} genomes in common\")\n",
    "print()\n",
    "\n",
    "gbnotdone = gbkaa - phagesaa\n",
    "gtnotdone = gtdbaa - phagesaa\n",
    "ranotdone = rastaa - phagesaa\n",
    "print(f\"There are {len(gbnotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print(f\"There are {len(gtnotdone):,} phages in GTDB that have not been analyzed\")\n",
    "print(f\"There are {len(ranotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print()\n",
    "\n",
    "allmissing = gbnotdone.intersection(gtnotdone).intersection(ranotdone)\n",
    "print(f\"There are {len(allmissing):,} phages in all three that have not been analyzed\")\n",
    "\n",
    "if False:\n",
    "    with open(\"../data/unprocessed_phages.txt\", 'w') as out:\n",
    "        for o in allmissing:\n",
    "            gbk[gbk['assembly_accession'] == 'GCA_002129805.1'][['assembly_accession', 'ftp_path']].to_csv(out, sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "surrounded-fireplace",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IF WE IGNORE GENOME VERSIONS\n",
      "There are 904,681 genomes in GenBank\n",
      "There are 320,033 genomes in PATRIC\n",
      "There are 191,521 genomes in GTDB\n",
      "Between GenBank and PATRIC there are 249,334 genomes in common\n",
      "Between GenBank and GTDB there are 191,370 genomes in common\n",
      "Between GTDB and PATRIC there are 143,445 genomes in common\n",
      "\n",
      "There are 337,599 phages in Genbank that have not been analyzed\n",
      "There are 64,358 phages in GTDB that have not been analyzed\n",
      "There are 183,308 phages in PATRIC that have not been analyzed\n",
      "\n",
      "There are 53,262 phages in all three that have not been analyzed\n"
     ]
    }
   ],
   "source": [
    "print(\"IF WE IGNORE GENOME VERSIONS\")\n",
    "gbkaa=set(gbk['assembly_nover'])\n",
    "rastaa=set(rast['assembly_nover'])\n",
    "gtdbaa=set(gtdb['assembly_nover'])\n",
    "phagesaa=set(phagesdf['assembly_nover'])\n",
    "\n",
    "\n",
    "gbr = gbkaa.intersection(rastaa)\n",
    "gbg = gbkaa.intersection(gtdbaa)\n",
    "gtr = gtdbaa.intersection(rastaa)\n",
    "\n",
    "print(f\"There are {len(gbkaa):,} genomes in GenBank\")\n",
    "print(f\"There are {len(rastaa):,} genomes in PATRIC\")\n",
    "print(f\"There are {len(gtdbaa):,} genomes in GTDB\")\n",
    "\n",
    "print(f\"Between GenBank and PATRIC there are {len(gbr):,} genomes in common\")\n",
    "print(f\"Between GenBank and GTDB there are {len(gbg):,} genomes in common\")\n",
    "print(f\"Between GTDB and PATRIC there are {len(gtr):,} genomes in common\")\n",
    "print()\n",
    "\n",
    "gbnotdone = gbkaa - phagesaa\n",
    "gtnotdone = gtdbaa - phagesaa\n",
    "ranotdone = rastaa - phagesaa\n",
    "print(f\"There are {len(gbnotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print(f\"There are {len(gtnotdone):,} phages in GTDB that have not been analyzed\")\n",
    "print(f\"There are {len(ranotdone):,} phages in PATRIC that have not been analyzed\")\n",
    "print()\n",
    "\n",
    "allmissing = gbnotdone.intersection(gtnotdone).intersection(ranotdone)\n",
    "print(f\"There are {len(allmissing):,} phages in all three that have not been analyzed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "incorrect-following",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
