{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "internal-anchor",
   "metadata": {},
   "source": [
    "## Compare the assemblies in GenBank, GTDB, and RAST\n",
    "\n",
    "Each has a different set. What are the unions and intersections?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "acute-prospect",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "from scipy.stats import pearsonr\n",
    "from sklearn.linear_model import LinearRegression\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import statsmodels.api as sm\n",
    "import subprocess\n",
    "import gzip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "aquatic-courage",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this is a neat trick for getting markdown in our output\n",
    "# see https://stackoverflow.com/questions/23271575/printing-bold-colored-etc-text-in-ipython-qtconsole\n",
    "# for the inspiration\n",
    "from IPython.display import Markdown, display\n",
    "def printmd(string, color=\"black\"):\n",
    "    colorstr = \"<span style='color:{}'>{}</span>\".format(color, string)\n",
    "    display(Markdown(colorstr))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "statewide-terror",
   "metadata": {},
   "source": [
    "# GTDB\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "compact-shell",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3165: DtypeWarning: Columns (61,65,74,82,83) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_lineage</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>coding_density</th>\n",
       "      <th>...</th>\n",
       "      <th>ssu_silva_blast_align_len</th>\n",
       "      <th>ssu_silva_blast_bitscore</th>\n",
       "      <th>ssu_silva_blast_evalue</th>\n",
       "      <th>ssu_silva_blast_perc_identity</th>\n",
       "      <th>ssu_silva_blast_subject_id</th>\n",
       "      <th>ssu_silva_taxonomy</th>\n",
       "      <th>total_gap_length</th>\n",
       "      <th>trna_aa_count</th>\n",
       "      <th>trna_count</th>\n",
       "      <th>trna_selenocysteine_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GB_GCA_000006155.2</td>\n",
       "      <td>1916</td>\n",
       "      <td>93.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1171</td>\n",
       "      <td>g__Bacillus (UID902)</td>\n",
       "      <td>324</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4305660</td>\n",
       "      <td>80.178992</td>\n",
       "      <td>...</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "      <td>42300</td>\n",
       "      <td>16</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GB_GCA_000007385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.82</td>\n",
       "      <td>0.00</td>\n",
       "      <td>481</td>\n",
       "      <td>c__Gammaproteobacteria (UID4202)</td>\n",
       "      <td>276</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4190634</td>\n",
       "      <td>84.805944</td>\n",
       "      <td>...</td>\n",
       "      <td>1541</td>\n",
       "      <td>2846</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>JXEG01000201.4293.5839</td>\n",
       "      <td>Bacteria;Proteobacteria;Gammaproteobacteria;Xa...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GB_GCA_000008605.1</td>\n",
       "      <td>67</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>235</td>\n",
       "      <td>f__Spirochaetaceae (UID2512)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1048744</td>\n",
       "      <td>92.155875</td>\n",
       "      <td>...</td>\n",
       "      <td>1545</td>\n",
       "      <td>2854</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>CP003679.231299.232859</td>\n",
       "      <td>Bacteria;Spirochaetes;Spirochaetia;Spirochaeta...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GB_GCA_000010565.1</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.63</td>\n",
       "      <td>295</td>\n",
       "      <td>p__Firmicutes (UID1022)</td>\n",
       "      <td>158</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2608397</td>\n",
       "      <td>86.217312</td>\n",
       "      <td>...</td>\n",
       "      <td>1526</td>\n",
       "      <td>2819</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>AP009389.1049070.1050595</td>\n",
       "      <td>Bacteria;Firmicutes;Clostridia;Clostridiales;P...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>51</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GB_GCA_000013845.2</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>332</td>\n",
       "      <td>o__Clostridiales (UID1375)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2428396</td>\n",
       "      <td>82.037966</td>\n",
       "      <td>...</td>\n",
       "      <td>1509</td>\n",
       "      <td>2787</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>CP000312.233320.234832</td>\n",
       "      <td>Bacteria;Firmicutes;Clostridia;Clostridiales;C...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191522</th>\n",
       "      <td>RS_GCF_902166935.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.62</td>\n",
       "      <td>0.09</td>\n",
       "      <td>1312</td>\n",
       "      <td>g__Klebsiella (UID5140)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4894244</td>\n",
       "      <td>87.293148</td>\n",
       "      <td>...</td>\n",
       "      <td>1536</td>\n",
       "      <td>2837</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>MAPL01000082.143.1696</td>\n",
       "      <td>Bacteria;Proteobacteria;Gammaproteobacteria;En...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>87</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191523</th>\n",
       "      <td>RS_GCF_902166945.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.94</td>\n",
       "      <td>0.72</td>\n",
       "      <td>1162</td>\n",
       "      <td>f__Enterobacteriaceae (UID5121)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5162721</td>\n",
       "      <td>86.540042</td>\n",
       "      <td>...</td>\n",
       "      <td>1536</td>\n",
       "      <td>2837</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>CP016813.16076.17629</td>\n",
       "      <td>Bacteria;Proteobacteria;Gammaproteobacteria;En...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191524</th>\n",
       "      <td>RS_GCF_902167295.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5737447</td>\n",
       "      <td>90.122088</td>\n",
       "      <td>...</td>\n",
       "      <td>1490</td>\n",
       "      <td>2747</td>\n",
       "      <td>0</td>\n",
       "      <td>99.933</td>\n",
       "      <td>JX101438.1.1490</td>\n",
       "      <td>Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191525</th>\n",
       "      <td>RS_GCF_902167305.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5738096</td>\n",
       "      <td>90.147873</td>\n",
       "      <td>...</td>\n",
       "      <td>1490</td>\n",
       "      <td>2747</td>\n",
       "      <td>0</td>\n",
       "      <td>99.933</td>\n",
       "      <td>JX101438.1.1490</td>\n",
       "      <td>Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191526</th>\n",
       "      <td>RS_GCF_902167325.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5671443</td>\n",
       "      <td>90.098258</td>\n",
       "      <td>...</td>\n",
       "      <td>1490</td>\n",
       "      <td>2741</td>\n",
       "      <td>0</td>\n",
       "      <td>99.866</td>\n",
       "      <td>JX101438.1.1490</td>\n",
       "      <td>Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191527 rows × 110 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 accession  ambiguous_bases  checkm_completeness  \\\n",
       "0       GB_GCA_000006155.2             1916                93.12   \n",
       "1       GB_GCA_000007385.1                0                99.82   \n",
       "2       GB_GCA_000008605.1               67               100.00   \n",
       "3       GB_GCA_000010565.1                0               100.00   \n",
       "4       GB_GCA_000013845.2                0               100.00   \n",
       "...                    ...              ...                  ...   \n",
       "191522  RS_GCF_902166935.1                0                99.62   \n",
       "191523  RS_GCF_902166945.1                0                99.94   \n",
       "191524  RS_GCF_902167295.1                0                99.51   \n",
       "191525  RS_GCF_902167305.1                0                99.51   \n",
       "191526  RS_GCF_902167325.1                0                99.51   \n",
       "\n",
       "        checkm_contamination  checkm_marker_count  \\\n",
       "0                       0.00                 1171   \n",
       "1                       0.00                  481   \n",
       "2                       0.00                  235   \n",
       "3                       0.63                  295   \n",
       "4                       0.00                  332   \n",
       "...                      ...                  ...   \n",
       "191522                  0.09                 1312   \n",
       "191523                  0.72                 1162   \n",
       "191524                  0.00                  303   \n",
       "191525                  0.00                  303   \n",
       "191526                  0.00                  303   \n",
       "\n",
       "                   checkm_marker_lineage  checkm_marker_set_count  \\\n",
       "0                   g__Bacillus (UID902)                      324   \n",
       "1       c__Gammaproteobacteria (UID4202)                      276   \n",
       "2           f__Spirochaetaceae (UID2512)                      124   \n",
       "3                p__Firmicutes (UID1022)                      158   \n",
       "4             o__Clostridiales (UID1375)                      124   \n",
       "...                                  ...                      ...   \n",
       "191522           g__Klebsiella (UID5140)                      336   \n",
       "191523   f__Enterobacteriaceae (UID5121)                      336   \n",
       "191524        p__Bacteroidetes (UID2591)                      203   \n",
       "191525        p__Bacteroidetes (UID2591)                      203   \n",
       "191526        p__Bacteroidetes (UID2591)                      203   \n",
       "\n",
       "        checkm_strain_heterogeneity  coding_bases  coding_density  ...  \\\n",
       "0                               0.0       4305660       80.178992  ...   \n",
       "1                               0.0       4190634       84.805944  ...   \n",
       "2                               0.0       1048744       92.155875  ...   \n",
       "3                               0.0       2608397       86.217312  ...   \n",
       "4                               0.0       2428396       82.037966  ...   \n",
       "...                             ...           ...             ...  ...   \n",
       "191522                          0.0       4894244       87.293148  ...   \n",
       "191523                          0.0       5162721       86.540042  ...   \n",
       "191524                          0.0       5737447       90.122088  ...   \n",
       "191525                          0.0       5738096       90.147873  ...   \n",
       "191526                          0.0       5671443       90.098258  ...   \n",
       "\n",
       "        ssu_silva_blast_align_len  ssu_silva_blast_bitscore  \\\n",
       "0                            none                      none   \n",
       "1                            1541                      2846   \n",
       "2                            1545                      2854   \n",
       "3                            1526                      2819   \n",
       "4                            1509                      2787   \n",
       "...                           ...                       ...   \n",
       "191522                       1536                      2837   \n",
       "191523                       1536                      2837   \n",
       "191524                       1490                      2747   \n",
       "191525                       1490                      2747   \n",
       "191526                       1490                      2741   \n",
       "\n",
       "        ssu_silva_blast_evalue  ssu_silva_blast_perc_identity  \\\n",
       "0                         none                           none   \n",
       "1                            0                            100   \n",
       "2                            0                            100   \n",
       "3                            0                            100   \n",
       "4                            0                            100   \n",
       "...                        ...                            ...   \n",
       "191522                       0                            100   \n",
       "191523                       0                            100   \n",
       "191524                       0                         99.933   \n",
       "191525                       0                         99.933   \n",
       "191526                       0                         99.866   \n",
       "\n",
       "       ssu_silva_blast_subject_id  \\\n",
       "0                            none   \n",
       "1          JXEG01000201.4293.5839   \n",
       "2          CP003679.231299.232859   \n",
       "3        AP009389.1049070.1050595   \n",
       "4          CP000312.233320.234832   \n",
       "...                           ...   \n",
       "191522      MAPL01000082.143.1696   \n",
       "191523       CP016813.16076.17629   \n",
       "191524            JX101438.1.1490   \n",
       "191525            JX101438.1.1490   \n",
       "191526            JX101438.1.1490   \n",
       "\n",
       "                                       ssu_silva_taxonomy total_gap_length  \\\n",
       "0                                                    none            42300   \n",
       "1       Bacteria;Proteobacteria;Gammaproteobacteria;Xa...                0   \n",
       "2       Bacteria;Spirochaetes;Spirochaetia;Spirochaeta...                0   \n",
       "3       Bacteria;Firmicutes;Clostridia;Clostridiales;P...                0   \n",
       "4       Bacteria;Firmicutes;Clostridia;Clostridiales;C...                0   \n",
       "...                                                   ...              ...   \n",
       "191522  Bacteria;Proteobacteria;Gammaproteobacteria;En...                0   \n",
       "191523  Bacteria;Proteobacteria;Gammaproteobacteria;En...                0   \n",
       "191524  Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...                0   \n",
       "191525  Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...                0   \n",
       "191526  Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...                0   \n",
       "\n",
       "       trna_aa_count trna_count trna_selenocysteine_count  \n",
       "0                 16         31                         0  \n",
       "1                 20         53                         0  \n",
       "2                 20         45                         0  \n",
       "3                 20         51                         1  \n",
       "4                 20         95                         1  \n",
       "...              ...        ...                       ...  \n",
       "191522            20         87                         1  \n",
       "191523            20         86                         1  \n",
       "191524            20         58                         0  \n",
       "191525            20         58                         0  \n",
       "191526            20         59                         0  \n",
       "\n",
       "[191527 rows x 110 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GTDB\n",
    "gtdb = pd.read_csv(\"../data/bac120_metadata_r95.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "gtdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "textile-software",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3165: DtypeWarning: Columns (2,6,7,8,9,10,11,12,14,19,21,34,40,41,42,43,44,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,65) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>mlst</th>\n",
       "      <th>...</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>469009.4</td>\n",
       "      <td>\"'Brassica napus' phytoplasma strain TW1\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>469009</td>\n",
       "      <td>WGS</td>\n",
       "      <td>TW1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequence of a strain of bacteria that c...</td>\n",
       "      <td>sample_type:metagenomic assembly;collected_by:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1309411.5</td>\n",
       "      <td>\"'Deinococcus soli' Cha et al. 2014 strain N5\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1309411</td>\n",
       "      <td>Complete</td>\n",
       "      <td>N5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing of a Gamma-Radiation-Resista...</td>\n",
       "      <td>sample_type:bacterial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1123738.3</td>\n",
       "      <td>\"'Echinacea purpurea' witches'-broom phytoplas...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1123738</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NCHU2014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'Echinacea purpurea' witches'-broom phytoplasm...</td>\n",
       "      <td>lab_host:Catharanthus roseus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>551115.6</td>\n",
       "      <td>\"'Nostoc azollae' 0708\"</td>\n",
       "      <td>'Nostoc azollae' 0708</td>\n",
       "      <td>551115</td>\n",
       "      <td>Complete</td>\n",
       "      <td>708</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nostoc azollae 0708. Nostoc azollae 0708, also...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1856298.3</td>\n",
       "      <td>\"'Osedax' symbiont bacterium Rs2_46_30_T18 str...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1856298</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Rs2_46_30_T18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In this study, we simulate the Deepwater Horiz...</td>\n",
       "      <td>sample_type:metagenomic assembly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433517</th>\n",
       "      <td>1131286.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>1131286</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-137-J06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433518</th>\n",
       "      <td>1131287.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>1131287</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-C20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433519</th>\n",
       "      <td>1131288.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>1131288</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-E04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433520</th>\n",
       "      <td>1131289.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>1131289</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-B04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433521</th>\n",
       "      <td>1131290.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>1131290</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-O16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433522 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        genome_id                                        genome_name  \\\n",
       "0        469009.4          \"'Brassica napus' phytoplasma strain TW1\"   \n",
       "1       1309411.5     \"'Deinococcus soli' Cha et al. 2014 strain N5\"   \n",
       "2       1123738.3  \"'Echinacea purpurea' witches'-broom phytoplas...   \n",
       "3        551115.6                            \"'Nostoc azollae' 0708\"   \n",
       "4       1856298.3  \"'Osedax' symbiont bacterium Rs2_46_30_T18 str...   \n",
       "...           ...                                                ...   \n",
       "433517  1131286.3               zeta proteobacterium SCGC AB-137-J06   \n",
       "433518  1131287.3               zeta proteobacterium SCGC AB-602-C20   \n",
       "433519  1131288.3               zeta proteobacterium SCGC AB-602-E04   \n",
       "433520  1131289.3               zeta proteobacterium SCGC AB-604-B04   \n",
       "433521  1131290.3               zeta proteobacterium SCGC AB-604-O16   \n",
       "\n",
       "                               organism_name  taxon_id genome_status  \\\n",
       "0                                        NaN    469009           WGS   \n",
       "1                                        NaN   1309411      Complete   \n",
       "2                                        NaN   1123738           WGS   \n",
       "3                      'Nostoc azollae' 0708    551115      Complete   \n",
       "4                                        NaN   1856298           WGS   \n",
       "...                                      ...       ...           ...   \n",
       "433517  zeta proteobacterium SCGC AB-137-J06   1131286           WGS   \n",
       "433518  zeta proteobacterium SCGC AB-602-C20   1131287           WGS   \n",
       "433519  zeta proteobacterium SCGC AB-602-E04   1131288           WGS   \n",
       "433520  zeta proteobacterium SCGC AB-604-B04   1131289           WGS   \n",
       "433521  zeta proteobacterium SCGC AB-604-O16   1131290           WGS   \n",
       "\n",
       "                 strain serovar biovar pathovar mlst  ... motility  \\\n",
       "0                   TW1     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1                    N5     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "2              NCHU2014     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "3                   708     NaN    NaN      NaN  NaN  ...      Yes   \n",
       "4         Rs2_46_30_T18     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "...                 ...     ...    ...      ...  ...  ...      ...   \n",
       "433517  SCGC AB-137-J06     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433518  SCGC AB-602-C20     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433519  SCGC AB-602-E04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433520  SCGC AB-604-B04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433521  SCGC AB-604-O16     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "\n",
       "       sporulation temperature_range optimal_temperature salinity  \\\n",
       "0              NaN               NaN                 NaN      NaN   \n",
       "1              NaN               NaN                 NaN      NaN   \n",
       "2              NaN               NaN                   C      NaN   \n",
       "3              NaN        Mesophilic                   -      NaN   \n",
       "4              NaN               NaN                 NaN      NaN   \n",
       "...            ...               ...                 ...      ...   \n",
       "433517         NaN               NaN                 NaN      NaN   \n",
       "433518         NaN               NaN                 NaN      NaN   \n",
       "433519         NaN               NaN                 NaN      NaN   \n",
       "433520         NaN               NaN                 NaN      NaN   \n",
       "433521         NaN               NaN                 NaN      NaN   \n",
       "\n",
       "       oxygen_requirement   habitat disease  \\\n",
       "0                     NaN       NaN     NaN   \n",
       "1                     NaN       NaN     NaN   \n",
       "2                     NaN       NaN     NaN   \n",
       "3                 Aerobic  Multiple     NaN   \n",
       "4                     NaN       NaN     NaN   \n",
       "...                   ...       ...     ...   \n",
       "433517                NaN       NaN     NaN   \n",
       "433518                NaN       NaN     NaN   \n",
       "433519                NaN       NaN     NaN   \n",
       "433520                NaN       NaN     NaN   \n",
       "433521                NaN       NaN     NaN   \n",
       "\n",
       "                                                 comments  \\\n",
       "0       Genome sequence of a strain of bacteria that c...   \n",
       "1       Genome sequencing of a Gamma-Radiation-Resista...   \n",
       "2       'Echinacea purpurea' witches'-broom phytoplasm...   \n",
       "3       Nostoc azollae 0708. Nostoc azollae 0708, also...   \n",
       "4       In this study, we simulate the Deepwater Horiz...   \n",
       "...                                                   ...   \n",
       "433517  Single cell genome sequencing of biomineralizi...   \n",
       "433518  Single cell genome sequencing of biomineralizi...   \n",
       "433519  Single cell genome sequencing of biomineralizi...   \n",
       "433520  Single cell genome sequencing of biomineralizi...   \n",
       "433521  Single cell genome sequencing of biomineralizi...   \n",
       "\n",
       "                                      additional_metadata  \n",
       "0       sample_type:metagenomic assembly;collected_by:...  \n",
       "1                                   sample_type:bacterial  \n",
       "2                            lab_host:Catharanthus roseus  \n",
       "3                                                     NaN  \n",
       "4                        sample_type:metagenomic assembly  \n",
       "...                                                   ...  \n",
       "433517                                                NaN  \n",
       "433518                                                NaN  \n",
       "433519                                                NaN  \n",
       "433520                                                NaN  \n",
       "433521                                                NaN  \n",
       "\n",
       "[433522 rows x 66 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# RAST\n",
    "# the full data set. Don't try this at home!\n",
    "# metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "rast = pd.read_csv(\"../data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "rast"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "considerable-brief",
   "metadata": {},
   "source": [
    "# GenBank\n",
    "\n",
    "This assembly summary comes from [GenBank ftp site](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/) and you want the [assembly_summary.txt](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/assembly_summary.txt) file explicitly from bacteria (but don't try and open the bacteria list in your browser!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "curious-compatibility",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../data/assembly_summary_.txt.gz'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-6-399a01a2bbc3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m# GenBank\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0mgbk\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"../data/assembly_summary_.txt.gz\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcompression\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'gzip'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdelimiter\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"\\t\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      3\u001b[0m \u001b[0mgbk\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[0;32m    608\u001b[0m     \u001b[0mkwds\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkwds_defaults\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    609\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 610\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    611\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    612\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    460\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    461\u001b[0m     \u001b[1;31m# Create the parser.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 462\u001b[1;33m     \u001b[0mparser\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    463\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    464\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[0;32m    817\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"has_index_names\"\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"has_index_names\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    818\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 819\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    820\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    821\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[1;34m(self, engine)\u001b[0m\n\u001b[0;32m   1048\u001b[0m             )\n\u001b[0;32m   1049\u001b[0m         \u001b[1;31m# error: Too many arguments for \"ParserBase\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1050\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mmapping\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mengine\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[1;33m)\u001b[0m  \u001b[1;31m# type: ignore[call-arg]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1051\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1052\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_failover_to_python\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, src, **kwds)\u001b[0m\n\u001b[0;32m   1865\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1866\u001b[0m         \u001b[1;31m# open handles\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1867\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_open_handles\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1868\u001b[0m         \u001b[1;32massert\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mhandles\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1869\u001b[0m         \u001b[1;32mfor\u001b[0m \u001b[0mkey\u001b[0m \u001b[1;32min\u001b[0m \u001b[1;33m(\u001b[0m\u001b[1;34m\"storage_options\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"encoding\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"memory_map\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"compression\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36m_open_handles\u001b[1;34m(self, src, kwds)\u001b[0m\n\u001b[0;32m   1360\u001b[0m         \u001b[0mLet\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mreaders\u001b[0m \u001b[0mopen\u001b[0m \u001b[0mIOHanldes\u001b[0m \u001b[0mafter\u001b[0m \u001b[0mthey\u001b[0m \u001b[0mare\u001b[0m \u001b[0mdone\u001b[0m \u001b[1;32mwith\u001b[0m \u001b[0mtheir\u001b[0m \u001b[0mpotential\u001b[0m \u001b[0mraises\u001b[0m\u001b[1;33m.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1361\u001b[0m         \"\"\"\n\u001b[1;32m-> 1362\u001b[1;33m         self.handles = get_handle(\n\u001b[0m\u001b[0;32m   1363\u001b[0m             \u001b[0msrc\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1364\u001b[0m             \u001b[1;34m\"r\"\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mc:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\pandas\\io\\common.py\u001b[0m in \u001b[0;36mget_handle\u001b[1;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[0;32m    584\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mis_path\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    585\u001b[0m                 \u001b[1;32massert\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 586\u001b[1;33m                 handle = gzip.GzipFile(\n\u001b[0m\u001b[0;32m    587\u001b[0m                     \u001b[0mfilename\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    588\u001b[0m                     \u001b[0mmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mioargs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmode\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.8_3.8.2288.0_x64__qbz5n2kfra8p0\\lib\\gzip.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, filename, mode, compresslevel, fileobj, mtime)\u001b[0m\n\u001b[0;32m    171\u001b[0m             \u001b[0mmode\u001b[0m \u001b[1;33m+=\u001b[0m \u001b[1;34m'b'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    172\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mfileobj\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 173\u001b[1;33m             \u001b[0mfileobj\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmyfileobj\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbuiltins\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmode\u001b[0m \u001b[1;32mor\u001b[0m \u001b[1;34m'rb'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    174\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mfilename\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    175\u001b[0m             \u001b[0mfilename\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfileobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'name'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m''\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../data/assembly_summary_.txt.gz'"
     ]
    }
   ],
   "source": [
    "# GenBank\n",
    "gbk = pd.read_csv(\"../data/assembly_summary_.txt.gz\", compression='gzip', header=1, delimiter=\"\\t\")\n",
    "gbk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "limiting-muslim",
   "metadata": {},
   "outputs": [],
   "source": [
    "def split_aa(x):\n",
    "    return x.split(\".\")\n",
    "\n",
    "# phagesdf = pd.concat([pd.DataFrame.from_records(phagesdf['Contig'].apply(get_acc_name), columns=[acccol, 'Name']), phagesdf], axis=1)\n",
    "# phagesdf\n",
    "\n",
    "gbkaa =  pd.concat([pd.DataFrame.from_records(gbk['# assembly_accession'].apply(split_aa), columns=['assembly', 'accession']), gbk], axis=1)\n",
    "gbkaa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "pregnant-edwards",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.unique(gbkaa['assembly']).shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "quick-gibson",
   "metadata": {},
   "outputs": [],
   "source": [
    "gtdb['ncbi_genbank_assembly_accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "wanted-snake",
   "metadata": {},
   "outputs": [],
   "source": [
    "rast['assembly_accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "round-andrews",
   "metadata": {},
   "outputs": [],
   "source": [
    "gbk['# assembly_accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "hybrid-reverse",
   "metadata": {},
   "outputs": [],
   "source": [
    "gbkaa[gbkaa['assembly'] == 'GCA_000001405']"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
