{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "internal-anchor",
   "metadata": {},
   "source": [
    "## Compare the assemblies in GenBank, GTDB, and RAST\n",
    "\n",
    "Each has a different set. What are the unions and intersections?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "acute-prospect",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A lot of this is not used, but we import it so we have it later!\n",
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from PhiSpyAnalysis import theils_u, DateConverter, printmd\n",
    "from PhiSpyAnalysis import read_phages, read_gtdb, read_checkv, read_base_pp, read_categories, read_metadata, read_gbk_metadata\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "\n",
    "\n",
    "import subprocess\n",
    "import gzip\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "digital-cancellation",
   "metadata": {},
   "outputs": [],
   "source": [
    "rv = re.compile('^\\\\w+')\n",
    "def remove_ver(x):\n",
    "    return rv.search(str(x)).group()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "statewide-terror",
   "metadata": {},
   "source": [
    "# GTDB\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "compact-shell",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000006155.2</td>\n",
       "      <td>GCA_000006155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000007385.1</td>\n",
       "      <td>GCA_000007385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000008605.1</td>\n",
       "      <td>GCA_000008605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000010565.1</td>\n",
       "      <td>GCA_000010565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000013845.2</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191522</th>\n",
       "      <td>GCA_902166935.1</td>\n",
       "      <td>GCA_902166935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191523</th>\n",
       "      <td>GCA_902166945.1</td>\n",
       "      <td>GCA_902166945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191524</th>\n",
       "      <td>GCA_902167295.1</td>\n",
       "      <td>GCA_902167295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191525</th>\n",
       "      <td>GCA_902167305.1</td>\n",
       "      <td>GCA_902167305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191526</th>\n",
       "      <td>GCA_902167325.1</td>\n",
       "      <td>GCA_902167325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191527 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000006155.2  GCA_000006155\n",
       "1         GCA_000007385.1  GCA_000007385\n",
       "2         GCA_000008605.1  GCA_000008605\n",
       "3         GCA_000010565.1  GCA_000010565\n",
       "4         GCA_000013845.2  GCA_000013845\n",
       "...                   ...            ...\n",
       "191522    GCA_902166935.1  GCA_902166935\n",
       "191523    GCA_902166945.1  GCA_902166945\n",
       "191524    GCA_902167295.1  GCA_902167295\n",
       "191525    GCA_902167305.1  GCA_902167305\n",
       "191526    GCA_902167325.1  GCA_902167325\n",
       "\n",
       "[191527 rows x 2 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GTDB\n",
    "gtdb = read_gtdb()\n",
    "gtdb['assembly_nover'] = gtdb['assembly_accession'].apply(remove_ver)\n",
    "gtdb[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "pediatric-seven",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_lineage</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>coding_density</th>\n",
       "      <th>...</th>\n",
       "      <th>ssu_silva_blast_bitscore</th>\n",
       "      <th>ssu_silva_blast_evalue</th>\n",
       "      <th>ssu_silva_blast_perc_identity</th>\n",
       "      <th>ssu_silva_blast_subject_id</th>\n",
       "      <th>ssu_silva_taxonomy</th>\n",
       "      <th>total_gap_length</th>\n",
       "      <th>trna_aa_count</th>\n",
       "      <th>trna_count</th>\n",
       "      <th>trna_selenocysteine_count</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GB_GCA_000006155.2</td>\n",
       "      <td>1916</td>\n",
       "      <td>93.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1171</td>\n",
       "      <td>g__Bacillus (UID902)</td>\n",
       "      <td>324</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4305660</td>\n",
       "      <td>80.178992</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42300</td>\n",
       "      <td>16</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>GCA_000006155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GB_GCA_000007385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.82</td>\n",
       "      <td>0.00</td>\n",
       "      <td>481</td>\n",
       "      <td>c__Gammaproteobacteria (UID4202)</td>\n",
       "      <td>276</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4190634</td>\n",
       "      <td>84.805944</td>\n",
       "      <td>...</td>\n",
       "      <td>2846.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.000</td>\n",
       "      <td>JXEG01000201.4293.5839</td>\n",
       "      <td>Bacteria;Proteobacteria;Gammaproteobacteria;Xa...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>GCA_000007385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GB_GCA_000008605.1</td>\n",
       "      <td>67</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>235</td>\n",
       "      <td>f__Spirochaetaceae (UID2512)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1048744</td>\n",
       "      <td>92.155875</td>\n",
       "      <td>...</td>\n",
       "      <td>2854.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.000</td>\n",
       "      <td>CP003679.231299.232859</td>\n",
       "      <td>Bacteria;Spirochaetes;Spirochaetia;Spirochaeta...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>GCA_000008605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GB_GCA_000010565.1</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.63</td>\n",
       "      <td>295</td>\n",
       "      <td>p__Firmicutes (UID1022)</td>\n",
       "      <td>158</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2608397</td>\n",
       "      <td>86.217312</td>\n",
       "      <td>...</td>\n",
       "      <td>2819.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.000</td>\n",
       "      <td>AP009389.1049070.1050595</td>\n",
       "      <td>Bacteria;Firmicutes;Clostridia;Clostridiales;P...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>51</td>\n",
       "      <td>1</td>\n",
       "      <td>GCA_000010565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GB_GCA_000013845.2</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>332</td>\n",
       "      <td>o__Clostridiales (UID1375)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2428396</td>\n",
       "      <td>82.037966</td>\n",
       "      <td>...</td>\n",
       "      <td>2787.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.000</td>\n",
       "      <td>CP000312.233320.234832</td>\n",
       "      <td>Bacteria;Firmicutes;Clostridia;Clostridiales;C...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>1</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191522</th>\n",
       "      <td>RS_GCF_902166935.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.62</td>\n",
       "      <td>0.09</td>\n",
       "      <td>1312</td>\n",
       "      <td>g__Klebsiella (UID5140)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4894244</td>\n",
       "      <td>87.293148</td>\n",
       "      <td>...</td>\n",
       "      <td>2837.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.000</td>\n",
       "      <td>MAPL01000082.143.1696</td>\n",
       "      <td>Bacteria;Proteobacteria;Gammaproteobacteria;En...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>87</td>\n",
       "      <td>1</td>\n",
       "      <td>GCA_902166935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191523</th>\n",
       "      <td>RS_GCF_902166945.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.94</td>\n",
       "      <td>0.72</td>\n",
       "      <td>1162</td>\n",
       "      <td>f__Enterobacteriaceae (UID5121)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5162721</td>\n",
       "      <td>86.540042</td>\n",
       "      <td>...</td>\n",
       "      <td>2837.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.000</td>\n",
       "      <td>CP016813.16076.17629</td>\n",
       "      <td>Bacteria;Proteobacteria;Gammaproteobacteria;En...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>GCA_902166945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191524</th>\n",
       "      <td>RS_GCF_902167295.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5737447</td>\n",
       "      <td>90.122088</td>\n",
       "      <td>...</td>\n",
       "      <td>2747.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>99.933</td>\n",
       "      <td>JX101438.1.1490</td>\n",
       "      <td>Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>GCA_902167295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191525</th>\n",
       "      <td>RS_GCF_902167305.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5738096</td>\n",
       "      <td>90.147873</td>\n",
       "      <td>...</td>\n",
       "      <td>2747.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>99.933</td>\n",
       "      <td>JX101438.1.1490</td>\n",
       "      <td>Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>GCA_902167305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191526</th>\n",
       "      <td>RS_GCF_902167325.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5671443</td>\n",
       "      <td>90.098258</td>\n",
       "      <td>...</td>\n",
       "      <td>2741.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>99.866</td>\n",
       "      <td>JX101438.1.1490</td>\n",
       "      <td>Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "      <td>GCA_902167325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191527 rows × 111 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 accession  ambiguous_bases  checkm_completeness  \\\n",
       "0       GB_GCA_000006155.2             1916                93.12   \n",
       "1       GB_GCA_000007385.1                0                99.82   \n",
       "2       GB_GCA_000008605.1               67               100.00   \n",
       "3       GB_GCA_000010565.1                0               100.00   \n",
       "4       GB_GCA_000013845.2                0               100.00   \n",
       "...                    ...              ...                  ...   \n",
       "191522  RS_GCF_902166935.1                0                99.62   \n",
       "191523  RS_GCF_902166945.1                0                99.94   \n",
       "191524  RS_GCF_902167295.1                0                99.51   \n",
       "191525  RS_GCF_902167305.1                0                99.51   \n",
       "191526  RS_GCF_902167325.1                0                99.51   \n",
       "\n",
       "        checkm_contamination  checkm_marker_count  \\\n",
       "0                       0.00                 1171   \n",
       "1                       0.00                  481   \n",
       "2                       0.00                  235   \n",
       "3                       0.63                  295   \n",
       "4                       0.00                  332   \n",
       "...                      ...                  ...   \n",
       "191522                  0.09                 1312   \n",
       "191523                  0.72                 1162   \n",
       "191524                  0.00                  303   \n",
       "191525                  0.00                  303   \n",
       "191526                  0.00                  303   \n",
       "\n",
       "                   checkm_marker_lineage  checkm_marker_set_count  \\\n",
       "0                   g__Bacillus (UID902)                      324   \n",
       "1       c__Gammaproteobacteria (UID4202)                      276   \n",
       "2           f__Spirochaetaceae (UID2512)                      124   \n",
       "3                p__Firmicutes (UID1022)                      158   \n",
       "4             o__Clostridiales (UID1375)                      124   \n",
       "...                                  ...                      ...   \n",
       "191522           g__Klebsiella (UID5140)                      336   \n",
       "191523   f__Enterobacteriaceae (UID5121)                      336   \n",
       "191524        p__Bacteroidetes (UID2591)                      203   \n",
       "191525        p__Bacteroidetes (UID2591)                      203   \n",
       "191526        p__Bacteroidetes (UID2591)                      203   \n",
       "\n",
       "        checkm_strain_heterogeneity  coding_bases  coding_density  ...  \\\n",
       "0                               0.0       4305660       80.178992  ...   \n",
       "1                               0.0       4190634       84.805944  ...   \n",
       "2                               0.0       1048744       92.155875  ...   \n",
       "3                               0.0       2608397       86.217312  ...   \n",
       "4                               0.0       2428396       82.037966  ...   \n",
       "...                             ...           ...             ...  ...   \n",
       "191522                          0.0       4894244       87.293148  ...   \n",
       "191523                          0.0       5162721       86.540042  ...   \n",
       "191524                          0.0       5737447       90.122088  ...   \n",
       "191525                          0.0       5738096       90.147873  ...   \n",
       "191526                          0.0       5671443       90.098258  ...   \n",
       "\n",
       "        ssu_silva_blast_bitscore  ssu_silva_blast_evalue  \\\n",
       "0                            NaN                     NaN   \n",
       "1                         2846.0                     0.0   \n",
       "2                         2854.0                     0.0   \n",
       "3                         2819.0                     0.0   \n",
       "4                         2787.0                     0.0   \n",
       "...                          ...                     ...   \n",
       "191522                    2837.0                     0.0   \n",
       "191523                    2837.0                     0.0   \n",
       "191524                    2747.0                     0.0   \n",
       "191525                    2747.0                     0.0   \n",
       "191526                    2741.0                     0.0   \n",
       "\n",
       "        ssu_silva_blast_perc_identity  ssu_silva_blast_subject_id  \\\n",
       "0                                 NaN                         NaN   \n",
       "1                             100.000      JXEG01000201.4293.5839   \n",
       "2                             100.000      CP003679.231299.232859   \n",
       "3                             100.000    AP009389.1049070.1050595   \n",
       "4                             100.000      CP000312.233320.234832   \n",
       "...                               ...                         ...   \n",
       "191522                        100.000       MAPL01000082.143.1696   \n",
       "191523                        100.000        CP016813.16076.17629   \n",
       "191524                         99.933             JX101438.1.1490   \n",
       "191525                         99.933             JX101438.1.1490   \n",
       "191526                         99.866             JX101438.1.1490   \n",
       "\n",
       "                                       ssu_silva_taxonomy total_gap_length  \\\n",
       "0                                                     NaN            42300   \n",
       "1       Bacteria;Proteobacteria;Gammaproteobacteria;Xa...                0   \n",
       "2       Bacteria;Spirochaetes;Spirochaetia;Spirochaeta...                0   \n",
       "3       Bacteria;Firmicutes;Clostridia;Clostridiales;P...                0   \n",
       "4       Bacteria;Firmicutes;Clostridia;Clostridiales;C...                0   \n",
       "...                                                   ...              ...   \n",
       "191522  Bacteria;Proteobacteria;Gammaproteobacteria;En...                0   \n",
       "191523  Bacteria;Proteobacteria;Gammaproteobacteria;En...                0   \n",
       "191524  Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...                0   \n",
       "191525  Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...                0   \n",
       "191526  Bacteria;Bacteroidetes;Bacteroidia;Chitinophag...                0   \n",
       "\n",
       "       trna_aa_count trna_count trna_selenocysteine_count assembly_nover  \n",
       "0                 16         31                         0  GCA_000006155  \n",
       "1                 20         53                         0  GCA_000007385  \n",
       "2                 20         45                         0  GCA_000008605  \n",
       "3                 20         51                         1  GCA_000010565  \n",
       "4                 20         95                         1  GCA_000013845  \n",
       "...              ...        ...                       ...            ...  \n",
       "191522            20         87                         1  GCA_902166935  \n",
       "191523            20         86                         1  GCA_902166945  \n",
       "191524            20         58                         0  GCA_902167295  \n",
       "191525            20         58                         0  GCA_902167305  \n",
       "191526            20         59                         0  GCA_902167325  \n",
       "\n",
       "[191527 rows x 111 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "textile-software",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>GCA_000003135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003215.1</td>\n",
       "      <td>GCA_000003215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>GCA_000003645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>GCA_000003925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>GCA_000003955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320171</th>\n",
       "      <td>GCF_900167595.1</td>\n",
       "      <td>GCF_900167595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320172</th>\n",
       "      <td>GCF_900167605.1</td>\n",
       "      <td>GCF_900167605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320173</th>\n",
       "      <td>GCF_900167615.1</td>\n",
       "      <td>GCF_900167615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320174</th>\n",
       "      <td>GCF_900167625.1</td>\n",
       "      <td>GCF_900167625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320175</th>\n",
       "      <td>GCF_900167635.1</td>\n",
       "      <td>GCF_900167635</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320176 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000003135.1  GCA_000003135\n",
       "1         GCA_000003215.1  GCA_000003215\n",
       "2         GCA_000003645.1  GCA_000003645\n",
       "3         GCA_000003925.1  GCA_000003925\n",
       "4         GCA_000003955.1  GCA_000003955\n",
       "...                   ...            ...\n",
       "320171    GCF_900167595.1  GCF_900167595\n",
       "320172    GCF_900167605.1  GCF_900167605\n",
       "320173    GCF_900167615.1  GCF_900167615\n",
       "320174    GCF_900167625.1  GCF_900167625\n",
       "320175    GCF_900167635.1  GCF_900167635\n",
       "\n",
       "[320176 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# RAST\n",
    "# the full data set. Don't try this at home!\n",
    "# metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "rast = read_metadata()\n",
    "rast['assembly_nover'] = rast['assembly_accession'].apply(remove_ver)\n",
    "rast[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "sustained-broadcast",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>...</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "      <th>isolation_date</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>5.484803e+05</td>\n",
       "      <td>Bifidobacterium longum subsp. longum ATCC 55813</td>\n",
       "      <td>Bifidobacterium longum subsp. longum ATCC 55813</td>\n",
       "      <td>548480</td>\n",
       "      <td>WGS</td>\n",
       "      <td>ATCC 55813</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Anaerobic</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bifidobacterium longum subsp. longum ATCC 5581...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003215.1</td>\n",
       "      <td>4.556313e+05</td>\n",
       "      <td>Clostridium difficile QCD-66c26</td>\n",
       "      <td>Clostridium difficile QCD-66c26</td>\n",
       "      <td>455631</td>\n",
       "      <td>WGS</td>\n",
       "      <td>QCD-66c26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Anaerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Antibiotic-associated Diarrhea;Pseudomembraneo...</td>\n",
       "      <td>Clostridium difficile QCD-66c26. was isolated ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>5.269733e+05</td>\n",
       "      <td>Bacillus cereus m1293</td>\n",
       "      <td>Bacillus cereus m1293</td>\n",
       "      <td>526973</td>\n",
       "      <td>WGS</td>\n",
       "      <td>m1293</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>Bacillus cereus m1293. This strain, also know...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>5.269973e+05</td>\n",
       "      <td>Bacillus mycoides DSM 2048</td>\n",
       "      <td>Bacillus mycoides DSM 2048</td>\n",
       "      <td>526997</td>\n",
       "      <td>WGS</td>\n",
       "      <td>DSM 2048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacillus mycoides DSM 2048. Bacillus mycoides...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>5.269943e+05</td>\n",
       "      <td>Bacillus cereus AH1273</td>\n",
       "      <td>Bacillus cereus AH1273</td>\n",
       "      <td>526994</td>\n",
       "      <td>WGS</td>\n",
       "      <td>AH1273</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>Bacillus cereus AH1273. This strain was isola...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000003955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320171</th>\n",
       "      <td>GCF_900167595.1</td>\n",
       "      <td>1.496160e+03</td>\n",
       "      <td>Clostridioides difficile strain VRECD0186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1496</td>\n",
       "      <td>WGS</td>\n",
       "      <td>VRECD0186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>These data are part of a pre-publication relea...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015.294319</td>\n",
       "      <td>GCF_900167595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320172</th>\n",
       "      <td>GCF_900167605.1</td>\n",
       "      <td>1.147123e+06</td>\n",
       "      <td>Caloramator quimbayensis strain USBA 833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1147123</td>\n",
       "      <td>WGS</td>\n",
       "      <td>USBA 833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Caloramator quimbayensis USBA 833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_900167605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320173</th>\n",
       "      <td>GCF_900167615.1</td>\n",
       "      <td>1.496159e+03</td>\n",
       "      <td>Clostridioides difficile strain CD19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1496</td>\n",
       "      <td>WGS</td>\n",
       "      <td>CD19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>http://www.sanger.ac.uk/resources/downloads/ba...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014.295003</td>\n",
       "      <td>GCF_900167615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320174</th>\n",
       "      <td>GCF_900167625.1</td>\n",
       "      <td>1.496159e+03</td>\n",
       "      <td>Clostridioides difficile strain VRECD0076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1496</td>\n",
       "      <td>WGS</td>\n",
       "      <td>VRECD0076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>These data are part of a pre-publication relea...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2014.295003</td>\n",
       "      <td>GCF_900167625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320175</th>\n",
       "      <td>GCF_900167635.1</td>\n",
       "      <td>3.591109e+05</td>\n",
       "      <td>Pseudomonas extremaustralis strain USBA 515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>359110</td>\n",
       "      <td>WGS</td>\n",
       "      <td>USBA 515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pseudomonas extremaustralis USBA 515</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_900167635</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320176 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession     genome_id  \\\n",
       "0         GCA_000003135.1  5.484803e+05   \n",
       "1         GCA_000003215.1  4.556313e+05   \n",
       "2         GCA_000003645.1  5.269733e+05   \n",
       "3         GCA_000003925.1  5.269973e+05   \n",
       "4         GCA_000003955.1  5.269943e+05   \n",
       "...                   ...           ...   \n",
       "320171    GCF_900167595.1  1.496160e+03   \n",
       "320172    GCF_900167605.1  1.147123e+06   \n",
       "320173    GCF_900167615.1  1.496159e+03   \n",
       "320174    GCF_900167625.1  1.496159e+03   \n",
       "320175    GCF_900167635.1  3.591109e+05   \n",
       "\n",
       "                                            genome_name  \\\n",
       "0       Bifidobacterium longum subsp. longum ATCC 55813   \n",
       "1                       Clostridium difficile QCD-66c26   \n",
       "2                                 Bacillus cereus m1293   \n",
       "3                            Bacillus mycoides DSM 2048   \n",
       "4                                Bacillus cereus AH1273   \n",
       "...                                                 ...   \n",
       "320171        Clostridioides difficile strain VRECD0186   \n",
       "320172         Caloramator quimbayensis strain USBA 833   \n",
       "320173             Clostridioides difficile strain CD19   \n",
       "320174        Clostridioides difficile strain VRECD0076   \n",
       "320175      Pseudomonas extremaustralis strain USBA 515   \n",
       "\n",
       "                                          organism_name  taxon_id  \\\n",
       "0       Bifidobacterium longum subsp. longum ATCC 55813    548480   \n",
       "1                       Clostridium difficile QCD-66c26    455631   \n",
       "2                                 Bacillus cereus m1293    526973   \n",
       "3                            Bacillus mycoides DSM 2048    526997   \n",
       "4                                Bacillus cereus AH1273    526994   \n",
       "...                                                 ...       ...   \n",
       "320171                                              NaN      1496   \n",
       "320172                                              NaN   1147123   \n",
       "320173                                              NaN      1496   \n",
       "320174                                              NaN      1496   \n",
       "320175                                              NaN    359110   \n",
       "\n",
       "       genome_status      strain serovar biovar pathovar  ...  \\\n",
       "0                WGS  ATCC 55813     NaN    NaN      NaN  ...   \n",
       "1                WGS   QCD-66c26     NaN    NaN      NaN  ...   \n",
       "2                WGS       m1293     NaN    NaN      NaN  ...   \n",
       "3                WGS    DSM 2048     NaN    NaN      NaN  ...   \n",
       "4                WGS      AH1273     NaN    NaN      NaN  ...   \n",
       "...              ...         ...     ...    ...      ...  ...   \n",
       "320171           WGS   VRECD0186     NaN    NaN      NaN  ...   \n",
       "320172           WGS    USBA 833     NaN    NaN      NaN  ...   \n",
       "320173           WGS        CD19     NaN    NaN      NaN  ...   \n",
       "320174           WGS   VRECD0076     NaN    NaN      NaN  ...   \n",
       "320175           WGS    USBA 515     NaN    NaN      NaN  ...   \n",
       "\n",
       "       temperature_range optimal_temperature salinity oxygen_requirement  \\\n",
       "0             Mesophilic                   -      NaN          Anaerobic   \n",
       "1             Mesophilic                  37      NaN          Anaerobic   \n",
       "2             Mesophilic                   -      NaN            Aerobic   \n",
       "3                    NaN                   -      NaN                NaN   \n",
       "4             Mesophilic                   -      NaN            Aerobic   \n",
       "...                  ...                 ...      ...                ...   \n",
       "320171               NaN                 NaN      NaN                NaN   \n",
       "320172               NaN                 NaN      NaN                NaN   \n",
       "320173               NaN                 NaN      NaN                NaN   \n",
       "320174               NaN                 NaN      NaN                NaN   \n",
       "320175               NaN                 NaN      NaN                NaN   \n",
       "\n",
       "                habitat                                            disease  \\\n",
       "0       Host-associated                                                NaN   \n",
       "1              Multiple  Antibiotic-associated Diarrhea;Pseudomembraneo...   \n",
       "2              Multiple                                     Food poisoning   \n",
       "3                   NaN                                                NaN   \n",
       "4              Multiple                                     Food poisoning   \n",
       "...                 ...                                                ...   \n",
       "320171              NaN                                                NaN   \n",
       "320172              NaN                                                NaN   \n",
       "320173              NaN                                                NaN   \n",
       "320174              NaN                                                NaN   \n",
       "320175              NaN                                                NaN   \n",
       "\n",
       "                                                 comments additional_metadata  \\\n",
       "0       Bifidobacterium longum subsp. longum ATCC 5581...                 NaN   \n",
       "1       Clostridium difficile QCD-66c26. was isolated ...                 NaN   \n",
       "2        Bacillus cereus m1293. This strain, also know...                 NaN   \n",
       "3        Bacillus mycoides DSM 2048. Bacillus mycoides...                 NaN   \n",
       "4        Bacillus cereus AH1273. This strain was isola...                 NaN   \n",
       "...                                                   ...                 ...   \n",
       "320171  These data are part of a pre-publication relea...                 NaN   \n",
       "320172                  Caloramator quimbayensis USBA 833                 NaN   \n",
       "320173  http://www.sanger.ac.uk/resources/downloads/ba...                 NaN   \n",
       "320174  These data are part of a pre-publication relea...                 NaN   \n",
       "320175               Pseudomonas extremaustralis USBA 515                 NaN   \n",
       "\n",
       "       isolation_date assembly_nover  \n",
       "0                 NaN  GCA_000003135  \n",
       "1                 NaN  GCA_000003215  \n",
       "2                 NaN  GCA_000003645  \n",
       "3                 NaN  GCA_000003925  \n",
       "4                 NaN  GCA_000003955  \n",
       "...               ...            ...  \n",
       "320171    2015.294319  GCF_900167595  \n",
       "320172            NaN  GCF_900167605  \n",
       "320173    2014.295003  GCF_900167615  \n",
       "320174    2014.295003  GCF_900167625  \n",
       "320175            NaN  GCF_900167635  \n",
       "\n",
       "[320176 rows x 68 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rast"
   ]
  },
  {
   "cell_type": "raw",
   "id": "clinical-means",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "considerable-brief",
   "metadata": {},
   "source": [
    "# GenBank\n",
    "\n",
    "This assembly summary comes from [GenBank ftp site](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/) and you want the [assembly_summary.txt](ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/assembly_summary.txt) file explicitly from bacteria (but don't try and open the bacteria list in your browser!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "curious-compatibility",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3357: DtypeWarning: Columns (21) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>GCA_900128725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>GCA_008244535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_011046815.1</td>\n",
       "      <td>GCA_011046815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_011054035.1</td>\n",
       "      <td>GCA_011054035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_011331185.1</td>\n",
       "      <td>GCA_011331185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949928</th>\n",
       "      <td>GCA_017746755.1</td>\n",
       "      <td>GCA_017746755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949929</th>\n",
       "      <td>GCA_017746715.1</td>\n",
       "      <td>GCA_017746715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949930</th>\n",
       "      <td>GCA_013266695.1</td>\n",
       "      <td>GCA_013266695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949931</th>\n",
       "      <td>GCA_015831295.1</td>\n",
       "      <td>GCA_015831295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949932</th>\n",
       "      <td>GCA_015686645.1</td>\n",
       "      <td>GCA_015686645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>949933 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_900128725.1  GCA_900128725\n",
       "1         GCA_008244535.1  GCA_008244535\n",
       "2         GCA_011046815.1  GCA_011046815\n",
       "3         GCA_011054035.1  GCA_011054035\n",
       "4         GCA_011331185.1  GCA_011331185\n",
       "...                   ...            ...\n",
       "949928    GCA_017746755.1  GCA_017746755\n",
       "949929    GCA_017746715.1  GCA_017746715\n",
       "949930    GCA_013266695.1  GCA_013266695\n",
       "949931    GCA_015831295.1  GCA_015831295\n",
       "949932    GCA_015686645.1  GCA_015686645\n",
       "\n",
       "[949933 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GenBank\n",
    "gbk = read_gbk_metadata()\n",
    "gbk['assembly_nover'] = gbk['assembly_accession'].apply(remove_ver)\n",
    "gbk[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "toxic-applicant",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>genome_rep</th>\n",
       "      <th>seq_rel_date</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>submitter</th>\n",
       "      <th>gbrs_paired_asm</th>\n",
       "      <th>paired_asm_comp</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>PRJEB15400</td>\n",
       "      <td>SAMEA4556317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>Buchnera aphidicola</td>\n",
       "      <td>strain=BCifornacula</td>\n",
       "      <td>2912</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2016/11/25</td>\n",
       "      <td>BCifornacula_v1.0</td>\n",
       "      <td>Centre de Biologie pour la Gestion des Populat...</td>\n",
       "      <td>GCF_900128725.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_900128725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>PRJNA562146</td>\n",
       "      <td>SAMN12634886</td>\n",
       "      <td>VTFL00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "      <td>strain=PYS_80_B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2019/09/02</td>\n",
       "      <td>ASM824453v1</td>\n",
       "      <td>Agharkar Research Institute</td>\n",
       "      <td>GCF_008244535.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/008...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_008244535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_011046815.1</td>\n",
       "      <td>PRJNA480137</td>\n",
       "      <td>SAMN09639436</td>\n",
       "      <td>DSOW00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SpSt-35</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2020/03/02</td>\n",
       "      <td>ASM1104681v1</td>\n",
       "      <td>The University of Hong Kong</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/011...</td>\n",
       "      <td>derived from metagenome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_011046815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_011054035.1</td>\n",
       "      <td>PRJNA480137</td>\n",
       "      <td>SAMN09639292</td>\n",
       "      <td>DSJI00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SpSt-22</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2020/03/02</td>\n",
       "      <td>ASM1105403v1</td>\n",
       "      <td>The University of Hong Kong</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/011...</td>\n",
       "      <td>derived from metagenome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_011054035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_011331185.1</td>\n",
       "      <td>PRJNA480137</td>\n",
       "      <td>SAMN09639525</td>\n",
       "      <td>DSSH00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>14.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SpSt-43</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2020/03/16</td>\n",
       "      <td>ASM1133118v1</td>\n",
       "      <td>The University of Hong Kong</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/011...</td>\n",
       "      <td>derived from metagenome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_011331185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949928</th>\n",
       "      <td>GCA_017746755.1</td>\n",
       "      <td>PRJNA647383</td>\n",
       "      <td>SAMN18473588</td>\n",
       "      <td>JAGHZP000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2821504.0</td>\n",
       "      <td>2821504.0</td>\n",
       "      <td>Cyanobacteria bacterium SID2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SID2</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2021/04/04</td>\n",
       "      <td>ASM1774675v1</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/017...</td>\n",
       "      <td>derived from metagenome; genus undefined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_017746755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949929</th>\n",
       "      <td>GCA_017746715.1</td>\n",
       "      <td>PRJNA647383</td>\n",
       "      <td>SAMN18473589</td>\n",
       "      <td>JAGHZQ000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2821505.0</td>\n",
       "      <td>2821505.0</td>\n",
       "      <td>Cyanobacteria bacterium SBLK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SBLK</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2021/04/04</td>\n",
       "      <td>ASM1774671v1</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/017...</td>\n",
       "      <td>derived from metagenome; genus undefined</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_017746715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949930</th>\n",
       "      <td>GCA_013266695.1</td>\n",
       "      <td>PRJNA633965</td>\n",
       "      <td>SAMN14978626</td>\n",
       "      <td>JABRXF000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2821844.0</td>\n",
       "      <td>2821844.0</td>\n",
       "      <td>Hongsoonwoonella zoysiae</td>\n",
       "      <td>strain=SY4-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2020/06/03</td>\n",
       "      <td>ASM1326669v1</td>\n",
       "      <td>BioPS Co., Ltd.</td>\n",
       "      <td>GCF_013266695.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/013...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_013266695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949931</th>\n",
       "      <td>GCA_015831295.1</td>\n",
       "      <td>PRJNA679689</td>\n",
       "      <td>SAMN16843871</td>\n",
       "      <td>JADXDQ000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2822799.0</td>\n",
       "      <td>2822799.0</td>\n",
       "      <td>Streptococcus sp. NLN64</td>\n",
       "      <td>strain=NLN64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2020/12/07</td>\n",
       "      <td>ASM1583129v1</td>\n",
       "      <td>Hainan Medical University-University</td>\n",
       "      <td>GCF_015831295.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/015...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_015831295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949932</th>\n",
       "      <td>GCA_015686645.1</td>\n",
       "      <td>PRJNA679155</td>\n",
       "      <td>SAMN16824905</td>\n",
       "      <td>JADPMS000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>2822800.0</td>\n",
       "      <td>2822800.0</td>\n",
       "      <td>Streptococcus sp. NLN76</td>\n",
       "      <td>strain=NLN76</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Full</td>\n",
       "      <td>2020/11/24</td>\n",
       "      <td>ASM1568664v1</td>\n",
       "      <td>Hainan Medical University-University</td>\n",
       "      <td>GCF_015686645.1</td>\n",
       "      <td>identical</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/015...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_015686645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>949933 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession   bioproject     biosample         wgs_master  \\\n",
       "0         GCA_900128725.1   PRJEB15400  SAMEA4556317                NaN   \n",
       "1         GCA_008244535.1  PRJNA562146  SAMN12634886     VTFL00000000.1   \n",
       "2         GCA_011046815.1  PRJNA480137  SAMN09639436     DSOW00000000.1   \n",
       "3         GCA_011054035.1  PRJNA480137  SAMN09639292     DSJI00000000.1   \n",
       "4         GCA_011331185.1  PRJNA480137  SAMN09639525     DSSH00000000.1   \n",
       "...                   ...          ...           ...                ...   \n",
       "949928    GCA_017746755.1  PRJNA647383  SAMN18473588  JAGHZP000000000.1   \n",
       "949929    GCA_017746715.1  PRJNA647383  SAMN18473589  JAGHZQ000000000.1   \n",
       "949930    GCA_013266695.1  PRJNA633965  SAMN14978626  JABRXF000000000.1   \n",
       "949931    GCA_015831295.1  PRJNA679689  SAMN16843871  JADXDQ000000000.1   \n",
       "949932    GCA_015686645.1  PRJNA679155  SAMN16824905  JADPMS000000000.1   \n",
       "\n",
       "       refseq_category      taxid  species_taxid  \\\n",
       "0                   na        9.0            9.0   \n",
       "1                   na       14.0           14.0   \n",
       "2                   na       14.0           14.0   \n",
       "3                   na       14.0           14.0   \n",
       "4                   na       14.0           14.0   \n",
       "...                ...        ...            ...   \n",
       "949928              na  2821504.0      2821504.0   \n",
       "949929              na  2821505.0      2821505.0   \n",
       "949930              na  2821844.0      2821844.0   \n",
       "949931              na  2822799.0      2822799.0   \n",
       "949932              na  2822800.0      2822800.0   \n",
       "\n",
       "                       organism_name   infraspecific_name  isolate  ...  \\\n",
       "0                Buchnera aphidicola  strain=BCifornacula     2912  ...   \n",
       "1          Dictyoglomus thermophilum      strain=PYS_80_B      NaN  ...   \n",
       "2          Dictyoglomus thermophilum                  NaN  SpSt-35  ...   \n",
       "3          Dictyoglomus thermophilum                  NaN  SpSt-22  ...   \n",
       "4          Dictyoglomus thermophilum                  NaN  SpSt-43  ...   \n",
       "...                              ...                  ...      ...  ...   \n",
       "949928  Cyanobacteria bacterium SID2                  NaN     SID2  ...   \n",
       "949929  Cyanobacteria bacterium SBLK                  NaN     SBLK  ...   \n",
       "949930      Hongsoonwoonella zoysiae         strain=SY4-7      NaN  ...   \n",
       "949931       Streptococcus sp. NLN64         strain=NLN64      NaN  ...   \n",
       "949932       Streptococcus sp. NLN76         strain=NLN76      NaN  ...   \n",
       "\n",
       "       genome_rep seq_rel_date           asm_name  \\\n",
       "0            Full   2016/11/25  BCifornacula_v1.0   \n",
       "1            Full   2019/09/02        ASM824453v1   \n",
       "2            Full   2020/03/02       ASM1104681v1   \n",
       "3            Full   2020/03/02       ASM1105403v1   \n",
       "4            Full   2020/03/16       ASM1133118v1   \n",
       "...           ...          ...                ...   \n",
       "949928       Full   2021/04/04       ASM1774675v1   \n",
       "949929       Full   2021/04/04       ASM1774671v1   \n",
       "949930       Full   2020/06/03       ASM1326669v1   \n",
       "949931       Full   2020/12/07       ASM1583129v1   \n",
       "949932       Full   2020/11/24       ASM1568664v1   \n",
       "\n",
       "                                                submitter  gbrs_paired_asm  \\\n",
       "0       Centre de Biologie pour la Gestion des Populat...  GCF_900128725.1   \n",
       "1                             Agharkar Research Institute  GCF_008244535.1   \n",
       "2                             The University of Hong Kong               na   \n",
       "3                             The University of Hong Kong               na   \n",
       "4                             The University of Hong Kong               na   \n",
       "...                                                   ...              ...   \n",
       "949928                              University of Florida               na   \n",
       "949929                              University of Florida               na   \n",
       "949930                                    BioPS Co., Ltd.  GCF_013266695.1   \n",
       "949931               Hainan Medical University-University  GCF_015831295.1   \n",
       "949932               Hainan Medical University-University  GCF_015686645.1   \n",
       "\n",
       "       paired_asm_comp                                           ftp_path  \\\n",
       "0            identical  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...   \n",
       "1            identical  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/008...   \n",
       "2                   na  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/011...   \n",
       "3                   na  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/011...   \n",
       "4                   na  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/011...   \n",
       "...                ...                                                ...   \n",
       "949928              na  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/017...   \n",
       "949929              na  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/017...   \n",
       "949930       identical  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/013...   \n",
       "949931       identical  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/015...   \n",
       "949932       identical  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/015...   \n",
       "\n",
       "                            excluded_from_refseq relation_to_type_material  \\\n",
       "0                                            NaN                       NaN   \n",
       "1                                            NaN                       NaN   \n",
       "2                        derived from metagenome                       NaN   \n",
       "3                        derived from metagenome                       NaN   \n",
       "4                        derived from metagenome                       NaN   \n",
       "...                                          ...                       ...   \n",
       "949928  derived from metagenome; genus undefined                       NaN   \n",
       "949929  derived from metagenome; genus undefined                       NaN   \n",
       "949930                                       NaN                       NaN   \n",
       "949931                                       NaN                       NaN   \n",
       "949932                                       NaN                       NaN   \n",
       "\n",
       "       assembly_nover  \n",
       "0       GCA_900128725  \n",
       "1       GCA_008244535  \n",
       "2       GCA_011046815  \n",
       "3       GCA_011054035  \n",
       "4       GCA_011331185  \n",
       "...               ...  \n",
       "949928  GCA_017746755  \n",
       "949929  GCA_017746715  \n",
       "949930  GCA_013266695  \n",
       "949931  GCA_015831295  \n",
       "949932  GCA_015686645  \n",
       "\n",
       "[949933 rows x 23 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "endless-entrance",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Please note that this was run with git commit 86420e1 that has 567,404 genomes parsed.\n",
      "Initially there were 3,265,453 kept phages,but after filtering we kept 3,265,453 prophages from 567,399 genomes"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000043285.1</td>\n",
       "      <td>ASM4328v1</td>\n",
       "      <td>705557.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000043285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000046685.1</td>\n",
       "      <td>ASM4668v1</td>\n",
       "      <td>1581384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>97011.0</td>\n",
       "      <td>GCA_000046685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000046705.1</td>\n",
       "      <td>ASM4670v1</td>\n",
       "      <td>1931047.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>278490.0</td>\n",
       "      <td>GCA_000046705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000046845.1</td>\n",
       "      <td>ASM4684v1</td>\n",
       "      <td>3598621.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>83112.0</td>\n",
       "      <td>GCA_000046845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000047365.1</td>\n",
       "      <td>ASM4736v1</td>\n",
       "      <td>4840898.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>213427.0</td>\n",
       "      <td>GCA_000047365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567399</th>\n",
       "      <td>GCA_905187425.1</td>\n",
       "      <td>Xanthomonas_sp._CPBF_424_-_hybrid_assembly</td>\n",
       "      <td>4900930.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>214278.0</td>\n",
       "      <td>GCA_905187425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567400</th>\n",
       "      <td>GCA_905188235.1</td>\n",
       "      <td>ASM90518823v1</td>\n",
       "      <td>4963609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>178993.0</td>\n",
       "      <td>GCA_905188235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567401</th>\n",
       "      <td>GCA_905219375.1</td>\n",
       "      <td>QI0054</td>\n",
       "      <td>3034314.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>178527.0</td>\n",
       "      <td>GCA_905219375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567402</th>\n",
       "      <td>GCA_905219385.1</td>\n",
       "      <td>QI0055</td>\n",
       "      <td>3034113.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>178526.0</td>\n",
       "      <td>GCA_905219385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567403</th>\n",
       "      <td>GCA_905220645.1</td>\n",
       "      <td>NMYAN.1</td>\n",
       "      <td>3400020.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>65454.0</td>\n",
       "      <td>GCA_905220645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>567399 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession                               assembly_name  \\\n",
       "0         GCA_000043285.1                                   ASM4328v1   \n",
       "1         GCA_000046685.1                                   ASM4668v1   \n",
       "2         GCA_000046705.1                                   ASM4670v1   \n",
       "3         GCA_000046845.1                                   ASM4684v1   \n",
       "4         GCA_000047365.1                                   ASM4736v1   \n",
       "...                   ...                                         ...   \n",
       "567399    GCA_905187425.1  Xanthomonas_sp._CPBF_424_-_hybrid_assembly   \n",
       "567400    GCA_905188235.1                               ASM90518823v1   \n",
       "567401    GCA_905219375.1                                      QI0054   \n",
       "567402    GCA_905219385.1                                      QI0055   \n",
       "567403    GCA_905220645.1                                     NMYAN.1   \n",
       "\n",
       "        Genome length  Contigs  Phage Contigs  Total Predicted Prophages  \\\n",
       "0            705557.0      1.0            1.0                        2.0   \n",
       "1           1581384.0      1.0            1.0                       18.0   \n",
       "2           1931047.0      1.0            1.0                       36.0   \n",
       "3           3598621.0      1.0            1.0                       13.0   \n",
       "4           4840898.0      3.0            3.0                       22.0   \n",
       "...               ...      ...            ...                        ...   \n",
       "567399      4900930.0      1.0            1.0                       46.0   \n",
       "567400      4963609.0      1.0            1.0                       29.0   \n",
       "567401      3034314.0      1.0            1.0                       12.0   \n",
       "567402      3034113.0      1.0            1.0                       12.0   \n",
       "567403      3400020.0     55.0           33.0                       31.0   \n",
       "\n",
       "        Kept  No phage genes  Not enough genes  bp prophage assembly_nover  \n",
       "0        0.0             2.0               0.0          0.0  GCA_000043285  \n",
       "1        3.0             0.0              15.0      97011.0  GCA_000046685  \n",
       "2        7.0             0.0              29.0     278490.0  GCA_000046705  \n",
       "3        3.0             4.0               6.0      83112.0  GCA_000046845  \n",
       "4        3.0             3.0              16.0     213427.0  GCA_000047365  \n",
       "...      ...             ...               ...          ...            ...  \n",
       "567399   7.0             9.0              30.0     214278.0  GCA_905187425  \n",
       "567400   4.0             4.0              21.0     178993.0  GCA_905188235  \n",
       "567401   4.0             3.0               5.0     178527.0  GCA_905219375  \n",
       "567402   4.0             3.0               5.0     178526.0  GCA_905219385  \n",
       "567403   4.0             5.0              22.0      65454.0  GCA_905220645  \n",
       "\n",
       "[567399 rows x 11 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagesdf = read_phages(maxcontigs=-1) # this disables contig length filtering\n",
    "phagesdf['assembly_nover'] = phagesdf['assembly_accession'].apply(remove_ver)\n",
    "phagesdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "anonymous-president",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000043285.1</td>\n",
       "      <td>ASM4328v1</td>\n",
       "      <td>705557.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000043285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000048645.1</td>\n",
       "      <td>ASM4864v1</td>\n",
       "      <td>3635495.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000048645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000048665.1</td>\n",
       "      <td>ASM4866v1</td>\n",
       "      <td>3405519.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000048665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_000020245.1</td>\n",
       "      <td>ASM2024v1</td>\n",
       "      <td>1608548.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000020245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_000020305.1</td>\n",
       "      <td>ASM2030v1</td>\n",
       "      <td>1202435.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000020305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567245</th>\n",
       "      <td>GCA_904067595.1</td>\n",
       "      <td>ASM90406759v1</td>\n",
       "      <td>6794181.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_904067595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567392</th>\n",
       "      <td>GCA_905143135.1</td>\n",
       "      <td>15-58d44</td>\n",
       "      <td>1158207.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_905143135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567394</th>\n",
       "      <td>GCA_905176665.1</td>\n",
       "      <td>D4_OalgB6SA</td>\n",
       "      <td>5465468.0</td>\n",
       "      <td>286.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_905176665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567395</th>\n",
       "      <td>GCA_905176675.1</td>\n",
       "      <td>G3_OalgB6SA</td>\n",
       "      <td>4272394.0</td>\n",
       "      <td>1048.0</td>\n",
       "      <td>179.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_905176675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567396</th>\n",
       "      <td>GCA_905176695.1</td>\n",
       "      <td>G1_OalgB6SA</td>\n",
       "      <td>3819172.0</td>\n",
       "      <td>8736.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_905176695</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34429 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession  assembly_name  Genome length  Contigs  \\\n",
       "0         GCA_000043285.1      ASM4328v1       705557.0      1.0   \n",
       "5         GCA_000048645.1      ASM4864v1      3635495.0      2.0   \n",
       "6         GCA_000048665.1      ASM4866v1      3405519.0      2.0   \n",
       "19        GCA_000020245.1      ASM2024v1      1608548.0      1.0   \n",
       "22        GCA_000020305.1      ASM2030v1      1202435.0      1.0   \n",
       "...                   ...            ...            ...      ...   \n",
       "567245    GCA_904067595.1  ASM90406759v1      6794181.0     68.0   \n",
       "567392    GCA_905143135.1       15-58d44      1158207.0      5.0   \n",
       "567394    GCA_905176665.1    D4_OalgB6SA      5465468.0    286.0   \n",
       "567395    GCA_905176675.1    G3_OalgB6SA      4272394.0   1048.0   \n",
       "567396    GCA_905176695.1    G1_OalgB6SA      3819172.0   8736.0   \n",
       "\n",
       "        Phage Contigs  Total Predicted Prophages  Kept  No phage genes  \\\n",
       "0                 1.0                        2.0   0.0             2.0   \n",
       "5                 2.0                        4.0   0.0             4.0   \n",
       "6                 2.0                        5.0   0.0             5.0   \n",
       "19                1.0                       11.0   0.0             6.0   \n",
       "22                1.0                        4.0   0.0             0.0   \n",
       "...               ...                        ...   ...             ...   \n",
       "567245           55.0                       46.0   0.0            21.0   \n",
       "567392            3.0                        1.0   0.0             1.0   \n",
       "567394           71.0                       10.0   0.0            10.0   \n",
       "567395          179.0                       11.0   0.0             9.0   \n",
       "567396           68.0                        2.0   0.0             2.0   \n",
       "\n",
       "        Not enough genes  bp prophage assembly_nover  \n",
       "0                    0.0          0.0  GCA_000043285  \n",
       "5                    0.0          0.0  GCA_000048645  \n",
       "6                    0.0          0.0  GCA_000048665  \n",
       "19                   5.0          0.0  GCA_000020245  \n",
       "22                   4.0          0.0  GCA_000020305  \n",
       "...                  ...          ...            ...  \n",
       "567245              25.0          0.0  GCA_904067595  \n",
       "567392               0.0          0.0  GCA_905143135  \n",
       "567394               0.0          0.0  GCA_905176665  \n",
       "567395               2.0          0.0  GCA_905176675  \n",
       "567396               0.0          0.0  GCA_905176695  \n",
       "\n",
       "[34429 rows x 11 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagesdf[phagesdf['Kept'] == 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "downtown-subject",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Total Predicted Prophages    21562059.0\n",
       "Not enough genes             15047785.0\n",
       "No phage genes                3248821.0\n",
       "Kept                          3265453.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagesdf[['Total Predicted Prophages', 'Not enough genes', 'No phage genes', 'Kept']].agg('sum')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "medieval-canyon",
   "metadata": {},
   "source": [
    "# What is in common between the groups?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "changing-president",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Between GBK and RAST there are 244,870 genomes in common\n",
      "Between GBK and GTDB there are 189,704 genomes in common\n",
      "Between GTDB and RAST there are 140,463 genomes in common\n",
      "\n",
      "There are 339,231 phages in Genbank that have not been analyzed\n",
      "There are 65,392 phages in GTDB that have not been analyzed\n",
      "There are 186,485 phages in Genbank that have not been analyzed\n",
      "\n",
      "There are 52,219 phages in all three that have not been analyzed\n"
     ]
    }
   ],
   "source": [
    "gbkaa=set(gbk['assembly_accession'])\n",
    "rastaa=set(rast['assembly_accession'])\n",
    "gtdbaa=set(gtdb['assembly_accession'])\n",
    "phagesaa=set(phagesdf['assembly_accession'])\n",
    "\n",
    "\n",
    "gbr = gbkaa.intersection(rastaa)\n",
    "gbg = gbkaa.intersection(gtdbaa)\n",
    "gtr = gtdbaa.intersection(rastaa)\n",
    "\n",
    "print(f\"Between GBK and RAST there are {len(gbr):,} genomes in common\")\n",
    "print(f\"Between GBK and GTDB there are {len(gbg):,} genomes in common\")\n",
    "print(f\"Between GTDB and RAST there are {len(gtr):,} genomes in common\")\n",
    "print()\n",
    "\n",
    "gbnotdone = gbkaa - phagesaa\n",
    "gtnotdone = gtdbaa - phagesaa\n",
    "ranotdone = rastaa - phagesaa\n",
    "print(f\"There are {len(gbnotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print(f\"There are {len(gtnotdone):,} phages in GTDB that have not been analyzed\")\n",
    "print(f\"There are {len(ranotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print()\n",
    "\n",
    "allmissing = gbnotdone.intersection(gtnotdone).intersection(ranotdone)\n",
    "print(f\"There are {len(allmissing):,} phages in all three that have not been analyzed\")\n",
    "\n",
    "if False:\n",
    "    with open(\"../data/unprocessed_phages.txt\", 'w') as out:\n",
    "        for o in allmissing:\n",
    "            gbk[gbk['assembly_accession'] == 'GCA_002129805.1'][['assembly_accession', 'ftp_path']].to_csv(out, sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "surrounded-fireplace",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IF WE IGNORE GENOME VERSIONS\n",
      "There are 904,681 genomes in GenBank\n",
      "There are 320,033 genomes in PATRIC\n",
      "There are 191,521 genomes in GTDB\n",
      "Between GenBank and PATRIC there are 249,334 genomes in common\n",
      "Between GenBank and GTDB there are 191,370 genomes in common\n",
      "Between GTDB and PATRIC there are 143,445 genomes in common\n",
      "\n",
      "There are 337,599 phages in Genbank that have not been analyzed\n",
      "There are 64,358 phages in GTDB that have not been analyzed\n",
      "There are 183,308 phages in PATRIC that have not been analyzed\n",
      "\n",
      "There are 53,262 phages in all three that have not been analyzed\n"
     ]
    }
   ],
   "source": [
    "print(\"IF WE IGNORE GENOME VERSIONS\")\n",
    "gbkaa=set(gbk['assembly_nover'])\n",
    "rastaa=set(rast['assembly_nover'])\n",
    "gtdbaa=set(gtdb['assembly_nover'])\n",
    "phagesaa=set(phagesdf['assembly_nover'])\n",
    "\n",
    "\n",
    "gbr = gbkaa.intersection(rastaa)\n",
    "gbg = gbkaa.intersection(gtdbaa)\n",
    "gtr = gtdbaa.intersection(rastaa)\n",
    "\n",
    "print(f\"There are {len(gbkaa):,} genomes in GenBank\")\n",
    "print(f\"There are {len(rastaa):,} genomes in PATRIC\")\n",
    "print(f\"There are {len(gtdbaa):,} genomes in GTDB\")\n",
    "\n",
    "print(f\"Between GenBank and PATRIC there are {len(gbr):,} genomes in common\")\n",
    "print(f\"Between GenBank and GTDB there are {len(gbg):,} genomes in common\")\n",
    "print(f\"Between GTDB and PATRIC there are {len(gtr):,} genomes in common\")\n",
    "print()\n",
    "\n",
    "gbnotdone = gbkaa - phagesaa\n",
    "gtnotdone = gtdbaa - phagesaa\n",
    "ranotdone = rastaa - phagesaa\n",
    "print(f\"There are {len(gbnotdone):,} phages in Genbank that have not been analyzed\")\n",
    "print(f\"There are {len(gtnotdone):,} phages in GTDB that have not been analyzed\")\n",
    "print(f\"There are {len(ranotdone):,} phages in PATRIC that have not been analyzed\")\n",
    "print()\n",
    "\n",
    "allmissing = gbnotdone.intersection(gtnotdone).intersection(ranotdone)\n",
    "print(f\"There are {len(allmissing):,} phages in all three that have not been analyzed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "incorrect-following",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}