{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "empirical-crystal",
   "metadata": {},
   "source": [
    "# Merged Metadata Analysis\n",
    "\n",
    "In this analysis we are going to try and merge:\n",
    "- Prophage counts\n",
    "- GenBank data\n",
    "- RAST data\n",
    "- GTDB data\n",
    "- CheckV predictions\n",
    "\n",
    "So that we can filter by different things and identify interesting characters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "killing-degree",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from PhiSpyAnalysis import theils_u, DateConverter, file_to_accession\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "from sklearn.model_selection import train_test_split, LeaveOneGroupOut\n",
    "from sklearn import metrics\n",
    "\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "# for parsing collection dates\n",
    "from dateutil.parser import parse, ParserError\n",
    "import pytz\n",
    "\n",
    "import subprocess\n",
    "import gzip\n",
    "\n",
    "# this is a neat trick for getting markdown in our output\n",
    "# see https://stackoverflow.com/questions/23271575/printing-bold-colored-etc-text-in-ipython-qtconsole\n",
    "# for the inspiration\n",
    "from IPython.display import Markdown, display\n",
    "def printmd(string, color=\"black\"):\n",
    "    colorstr = \"<span style='color:{}'>{}</span>\".format(color, string)\n",
    "    display(Markdown(colorstr))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "mighty-clinic",
   "metadata": {},
   "source": [
    "# Read the phage data. Check the version!\n",
    "\n",
    "We have two data sets: `small` is just 99 genomes and 1,561 phages and should run quickly for development. `not small` is all the data!\n",
    "\n",
    "Here we also convert all file names to just the `accession` and name the column `assembly_accession` so we can merge everything as needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "proof-honor",
   "metadata": {},
   "outputs": [],
   "source": [
    "use_small_data = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "desperate-arabic",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1_ASM313v1_genomic.gbff.gz</td>\n",
       "      <td>2396359</td>\n",
       "      <td>114</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>48916</td>\n",
       "      <td>GCA_000003135.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003645.1_ASM364v1_genomic.gbff.gz</td>\n",
       "      <td>5269725</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>40297</td>\n",
       "      <td>GCA_000003645.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003925.1_ASM392v1_genomic.gbff.gz</td>\n",
       "      <td>5561906</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>268081</td>\n",
       "      <td>GCA_000003925.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003955.1_ASM395v1_genomic.gbff.gz</td>\n",
       "      <td>5790501</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>166286</td>\n",
       "      <td>GCA_000003955.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000005825.2_ASM582v2_genomic.gbff.gz</td>\n",
       "      <td>4249248</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>93416</td>\n",
       "      <td>GCA_000005825.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553077</th>\n",
       "      <td>GCA_902860175.1_LMG_5997_genomic.gbff.gz</td>\n",
       "      <td>7197255</td>\n",
       "      <td>38</td>\n",
       "      <td>21</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>17</td>\n",
       "      <td>69051</td>\n",
       "      <td>GCA_902860175.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553078</th>\n",
       "      <td>GCA_902860185.1_LMG_6103_genomic.gbff.gz</td>\n",
       "      <td>6497464</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>GCA_902860185.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553079</th>\n",
       "      <td>GCA_902860195.1_LMG_7053_genomic.gbff.gz</td>\n",
       "      <td>6702936</td>\n",
       "      <td>200</td>\n",
       "      <td>148</td>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>12819</td>\n",
       "      <td>GCA_902860195.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553080</th>\n",
       "      <td>GCA_902860205.1_LMG_6001_genomic.gbff.gz</td>\n",
       "      <td>6320373</td>\n",
       "      <td>36</td>\n",
       "      <td>19</td>\n",
       "      <td>35</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>12</td>\n",
       "      <td>41572</td>\n",
       "      <td>GCA_902860205.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553081</th>\n",
       "      <td>GCA_902860235.1_LMG_9964_genomic.gbff.gz</td>\n",
       "      <td>7602199</td>\n",
       "      <td>42</td>\n",
       "      <td>26</td>\n",
       "      <td>37</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>71710</td>\n",
       "      <td>GCA_902860235.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>553082 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Contig  Genome length  Contigs  \\\n",
       "0       GCA_000003135.1_ASM313v1_genomic.gbff.gz        2396359      114   \n",
       "1       GCA_000003645.1_ASM364v1_genomic.gbff.gz        5269725        1   \n",
       "2       GCA_000003925.1_ASM392v1_genomic.gbff.gz        5561906        1   \n",
       "3       GCA_000003955.1_ASM395v1_genomic.gbff.gz        5790501        1   \n",
       "4       GCA_000005825.2_ASM582v2_genomic.gbff.gz        4249248        3   \n",
       "...                                          ...            ...      ...   \n",
       "553077  GCA_902860175.1_LMG_5997_genomic.gbff.gz        7197255       38   \n",
       "553078  GCA_902860185.1_LMG_6103_genomic.gbff.gz        6497464       13   \n",
       "553079  GCA_902860195.1_LMG_7053_genomic.gbff.gz        6702936      200   \n",
       "553080  GCA_902860205.1_LMG_6001_genomic.gbff.gz        6320373       36   \n",
       "553081  GCA_902860235.1_LMG_9964_genomic.gbff.gz        7602199       42   \n",
       "\n",
       "        Phage Contigs  Total Predicted Prophages  Kept  No phage genes  \\\n",
       "0                  10                         16     2               1   \n",
       "1                   1                         31     1              10   \n",
       "2                   1                         38     6              13   \n",
       "3                   1                         46     6              11   \n",
       "4                   3                         33     3               9   \n",
       "...               ...                        ...   ...             ...   \n",
       "553077             21                         33     2              14   \n",
       "553078              8                         22     0              10   \n",
       "553079            148                         33     1              11   \n",
       "553080             19                         35     2              21   \n",
       "553081             26                         37     2              20   \n",
       "\n",
       "        Not enough genes  bp prophage assembly_accession  \n",
       "0                     13        48916    GCA_000003135.1  \n",
       "1                     20        40297    GCA_000003645.1  \n",
       "2                     19       268081    GCA_000003925.1  \n",
       "3                     29       166286    GCA_000003955.1  \n",
       "4                     21        93416    GCA_000005825.2  \n",
       "...                  ...          ...                ...  \n",
       "553077                17        69051    GCA_902860175.1  \n",
       "553078                12            0    GCA_902860185.1  \n",
       "553079                21        12819    GCA_902860195.1  \n",
       "553080                12        41572    GCA_902860205.1  \n",
       "553081                15        71710    GCA_902860235.1  \n",
       "\n",
       "[553082 rows x 10 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if use_small_data:\n",
    "    phagesdf = pd.read_csv(\"../small_data/phages_per_genome.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "else:\n",
    "    phagesdf = pd.read_csv(\"../data/phages_per_genome.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "phagesdf['assembly_accession'] = phagesdf['Contig'].apply(file_to_accession)\n",
    "phagesdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "genuine-trace",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Please note that this was run with git commit 71da118 that has 553,082 genomes parsed and 20,946,107 total prophages\n"
     ]
    }
   ],
   "source": [
    "githash = subprocess.check_output([\"git\", \"describe\", \"--always\"]).strip().decode()\n",
    "print(f\"Please note that this was run with git commit {githash} that has {phagesdf.shape[0]:,} genomes parsed and {phagesdf['Total Predicted Prophages'].sum():,} total prophages\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "improving-vehicle",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3165: DtypeWarning: Columns (2,6,7,8,9,10,11,12,14,19,21,34,40,41,42,43,44,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,65) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>mlst</th>\n",
       "      <th>...</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "      <th>isolation_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>469009.4</td>\n",
       "      <td>\"'Brassica napus' phytoplasma strain TW1\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>469009</td>\n",
       "      <td>WGS</td>\n",
       "      <td>TW1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequence of a strain of bacteria that c...</td>\n",
       "      <td>sample_type:metagenomic assembly;collected_by:...</td>\n",
       "      <td>2017.577687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1309411.5</td>\n",
       "      <td>\"'Deinococcus soli' Cha et al. 2014 strain N5\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1309411</td>\n",
       "      <td>Complete</td>\n",
       "      <td>N5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing of a Gamma-Radiation-Resista...</td>\n",
       "      <td>sample_type:bacterial</td>\n",
       "      <td>2013.260096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1123738.3</td>\n",
       "      <td>\"'Echinacea purpurea' witches'-broom phytoplas...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1123738</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NCHU2014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'Echinacea purpurea' witches'-broom phytoplasm...</td>\n",
       "      <td>lab_host:Catharanthus roseus</td>\n",
       "      <td>2014.371663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>551115.6</td>\n",
       "      <td>\"'Nostoc azollae' 0708\"</td>\n",
       "      <td>'Nostoc azollae' 0708</td>\n",
       "      <td>551115</td>\n",
       "      <td>Complete</td>\n",
       "      <td>708</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nostoc azollae 0708. Nostoc azollae 0708, also...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1856298.3</td>\n",
       "      <td>\"'Osedax' symbiont bacterium Rs2_46_30_T18 str...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1856298</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Rs2_46_30_T18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In this study, we simulate the Deepwater Horiz...</td>\n",
       "      <td>sample_type:metagenomic assembly</td>\n",
       "      <td>2013.525667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433517</th>\n",
       "      <td>1131286.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>1131286</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-137-J06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433518</th>\n",
       "      <td>1131287.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>1131287</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-C20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433519</th>\n",
       "      <td>1131288.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>1131288</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-E04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433520</th>\n",
       "      <td>1131289.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>1131289</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-B04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433521</th>\n",
       "      <td>1131290.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>1131290</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-O16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433522 rows Ã— 67 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        genome_id                                        genome_name  \\\n",
       "0        469009.4          \"'Brassica napus' phytoplasma strain TW1\"   \n",
       "1       1309411.5     \"'Deinococcus soli' Cha et al. 2014 strain N5\"   \n",
       "2       1123738.3  \"'Echinacea purpurea' witches'-broom phytoplas...   \n",
       "3        551115.6                            \"'Nostoc azollae' 0708\"   \n",
       "4       1856298.3  \"'Osedax' symbiont bacterium Rs2_46_30_T18 str...   \n",
       "...           ...                                                ...   \n",
       "433517  1131286.3               zeta proteobacterium SCGC AB-137-J06   \n",
       "433518  1131287.3               zeta proteobacterium SCGC AB-602-C20   \n",
       "433519  1131288.3               zeta proteobacterium SCGC AB-602-E04   \n",
       "433520  1131289.3               zeta proteobacterium SCGC AB-604-B04   \n",
       "433521  1131290.3               zeta proteobacterium SCGC AB-604-O16   \n",
       "\n",
       "                               organism_name  taxon_id genome_status  \\\n",
       "0                                        NaN    469009           WGS   \n",
       "1                                        NaN   1309411      Complete   \n",
       "2                                        NaN   1123738           WGS   \n",
       "3                      'Nostoc azollae' 0708    551115      Complete   \n",
       "4                                        NaN   1856298           WGS   \n",
       "...                                      ...       ...           ...   \n",
       "433517  zeta proteobacterium SCGC AB-137-J06   1131286           WGS   \n",
       "433518  zeta proteobacterium SCGC AB-602-C20   1131287           WGS   \n",
       "433519  zeta proteobacterium SCGC AB-602-E04   1131288           WGS   \n",
       "433520  zeta proteobacterium SCGC AB-604-B04   1131289           WGS   \n",
       "433521  zeta proteobacterium SCGC AB-604-O16   1131290           WGS   \n",
       "\n",
       "                 strain serovar biovar pathovar mlst  ... sporulation  \\\n",
       "0                   TW1     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "1                    N5     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "2              NCHU2014     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "3                   708     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "4         Rs2_46_30_T18     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "...                 ...     ...    ...      ...  ...  ...         ...   \n",
       "433517  SCGC AB-137-J06     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "433518  SCGC AB-602-C20     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "433519  SCGC AB-602-E04     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "433520  SCGC AB-604-B04     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "433521  SCGC AB-604-O16     NaN    NaN      NaN  NaN  ...         NaN   \n",
       "\n",
       "       temperature_range optimal_temperature salinity oxygen_requirement  \\\n",
       "0                    NaN                 NaN      NaN                NaN   \n",
       "1                    NaN                 NaN      NaN                NaN   \n",
       "2                    NaN                   C      NaN                NaN   \n",
       "3             Mesophilic                   -      NaN            Aerobic   \n",
       "4                    NaN                 NaN      NaN                NaN   \n",
       "...                  ...                 ...      ...                ...   \n",
       "433517               NaN                 NaN      NaN                NaN   \n",
       "433518               NaN                 NaN      NaN                NaN   \n",
       "433519               NaN                 NaN      NaN                NaN   \n",
       "433520               NaN                 NaN      NaN                NaN   \n",
       "433521               NaN                 NaN      NaN                NaN   \n",
       "\n",
       "         habitat disease                                           comments  \\\n",
       "0            NaN     NaN  Genome sequence of a strain of bacteria that c...   \n",
       "1            NaN     NaN  Genome sequencing of a Gamma-Radiation-Resista...   \n",
       "2            NaN     NaN  'Echinacea purpurea' witches'-broom phytoplasm...   \n",
       "3       Multiple     NaN  Nostoc azollae 0708. Nostoc azollae 0708, also...   \n",
       "4            NaN     NaN  In this study, we simulate the Deepwater Horiz...   \n",
       "...          ...     ...                                                ...   \n",
       "433517       NaN     NaN  Single cell genome sequencing of biomineralizi...   \n",
       "433518       NaN     NaN  Single cell genome sequencing of biomineralizi...   \n",
       "433519       NaN     NaN  Single cell genome sequencing of biomineralizi...   \n",
       "433520       NaN     NaN  Single cell genome sequencing of biomineralizi...   \n",
       "433521       NaN     NaN  Single cell genome sequencing of biomineralizi...   \n",
       "\n",
       "                                      additional_metadata isolation_date  \n",
       "0       sample_type:metagenomic assembly;collected_by:...    2017.577687  \n",
       "1                                   sample_type:bacterial    2013.260096  \n",
       "2                            lab_host:Catharanthus roseus    2014.371663  \n",
       "3                                                     NaN            NaN  \n",
       "4                        sample_type:metagenomic assembly    2013.525667  \n",
       "...                                                   ...            ...  \n",
       "433517                                                NaN            NaN  \n",
       "433518                                                NaN            NaN  \n",
       "433519                                                NaN            NaN  \n",
       "433520                                                NaN            NaN  \n",
       "433521                                                NaN            NaN  \n",
       "\n",
       "[433522 rows x 67 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if use_small_data:\n",
    "    metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "else:\n",
    "    metadf = pd.read_csv(\"../data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "dc = DateConverter()\n",
    "metadf['isolation_date'] = metadf.collection_date.apply(dc.convert_date)\n",
    "metadf"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
