{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8c15ee07",
   "metadata": {},
   "source": [
    "# Phages from Australia\n",
    "\n",
    "What are the unique and wonderful phage from Australia that we should be studying?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "155db554",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A lot of this is not used, but we import it so we have it later!\n",
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from PhiSpyAnalysis import theils_u, DateConverter, printmd\n",
    "from PhiSpyAnalysis import read_phages, read_gtdb, read_checkv, read_base_pp, read_categories, read_metadata, read_gbk_metadata\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "\n",
    "\n",
    "import subprocess\n",
    "import gzip\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "40b378b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "rv = re.compile('^\\\\w+')\n",
    "def remove_ver(x):\n",
    "    return rv.search(str(x)).group()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "501c9f2a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000006155.2</td>\n",
       "      <td>GCA_000006155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000007385.1</td>\n",
       "      <td>GCA_000007385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000008605.1</td>\n",
       "      <td>GCA_000008605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000010565.1</td>\n",
       "      <td>GCA_000010565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000013845.2</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191522</th>\n",
       "      <td>GCA_902166935.1</td>\n",
       "      <td>GCA_902166935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191523</th>\n",
       "      <td>GCA_902166945.1</td>\n",
       "      <td>GCA_902166945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191524</th>\n",
       "      <td>GCA_902167295.1</td>\n",
       "      <td>GCA_902167295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191525</th>\n",
       "      <td>GCA_902167305.1</td>\n",
       "      <td>GCA_902167305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191526</th>\n",
       "      <td>GCA_902167325.1</td>\n",
       "      <td>GCA_902167325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191527 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000006155.2  GCA_000006155\n",
       "1         GCA_000007385.1  GCA_000007385\n",
       "2         GCA_000008605.1  GCA_000008605\n",
       "3         GCA_000010565.1  GCA_000010565\n",
       "4         GCA_000013845.2  GCA_000013845\n",
       "...                   ...            ...\n",
       "191522    GCA_902166935.1  GCA_902166935\n",
       "191523    GCA_902166945.1  GCA_902166945\n",
       "191524    GCA_902167295.1  GCA_902167295\n",
       "191525    GCA_902167305.1  GCA_902167305\n",
       "191526    GCA_902167325.1  GCA_902167325\n",
       "\n",
       "[191527 rows x 2 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GTDB\n",
    "gtdb = read_gtdb()\n",
    "gtdb['assembly_nover'] = gtdb['assembly_accession'].apply(remove_ver)\n",
    "gtdb[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2268410f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>GCA_000003135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003215.1</td>\n",
       "      <td>GCA_000003215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>GCA_000003645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>GCA_000003925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>GCA_000003955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320171</th>\n",
       "      <td>GCF_900167595.1</td>\n",
       "      <td>GCF_900167595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320172</th>\n",
       "      <td>GCF_900167605.1</td>\n",
       "      <td>GCF_900167605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320173</th>\n",
       "      <td>GCF_900167615.1</td>\n",
       "      <td>GCF_900167615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320174</th>\n",
       "      <td>GCF_900167625.1</td>\n",
       "      <td>GCF_900167625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320175</th>\n",
       "      <td>GCF_900167635.1</td>\n",
       "      <td>GCF_900167635</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320176 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_000003135.1  GCA_000003135\n",
       "1         GCA_000003215.1  GCA_000003215\n",
       "2         GCA_000003645.1  GCA_000003645\n",
       "3         GCA_000003925.1  GCA_000003925\n",
       "4         GCA_000003955.1  GCA_000003955\n",
       "...                   ...            ...\n",
       "320171    GCF_900167595.1  GCF_900167595\n",
       "320172    GCF_900167605.1  GCF_900167605\n",
       "320173    GCF_900167615.1  GCF_900167615\n",
       "320174    GCF_900167625.1  GCF_900167625\n",
       "320175    GCF_900167635.1  GCF_900167635\n",
       "\n",
       "[320176 rows x 2 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# RAST\n",
    "# the full data set. Don't try this at home!\n",
    "# metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "rast = read_metadata()\n",
    "rast['assembly_nover'] = rast['assembly_accession'].apply(remove_ver)\n",
    "rast[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d990e411",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3361: DtypeWarning: Columns (21) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>GCA_900128725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>GCA_008244535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_011046815.1</td>\n",
       "      <td>GCA_011046815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_011054035.1</td>\n",
       "      <td>GCA_011054035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_011331185.1</td>\n",
       "      <td>GCA_011331185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949928</th>\n",
       "      <td>GCA_017746755.1</td>\n",
       "      <td>GCA_017746755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949929</th>\n",
       "      <td>GCA_017746715.1</td>\n",
       "      <td>GCA_017746715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949930</th>\n",
       "      <td>GCA_013266695.1</td>\n",
       "      <td>GCA_013266695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949931</th>\n",
       "      <td>GCA_015831295.1</td>\n",
       "      <td>GCA_015831295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949932</th>\n",
       "      <td>GCA_015686645.1</td>\n",
       "      <td>GCA_015686645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>949933 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession assembly_nover\n",
       "0         GCA_900128725.1  GCA_900128725\n",
       "1         GCA_008244535.1  GCA_008244535\n",
       "2         GCA_011046815.1  GCA_011046815\n",
       "3         GCA_011054035.1  GCA_011054035\n",
       "4         GCA_011331185.1  GCA_011331185\n",
       "...                   ...            ...\n",
       "949928    GCA_017746755.1  GCA_017746755\n",
       "949929    GCA_017746715.1  GCA_017746715\n",
       "949930    GCA_013266695.1  GCA_013266695\n",
       "949931    GCA_015831295.1  GCA_015831295\n",
       "949932    GCA_015686645.1  GCA_015686645\n",
       "\n",
       "[949933 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# GenBank\n",
    "gbk = read_gbk_metadata()\n",
    "gbk['assembly_nover'] = gbk['assembly_accession'].apply(remove_ver)\n",
    "gbk[['assembly_accession', 'assembly_nover']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e0cdc003",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Please note that this was run with git commit e73598a that has 567,404 genomes parsed.\n",
      "Initially there were 3,265,453 kept phages, but after filtering we kept 3,265,453 prophages from 567,399 genomes"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000043285.1</td>\n",
       "      <td>ASM4328v1</td>\n",
       "      <td>705557.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000043285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000046685.1</td>\n",
       "      <td>ASM4668v1</td>\n",
       "      <td>1581384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>97011.0</td>\n",
       "      <td>GCA_000046685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000046705.1</td>\n",
       "      <td>ASM4670v1</td>\n",
       "      <td>1931047.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>278490.0</td>\n",
       "      <td>GCA_000046705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000046845.1</td>\n",
       "      <td>ASM4684v1</td>\n",
       "      <td>3598621.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>83112.0</td>\n",
       "      <td>GCA_000046845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000047365.1</td>\n",
       "      <td>ASM4736v1</td>\n",
       "      <td>4840898.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>213427.0</td>\n",
       "      <td>GCA_000047365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567399</th>\n",
       "      <td>GCA_905187425.1</td>\n",
       "      <td>Xanthomonas_sp._CPBF_424_-_hybrid_assembly</td>\n",
       "      <td>4900930.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>214278.0</td>\n",
       "      <td>GCA_905187425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567400</th>\n",
       "      <td>GCA_905188235.1</td>\n",
       "      <td>ASM90518823v1</td>\n",
       "      <td>4963609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>178993.0</td>\n",
       "      <td>GCA_905188235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567401</th>\n",
       "      <td>GCA_905219375.1</td>\n",
       "      <td>QI0054</td>\n",
       "      <td>3034314.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>178527.0</td>\n",
       "      <td>GCA_905219375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567402</th>\n",
       "      <td>GCA_905219385.1</td>\n",
       "      <td>QI0055</td>\n",
       "      <td>3034113.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>178526.0</td>\n",
       "      <td>GCA_905219385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567403</th>\n",
       "      <td>GCA_905220645.1</td>\n",
       "      <td>NMYAN.1</td>\n",
       "      <td>3400020.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>65454.0</td>\n",
       "      <td>GCA_905220645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>567399 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession                               assembly_name  \\\n",
       "0         GCA_000043285.1                                   ASM4328v1   \n",
       "1         GCA_000046685.1                                   ASM4668v1   \n",
       "2         GCA_000046705.1                                   ASM4670v1   \n",
       "3         GCA_000046845.1                                   ASM4684v1   \n",
       "4         GCA_000047365.1                                   ASM4736v1   \n",
       "...                   ...                                         ...   \n",
       "567399    GCA_905187425.1  Xanthomonas_sp._CPBF_424_-_hybrid_assembly   \n",
       "567400    GCA_905188235.1                               ASM90518823v1   \n",
       "567401    GCA_905219375.1                                      QI0054   \n",
       "567402    GCA_905219385.1                                      QI0055   \n",
       "567403    GCA_905220645.1                                     NMYAN.1   \n",
       "\n",
       "        Genome length  Contigs  Phage Contigs  Total Predicted Prophages  \\\n",
       "0            705557.0      1.0            1.0                        2.0   \n",
       "1           1581384.0      1.0            1.0                       18.0   \n",
       "2           1931047.0      1.0            1.0                       36.0   \n",
       "3           3598621.0      1.0            1.0                       13.0   \n",
       "4           4840898.0      3.0            3.0                       22.0   \n",
       "...               ...      ...            ...                        ...   \n",
       "567399      4900930.0      1.0            1.0                       46.0   \n",
       "567400      4963609.0      1.0            1.0                       29.0   \n",
       "567401      3034314.0      1.0            1.0                       12.0   \n",
       "567402      3034113.0      1.0            1.0                       12.0   \n",
       "567403      3400020.0     55.0           33.0                       31.0   \n",
       "\n",
       "        Kept  No phage genes  Not enough genes  bp prophage assembly_nover  \n",
       "0        0.0             2.0               0.0          0.0  GCA_000043285  \n",
       "1        3.0             0.0              15.0      97011.0  GCA_000046685  \n",
       "2        7.0             0.0              29.0     278490.0  GCA_000046705  \n",
       "3        3.0             4.0               6.0      83112.0  GCA_000046845  \n",
       "4        3.0             3.0              16.0     213427.0  GCA_000047365  \n",
       "...      ...             ...               ...          ...            ...  \n",
       "567399   7.0             9.0              30.0     214278.0  GCA_905187425  \n",
       "567400   4.0             4.0              21.0     178993.0  GCA_905188235  \n",
       "567401   4.0             3.0               5.0     178527.0  GCA_905219375  \n",
       "567402   4.0             3.0               5.0     178526.0  GCA_905219385  \n",
       "567403   4.0             5.0              22.0      65454.0  GCA_905220645  \n",
       "\n",
       "[567399 rows x 11 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagesdf = read_phages(maxcontigs=-1) # this disables contig length filtering\n",
    "phagesdf['assembly_nover'] = phagesdf['assembly_accession'].apply(remove_ver)\n",
    "phagesdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f75b6ba7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['assembly_accession', 'genome_id', 'genome_name', 'organism_name',\n",
       "       'taxon_id', 'genome_status', 'strain', 'serovar', 'biovar', 'pathovar',\n",
       "       'mlst', 'other_typing', 'culture_collection', 'type_strain',\n",
       "       'completion_date', 'publication', 'bioproject_accession',\n",
       "       'biosample_accession', 'genbank_accessions', 'refseq_accessions',\n",
       "       'sequencing_centers', 'sequencing_status', 'sequencing_platform',\n",
       "       'sequencing_depth', 'assembly_method', 'chromosomes', 'plasmids',\n",
       "       'contigs', 'sequences', 'genome_length', 'gc_content', 'patric_cds',\n",
       "       'brc1_cds', 'refseq_cds', 'isolation_site', 'isolation_source',\n",
       "       'isolation_comments', 'collection_date', 'isolation_country',\n",
       "       'geographic_location', 'latitude', 'longitude', 'altitude', 'depth',\n",
       "       'other_environmental', 'host_name', 'host_gender', 'host_age',\n",
       "       'host_health', 'body_sample_site', 'body_sample_subsite',\n",
       "       'other_clinical', 'antimicrobial_resistance',\n",
       "       'antimicrobial_resistance_evidence', 'gram_stain', 'cell_shape',\n",
       "       'motility', 'sporulation', 'temperature_range', 'optimal_temperature',\n",
       "       'salinity', 'oxygen_requirement', 'habitat', 'disease', 'comments',\n",
       "       'additional_metadata', 'isolation_date', 'assembly_nover'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rast.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "868abeff",
   "metadata": {},
   "source": [
    "# Kangaroos and Koalas\n",
    "\n",
    "Here, we limit the data to bacteria from kangaroos and koalas to see if there are some interesting things. \n",
    "\n",
    "Unfortunately, it seems they are mostly Proteobacteria (yawn) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "dd56a90b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>...</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "      <th>isolation_date</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>GCA_000017105.1</td>\n",
       "      <td>4.349225e+05</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111</td>\n",
       "      <td>434922</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Dugway 5J108-111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Facultative</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Q fever</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111. Coxiella b...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000017105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8863</th>\n",
       "      <td>GCA_000341355.1</td>\n",
       "      <td>1.094489e+06</td>\n",
       "      <td>Bartonella australis Aust/NH1</td>\n",
       "      <td>Bartonella australis Aust/NH1</td>\n",
       "      <td>1094489</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Aust/NH1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>HostAssociated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Whole-genome sequencing of Bartonella australi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1999.683094</td>\n",
       "      <td>GCA_000341355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24579</th>\n",
       "      <td>GCA_000689835.1</td>\n",
       "      <td>1.331229e+06</td>\n",
       "      <td>Bordetella bronchiseptica MBORD681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1331229</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MBORD681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In order to understand the speciation and evol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000689835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24580</th>\n",
       "      <td>GCA_000689855.1</td>\n",
       "      <td>1.331230e+06</td>\n",
       "      <td>Bordetella bronchiseptica MBORD698</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1331230</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MBORD698</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In order to understand the speciation and evol...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000689855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37899</th>\n",
       "      <td>GCA_001689645.1</td>\n",
       "      <td>1.807743e+06</td>\n",
       "      <td>Bacteroidales bacterium K1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1807743</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The Bacteroidales group S24-7 is recognized as...</td>\n",
       "      <td>sample_type:metagenomic assembly</td>\n",
       "      <td>2015.017796</td>\n",
       "      <td>GCA_001689645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37900</th>\n",
       "      <td>GCA_001689655.1</td>\n",
       "      <td>1.807744e+06</td>\n",
       "      <td>Bacteroidales bacterium K10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1807744</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The Bacteroidales group S24-7 is recognized as...</td>\n",
       "      <td>sample_type:metagenomic assembly</td>\n",
       "      <td>2015.017796</td>\n",
       "      <td>GCA_001689655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38407</th>\n",
       "      <td>GCA_001729965.1</td>\n",
       "      <td>1.561964e+06</td>\n",
       "      <td>Methanosphaera sp. WGK6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1561964</td>\n",
       "      <td>WGS</td>\n",
       "      <td>WGK6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aim to isolate and sequence methanogens origin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008.685147</td>\n",
       "      <td>GCA_001729965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42914</th>\n",
       "      <td>GCA_001941025.1</td>\n",
       "      <td>1.930016e+06</td>\n",
       "      <td>Lachnospiraceae bacterium Zagget3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930016</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42915</th>\n",
       "      <td>GCA_001941045.1</td>\n",
       "      <td>1.930015e+06</td>\n",
       "      <td>Lachnospiraceae bacterium Zagget2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930015</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42916</th>\n",
       "      <td>GCA_001941055.1</td>\n",
       "      <td>1.930001e+06</td>\n",
       "      <td>Ruminococcus sp. Zagget7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930001</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42917</th>\n",
       "      <td>GCA_001941065.1</td>\n",
       "      <td>1.930002e+06</td>\n",
       "      <td>Alistipes sp. Zagget8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930002</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42918</th>\n",
       "      <td>GCA_001941075.1</td>\n",
       "      <td>1.930019e+06</td>\n",
       "      <td>Oscillospiraceae bacterium Zagget6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930019</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42919</th>\n",
       "      <td>GCA_001941125.1</td>\n",
       "      <td>1.930017e+06</td>\n",
       "      <td>Lachnospiraceae bacterium Zagget13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930017</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42920</th>\n",
       "      <td>GCA_001941135.1</td>\n",
       "      <td>1.930004e+06</td>\n",
       "      <td>Ruminococcus sp. Zagget11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930004</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42921</th>\n",
       "      <td>GCA_001941165.1</td>\n",
       "      <td>1.930018e+06</td>\n",
       "      <td>Oscillospiraceae bacterium Zagget4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930018</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42922</th>\n",
       "      <td>GCA_001941185.1</td>\n",
       "      <td>1.930021e+06</td>\n",
       "      <td>Rhodocyclaceae bacterium Zagget5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930021</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42923</th>\n",
       "      <td>GCA_001941205.1</td>\n",
       "      <td>1.930003e+06</td>\n",
       "      <td>Synergistes sp. Zagget9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930003</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42924</th>\n",
       "      <td>GCA_001941225.1</td>\n",
       "      <td>1.930020e+06</td>\n",
       "      <td>Oscillospiraceae bacterium Zagget12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930020</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42925</th>\n",
       "      <td>GCA_001941235.1</td>\n",
       "      <td>1.930005e+06</td>\n",
       "      <td>Desulfovibrio sp. Zagget14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930005</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_001941235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46707</th>\n",
       "      <td>GCA_002097375.1</td>\n",
       "      <td>1.280123e+03</td>\n",
       "      <td>Staphylococcus aureus strain N13/1/662</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1280</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MRSA isolates were collected from Australian a...</td>\n",
       "      <td>collected_by:University of Adelaide</td>\n",
       "      <td>2013.684463</td>\n",
       "      <td>GCA_002097375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47561</th>\n",
       "      <td>GCA_002117465.1</td>\n",
       "      <td>1.396189e+03</td>\n",
       "      <td>Bacillus cereus strain BC-AK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1396</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>BC-AK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>It's a novel Bacillus anthracis-like strain.</td>\n",
       "      <td>sample_type:tissue sample</td>\n",
       "      <td>2016.548255</td>\n",
       "      <td>GCA_002117465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50807</th>\n",
       "      <td>GCA_002190685.1</td>\n",
       "      <td>5.621760e+02</td>\n",
       "      <td>Escherichia coli strain ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>562</td>\n",
       "      <td>WGS</td>\n",
       "      <td>ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Investigation of Horizontal Gene Transfer of P...</td>\n",
       "      <td>sample_type:whole organism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_002190685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71669</th>\n",
       "      <td>GCA_002633115.1</td>\n",
       "      <td>1.958821e+06</td>\n",
       "      <td>Muribaculaceae bacterium Zag1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1958821</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zag1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_002633115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71671</th>\n",
       "      <td>GCA_002633155.1</td>\n",
       "      <td>1.958823e+06</td>\n",
       "      <td>Candidatus Melainabacteria bacterium Zag15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1958823</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zag15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_002633155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71679</th>\n",
       "      <td>GCA_002633305.1</td>\n",
       "      <td>1.958822e+06</td>\n",
       "      <td>Muribaculaceae bacterium Zag10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1958822</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zag10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The study goal was to identify the top 15 popu...</td>\n",
       "      <td>sample_type:Population genome extracted from m...</td>\n",
       "      <td>2011.316222</td>\n",
       "      <td>GCA_002633305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99046</th>\n",
       "      <td>GCA_003334005.1</td>\n",
       "      <td>5.623333e+02</td>\n",
       "      <td>Escherichia coli strain MOD1-ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>562</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MOD1-ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Whole genome sequencing of cultured E. coli as...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_003334005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130047</th>\n",
       "      <td>GCA_004155475.1</td>\n",
       "      <td>1.690360e+03</td>\n",
       "      <td>Bifidobacterium pseudolongum subsp. globosum s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1690</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2109B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing and assembly of strains belo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_004155475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130055</th>\n",
       "      <td>GCA_004155625.1</td>\n",
       "      <td>1.690450e+03</td>\n",
       "      <td>Bifidobacterium pseudolongum subsp. globosum s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1690</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2009B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing and assembly of strains belo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_004155625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130180</th>\n",
       "      <td>GCA_004168465.1</td>\n",
       "      <td>3.572332e+05</td>\n",
       "      <td>Siccibacter turicensis strain #493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>357233</td>\n",
       "      <td>WGS</td>\n",
       "      <td>#493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The goal of this project was to find the kanga...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018.180014</td>\n",
       "      <td>GCA_004168465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284627</th>\n",
       "      <td>GCF_001211245.1</td>\n",
       "      <td>2.845022e+04</td>\n",
       "      <td>Burkholderia pseudomallei strain 561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28450</td>\n",
       "      <td>WGS</td>\n",
       "      <td>561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"Microevolution of Burkholderia pseudomallei d...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1996.685147</td>\n",
       "      <td>GCF_001211245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310046</th>\n",
       "      <td>GCF_002000985.1</td>\n",
       "      <td>1.258574e+06</td>\n",
       "      <td>Streptococcus gallolyticus subsp. gallolyticus...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1258574</td>\n",
       "      <td>Complete</td>\n",
       "      <td>DSM 16831</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Streptococcus gallolyticus subsp. gallolyticus...</td>\n",
       "      <td>sample_type:Bacteria;biomaterial_provider:DSMZ...</td>\n",
       "      <td>1992.685147</td>\n",
       "      <td>GCF_002000985</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession     genome_id  \\\n",
       "528       GCA_000017105.1  4.349225e+05   \n",
       "8863      GCA_000341355.1  1.094489e+06   \n",
       "24579     GCA_000689835.1  1.331229e+06   \n",
       "24580     GCA_000689855.1  1.331230e+06   \n",
       "37899     GCA_001689645.1  1.807743e+06   \n",
       "37900     GCA_001689655.1  1.807744e+06   \n",
       "38407     GCA_001729965.1  1.561964e+06   \n",
       "42914     GCA_001941025.1  1.930016e+06   \n",
       "42915     GCA_001941045.1  1.930015e+06   \n",
       "42916     GCA_001941055.1  1.930001e+06   \n",
       "42917     GCA_001941065.1  1.930002e+06   \n",
       "42918     GCA_001941075.1  1.930019e+06   \n",
       "42919     GCA_001941125.1  1.930017e+06   \n",
       "42920     GCA_001941135.1  1.930004e+06   \n",
       "42921     GCA_001941165.1  1.930018e+06   \n",
       "42922     GCA_001941185.1  1.930021e+06   \n",
       "42923     GCA_001941205.1  1.930003e+06   \n",
       "42924     GCA_001941225.1  1.930020e+06   \n",
       "42925     GCA_001941235.1  1.930005e+06   \n",
       "46707     GCA_002097375.1  1.280123e+03   \n",
       "47561     GCA_002117465.1  1.396189e+03   \n",
       "50807     GCA_002190685.1  5.621760e+02   \n",
       "71669     GCA_002633115.1  1.958821e+06   \n",
       "71671     GCA_002633155.1  1.958823e+06   \n",
       "71679     GCA_002633305.1  1.958822e+06   \n",
       "99046     GCA_003334005.1  5.623333e+02   \n",
       "130047    GCA_004155475.1  1.690360e+03   \n",
       "130055    GCA_004155625.1  1.690450e+03   \n",
       "130180    GCA_004168465.1  3.572332e+05   \n",
       "284627    GCF_001211245.1  2.845022e+04   \n",
       "310046    GCF_002000985.1  1.258574e+06   \n",
       "\n",
       "                                              genome_name  \\\n",
       "528                    Coxiella burnetii Dugway 5J108-111   \n",
       "8863                        Bartonella australis Aust/NH1   \n",
       "24579                  Bordetella bronchiseptica MBORD681   \n",
       "24580                  Bordetella bronchiseptica MBORD698   \n",
       "37899                          Bacteroidales bacterium K1   \n",
       "37900                         Bacteroidales bacterium K10   \n",
       "38407                             Methanosphaera sp. WGK6   \n",
       "42914                   Lachnospiraceae bacterium Zagget3   \n",
       "42915                   Lachnospiraceae bacterium Zagget2   \n",
       "42916                            Ruminococcus sp. Zagget7   \n",
       "42917                               Alistipes sp. Zagget8   \n",
       "42918                  Oscillospiraceae bacterium Zagget6   \n",
       "42919                  Lachnospiraceae bacterium Zagget13   \n",
       "42920                           Ruminococcus sp. Zagget11   \n",
       "42921                  Oscillospiraceae bacterium Zagget4   \n",
       "42922                    Rhodocyclaceae bacterium Zagget5   \n",
       "42923                             Synergistes sp. Zagget9   \n",
       "42924                 Oscillospiraceae bacterium Zagget12   \n",
       "42925                          Desulfovibrio sp. Zagget14   \n",
       "46707              Staphylococcus aureus strain N13/1/662   \n",
       "47561                        Bacillus cereus strain BC-AK   \n",
       "50807                      Escherichia coli strain ECOR29   \n",
       "71669                       Muribaculaceae bacterium Zag1   \n",
       "71671          Candidatus Melainabacteria bacterium Zag15   \n",
       "71679                      Muribaculaceae bacterium Zag10   \n",
       "99046                 Escherichia coli strain MOD1-ECOR29   \n",
       "130047  Bifidobacterium pseudolongum subsp. globosum s...   \n",
       "130055  Bifidobacterium pseudolongum subsp. globosum s...   \n",
       "130180                 Siccibacter turicensis strain #493   \n",
       "284627               Burkholderia pseudomallei strain 561   \n",
       "310046  Streptococcus gallolyticus subsp. gallolyticus...   \n",
       "\n",
       "                             organism_name  taxon_id genome_status  \\\n",
       "528     Coxiella burnetii Dugway 5J108-111    434922      Complete   \n",
       "8863         Bartonella australis Aust/NH1   1094489      Complete   \n",
       "24579                                  NaN   1331229           WGS   \n",
       "24580                                  NaN   1331230           WGS   \n",
       "37899                                  NaN   1807743           WGS   \n",
       "37900                                  NaN   1807744           WGS   \n",
       "38407                                  NaN   1561964           WGS   \n",
       "42914                                  NaN   1930016           WGS   \n",
       "42915                                  NaN   1930015           WGS   \n",
       "42916                                  NaN   1930001           WGS   \n",
       "42917                                  NaN   1930002           WGS   \n",
       "42918                                  NaN   1930019           WGS   \n",
       "42919                                  NaN   1930017           WGS   \n",
       "42920                                  NaN   1930004           WGS   \n",
       "42921                                  NaN   1930018           WGS   \n",
       "42922                                  NaN   1930021           WGS   \n",
       "42923                                  NaN   1930003           WGS   \n",
       "42924                                  NaN   1930020           WGS   \n",
       "42925                                  NaN   1930005           WGS   \n",
       "46707                                  NaN      1280           WGS   \n",
       "47561                                  NaN      1396       Plasmid   \n",
       "50807                                  NaN       562           WGS   \n",
       "71669                                  NaN   1958821           WGS   \n",
       "71671                                  NaN   1958823           WGS   \n",
       "71679                                  NaN   1958822           WGS   \n",
       "99046                                  NaN       562           WGS   \n",
       "130047                                 NaN      1690           WGS   \n",
       "130055                                 NaN      1690           WGS   \n",
       "130180                                 NaN    357233           WGS   \n",
       "284627                                 NaN     28450           WGS   \n",
       "310046                                 NaN   1258574      Complete   \n",
       "\n",
       "                  strain serovar biovar pathovar  ... temperature_range  \\\n",
       "528     Dugway 5J108-111     NaN    NaN      NaN  ...        Mesophilic   \n",
       "8863            Aust/NH1     NaN    NaN      NaN  ...           Unknown   \n",
       "24579           MBORD681     NaN    NaN      NaN  ...               NaN   \n",
       "24580           MBORD698     NaN    NaN      NaN  ...               NaN   \n",
       "37899                NaN     NaN    NaN      NaN  ...               NaN   \n",
       "37900                NaN     NaN    NaN      NaN  ...               NaN   \n",
       "38407               WGK6     NaN    NaN      NaN  ...               NaN   \n",
       "42914            Zagget3     NaN    NaN      NaN  ...               NaN   \n",
       "42915            Zagget2     NaN    NaN      NaN  ...               NaN   \n",
       "42916            Zagget7     NaN    NaN      NaN  ...               NaN   \n",
       "42917            Zagget8     NaN    NaN      NaN  ...               NaN   \n",
       "42918            Zagget6     NaN    NaN      NaN  ...               NaN   \n",
       "42919           Zagget13     NaN    NaN      NaN  ...               NaN   \n",
       "42920           Zagget11     NaN    NaN      NaN  ...               NaN   \n",
       "42921            Zagget4     NaN    NaN      NaN  ...               NaN   \n",
       "42922            Zagget5     NaN    NaN      NaN  ...               NaN   \n",
       "42923            Zagget9     NaN    NaN      NaN  ...               NaN   \n",
       "42924           Zagget12     NaN    NaN      NaN  ...               NaN   \n",
       "42925           Zagget14     NaN    NaN      NaN  ...               NaN   \n",
       "46707                NaN     NaN    NaN      NaN  ...               NaN   \n",
       "47561              BC-AK     NaN    NaN      NaN  ...               NaN   \n",
       "50807             ECOR29     NaN    NaN      NaN  ...               NaN   \n",
       "71669               Zag1     NaN    NaN      NaN  ...               NaN   \n",
       "71671              Zag15     NaN    NaN      NaN  ...               NaN   \n",
       "71679              Zag10     NaN    NaN      NaN  ...               NaN   \n",
       "99046        MOD1-ECOR29     NaN    NaN      NaN  ...               NaN   \n",
       "130047             2109B     NaN    NaN      NaN  ...               NaN   \n",
       "130055             2009B     NaN    NaN      NaN  ...               NaN   \n",
       "130180              #493     NaN    NaN      NaN  ...               NaN   \n",
       "284627               561     NaN    NaN      NaN  ...               NaN   \n",
       "310046         DSM 16831     NaN    NaN      NaN  ...               NaN   \n",
       "\n",
       "       optimal_temperature salinity oxygen_requirement          habitat  \\\n",
       "528                     37      NaN        Facultative  Host-associated   \n",
       "8863                   NaN  Unknown            Unknown   HostAssociated   \n",
       "24579                  NaN      NaN                NaN              NaN   \n",
       "24580                  NaN      NaN                NaN              NaN   \n",
       "37899                  NaN      NaN                NaN              NaN   \n",
       "37900                  NaN      NaN                NaN              NaN   \n",
       "38407                  NaN      NaN                NaN              NaN   \n",
       "42914                  NaN      NaN                NaN              NaN   \n",
       "42915                  NaN      NaN                NaN              NaN   \n",
       "42916                  NaN      NaN                NaN              NaN   \n",
       "42917                  NaN      NaN                NaN              NaN   \n",
       "42918                  NaN      NaN                NaN              NaN   \n",
       "42919                  NaN      NaN                NaN              NaN   \n",
       "42920                  NaN      NaN                NaN              NaN   \n",
       "42921                  NaN      NaN                NaN              NaN   \n",
       "42922                  NaN      NaN                NaN              NaN   \n",
       "42923                  NaN      NaN                NaN              NaN   \n",
       "42924                  NaN      NaN                NaN              NaN   \n",
       "42925                  NaN      NaN                NaN              NaN   \n",
       "46707                  NaN      NaN                NaN              NaN   \n",
       "47561                  NaN      NaN                NaN              NaN   \n",
       "50807                  NaN      NaN                NaN              NaN   \n",
       "71669                  NaN      NaN                NaN              NaN   \n",
       "71671                  NaN      NaN                NaN              NaN   \n",
       "71679                  NaN      NaN                NaN              NaN   \n",
       "99046                  NaN      NaN                NaN              NaN   \n",
       "130047                 NaN      NaN                NaN              NaN   \n",
       "130055                 NaN      NaN                NaN              NaN   \n",
       "130180                 NaN      NaN                NaN              NaN   \n",
       "284627                 NaN      NaN                NaN              NaN   \n",
       "310046                 NaN      NaN                NaN              NaN   \n",
       "\n",
       "        disease                                           comments  \\\n",
       "528     Q fever  Coxiella burnetii Dugway 5J108-111. Coxiella b...   \n",
       "8863        NaN  Whole-genome sequencing of Bartonella australi...   \n",
       "24579       NaN  In order to understand the speciation and evol...   \n",
       "24580       NaN  In order to understand the speciation and evol...   \n",
       "37899       NaN  The Bacteroidales group S24-7 is recognized as...   \n",
       "37900       NaN  The Bacteroidales group S24-7 is recognized as...   \n",
       "38407       NaN  Aim to isolate and sequence methanogens origin...   \n",
       "42914       NaN  The study goal was to identify the top 15 popu...   \n",
       "42915       NaN  The study goal was to identify the top 15 popu...   \n",
       "42916       NaN  The study goal was to identify the top 15 popu...   \n",
       "42917       NaN  The study goal was to identify the top 15 popu...   \n",
       "42918       NaN  The study goal was to identify the top 15 popu...   \n",
       "42919       NaN  The study goal was to identify the top 15 popu...   \n",
       "42920       NaN  The study goal was to identify the top 15 popu...   \n",
       "42921       NaN  The study goal was to identify the top 15 popu...   \n",
       "42922       NaN  The study goal was to identify the top 15 popu...   \n",
       "42923       NaN  The study goal was to identify the top 15 popu...   \n",
       "42924       NaN  The study goal was to identify the top 15 popu...   \n",
       "42925       NaN  The study goal was to identify the top 15 popu...   \n",
       "46707       NaN  MRSA isolates were collected from Australian a...   \n",
       "47561       NaN       It's a novel Bacillus anthracis-like strain.   \n",
       "50807       NaN  Investigation of Horizontal Gene Transfer of P...   \n",
       "71669       NaN  The study goal was to identify the top 15 popu...   \n",
       "71671       NaN  The study goal was to identify the top 15 popu...   \n",
       "71679       NaN  The study goal was to identify the top 15 popu...   \n",
       "99046       NaN  Whole genome sequencing of cultured E. coli as...   \n",
       "130047      NaN  Genome sequencing and assembly of strains belo...   \n",
       "130055      NaN  Genome sequencing and assembly of strains belo...   \n",
       "130180      NaN  The goal of this project was to find the kanga...   \n",
       "284627      NaN  \"Microevolution of Burkholderia pseudomallei d...   \n",
       "310046      NaN  Streptococcus gallolyticus subsp. gallolyticus...   \n",
       "\n",
       "                                      additional_metadata isolation_date  \\\n",
       "528                                                   NaN            NaN   \n",
       "8863                                                  NaN    1999.683094   \n",
       "24579                                                 NaN            NaN   \n",
       "24580                                                 NaN            NaN   \n",
       "37899                    sample_type:metagenomic assembly    2015.017796   \n",
       "37900                    sample_type:metagenomic assembly    2015.017796   \n",
       "38407                                                 NaN    2008.685147   \n",
       "42914   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42915   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42916   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42917   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42918   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42919   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42920   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42921   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42922   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42923   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42924   sample_type:Population genome extracted from m...    2011.316222   \n",
       "42925   sample_type:Population genome extracted from m...    2011.316222   \n",
       "46707                 collected_by:University of Adelaide    2013.684463   \n",
       "47561                           sample_type:tissue sample    2016.548255   \n",
       "50807                          sample_type:whole organism            NaN   \n",
       "71669   sample_type:Population genome extracted from m...    2011.316222   \n",
       "71671   sample_type:Population genome extracted from m...    2011.316222   \n",
       "71679   sample_type:Population genome extracted from m...    2011.316222   \n",
       "99046                                                 NaN            NaN   \n",
       "130047                                                NaN            NaN   \n",
       "130055                                                NaN            NaN   \n",
       "130180                                                NaN    2018.180014   \n",
       "284627                                                NaN    1996.685147   \n",
       "310046  sample_type:Bacteria;biomaterial_provider:DSMZ...    1992.685147   \n",
       "\n",
       "       assembly_nover  \n",
       "528     GCA_000017105  \n",
       "8863    GCA_000341355  \n",
       "24579   GCA_000689835  \n",
       "24580   GCA_000689855  \n",
       "37899   GCA_001689645  \n",
       "37900   GCA_001689655  \n",
       "38407   GCA_001729965  \n",
       "42914   GCA_001941025  \n",
       "42915   GCA_001941045  \n",
       "42916   GCA_001941055  \n",
       "42917   GCA_001941065  \n",
       "42918   GCA_001941075  \n",
       "42919   GCA_001941125  \n",
       "42920   GCA_001941135  \n",
       "42921   GCA_001941165  \n",
       "42922   GCA_001941185  \n",
       "42923   GCA_001941205  \n",
       "42924   GCA_001941225  \n",
       "42925   GCA_001941235  \n",
       "46707   GCA_002097375  \n",
       "47561   GCA_002117465  \n",
       "50807   GCA_002190685  \n",
       "71669   GCA_002633115  \n",
       "71671   GCA_002633155  \n",
       "71679   GCA_002633305  \n",
       "99046   GCA_003334005  \n",
       "130047  GCA_004155475  \n",
       "130055  GCA_004155625  \n",
       "130180  GCA_004168465  \n",
       "284627  GCF_001211245  \n",
       "310046  GCF_002000985  \n",
       "\n",
       "[31 rows x 68 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kk = rast[(rast['host_name'].str.contains('koala', case=False, na=False)) | (rast['host_name'].str.contains('kangaroo', case=False, na=False))]\n",
    "kk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "60e23a0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>genome_size</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>same</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [assembly_accession, contig_count, genome_size, Contigs, Genome length, same]\n",
       "Index: []"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acccol = 'assembly_accession'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a14d1e30",
   "metadata": {},
   "source": [
    "### Merge the tables together\n",
    "\n",
    "First we merge the gtdb taxonomy with the RAST data just for kangaroos and koalas, and then we merge with the number of prophages."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "17d76395",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>...</th>\n",
       "      <th>isolation_date</th>\n",
       "      <th>assembly_nover</th>\n",
       "      <th>gtdb_taxonomy</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000017105.1</td>\n",
       "      <td>434922.5</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111</td>\n",
       "      <td>434922</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Dugway 5J108-111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000017105</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Coxiellales</td>\n",
       "      <td>Coxiellaceae</td>\n",
       "      <td>Coxiella</td>\n",
       "      <td>Coxiella burnetii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000341355.1</td>\n",
       "      <td>1094489.3</td>\n",
       "      <td>Bartonella australis Aust/NH1</td>\n",
       "      <td>Bartonella australis Aust/NH1</td>\n",
       "      <td>1094489</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Aust/NH1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1999.683094</td>\n",
       "      <td>GCA_000341355</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Bartonella</td>\n",
       "      <td>Bartonella australis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000689835.1</td>\n",
       "      <td>1331229.3</td>\n",
       "      <td>Bordetella bronchiseptica MBORD681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1331229</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MBORD681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000689835</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Bordetella</td>\n",
       "      <td>Bordetella pertussis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000689855.1</td>\n",
       "      <td>1331230.3</td>\n",
       "      <td>Bordetella bronchiseptica MBORD698</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1331230</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MBORD698</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_000689855</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Bordetella</td>\n",
       "      <td>Bordetella pertussis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001689645.1</td>\n",
       "      <td>1807743.4</td>\n",
       "      <td>Bacteroidales bacterium K1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1807743</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2015.017796</td>\n",
       "      <td>GCA_001689645</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Muribaculaceae</td>\n",
       "      <td>CAG-873</td>\n",
       "      <td>CAG-873 sp002633115</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession  genome_id                         genome_name  \\\n",
       "0    GCA_000017105.1   434922.5  Coxiella burnetii Dugway 5J108-111   \n",
       "1    GCA_000341355.1  1094489.3       Bartonella australis Aust/NH1   \n",
       "2    GCA_000689835.1  1331229.3  Bordetella bronchiseptica MBORD681   \n",
       "3    GCA_000689855.1  1331230.3  Bordetella bronchiseptica MBORD698   \n",
       "4    GCA_001689645.1  1807743.4          Bacteroidales bacterium K1   \n",
       "\n",
       "                        organism_name  taxon_id genome_status  \\\n",
       "0  Coxiella burnetii Dugway 5J108-111    434922      Complete   \n",
       "1       Bartonella australis Aust/NH1   1094489      Complete   \n",
       "2                                 NaN   1331229           WGS   \n",
       "3                                 NaN   1331230           WGS   \n",
       "4                                 NaN   1807743           WGS   \n",
       "\n",
       "             strain serovar biovar pathovar  ... isolation_date  \\\n",
       "0  Dugway 5J108-111     NaN    NaN      NaN  ...            NaN   \n",
       "1          Aust/NH1     NaN    NaN      NaN  ...    1999.683094   \n",
       "2          MBORD681     NaN    NaN      NaN  ...            NaN   \n",
       "3          MBORD698     NaN    NaN      NaN  ...            NaN   \n",
       "4               NaN     NaN    NaN      NaN  ...    2015.017796   \n",
       "\n",
       "  assembly_nover                                      gtdb_taxonomy    domain  \\\n",
       "0  GCA_000017105  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  Bacteria   \n",
       "1  GCA_000341355  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  Bacteria   \n",
       "2  GCA_000689835  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  Bacteria   \n",
       "3  GCA_000689855  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  Bacteria   \n",
       "4  GCA_001689645  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...  Bacteria   \n",
       "\n",
       "           phylum                class            order            family  \\\n",
       "0  Proteobacteria  Gammaproteobacteria      Coxiellales      Coxiellaceae   \n",
       "1  Proteobacteria  Alphaproteobacteria      Rhizobiales      Rhizobiaceae   \n",
       "2  Proteobacteria  Gammaproteobacteria  Burkholderiales  Burkholderiaceae   \n",
       "3  Proteobacteria  Gammaproteobacteria  Burkholderiales  Burkholderiaceae   \n",
       "4    Bacteroidota          Bacteroidia    Bacteroidales    Muribaculaceae   \n",
       "\n",
       "        genus               species  \n",
       "0    Coxiella     Coxiella burnetii  \n",
       "1  Bartonella  Bartonella australis  \n",
       "2  Bordetella  Bordetella pertussis  \n",
       "3  Bordetella  Bordetella pertussis  \n",
       "4     CAG-873   CAG-873 sp002633115  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kk_tax = pd.merge(kk,\n",
    "                  gtdb[[acccol, 'gtdb_taxonomy', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'species']],\n",
    "                  how='inner', left_on=acccol, right_on=acccol)\n",
    "kk_tax.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "67014fe6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>...</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_nover_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000017105.1</td>\n",
       "      <td>4.349225e+05</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111</td>\n",
       "      <td>434922</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Dugway 5J108-111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM1710v1</td>\n",
       "      <td>2212937.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000017105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000341355.1</td>\n",
       "      <td>1.094489e+06</td>\n",
       "      <td>Bartonella australis Aust/NH1</td>\n",
       "      <td>Bartonella australis Aust/NH1</td>\n",
       "      <td>1094489</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Aust/NH1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM34135v1</td>\n",
       "      <td>1596490.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>79039.0</td>\n",
       "      <td>GCA_000341355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000689835.1</td>\n",
       "      <td>1.331229e+06</td>\n",
       "      <td>Bordetella bronchiseptica MBORD681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1331229</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MBORD681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>gbb33v01</td>\n",
       "      <td>5158310.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>126457.0</td>\n",
       "      <td>GCA_000689835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000689855.1</td>\n",
       "      <td>1.331230e+06</td>\n",
       "      <td>Bordetella bronchiseptica MBORD698</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1331230</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MBORD698</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>gbb34v01</td>\n",
       "      <td>5151735.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>111816.0</td>\n",
       "      <td>GCA_000689855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001689645.1</td>\n",
       "      <td>1.807743e+06</td>\n",
       "      <td>Bacteroidales bacterium K1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1807743</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_001689655.1</td>\n",
       "      <td>1.807744e+06</td>\n",
       "      <td>Bacteroidales bacterium K10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1807744</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_001941025.1</td>\n",
       "      <td>1.930016e+06</td>\n",
       "      <td>Lachnospiraceae bacterium Zagget3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930016</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_001941045.1</td>\n",
       "      <td>1.930015e+06</td>\n",
       "      <td>Lachnospiraceae bacterium Zagget2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930015</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_001941065.1</td>\n",
       "      <td>1.930002e+06</td>\n",
       "      <td>Alistipes sp. Zagget8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930002</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_001941075.1</td>\n",
       "      <td>1.930019e+06</td>\n",
       "      <td>Oscillospiraceae bacterium Zagget6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930019</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_001941135.1</td>\n",
       "      <td>1.930004e+06</td>\n",
       "      <td>Ruminococcus sp. Zagget11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930004</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_001941205.1</td>\n",
       "      <td>1.930003e+06</td>\n",
       "      <td>Synergistes sp. Zagget9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930003</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCA_001941225.1</td>\n",
       "      <td>1.930020e+06</td>\n",
       "      <td>Oscillospiraceae bacterium Zagget12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1930020</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zagget12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_002097375.1</td>\n",
       "      <td>1.280123e+03</td>\n",
       "      <td>Staphylococcus aureus strain N13/1/662</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1280</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM209737v1</td>\n",
       "      <td>2589863.0</td>\n",
       "      <td>366.0</td>\n",
       "      <td>157.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>68577.0</td>\n",
       "      <td>GCA_002097375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_002117465.1</td>\n",
       "      <td>1.396189e+03</td>\n",
       "      <td>Bacillus cereus strain BC-AK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1396</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>BC-AK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM211746v1</td>\n",
       "      <td>5786333.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>191182.0</td>\n",
       "      <td>GCA_002117465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_002190685.1</td>\n",
       "      <td>5.621760e+02</td>\n",
       "      <td>Escherichia coli strain ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>562</td>\n",
       "      <td>WGS</td>\n",
       "      <td>ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM219068v1</td>\n",
       "      <td>4952372.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>166020.0</td>\n",
       "      <td>GCA_002190685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_002633115.1</td>\n",
       "      <td>1.958821e+06</td>\n",
       "      <td>Muribaculaceae bacterium Zag1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1958821</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zag1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_002633155.1</td>\n",
       "      <td>1.958823e+06</td>\n",
       "      <td>Candidatus Melainabacteria bacterium Zag15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1958823</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zag15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_002633305.1</td>\n",
       "      <td>1.958822e+06</td>\n",
       "      <td>Muribaculaceae bacterium Zag10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1958822</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Zag10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_003334005.1</td>\n",
       "      <td>5.623333e+02</td>\n",
       "      <td>Escherichia coli strain MOD1-ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>562</td>\n",
       "      <td>WGS</td>\n",
       "      <td>MOD1-ECOR29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM333400v1</td>\n",
       "      <td>4928564.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>165058.0</td>\n",
       "      <td>GCA_003334005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_004155475.1</td>\n",
       "      <td>1.690360e+03</td>\n",
       "      <td>Bifidobacterium pseudolongum subsp. globosum s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1690</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2109B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM415547v1</td>\n",
       "      <td>2039216.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>35149.0</td>\n",
       "      <td>GCA_004155475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_004155625.1</td>\n",
       "      <td>1.690450e+03</td>\n",
       "      <td>Bifidobacterium pseudolongum subsp. globosum s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1690</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2009B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM415562v1</td>\n",
       "      <td>2107529.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>45146.0</td>\n",
       "      <td>GCA_004155625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_004168465.1</td>\n",
       "      <td>3.572332e+05</td>\n",
       "      <td>Siccibacter turicensis strain #493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>357233</td>\n",
       "      <td>WGS</td>\n",
       "      <td>#493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23 rows × 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession     genome_id  \\\n",
       "0     GCA_000017105.1  4.349225e+05   \n",
       "1     GCA_000341355.1  1.094489e+06   \n",
       "2     GCA_000689835.1  1.331229e+06   \n",
       "3     GCA_000689855.1  1.331230e+06   \n",
       "4     GCA_001689645.1  1.807743e+06   \n",
       "5     GCA_001689655.1  1.807744e+06   \n",
       "6     GCA_001941025.1  1.930016e+06   \n",
       "7     GCA_001941045.1  1.930015e+06   \n",
       "8     GCA_001941065.1  1.930002e+06   \n",
       "9     GCA_001941075.1  1.930019e+06   \n",
       "10    GCA_001941135.1  1.930004e+06   \n",
       "11    GCA_001941205.1  1.930003e+06   \n",
       "12    GCA_001941225.1  1.930020e+06   \n",
       "13    GCA_002097375.1  1.280123e+03   \n",
       "14    GCA_002117465.1  1.396189e+03   \n",
       "15    GCA_002190685.1  5.621760e+02   \n",
       "16    GCA_002633115.1  1.958821e+06   \n",
       "17    GCA_002633155.1  1.958823e+06   \n",
       "18    GCA_002633305.1  1.958822e+06   \n",
       "19    GCA_003334005.1  5.623333e+02   \n",
       "20    GCA_004155475.1  1.690360e+03   \n",
       "21    GCA_004155625.1  1.690450e+03   \n",
       "22    GCA_004168465.1  3.572332e+05   \n",
       "\n",
       "                                          genome_name  \\\n",
       "0                  Coxiella burnetii Dugway 5J108-111   \n",
       "1                       Bartonella australis Aust/NH1   \n",
       "2                  Bordetella bronchiseptica MBORD681   \n",
       "3                  Bordetella bronchiseptica MBORD698   \n",
       "4                          Bacteroidales bacterium K1   \n",
       "5                         Bacteroidales bacterium K10   \n",
       "6                   Lachnospiraceae bacterium Zagget3   \n",
       "7                   Lachnospiraceae bacterium Zagget2   \n",
       "8                               Alistipes sp. Zagget8   \n",
       "9                  Oscillospiraceae bacterium Zagget6   \n",
       "10                          Ruminococcus sp. Zagget11   \n",
       "11                            Synergistes sp. Zagget9   \n",
       "12                Oscillospiraceae bacterium Zagget12   \n",
       "13             Staphylococcus aureus strain N13/1/662   \n",
       "14                       Bacillus cereus strain BC-AK   \n",
       "15                     Escherichia coli strain ECOR29   \n",
       "16                      Muribaculaceae bacterium Zag1   \n",
       "17         Candidatus Melainabacteria bacterium Zag15   \n",
       "18                     Muribaculaceae bacterium Zag10   \n",
       "19                Escherichia coli strain MOD1-ECOR29   \n",
       "20  Bifidobacterium pseudolongum subsp. globosum s...   \n",
       "21  Bifidobacterium pseudolongum subsp. globosum s...   \n",
       "22                 Siccibacter turicensis strain #493   \n",
       "\n",
       "                         organism_name  taxon_id genome_status  \\\n",
       "0   Coxiella burnetii Dugway 5J108-111    434922      Complete   \n",
       "1        Bartonella australis Aust/NH1   1094489      Complete   \n",
       "2                                  NaN   1331229           WGS   \n",
       "3                                  NaN   1331230           WGS   \n",
       "4                                  NaN   1807743           WGS   \n",
       "5                                  NaN   1807744           WGS   \n",
       "6                                  NaN   1930016           WGS   \n",
       "7                                  NaN   1930015           WGS   \n",
       "8                                  NaN   1930002           WGS   \n",
       "9                                  NaN   1930019           WGS   \n",
       "10                                 NaN   1930004           WGS   \n",
       "11                                 NaN   1930003           WGS   \n",
       "12                                 NaN   1930020           WGS   \n",
       "13                                 NaN      1280           WGS   \n",
       "14                                 NaN      1396       Plasmid   \n",
       "15                                 NaN       562           WGS   \n",
       "16                                 NaN   1958821           WGS   \n",
       "17                                 NaN   1958823           WGS   \n",
       "18                                 NaN   1958822           WGS   \n",
       "19                                 NaN       562           WGS   \n",
       "20                                 NaN      1690           WGS   \n",
       "21                                 NaN      1690           WGS   \n",
       "22                                 NaN    357233           WGS   \n",
       "\n",
       "              strain serovar biovar pathovar  ... assembly_name Genome length  \\\n",
       "0   Dugway 5J108-111     NaN    NaN      NaN  ...     ASM1710v1     2212937.0   \n",
       "1           Aust/NH1     NaN    NaN      NaN  ...    ASM34135v1     1596490.0   \n",
       "2           MBORD681     NaN    NaN      NaN  ...      gbb33v01     5158310.0   \n",
       "3           MBORD698     NaN    NaN      NaN  ...      gbb34v01     5151735.0   \n",
       "4                NaN     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "5                NaN     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "6            Zagget3     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "7            Zagget2     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "8            Zagget8     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "9            Zagget6     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "10          Zagget11     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "11           Zagget9     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "12          Zagget12     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "13               NaN     NaN    NaN      NaN  ...   ASM209737v1     2589863.0   \n",
       "14             BC-AK     NaN    NaN      NaN  ...   ASM211746v1     5786333.0   \n",
       "15            ECOR29     NaN    NaN      NaN  ...   ASM219068v1     4952372.0   \n",
       "16              Zag1     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "17             Zag15     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "18             Zag10     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "19       MOD1-ECOR29     NaN    NaN      NaN  ...   ASM333400v1     4928564.0   \n",
       "20             2109B     NaN    NaN      NaN  ...   ASM415547v1     2039216.0   \n",
       "21             2009B     NaN    NaN      NaN  ...   ASM415562v1     2107529.0   \n",
       "22              #493     NaN    NaN      NaN  ...           NaN           NaN   \n",
       "\n",
       "   Contigs Phage Contigs Total Predicted Prophages  Kept No phage genes  \\\n",
       "0      2.0           2.0                      10.0   0.0            6.0   \n",
       "1      1.0           1.0                       8.0   3.0            0.0   \n",
       "2     70.0          34.0                      58.0   5.0           14.0   \n",
       "3     52.0          39.0                      55.0   5.0           11.0   \n",
       "4      NaN           NaN                       NaN   NaN            NaN   \n",
       "5      NaN           NaN                       NaN   NaN            NaN   \n",
       "6      NaN           NaN                       NaN   NaN            NaN   \n",
       "7      NaN           NaN                       NaN   NaN            NaN   \n",
       "8      NaN           NaN                       NaN   NaN            NaN   \n",
       "9      NaN           NaN                       NaN   NaN            NaN   \n",
       "10     NaN           NaN                       NaN   NaN            NaN   \n",
       "11     NaN           NaN                       NaN   NaN            NaN   \n",
       "12     NaN           NaN                       NaN   NaN            NaN   \n",
       "13   366.0         157.0                      49.0   4.0           27.0   \n",
       "14     5.0           5.0                      46.0   4.0           10.0   \n",
       "15   301.0          78.0                      40.0  11.0            1.0   \n",
       "16     NaN           NaN                       NaN   NaN            NaN   \n",
       "17     NaN           NaN                       NaN   NaN            NaN   \n",
       "18     NaN           NaN                       NaN   NaN            NaN   \n",
       "19   112.0          49.0                      36.0   5.0            3.0   \n",
       "20     6.0           6.0                      10.0   1.0            0.0   \n",
       "21    42.0          33.0                      24.0   3.0            3.0   \n",
       "22     NaN           NaN                       NaN   NaN            NaN   \n",
       "\n",
       "   Not enough genes bp prophage assembly_nover_y  \n",
       "0               4.0         0.0    GCA_000017105  \n",
       "1               5.0     79039.0    GCA_000341355  \n",
       "2              39.0    126457.0    GCA_000689835  \n",
       "3              39.0    111816.0    GCA_000689855  \n",
       "4               NaN         NaN              NaN  \n",
       "5               NaN         NaN              NaN  \n",
       "6               NaN         NaN              NaN  \n",
       "7               NaN         NaN              NaN  \n",
       "8               NaN         NaN              NaN  \n",
       "9               NaN         NaN              NaN  \n",
       "10              NaN         NaN              NaN  \n",
       "11              NaN         NaN              NaN  \n",
       "12              NaN         NaN              NaN  \n",
       "13             18.0     68577.0    GCA_002097375  \n",
       "14             32.0    191182.0    GCA_002117465  \n",
       "15             28.0    166020.0    GCA_002190685  \n",
       "16              NaN         NaN              NaN  \n",
       "17              NaN         NaN              NaN  \n",
       "18              NaN         NaN              NaN  \n",
       "19             28.0    165058.0    GCA_003334005  \n",
       "20              9.0     35149.0    GCA_004155475  \n",
       "21             18.0     45146.0    GCA_004155625  \n",
       "22              NaN         NaN              NaN  \n",
       "\n",
       "[23 rows x 86 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kk_tp = pd.merge(kk_tax,\n",
    "                  phagesdf,\n",
    "                  how='left', left_on=acccol, right_on=acccol)\n",
    "kk_tp"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97d896ad",
   "metadata": {},
   "source": [
    "## NOTE: We don't have prophage predictions for all Australian isolates!\n",
    "\n",
    "We only have pp predictions for 10 isolates, and so for some of these kept is NaN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e9b7939a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(10, 86)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kk_tp.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "296b24bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>host_name</th>\n",
       "      <th>isolation_country</th>\n",
       "      <th>phylum</th>\n",
       "      <th>species</th>\n",
       "      <th>Kept</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_004155475.1</td>\n",
       "      <td>Bifidobacterium pseudolongum subsp. globosum s...</td>\n",
       "      <td>Kangaroo</td>\n",
       "      <td>Italy</td>\n",
       "      <td>Actinobacteriota</td>\n",
       "      <td>Bifidobacterium globosum</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_004155625.1</td>\n",
       "      <td>Bifidobacterium pseudolongum subsp. globosum s...</td>\n",
       "      <td>Kangaroo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Actinobacteriota</td>\n",
       "      <td>Bifidobacterium globosum</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_001941205.1</td>\n",
       "      <td>Synergistes sp. Zagget9</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Amulumruptor caecigallinarius</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_001689655.1</td>\n",
       "      <td>Bacteroidales bacterium K10</td>\n",
       "      <td>Koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Amulumruptor caecigallinarius</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_001941065.1</td>\n",
       "      <td>Alistipes sp. Zagget8</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Alistipes sp001941065</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_002633305.1</td>\n",
       "      <td>Muribaculaceae bacterium Zag10</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Amulumruptor caecigallinarius</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001689645.1</td>\n",
       "      <td>Bacteroidales bacterium K1</td>\n",
       "      <td>Koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>CAG-873 sp002633115</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_002633115.1</td>\n",
       "      <td>Muribaculaceae bacterium Zag1</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>CAG-873 sp002633115</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_002633155.1</td>\n",
       "      <td>Candidatus Melainabacteria bacterium Zag15</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>Zag1 sp001917115</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_002097375.1</td>\n",
       "      <td>Staphylococcus aureus strain N13/1/662</td>\n",
       "      <td>kangaroo</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_002117465.1</td>\n",
       "      <td>Bacillus cereus strain BC-AK</td>\n",
       "      <td>kangaroo</td>\n",
       "      <td>China</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacillus_A tropicus</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_001941025.1</td>\n",
       "      <td>Lachnospiraceae bacterium Zagget3</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Eubacterium_I sp001941025</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_001941045.1</td>\n",
       "      <td>Lachnospiraceae bacterium Zagget2</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>UBA7182 sp001941045</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_001941075.1</td>\n",
       "      <td>Oscillospiraceae bacterium Zagget6</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>NK3B98 sp001941075</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_001941135.1</td>\n",
       "      <td>Ruminococcus sp. Zagget11</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>CAG-353 sp001941135</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCA_001941225.1</td>\n",
       "      <td>Oscillospiraceae bacterium Zagget12</td>\n",
       "      <td>koala</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>CAG-41 sp001941225</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_003334005.1</td>\n",
       "      <td>Escherichia coli strain MOD1-ECOR29</td>\n",
       "      <td>Dipodomys (kangaroo rat)</td>\n",
       "      <td>United States</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Escherichia flexneri</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000017105.1</td>\n",
       "      <td>Coxiella burnetii Dugway 5J108-111</td>\n",
       "      <td>Chisel-toothed kangaroo rat, Dipodomys microps</td>\n",
       "      <td>United States</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Coxiella burnetii</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000689855.1</td>\n",
       "      <td>Bordetella bronchiseptica MBORD698</td>\n",
       "      <td>Koala, Phascolarctos cinereus</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Bordetella pertussis</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000689835.1</td>\n",
       "      <td>Bordetella bronchiseptica MBORD681</td>\n",
       "      <td>Koala, Phascolarctos cinereus</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Bordetella pertussis</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000341355.1</td>\n",
       "      <td>Bartonella australis Aust/NH1</td>\n",
       "      <td>Eastern grey kangaroo, Macropus giganteus</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Bartonella australis</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_002190685.1</td>\n",
       "      <td>Escherichia coli strain ECOR29</td>\n",
       "      <td>kangaroo rat</td>\n",
       "      <td>United States</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Escherichia flexneri</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_004168465.1</td>\n",
       "      <td>Siccibacter turicensis strain #493</td>\n",
       "      <td>wild eastern grey kangaroo</td>\n",
       "      <td>Australia</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Siccibacter turicensis</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession                                        genome_name  \\\n",
       "20    GCA_004155475.1  Bifidobacterium pseudolongum subsp. globosum s...   \n",
       "21    GCA_004155625.1  Bifidobacterium pseudolongum subsp. globosum s...   \n",
       "11    GCA_001941205.1                            Synergistes sp. Zagget9   \n",
       "5     GCA_001689655.1                        Bacteroidales bacterium K10   \n",
       "8     GCA_001941065.1                              Alistipes sp. Zagget8   \n",
       "18    GCA_002633305.1                     Muribaculaceae bacterium Zag10   \n",
       "4     GCA_001689645.1                         Bacteroidales bacterium K1   \n",
       "16    GCA_002633115.1                      Muribaculaceae bacterium Zag1   \n",
       "17    GCA_002633155.1         Candidatus Melainabacteria bacterium Zag15   \n",
       "13    GCA_002097375.1             Staphylococcus aureus strain N13/1/662   \n",
       "14    GCA_002117465.1                       Bacillus cereus strain BC-AK   \n",
       "6     GCA_001941025.1                  Lachnospiraceae bacterium Zagget3   \n",
       "7     GCA_001941045.1                  Lachnospiraceae bacterium Zagget2   \n",
       "9     GCA_001941075.1                 Oscillospiraceae bacterium Zagget6   \n",
       "10    GCA_001941135.1                          Ruminococcus sp. Zagget11   \n",
       "12    GCA_001941225.1                Oscillospiraceae bacterium Zagget12   \n",
       "19    GCA_003334005.1                Escherichia coli strain MOD1-ECOR29   \n",
       "0     GCA_000017105.1                 Coxiella burnetii Dugway 5J108-111   \n",
       "3     GCA_000689855.1                 Bordetella bronchiseptica MBORD698   \n",
       "2     GCA_000689835.1                 Bordetella bronchiseptica MBORD681   \n",
       "1     GCA_000341355.1                      Bartonella australis Aust/NH1   \n",
       "15    GCA_002190685.1                     Escherichia coli strain ECOR29   \n",
       "22    GCA_004168465.1                 Siccibacter turicensis strain #493   \n",
       "\n",
       "                                         host_name isolation_country  \\\n",
       "20                                        Kangaroo             Italy   \n",
       "21                                        Kangaroo               NaN   \n",
       "11                                           koala         Australia   \n",
       "5                                            Koala         Australia   \n",
       "8                                            koala         Australia   \n",
       "18                                           koala         Australia   \n",
       "4                                            Koala         Australia   \n",
       "16                                           koala         Australia   \n",
       "17                                           koala         Australia   \n",
       "13                                        kangaroo         Australia   \n",
       "14                                        kangaroo             China   \n",
       "6                                            koala         Australia   \n",
       "7                                            koala         Australia   \n",
       "9                                            koala         Australia   \n",
       "10                                           koala         Australia   \n",
       "12                                           koala         Australia   \n",
       "19                        Dipodomys (kangaroo rat)     United States   \n",
       "0   Chisel-toothed kangaroo rat, Dipodomys microps     United States   \n",
       "3                    Koala, Phascolarctos cinereus         Australia   \n",
       "2                    Koala, Phascolarctos cinereus         Australia   \n",
       "1        Eastern grey kangaroo, Macropus giganteus         Australia   \n",
       "15                                    kangaroo rat     United States   \n",
       "22                      wild eastern grey kangaroo         Australia   \n",
       "\n",
       "              phylum                        species  Kept  \n",
       "20  Actinobacteriota       Bifidobacterium globosum   1.0  \n",
       "21  Actinobacteriota       Bifidobacterium globosum   3.0  \n",
       "11      Bacteroidota  Amulumruptor caecigallinarius   NaN  \n",
       "5       Bacteroidota  Amulumruptor caecigallinarius   NaN  \n",
       "8       Bacteroidota          Alistipes sp001941065   NaN  \n",
       "18      Bacteroidota  Amulumruptor caecigallinarius   NaN  \n",
       "4       Bacteroidota            CAG-873 sp002633115   NaN  \n",
       "16      Bacteroidota            CAG-873 sp002633115   NaN  \n",
       "17     Cyanobacteria               Zag1 sp001917115   NaN  \n",
       "13        Firmicutes          Staphylococcus aureus   4.0  \n",
       "14        Firmicutes            Bacillus_A tropicus   4.0  \n",
       "6       Firmicutes_A      Eubacterium_I sp001941025   NaN  \n",
       "7       Firmicutes_A            UBA7182 sp001941045   NaN  \n",
       "9       Firmicutes_A             NK3B98 sp001941075   NaN  \n",
       "10      Firmicutes_A            CAG-353 sp001941135   NaN  \n",
       "12      Firmicutes_A             CAG-41 sp001941225   NaN  \n",
       "19    Proteobacteria           Escherichia flexneri   5.0  \n",
       "0     Proteobacteria              Coxiella burnetii   0.0  \n",
       "3     Proteobacteria           Bordetella pertussis   5.0  \n",
       "2     Proteobacteria           Bordetella pertussis   5.0  \n",
       "1     Proteobacteria           Bartonella australis   3.0  \n",
       "15    Proteobacteria           Escherichia flexneri  11.0  \n",
       "22    Proteobacteria         Siccibacter turicensis   NaN  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kk_tp[['assembly_accession', 'genome_name', 'host_name', 'isolation_country', 'phylum', 'species', 'Kept']].sort_values('phylum')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd3d7029",
   "metadata": {},
   "source": [
    "# Phages from Australia\n",
    "\n",
    "Repeat the above, but with isolation country = Australia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "d33b691d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>...</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_nover_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000014905.1</td>\n",
       "      <td>2.342671e+05</td>\n",
       "      <td>Candidatus Solibacter usitatus Ellin6076</td>\n",
       "      <td>Candidatus Solibacter usitatus Ellin6076</td>\n",
       "      <td>234267</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Ellin6076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM1490v1</td>\n",
       "      <td>9965640.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>202405.0</td>\n",
       "      <td>GCA_000014905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000016845.1</td>\n",
       "      <td>4.441783e+05</td>\n",
       "      <td>Brucella ovis ATCC 25840</td>\n",
       "      <td>Brucella ovis ATCC 25840</td>\n",
       "      <td>444178</td>\n",
       "      <td>Complete</td>\n",
       "      <td>ATCC25840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM1684v1</td>\n",
       "      <td>3275590.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>82310.0</td>\n",
       "      <td>GCA_000016845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000024005.1</td>\n",
       "      <td>4.859186e+05</td>\n",
       "      <td>Chitinophaga pinensis DSM 2588</td>\n",
       "      <td>Chitinophaga pinensis DSM 2588</td>\n",
       "      <td>485918</td>\n",
       "      <td>Complete</td>\n",
       "      <td>DSM 2588</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM2400v1</td>\n",
       "      <td>9127347.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>52154.0</td>\n",
       "      <td>GCA_000024005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000026185.1</td>\n",
       "      <td>4.658172e+05</td>\n",
       "      <td>Erwinia tasmaniensis Et1/99</td>\n",
       "      <td>Erwinia tasmaniensis Et1/99</td>\n",
       "      <td>465817</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>Et1/99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM2618v1</td>\n",
       "      <td>4067864.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>144087.0</td>\n",
       "      <td>GCA_000026185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000143865.1</td>\n",
       "      <td>8.658673e+05</td>\n",
       "      <td>Mycoplasma mycoides subsp. mycoides SC str. Gl...</td>\n",
       "      <td>Mycoplasma mycoides subsp. mycoides SC str. Gl...</td>\n",
       "      <td>865867</td>\n",
       "      <td>Complete</td>\n",
       "      <td>Gladysdale</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM14386v1</td>\n",
       "      <td>1193808.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>13721.0</td>\n",
       "      <td>GCA_000143865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2709</th>\n",
       "      <td>GCA_901212295.1</td>\n",
       "      <td>1.145171e+05</td>\n",
       "      <td>Aeromonas veronii bv. sobria strain BC88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>114517</td>\n",
       "      <td>WGS</td>\n",
       "      <td>BC88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2710</th>\n",
       "      <td>GCA_901482475.1</td>\n",
       "      <td>1.502107e+03</td>\n",
       "      <td>Clostridium perfringens strain NCTC8503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1502</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NCTC8503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>44738_A01</td>\n",
       "      <td>3574234.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>131311.0</td>\n",
       "      <td>GCA_901482475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2711</th>\n",
       "      <td>GCA_901482565.1</td>\n",
       "      <td>1.719267e+03</td>\n",
       "      <td>Corynebacterium pseudotuberculosis strain NCTC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1719</td>\n",
       "      <td>Complete</td>\n",
       "      <td>NCTC4681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>45076_G01</td>\n",
       "      <td>2337798.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>18350.0</td>\n",
       "      <td>GCA_901482565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2712</th>\n",
       "      <td>GCA_901687245.1</td>\n",
       "      <td>1.431451e+06</td>\n",
       "      <td>Helicobacter pylori PMSS1 strain Hp GFP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1431450</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Hp GFP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>25455_8_248</td>\n",
       "      <td>1602364.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_901687245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2713</th>\n",
       "      <td>GCA_901687255.1</td>\n",
       "      <td>1.431451e+06</td>\n",
       "      <td>Helicobacter pylori PMSS1 strain Hp td strain ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1431450</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Hp tdT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>25455_8_249</td>\n",
       "      <td>1598890.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_901687255</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2714 rows × 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     assembly_accession     genome_id  \\\n",
       "0       GCA_000014905.1  2.342671e+05   \n",
       "1       GCA_000016845.1  4.441783e+05   \n",
       "2       GCA_000024005.1  4.859186e+05   \n",
       "3       GCA_000026185.1  4.658172e+05   \n",
       "4       GCA_000143865.1  8.658673e+05   \n",
       "...                 ...           ...   \n",
       "2709    GCA_901212295.1  1.145171e+05   \n",
       "2710    GCA_901482475.1  1.502107e+03   \n",
       "2711    GCA_901482565.1  1.719267e+03   \n",
       "2712    GCA_901687245.1  1.431451e+06   \n",
       "2713    GCA_901687255.1  1.431451e+06   \n",
       "\n",
       "                                            genome_name  \\\n",
       "0              Candidatus Solibacter usitatus Ellin6076   \n",
       "1                              Brucella ovis ATCC 25840   \n",
       "2                        Chitinophaga pinensis DSM 2588   \n",
       "3                           Erwinia tasmaniensis Et1/99   \n",
       "4     Mycoplasma mycoides subsp. mycoides SC str. Gl...   \n",
       "...                                                 ...   \n",
       "2709           Aeromonas veronii bv. sobria strain BC88   \n",
       "2710            Clostridium perfringens strain NCTC8503   \n",
       "2711  Corynebacterium pseudotuberculosis strain NCTC...   \n",
       "2712            Helicobacter pylori PMSS1 strain Hp GFP   \n",
       "2713  Helicobacter pylori PMSS1 strain Hp td strain ...   \n",
       "\n",
       "                                          organism_name  taxon_id  \\\n",
       "0              Candidatus Solibacter usitatus Ellin6076    234267   \n",
       "1                              Brucella ovis ATCC 25840    444178   \n",
       "2                        Chitinophaga pinensis DSM 2588    485918   \n",
       "3                           Erwinia tasmaniensis Et1/99    465817   \n",
       "4     Mycoplasma mycoides subsp. mycoides SC str. Gl...    865867   \n",
       "...                                                 ...       ...   \n",
       "2709                                                NaN    114517   \n",
       "2710                                                NaN      1502   \n",
       "2711                                                NaN      1719   \n",
       "2712                                                NaN   1431450   \n",
       "2713                                                NaN   1431450   \n",
       "\n",
       "     genome_status      strain serovar biovar pathovar  ... assembly_name  \\\n",
       "0         Complete   Ellin6076     NaN    NaN      NaN  ...     ASM1490v1   \n",
       "1         Complete   ATCC25840     NaN    NaN      NaN  ...     ASM1684v1   \n",
       "2         Complete    DSM 2588     NaN    NaN      NaN  ...     ASM2400v1   \n",
       "3          Plasmid      Et1/99     NaN    NaN      NaN  ...     ASM2618v1   \n",
       "4         Complete  Gladysdale     NaN    NaN      NaN  ...    ASM14386v1   \n",
       "...            ...         ...     ...    ...      ...  ...           ...   \n",
       "2709           WGS        BC88     NaN    NaN      NaN  ...           NaN   \n",
       "2710           WGS    NCTC8503     NaN    NaN      NaN  ...     44738_A01   \n",
       "2711      Complete    NCTC4681     NaN    NaN      NaN  ...     45076_G01   \n",
       "2712           WGS      Hp GFP     NaN    NaN      NaN  ...   25455_8_248   \n",
       "2713           WGS      Hp tdT     NaN    NaN      NaN  ...   25455_8_249   \n",
       "\n",
       "     Genome length Contigs Phage Contigs Total Predicted Prophages Kept  \\\n",
       "0        9965640.0     1.0           1.0                      60.0  8.0   \n",
       "1        3275590.0     2.0           2.0                      16.0  2.0   \n",
       "2        9127347.0     1.0           1.0                      23.0  1.0   \n",
       "3        4067864.0     6.0           6.0                      34.0  4.0   \n",
       "4        1193808.0     1.0           1.0                       8.0  1.0   \n",
       "...            ...     ...           ...                       ...  ...   \n",
       "2709           NaN     NaN           NaN                       NaN  NaN   \n",
       "2710     3574234.0    10.0           9.0                      28.0  5.0   \n",
       "2711     2337798.0     1.0           1.0                      14.0  1.0   \n",
       "2712     1602364.0    40.0          18.0                       7.0  0.0   \n",
       "2713     1598890.0    37.0          19.0                       8.0  0.0   \n",
       "\n",
       "     No phage genes Not enough genes bp prophage assembly_nover_y  \n",
       "0              11.0             41.0    202405.0    GCA_000014905  \n",
       "1               5.0              9.0     82310.0    GCA_000016845  \n",
       "2               0.0             22.0     52154.0    GCA_000024005  \n",
       "3               6.0             24.0    144087.0    GCA_000026185  \n",
       "4               3.0              4.0     13721.0    GCA_000143865  \n",
       "...             ...              ...         ...              ...  \n",
       "2709            NaN              NaN         NaN              NaN  \n",
       "2710            3.0             20.0    131311.0    GCA_901482475  \n",
       "2711            4.0              9.0     18350.0    GCA_901482565  \n",
       "2712            3.0              4.0         0.0    GCA_901687245  \n",
       "2713            4.0              4.0         0.0    GCA_901687255  \n",
       "\n",
       "[2714 rows x 86 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aus = rast[rast['isolation_country'].str.contains('australia', case=False, na=False)]\n",
    "aus_tax = pd.merge(aus,\n",
    "                  gtdb[[acccol, 'gtdb_taxonomy', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'species']],\n",
    "                  how='inner', left_on=acccol, right_on=acccol)\n",
    "aus_tp = pd.merge(aus_tax,\n",
    "                  phagesdf,\n",
    "                  how='left', left_on=acccol, right_on=acccol)\n",
    "aus_tp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "d264bfb7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Acidobacteriota',\n",
       " 'Proteobacteria',\n",
       " 'Bacteroidota',\n",
       " 'Firmicutes',\n",
       " 'Planctomycetota',\n",
       " 'Verrucomicrobiota',\n",
       " 'Cyanobacteria',\n",
       " 'Firmicutes_E',\n",
       " 'Actinobacteriota',\n",
       " 'Spirochaetota',\n",
       " 'Campylobacterota',\n",
       " 'Desulfobacterota',\n",
       " 'Firmicutes_A',\n",
       " 'Thermotogota',\n",
       " 'Firmicutes_B',\n",
       " 'Chrysiogenetota',\n",
       " 'Verrucomicrobiota_A',\n",
       " 'Fibrobacterota',\n",
       " 'Dependentiae',\n",
       " 'Cloacimonadota',\n",
       " 'Chloroflexota',\n",
       " 'Deferribacterota',\n",
       " 'Synergistota',\n",
       " 'Fusobacteriota',\n",
       " 'Firmicutes_C',\n",
       " 'Patescibacteria',\n",
       " 'Bdellovibrionota',\n",
       " 'Armatimonadota',\n",
       " 'Bdellovibrionota_C',\n",
       " 'Myxococcota',\n",
       " 'Omnitrophota',\n",
       " 'UBP18',\n",
       " 'Firmicutes_D',\n",
       " 'Bipolaricaulota',\n",
       " 'SAR324',\n",
       " 'Methylomirabilota',\n",
       " 'Zixibacteria']"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(pd.unique(aus_tp['phylum']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2eaeeb2",
   "metadata": {},
   "source": [
    "# Ochre\n",
    "\n",
    "What is happening in the Ochre world?\n",
    "\n",
    "This set of bactereial orders comes from this paper: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0185252#sec008\n",
    "\n",
    "```\n",
    "Lenehan CE, Tobe SS, Smith RJ, Popelka-Filcoff RS. 2017. Microbial composition analyses by 16S rRNA sequencing: A proof of concept approach to provenance determination of archaeological ochre. PLoS One 12:e0185252.\n",
    "```\n",
    "\n",
    "I extracted the orders from `Table S2 - Order level relative proportion of matches to the Greengenes (13_08) database` by:\n",
    "1. convert pdf to text: `pdftotext -layout pone.0185252.s003.pdf`\n",
    "2. print the list of orders that end `ales`:  `perl -ne 'if (/(\\S+ales)/) {print \"$1\\n\"}' pone.0185252.s003.txt | sort -u | perl -ne 'chomp; print \"\\\"$_\\\", \"'`\n",
    "\n",
    "Note: Evan kindly converted that file into [an excel file](Australia/Lenehan_Table_S2.xlsx) and we read that below\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "0f914d9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ochre_orders = [\"Acidimicrobiales\", \"Actinomycetales\", \"Aeromonadales\", \"Alteromonadales\", \"Ardenscatenales\", \"Armatimonadales\", \"Bacillales\", \"Bacteroidales\", \"Burkholderiales\", \"Caldilineales\", \"Campylobacterales\", \"Caulobacterales\", \"Chloroflexales\", \"Chromatiales\", \"Chroococcales\", \"Chthoniobacterales\", \"Clostridiales\", \"Coriobacteriales\", \"Cytophagales\", \"Deinococcales\", \"Enterobacteriales\", \"Euzebyales\", \"Flavobacteriales\", \"Gaiellales\", \"Gemellales\", \"Gemmatales\", \"Gemmatimonadales\", \"Herpetosiphonales\", \"Lactobacillales\", \"Micrococcales\", \"Myxococcales\", \"Neisseriales\", \"Nitrosomonadales\", \"Nitrospirales\", \"Nostocales\", \"Pasteurellales\", \"Pirellulales\", \"Planctomycetales\", \"Pseudomonadales\", \"Rhizobiales\", \"Rhodobacterales\", \"Rhodospirillales\", \"Rhodothermales\", \"Rickettsiales\", \"Roseiflexales\", \"Rubrobacterales\", \"Saprospirales\", \"Solirubrobacterales\", \"Sphingobacteriales\", \"Sphingomonadales\", \"Spirobacillales\", \"Synergistales\", \"Syntrophobacterales\", \"Thermobaculales\", \"Thiotrichales\", \"Vibrionales\", \"Xanthomonadales\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "3a503704",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_lineage</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>coding_density</th>\n",
       "      <th>...</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "      <th>assembly_nover_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GB_GCA_000006155.2</td>\n",
       "      <td>1916</td>\n",
       "      <td>93.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1171</td>\n",
       "      <td>g__Bacillus (UID902)</td>\n",
       "      <td>324</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4305660</td>\n",
       "      <td>80.178992</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM615v2</td>\n",
       "      <td>5370060.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_000006155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GB_GCA_000007385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.82</td>\n",
       "      <td>0.00</td>\n",
       "      <td>481</td>\n",
       "      <td>c__Gammaproteobacteria (UID4202)</td>\n",
       "      <td>276</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4190634</td>\n",
       "      <td>84.805944</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM738v1</td>\n",
       "      <td>4941439.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>129141.0</td>\n",
       "      <td>GCA_000007385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GB_GCA_000013845.2</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>332</td>\n",
       "      <td>o__Clostridiales (UID1375)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2428396</td>\n",
       "      <td>82.037966</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM1384v2</td>\n",
       "      <td>2960088.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>56955.0</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GB_GCA_000014305.1</td>\n",
       "      <td>0</td>\n",
       "      <td>97.38</td>\n",
       "      <td>0.00</td>\n",
       "      <td>475</td>\n",
       "      <td>o__Lactobacillales (UID544)</td>\n",
       "      <td>267</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1810429</td>\n",
       "      <td>86.362697</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM1430v1</td>\n",
       "      <td>2096309.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>84555.0</td>\n",
       "      <td>GCA_000014305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GB_GCA_000014325.1</td>\n",
       "      <td>0</td>\n",
       "      <td>97.87</td>\n",
       "      <td>0.00</td>\n",
       "      <td>475</td>\n",
       "      <td>o__Lactobacillales (UID544)</td>\n",
       "      <td>267</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1808181</td>\n",
       "      <td>86.280609</td>\n",
       "      <td>...</td>\n",
       "      <td>ASM1432v1</td>\n",
       "      <td>2095698.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>74165.0</td>\n",
       "      <td>GCA_000014325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56758</th>\n",
       "      <td>RS_GCF_901971775.1</td>\n",
       "      <td>19</td>\n",
       "      <td>99.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>336</td>\n",
       "      <td>o__Lactobacillales (UID355)</td>\n",
       "      <td>184</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1864126</td>\n",
       "      <td>85.804187</td>\n",
       "      <td>...</td>\n",
       "      <td>INIA_P508</td>\n",
       "      <td>2172535.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_901971775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56759</th>\n",
       "      <td>RS_GCF_901971785.1</td>\n",
       "      <td>27</td>\n",
       "      <td>99.46</td>\n",
       "      <td>0.27</td>\n",
       "      <td>586</td>\n",
       "      <td>g__Lactobacillus (UID436)</td>\n",
       "      <td>184</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2781164</td>\n",
       "      <td>85.330494</td>\n",
       "      <td>...</td>\n",
       "      <td>INIA_P344</td>\n",
       "      <td>3259285.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>41804.0</td>\n",
       "      <td>GCA_901971785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56760</th>\n",
       "      <td>RS_GCF_901971795.1</td>\n",
       "      <td>29</td>\n",
       "      <td>99.46</td>\n",
       "      <td>1.27</td>\n",
       "      <td>586</td>\n",
       "      <td>g__Lactobacillus (UID436)</td>\n",
       "      <td>184</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2612902</td>\n",
       "      <td>85.095735</td>\n",
       "      <td>...</td>\n",
       "      <td>INIA_P540</td>\n",
       "      <td>3070544.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>57146.0</td>\n",
       "      <td>GCA_901971795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56761</th>\n",
       "      <td>RS_GCF_902141825.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.27</td>\n",
       "      <td>1.34</td>\n",
       "      <td>507</td>\n",
       "      <td>c__Gammaproteobacteria (UID4444)</td>\n",
       "      <td>232</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5117266</td>\n",
       "      <td>84.091758</td>\n",
       "      <td>...</td>\n",
       "      <td>CECT_9533</td>\n",
       "      <td>6085336.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GCA_902141825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56762</th>\n",
       "      <td>RS_GCF_902141845.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.00</td>\n",
       "      <td>0.16</td>\n",
       "      <td>481</td>\n",
       "      <td>o__Rhizobiales (UID3654)</td>\n",
       "      <td>319</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5188958</td>\n",
       "      <td>85.252203</td>\n",
       "      <td>...</td>\n",
       "      <td>CECT_9862</td>\n",
       "      <td>6086597.0</td>\n",
       "      <td>247.0</td>\n",
       "      <td>189.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>100308.0</td>\n",
       "      <td>GCA_902141845</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>56763 rows × 128 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                accession  ambiguous_bases  checkm_completeness  \\\n",
       "0      GB_GCA_000006155.2             1916                93.12   \n",
       "1      GB_GCA_000007385.1                0                99.82   \n",
       "2      GB_GCA_000013845.2                0               100.00   \n",
       "3      GB_GCA_000014305.1                0                97.38   \n",
       "4      GB_GCA_000014325.1                0                97.87   \n",
       "...                   ...              ...                  ...   \n",
       "56758  RS_GCF_901971775.1               19                99.18   \n",
       "56759  RS_GCF_901971785.1               27                99.46   \n",
       "56760  RS_GCF_901971795.1               29                99.46   \n",
       "56761  RS_GCF_902141825.1                0                99.27   \n",
       "56762  RS_GCF_902141845.1                0                99.00   \n",
       "\n",
       "       checkm_contamination  checkm_marker_count  \\\n",
       "0                      0.00                 1171   \n",
       "1                      0.00                  481   \n",
       "2                      0.00                  332   \n",
       "3                      0.00                  475   \n",
       "4                      0.00                  475   \n",
       "...                     ...                  ...   \n",
       "56758                  0.00                  336   \n",
       "56759                  0.27                  586   \n",
       "56760                  1.27                  586   \n",
       "56761                  1.34                  507   \n",
       "56762                  0.16                  481   \n",
       "\n",
       "                  checkm_marker_lineage  checkm_marker_set_count  \\\n",
       "0                  g__Bacillus (UID902)                      324   \n",
       "1      c__Gammaproteobacteria (UID4202)                      276   \n",
       "2            o__Clostridiales (UID1375)                      124   \n",
       "3           o__Lactobacillales (UID544)                      267   \n",
       "4           o__Lactobacillales (UID544)                      267   \n",
       "...                                 ...                      ...   \n",
       "56758       o__Lactobacillales (UID355)                      184   \n",
       "56759         g__Lactobacillus (UID436)                      184   \n",
       "56760         g__Lactobacillus (UID436)                      184   \n",
       "56761  c__Gammaproteobacteria (UID4444)                      232   \n",
       "56762          o__Rhizobiales (UID3654)                      319   \n",
       "\n",
       "       checkm_strain_heterogeneity  coding_bases  coding_density  ...  \\\n",
       "0                              0.0       4305660       80.178992  ...   \n",
       "1                              0.0       4190634       84.805944  ...   \n",
       "2                              0.0       2428396       82.037966  ...   \n",
       "3                              0.0       1810429       86.362697  ...   \n",
       "4                              0.0       1808181       86.280609  ...   \n",
       "...                            ...           ...             ...  ...   \n",
       "56758                          0.0       1864126       85.804187  ...   \n",
       "56759                        100.0       2781164       85.330494  ...   \n",
       "56760                        100.0       2612902       85.095735  ...   \n",
       "56761                          0.0       5117266       84.091758  ...   \n",
       "56762                          0.0       5188958       85.252203  ...   \n",
       "\n",
       "       assembly_name  Genome length  Contigs  Phage Contigs  \\\n",
       "0           ASM615v2      5370060.0      3.0            3.0   \n",
       "1           ASM738v1      4941439.0      1.0            1.0   \n",
       "2          ASM1384v2      2960088.0      4.0            4.0   \n",
       "3          ASM1430v1      2096309.0      1.0            1.0   \n",
       "4          ASM1432v1      2095698.0      1.0            1.0   \n",
       "...              ...            ...      ...            ...   \n",
       "56758      INIA_P508      2172535.0     67.0           42.0   \n",
       "56759      INIA_P344      3259285.0     88.0           45.0   \n",
       "56760      INIA_P540      3070544.0     80.0           51.0   \n",
       "56761      CECT_9533      6085336.0     49.0           20.0   \n",
       "56762      CECT_9862      6086597.0    247.0          189.0   \n",
       "\n",
       "      Total Predicted Prophages Kept No phage genes Not enough genes  \\\n",
       "0                           5.0  0.0            2.0              3.0   \n",
       "1                          29.0  4.0           12.0             13.0   \n",
       "2                          12.0  2.0            4.0              6.0   \n",
       "3                          19.0  4.0            9.0              6.0   \n",
       "4                          23.0  4.0           14.0              5.0   \n",
       "...                         ...  ...            ...              ...   \n",
       "56758                      18.0  0.0            7.0             11.0   \n",
       "56759                      20.0  2.0           11.0              7.0   \n",
       "56760                      12.0  2.0            4.0              6.0   \n",
       "56761                      15.0  0.0            7.0              8.0   \n",
       "56762                      43.0  4.0           22.0             17.0   \n",
       "\n",
       "      bp prophage assembly_nover_y  \n",
       "0             0.0    GCA_000006155  \n",
       "1        129141.0    GCA_000007385  \n",
       "2         56955.0    GCA_000013845  \n",
       "3         84555.0    GCA_000014305  \n",
       "4         74165.0    GCA_000014325  \n",
       "...           ...              ...  \n",
       "56758         0.0    GCA_901971775  \n",
       "56759     41804.0    GCA_901971785  \n",
       "56760     57146.0    GCA_901971795  \n",
       "56761         0.0    GCA_902141825  \n",
       "56762    100308.0    GCA_902141845  \n",
       "\n",
       "[56763 rows x 128 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ochre_gtdb = gtdb[gtdb['order'].isin(ochre_orders)]\n",
    "ochre_phage = pd.merge(ochre_gtdb,\n",
    "                  phagesdf,\n",
    "                  how='inner', left_on=acccol, right_on=acccol)\n",
    "ochre_phage"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2a81e43",
   "metadata": {},
   "source": [
    "### Read the Lenehan et al results\n",
    "\n",
    "This table has the results from Claire and Rachel's paper where they describe the microbiota of ochre using 16S. Evan converted it to an excel file, and I added average and sum columns to allow sorting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "bd075f34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Bookarto R1</th>\n",
       "      <th>Bookarto R2</th>\n",
       "      <th>Bookarto R3</th>\n",
       "      <th>Karrku R1</th>\n",
       "      <th>Karrku R2</th>\n",
       "      <th>Moana R1</th>\n",
       "      <th>Moana R2</th>\n",
       "      <th>Moana R3</th>\n",
       "      <th>Wilgie Mia R1</th>\n",
       "      <th>Wilgia Mia R2</th>\n",
       "      <th>Wilgia Mia R3</th>\n",
       "      <th>Average</th>\n",
       "      <th>Sum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Actinomycetales</td>\n",
       "      <td>0.304289</td>\n",
       "      <td>0.334295</td>\n",
       "      <td>0.382574</td>\n",
       "      <td>0.083093</td>\n",
       "      <td>0.417965</td>\n",
       "      <td>0.125024</td>\n",
       "      <td>0.152529</td>\n",
       "      <td>0.255049</td>\n",
       "      <td>0.070013</td>\n",
       "      <td>0.164839</td>\n",
       "      <td>0.046932</td>\n",
       "      <td>0.212418</td>\n",
       "      <td>2.336603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bacillales</td>\n",
       "      <td>0.417965</td>\n",
       "      <td>0.239084</td>\n",
       "      <td>0.283131</td>\n",
       "      <td>0.268898</td>\n",
       "      <td>0.028082</td>\n",
       "      <td>0.099635</td>\n",
       "      <td>0.038276</td>\n",
       "      <td>0.037507</td>\n",
       "      <td>0.108867</td>\n",
       "      <td>0.008655</td>\n",
       "      <td>0.019042</td>\n",
       "      <td>0.140831</td>\n",
       "      <td>1.549144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>0.015388</td>\n",
       "      <td>0.020196</td>\n",
       "      <td>0.018465</td>\n",
       "      <td>0.046740</td>\n",
       "      <td>0.075784</td>\n",
       "      <td>0.000962</td>\n",
       "      <td>0.006924</td>\n",
       "      <td>0.013272</td>\n",
       "      <td>0.193691</td>\n",
       "      <td>0.472014</td>\n",
       "      <td>0.408155</td>\n",
       "      <td>0.115599</td>\n",
       "      <td>1.271590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Rhizobiales</td>\n",
       "      <td>0.027890</td>\n",
       "      <td>0.086747</td>\n",
       "      <td>0.085593</td>\n",
       "      <td>0.031737</td>\n",
       "      <td>0.095403</td>\n",
       "      <td>0.109636</td>\n",
       "      <td>0.043854</td>\n",
       "      <td>0.006732</td>\n",
       "      <td>0.278515</td>\n",
       "      <td>0.173302</td>\n",
       "      <td>0.216772</td>\n",
       "      <td>0.105108</td>\n",
       "      <td>1.156183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Rhodobacterales</td>\n",
       "      <td>0.020004</td>\n",
       "      <td>0.065974</td>\n",
       "      <td>0.033276</td>\n",
       "      <td>0.001346</td>\n",
       "      <td>0.060781</td>\n",
       "      <td>0.149452</td>\n",
       "      <td>0.043278</td>\n",
       "      <td>0.358723</td>\n",
       "      <td>0.001923</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.100211</td>\n",
       "      <td>0.075941</td>\n",
       "      <td>0.835353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Sva0725</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Syntrophobacterales</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Thiotrichales</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>TK10 (unknown order)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>TM7SC3 (unknown order)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000192</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Taxon      Bookarto R1  Bookarto R2  Bookarto R3  \\\n",
       "0         Actinomycetales         0.304289     0.334295     0.382574   \n",
       "1              Bacillales         0.417965     0.239084     0.283131   \n",
       "2        Rhodospirillales         0.015388     0.020196     0.018465   \n",
       "3             Rhizobiales         0.027890     0.086747     0.085593   \n",
       "4         Rhodobacterales         0.020004     0.065974     0.033276   \n",
       "..                        ...          ...          ...          ...   \n",
       "96                Sva0725         0.000000     0.000192     0.000000   \n",
       "97    Syntrophobacterales         0.000000     0.000192     0.000000   \n",
       "98          Thiotrichales         0.000000     0.000192     0.000000   \n",
       "99   TK10 (unknown order)         0.000000     0.000192     0.000000   \n",
       "100    TM7SC3 (unknown order)     0.000000     0.000000     0.000192   \n",
       "\n",
       "     Karrku R1      Karrku R2      Moana R1      Moana R2      Moana R3      \\\n",
       "0         0.083093       0.417965      0.125024      0.152529      0.255049   \n",
       "1         0.268898       0.028082      0.099635      0.038276      0.037507   \n",
       "2         0.046740       0.075784      0.000962      0.006924      0.013272   \n",
       "3         0.031737       0.095403      0.109636      0.043854      0.006732   \n",
       "4         0.001346       0.060781      0.149452      0.043278      0.358723   \n",
       "..             ...            ...           ...           ...           ...   \n",
       "96        0.000000       0.000000      0.000000      0.000000      0.000000   \n",
       "97        0.000000       0.000000      0.000000      0.000000      0.000000   \n",
       "98        0.000000       0.000000      0.000000      0.000000      0.000000   \n",
       "99        0.000000       0.000000      0.000000      0.000000      0.000000   \n",
       "100       0.000000       0.000000      0.000000      0.000000      0.000000   \n",
       "\n",
       "     Wilgie Mia R1     Wilgia Mia R2     Wilgia Mia R3     Average       Sum  \n",
       "0            0.070013          0.164839         0.046932  0.212418  2.336603  \n",
       "1            0.108867          0.008655         0.019042  0.140831  1.549144  \n",
       "2            0.193691          0.472014         0.408155  0.115599  1.271590  \n",
       "3            0.278515          0.173302         0.216772  0.105108  1.156183  \n",
       "4            0.001923          0.000385         0.100211  0.075941  0.835353  \n",
       "..                ...               ...              ...       ...       ...  \n",
       "96           0.000000          0.000000         0.000000  0.000017  0.000192  \n",
       "97           0.000000          0.000000         0.000000  0.000017  0.000192  \n",
       "98           0.000000          0.000000         0.000000  0.000017  0.000192  \n",
       "99           0.000000          0.000000         0.000000  0.000017  0.000192  \n",
       "100          0.000000          0.000000         0.000000  0.000017  0.000192  \n",
       "\n",
       "[101 rows x 14 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xldata = pd.read_excel('Australia/Lenehan_Table_S2.xlsx', \"Sortable\")\n",
    "xldata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "bf3d9d6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Firmicutes', 'Proteobacteria', 'Actinobacteriota', 'Bacteroidota'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xlorders = [x.strip() for x in xldata[xldata['Average'] > 0.01].iloc[:,0]]\n",
    "xlgtdb = gtdb[gtdb['order'].isin(xlorders)]\n",
    "pd.unique(xlgtdb['phylum'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "eaa2aeae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_lineage</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>coding_density</th>\n",
       "      <th>...</th>\n",
       "      <th>trna_count</th>\n",
       "      <th>trna_selenocysteine_count</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>assembly_nover</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GB_GCA_000006155.2</td>\n",
       "      <td>1916</td>\n",
       "      <td>93.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1171</td>\n",
       "      <td>g__Bacillus (UID902)</td>\n",
       "      <td>324</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4305660</td>\n",
       "      <td>80.178992</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Bacillales</td>\n",
       "      <td>Bacillaceae_G</td>\n",
       "      <td>Bacillus_A</td>\n",
       "      <td>Bacillus_A anthracis</td>\n",
       "      <td>GCA_000006155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GB_GCA_000007385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.82</td>\n",
       "      <td>0.00</td>\n",
       "      <td>481</td>\n",
       "      <td>c__Gammaproteobacteria (UID4202)</td>\n",
       "      <td>276</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4190634</td>\n",
       "      <td>84.805944</td>\n",
       "      <td>...</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Xanthomonas</td>\n",
       "      <td>Xanthomonas oryzae</td>\n",
       "      <td>GCA_000007385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GB_GCA_000008605.1</td>\n",
       "      <td>67</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>235</td>\n",
       "      <td>f__Spirochaetaceae (UID2512)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1048744</td>\n",
       "      <td>92.155875</td>\n",
       "      <td>...</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Spirochaetota</td>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>Treponematales</td>\n",
       "      <td>Treponemataceae</td>\n",
       "      <td>Treponema</td>\n",
       "      <td>Treponema pallidum</td>\n",
       "      <td>GCA_000008605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GB_GCA_000010565.1</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.63</td>\n",
       "      <td>295</td>\n",
       "      <td>p__Firmicutes (UID1022)</td>\n",
       "      <td>158</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2608397</td>\n",
       "      <td>86.217312</td>\n",
       "      <td>...</td>\n",
       "      <td>51</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_B</td>\n",
       "      <td>Desulfotomaculia</td>\n",
       "      <td>Desulfotomaculales</td>\n",
       "      <td>Pelotomaculaceae</td>\n",
       "      <td>Pelotomaculum</td>\n",
       "      <td>Pelotomaculum thermopropionicum</td>\n",
       "      <td>GCA_000010565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GB_GCA_000013845.2</td>\n",
       "      <td>0</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>332</td>\n",
       "      <td>o__Clostridiales (UID1375)</td>\n",
       "      <td>124</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2428396</td>\n",
       "      <td>82.037966</td>\n",
       "      <td>...</td>\n",
       "      <td>95</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>Clostridiaceae</td>\n",
       "      <td>Clostridium_P</td>\n",
       "      <td>Clostridium_P perfringens</td>\n",
       "      <td>GCA_000013845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191522</th>\n",
       "      <td>RS_GCF_902166935.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.62</td>\n",
       "      <td>0.09</td>\n",
       "      <td>1312</td>\n",
       "      <td>g__Klebsiella (UID5140)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4894244</td>\n",
       "      <td>87.293148</td>\n",
       "      <td>...</td>\n",
       "      <td>87</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Klebsiella</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>GCA_902166935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191523</th>\n",
       "      <td>RS_GCF_902166945.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.94</td>\n",
       "      <td>0.72</td>\n",
       "      <td>1162</td>\n",
       "      <td>f__Enterobacteriaceae (UID5121)</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5162721</td>\n",
       "      <td>86.540042</td>\n",
       "      <td>...</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>Klebsiella</td>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>GCA_902166945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191524</th>\n",
       "      <td>RS_GCF_902167295.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5737447</td>\n",
       "      <td>90.122088</td>\n",
       "      <td>...</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagales</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Chitinophaga</td>\n",
       "      <td>Chitinophaga pinensis_A</td>\n",
       "      <td>GCA_902167295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191525</th>\n",
       "      <td>RS_GCF_902167305.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5738096</td>\n",
       "      <td>90.147873</td>\n",
       "      <td>...</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagales</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Chitinophaga</td>\n",
       "      <td>Chitinophaga pinensis_A</td>\n",
       "      <td>GCA_902167305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191526</th>\n",
       "      <td>RS_GCF_902167325.1</td>\n",
       "      <td>0</td>\n",
       "      <td>99.51</td>\n",
       "      <td>0.00</td>\n",
       "      <td>303</td>\n",
       "      <td>p__Bacteroidetes (UID2591)</td>\n",
       "      <td>203</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5671443</td>\n",
       "      <td>90.098258</td>\n",
       "      <td>...</td>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagales</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Chitinophaga</td>\n",
       "      <td>Chitinophaga pinensis_A</td>\n",
       "      <td>GCA_902167325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191527 rows × 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 accession  ambiguous_bases  checkm_completeness  \\\n",
       "0       GB_GCA_000006155.2             1916                93.12   \n",
       "1       GB_GCA_000007385.1                0                99.82   \n",
       "2       GB_GCA_000008605.1               67               100.00   \n",
       "3       GB_GCA_000010565.1                0               100.00   \n",
       "4       GB_GCA_000013845.2                0               100.00   \n",
       "...                    ...              ...                  ...   \n",
       "191522  RS_GCF_902166935.1                0                99.62   \n",
       "191523  RS_GCF_902166945.1                0                99.94   \n",
       "191524  RS_GCF_902167295.1                0                99.51   \n",
       "191525  RS_GCF_902167305.1                0                99.51   \n",
       "191526  RS_GCF_902167325.1                0                99.51   \n",
       "\n",
       "        checkm_contamination  checkm_marker_count  \\\n",
       "0                       0.00                 1171   \n",
       "1                       0.00                  481   \n",
       "2                       0.00                  235   \n",
       "3                       0.63                  295   \n",
       "4                       0.00                  332   \n",
       "...                      ...                  ...   \n",
       "191522                  0.09                 1312   \n",
       "191523                  0.72                 1162   \n",
       "191524                  0.00                  303   \n",
       "191525                  0.00                  303   \n",
       "191526                  0.00                  303   \n",
       "\n",
       "                   checkm_marker_lineage  checkm_marker_set_count  \\\n",
       "0                   g__Bacillus (UID902)                      324   \n",
       "1       c__Gammaproteobacteria (UID4202)                      276   \n",
       "2           f__Spirochaetaceae (UID2512)                      124   \n",
       "3                p__Firmicutes (UID1022)                      158   \n",
       "4             o__Clostridiales (UID1375)                      124   \n",
       "...                                  ...                      ...   \n",
       "191522           g__Klebsiella (UID5140)                      336   \n",
       "191523   f__Enterobacteriaceae (UID5121)                      336   \n",
       "191524        p__Bacteroidetes (UID2591)                      203   \n",
       "191525        p__Bacteroidetes (UID2591)                      203   \n",
       "191526        p__Bacteroidetes (UID2591)                      203   \n",
       "\n",
       "        checkm_strain_heterogeneity  coding_bases  coding_density  ...  \\\n",
       "0                               0.0       4305660       80.178992  ...   \n",
       "1                               0.0       4190634       84.805944  ...   \n",
       "2                               0.0       1048744       92.155875  ...   \n",
       "3                               0.0       2608397       86.217312  ...   \n",
       "4                               0.0       2428396       82.037966  ...   \n",
       "...                             ...           ...             ...  ...   \n",
       "191522                          0.0       4894244       87.293148  ...   \n",
       "191523                          0.0       5162721       86.540042  ...   \n",
       "191524                          0.0       5737447       90.122088  ...   \n",
       "191525                          0.0       5738096       90.147873  ...   \n",
       "191526                          0.0       5671443       90.098258  ...   \n",
       "\n",
       "        trna_count  trna_selenocysteine_count    domain          phylum  \\\n",
       "0               31                          0  Bacteria      Firmicutes   \n",
       "1               53                          0  Bacteria  Proteobacteria   \n",
       "2               45                          0  Bacteria   Spirochaetota   \n",
       "3               51                          1  Bacteria    Firmicutes_B   \n",
       "4               95                          1  Bacteria    Firmicutes_A   \n",
       "...            ...                        ...       ...             ...   \n",
       "191522          87                          1  Bacteria  Proteobacteria   \n",
       "191523          86                          1  Bacteria  Proteobacteria   \n",
       "191524          58                          0  Bacteria    Bacteroidota   \n",
       "191525          58                          0  Bacteria    Bacteroidota   \n",
       "191526          59                          0  Bacteria    Bacteroidota   \n",
       "\n",
       "                      class               order              family  \\\n",
       "0                   Bacilli          Bacillales       Bacillaceae_G   \n",
       "1       Gammaproteobacteria     Xanthomonadales    Xanthomonadaceae   \n",
       "2              Spirochaetia      Treponematales     Treponemataceae   \n",
       "3          Desulfotomaculia  Desulfotomaculales    Pelotomaculaceae   \n",
       "4                Clostridia       Clostridiales      Clostridiaceae   \n",
       "...                     ...                 ...                 ...   \n",
       "191522  Gammaproteobacteria    Enterobacterales  Enterobacteriaceae   \n",
       "191523  Gammaproteobacteria    Enterobacterales  Enterobacteriaceae   \n",
       "191524          Bacteroidia     Chitinophagales    Chitinophagaceae   \n",
       "191525          Bacteroidia     Chitinophagales    Chitinophagaceae   \n",
       "191526          Bacteroidia     Chitinophagales    Chitinophagaceae   \n",
       "\n",
       "                genus                          species assembly_nover  \n",
       "0          Bacillus_A             Bacillus_A anthracis  GCA_000006155  \n",
       "1         Xanthomonas               Xanthomonas oryzae  GCA_000007385  \n",
       "2           Treponema               Treponema pallidum  GCA_000008605  \n",
       "3       Pelotomaculum  Pelotomaculum thermopropionicum  GCA_000010565  \n",
       "4       Clostridium_P        Clostridium_P perfringens  GCA_000013845  \n",
       "...               ...                              ...            ...  \n",
       "191522     Klebsiella            Klebsiella pneumoniae  GCA_902166935  \n",
       "191523     Klebsiella            Klebsiella pneumoniae  GCA_902166945  \n",
       "191524   Chitinophaga          Chitinophaga pinensis_A  GCA_902167295  \n",
       "191525   Chitinophaga          Chitinophaga pinensis_A  GCA_902167305  \n",
       "191526   Chitinophaga          Chitinophaga pinensis_A  GCA_902167325  \n",
       "\n",
       "[191527 rows x 118 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ad7aae7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "183423c4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e50ba74",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e87abed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c603a360",
   "metadata": {},
   "source": [
    "### Select a few candidate examples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e94750be",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "47f1f4a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phylum</th>\n",
       "      <th>ambiguous_bases</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>checkm_marker_count</th>\n",
       "      <th>checkm_marker_set_count</th>\n",
       "      <th>checkm_strain_heterogeneity</th>\n",
       "      <th>coding_bases</th>\n",
       "      <th>coding_density</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>...</th>\n",
       "      <th>trna_count</th>\n",
       "      <th>trna_selenocysteine_count</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Desulfobacterota</td>\n",
       "      <td>870</td>\n",
       "      <td>837.67</td>\n",
       "      <td>17.16</td>\n",
       "      <td>2223</td>\n",
       "      <td>1395</td>\n",
       "      <td>240.00</td>\n",
       "      <td>27980313</td>\n",
       "      <td>7.734006e+02</td>\n",
       "      <td>1255</td>\n",
       "      <td>...</td>\n",
       "      <td>371</td>\n",
       "      <td>8</td>\n",
       "      <td>3.267283e+07</td>\n",
       "      <td>957.0</td>\n",
       "      <td>730.0</td>\n",
       "      <td>190.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>2.071120e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Chloroflexota</td>\n",
       "      <td>111</td>\n",
       "      <td>2346.18</td>\n",
       "      <td>42.55</td>\n",
       "      <td>3952</td>\n",
       "      <td>2692</td>\n",
       "      <td>70.24</td>\n",
       "      <td>112260883</td>\n",
       "      <td>2.193711e+03</td>\n",
       "      <td>11081</td>\n",
       "      <td>...</td>\n",
       "      <td>1161</td>\n",
       "      <td>12</td>\n",
       "      <td>1.286956e+08</td>\n",
       "      <td>9943.0</td>\n",
       "      <td>2810.0</td>\n",
       "      <td>862.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>445.0</td>\n",
       "      <td>7.922670e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Gemmatimonadota</td>\n",
       "      <td>1650</td>\n",
       "      <td>5732.92</td>\n",
       "      <td>166.68</td>\n",
       "      <td>9634</td>\n",
       "      <td>5932</td>\n",
       "      <td>1395.36</td>\n",
       "      <td>196296537</td>\n",
       "      <td>6.261433e+03</td>\n",
       "      <td>16835</td>\n",
       "      <td>...</td>\n",
       "      <td>2337</td>\n",
       "      <td>39</td>\n",
       "      <td>2.103923e+08</td>\n",
       "      <td>14156.0</td>\n",
       "      <td>9150.0</td>\n",
       "      <td>936.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>348.0</td>\n",
       "      <td>1.163162e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Synergistota</td>\n",
       "      <td>1975</td>\n",
       "      <td>3905.17</td>\n",
       "      <td>17.65</td>\n",
       "      <td>4301</td>\n",
       "      <td>2415</td>\n",
       "      <td>296.66</td>\n",
       "      <td>81842508</td>\n",
       "      <td>3.676171e+03</td>\n",
       "      <td>4170</td>\n",
       "      <td>...</td>\n",
       "      <td>1736</td>\n",
       "      <td>13</td>\n",
       "      <td>9.162332e+07</td>\n",
       "      <td>3306.0</td>\n",
       "      <td>1713.0</td>\n",
       "      <td>534.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>319.0</td>\n",
       "      <td>1.739537e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Verrucomicrobiota</td>\n",
       "      <td>3458</td>\n",
       "      <td>9846.03</td>\n",
       "      <td>343.13</td>\n",
       "      <td>26910</td>\n",
       "      <td>17316</td>\n",
       "      <td>3975.34</td>\n",
       "      <td>277225701</td>\n",
       "      <td>1.045643e+04</td>\n",
       "      <td>37096</td>\n",
       "      <td>...</td>\n",
       "      <td>4163</td>\n",
       "      <td>64</td>\n",
       "      <td>3.103284e+08</td>\n",
       "      <td>32445.0</td>\n",
       "      <td>18230.0</td>\n",
       "      <td>4528.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>3401.0</td>\n",
       "      <td>1067.0</td>\n",
       "      <td>1.177506e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Nitrospirota</td>\n",
       "      <td>5338</td>\n",
       "      <td>5826.35</td>\n",
       "      <td>196.93</td>\n",
       "      <td>11790</td>\n",
       "      <td>7154</td>\n",
       "      <td>736.06</td>\n",
       "      <td>178792106</td>\n",
       "      <td>5.765020e+03</td>\n",
       "      <td>15123</td>\n",
       "      <td>...</td>\n",
       "      <td>2483</td>\n",
       "      <td>3</td>\n",
       "      <td>2.052842e+08</td>\n",
       "      <td>9957.0</td>\n",
       "      <td>5806.0</td>\n",
       "      <td>1480.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>557.0</td>\n",
       "      <td>836.0</td>\n",
       "      <td>2.240749e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Deinococcota</td>\n",
       "      <td>12302</td>\n",
       "      <td>10940.03</td>\n",
       "      <td>150.95</td>\n",
       "      <td>35023</td>\n",
       "      <td>26143</td>\n",
       "      <td>1468.75</td>\n",
       "      <td>293564020</td>\n",
       "      <td>1.014214e+04</td>\n",
       "      <td>19884</td>\n",
       "      <td>...</td>\n",
       "      <td>5309</td>\n",
       "      <td>0</td>\n",
       "      <td>3.235184e+08</td>\n",
       "      <td>19543.0</td>\n",
       "      <td>8605.0</td>\n",
       "      <td>2076.0</td>\n",
       "      <td>140.0</td>\n",
       "      <td>563.0</td>\n",
       "      <td>1373.0</td>\n",
       "      <td>3.809000e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Planctomycetota</td>\n",
       "      <td>157602</td>\n",
       "      <td>10614.14</td>\n",
       "      <td>230.41</td>\n",
       "      <td>17232</td>\n",
       "      <td>10519</td>\n",
       "      <td>2020.66</td>\n",
       "      <td>579946278</td>\n",
       "      <td>1.070008e+04</td>\n",
       "      <td>29778</td>\n",
       "      <td>...</td>\n",
       "      <td>5355</td>\n",
       "      <td>38</td>\n",
       "      <td>6.631786e+08</td>\n",
       "      <td>25660.0</td>\n",
       "      <td>20452.0</td>\n",
       "      <td>8652.0</td>\n",
       "      <td>307.0</td>\n",
       "      <td>5594.0</td>\n",
       "      <td>2751.0</td>\n",
       "      <td>6.938966e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Myxococcota</td>\n",
       "      <td>2171</td>\n",
       "      <td>7593.06</td>\n",
       "      <td>175.63</td>\n",
       "      <td>19463</td>\n",
       "      <td>12200</td>\n",
       "      <td>601.23</td>\n",
       "      <td>614570629</td>\n",
       "      <td>7.205088e+03</td>\n",
       "      <td>24144</td>\n",
       "      <td>...</td>\n",
       "      <td>4178</td>\n",
       "      <td>61</td>\n",
       "      <td>6.746475e+08</td>\n",
       "      <td>22704.0</td>\n",
       "      <td>13890.0</td>\n",
       "      <td>7567.0</td>\n",
       "      <td>473.0</td>\n",
       "      <td>4494.0</td>\n",
       "      <td>2600.0</td>\n",
       "      <td>1.033916e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Firmicutes_A</td>\n",
       "      <td>4826</td>\n",
       "      <td>50833.03</td>\n",
       "      <td>407.04</td>\n",
       "      <td>186549</td>\n",
       "      <td>69248</td>\n",
       "      <td>3315.46</td>\n",
       "      <td>1605801224</td>\n",
       "      <td>4.367840e+04</td>\n",
       "      <td>47276</td>\n",
       "      <td>...</td>\n",
       "      <td>36259</td>\n",
       "      <td>238</td>\n",
       "      <td>1.920572e+09</td>\n",
       "      <td>40838.0</td>\n",
       "      <td>23596.0</td>\n",
       "      <td>13981.0</td>\n",
       "      <td>2122.0</td>\n",
       "      <td>3122.0</td>\n",
       "      <td>8737.0</td>\n",
       "      <td>7.250677e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Campylobacterota</td>\n",
       "      <td>75712</td>\n",
       "      <td>247549.00</td>\n",
       "      <td>1049.53</td>\n",
       "      <td>1776796</td>\n",
       "      <td>543639</td>\n",
       "      <td>33765.75</td>\n",
       "      <td>4044000669</td>\n",
       "      <td>2.308418e+05</td>\n",
       "      <td>153383</td>\n",
       "      <td>...</td>\n",
       "      <td>96364</td>\n",
       "      <td>1219</td>\n",
       "      <td>4.383539e+09</td>\n",
       "      <td>146410.0</td>\n",
       "      <td>68951.0</td>\n",
       "      <td>27366.0</td>\n",
       "      <td>2764.0</td>\n",
       "      <td>9275.0</td>\n",
       "      <td>15327.0</td>\n",
       "      <td>6.845617e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Actinobacteriota</td>\n",
       "      <td>491779</td>\n",
       "      <td>287616.58</td>\n",
       "      <td>2399.52</td>\n",
       "      <td>1104155</td>\n",
       "      <td>553644</td>\n",
       "      <td>30844.49</td>\n",
       "      <td>7648679388</td>\n",
       "      <td>2.613134e+05</td>\n",
       "      <td>219615</td>\n",
       "      <td>...</td>\n",
       "      <td>143651</td>\n",
       "      <td>276</td>\n",
       "      <td>8.593107e+09</td>\n",
       "      <td>174607.0</td>\n",
       "      <td>104834.0</td>\n",
       "      <td>63911.0</td>\n",
       "      <td>5408.0</td>\n",
       "      <td>18645.0</td>\n",
       "      <td>39858.0</td>\n",
       "      <td>1.369885e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>1420426</td>\n",
       "      <td>341502.51</td>\n",
       "      <td>3151.20</td>\n",
       "      <td>1557319</td>\n",
       "      <td>891153</td>\n",
       "      <td>41222.30</td>\n",
       "      <td>12697874268</td>\n",
       "      <td>3.202252e+05</td>\n",
       "      <td>406070</td>\n",
       "      <td>...</td>\n",
       "      <td>168253</td>\n",
       "      <td>50</td>\n",
       "      <td>1.422641e+10</td>\n",
       "      <td>323390.0</td>\n",
       "      <td>200750.0</td>\n",
       "      <td>115723.0</td>\n",
       "      <td>8920.0</td>\n",
       "      <td>30027.0</td>\n",
       "      <td>76776.0</td>\n",
       "      <td>2.372777e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>445416</td>\n",
       "      <td>2347447.73</td>\n",
       "      <td>12030.17</td>\n",
       "      <td>14660704</td>\n",
       "      <td>5538249</td>\n",
       "      <td>277328.74</td>\n",
       "      <td>56108892042</td>\n",
       "      <td>2.042376e+06</td>\n",
       "      <td>1921765</td>\n",
       "      <td>...</td>\n",
       "      <td>1231245</td>\n",
       "      <td>242</td>\n",
       "      <td>6.521015e+10</td>\n",
       "      <td>1760727.0</td>\n",
       "      <td>962285.0</td>\n",
       "      <td>575517.0</td>\n",
       "      <td>81867.0</td>\n",
       "      <td>136937.0</td>\n",
       "      <td>356713.0</td>\n",
       "      <td>2.064300e+09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>4051327</td>\n",
       "      <td>2259601.10</td>\n",
       "      <td>17569.10</td>\n",
       "      <td>16984390</td>\n",
       "      <td>6877996</td>\n",
       "      <td>331545.86</td>\n",
       "      <td>99711393779</td>\n",
       "      <td>2.006879e+06</td>\n",
       "      <td>2968989</td>\n",
       "      <td>...</td>\n",
       "      <td>1236840</td>\n",
       "      <td>4361</td>\n",
       "      <td>1.138997e+11</td>\n",
       "      <td>2661865.0</td>\n",
       "      <td>1499900.0</td>\n",
       "      <td>911514.0</td>\n",
       "      <td>98508.0</td>\n",
       "      <td>212374.0</td>\n",
       "      <td>600632.0</td>\n",
       "      <td>2.751603e+09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15 rows × 79 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               phylum  ambiguous_bases  checkm_completeness  \\\n",
       "5    Desulfobacterota              870               837.67   \n",
       "3       Chloroflexota              111              2346.18   \n",
       "8     Gemmatimonadota             1650              5732.92   \n",
       "13       Synergistota             1975              3905.17   \n",
       "14  Verrucomicrobiota             3458              9846.03   \n",
       "10       Nitrospirota             5338              5826.35   \n",
       "4        Deinococcota            12302             10940.03   \n",
       "11    Planctomycetota           157602             10614.14   \n",
       "9         Myxococcota             2171              7593.06   \n",
       "7        Firmicutes_A             4826             50833.03   \n",
       "2    Campylobacterota            75712            247549.00   \n",
       "0    Actinobacteriota           491779            287616.58   \n",
       "1        Bacteroidota          1420426            341502.51   \n",
       "6          Firmicutes           445416           2347447.73   \n",
       "12     Proteobacteria          4051327           2259601.10   \n",
       "\n",
       "    checkm_contamination  checkm_marker_count  checkm_marker_set_count  \\\n",
       "5                  17.16                 2223                     1395   \n",
       "3                  42.55                 3952                     2692   \n",
       "8                 166.68                 9634                     5932   \n",
       "13                 17.65                 4301                     2415   \n",
       "14                343.13                26910                    17316   \n",
       "10                196.93                11790                     7154   \n",
       "4                 150.95                35023                    26143   \n",
       "11                230.41                17232                    10519   \n",
       "9                 175.63                19463                    12200   \n",
       "7                 407.04               186549                    69248   \n",
       "2                1049.53              1776796                   543639   \n",
       "0                2399.52              1104155                   553644   \n",
       "1                3151.20              1557319                   891153   \n",
       "6               12030.17             14660704                  5538249   \n",
       "12              17569.10             16984390                  6877996   \n",
       "\n",
       "    checkm_strain_heterogeneity  coding_bases  coding_density  contig_count  \\\n",
       "5                        240.00      27980313    7.734006e+02          1255   \n",
       "3                         70.24     112260883    2.193711e+03         11081   \n",
       "8                       1395.36     196296537    6.261433e+03         16835   \n",
       "13                       296.66      81842508    3.676171e+03          4170   \n",
       "14                      3975.34     277225701    1.045643e+04         37096   \n",
       "10                       736.06     178792106    5.765020e+03         15123   \n",
       "4                       1468.75     293564020    1.014214e+04         19884   \n",
       "11                      2020.66     579946278    1.070008e+04         29778   \n",
       "9                        601.23     614570629    7.205088e+03         24144   \n",
       "7                       3315.46    1605801224    4.367840e+04         47276   \n",
       "2                      33765.75    4044000669    2.308418e+05        153383   \n",
       "0                      30844.49    7648679388    2.613134e+05        219615   \n",
       "1                      41222.30   12697874268    3.202252e+05        406070   \n",
       "6                     277328.74   56108892042    2.042376e+06       1921765   \n",
       "12                    331545.86   99711393779    2.006879e+06       2968989   \n",
       "\n",
       "    ...  trna_count  trna_selenocysteine_count  Genome length    Contigs  \\\n",
       "5   ...         371                          8   3.267283e+07      957.0   \n",
       "3   ...        1161                         12   1.286956e+08     9943.0   \n",
       "8   ...        2337                         39   2.103923e+08    14156.0   \n",
       "13  ...        1736                         13   9.162332e+07     3306.0   \n",
       "14  ...        4163                         64   3.103284e+08    32445.0   \n",
       "10  ...        2483                          3   2.052842e+08     9957.0   \n",
       "4   ...        5309                          0   3.235184e+08    19543.0   \n",
       "11  ...        5355                         38   6.631786e+08    25660.0   \n",
       "9   ...        4178                         61   6.746475e+08    22704.0   \n",
       "7   ...       36259                        238   1.920572e+09    40838.0   \n",
       "2   ...       96364                       1219   4.383539e+09   146410.0   \n",
       "0   ...      143651                        276   8.593107e+09   174607.0   \n",
       "1   ...      168253                         50   1.422641e+10   323390.0   \n",
       "6   ...     1231245                        242   6.521015e+10  1760727.0   \n",
       "12  ...     1236840                       4361   1.138997e+11  2661865.0   \n",
       "\n",
       "    Phage Contigs  Total Predicted Prophages     Kept  No phage genes  \\\n",
       "5           730.0                      190.0      8.0            62.0   \n",
       "3          2810.0                      862.0     26.0           391.0   \n",
       "8          9150.0                      936.0     42.0           546.0   \n",
       "13         1713.0                      534.0     54.0           161.0   \n",
       "14        18230.0                     4528.0     60.0          3401.0   \n",
       "10         5806.0                     1480.0     87.0           557.0   \n",
       "4          8605.0                     2076.0    140.0           563.0   \n",
       "11        20452.0                     8652.0    307.0          5594.0   \n",
       "9         13890.0                     7567.0    473.0          4494.0   \n",
       "7         23596.0                    13981.0   2122.0          3122.0   \n",
       "2         68951.0                    27366.0   2764.0          9275.0   \n",
       "0        104834.0                    63911.0   5408.0         18645.0   \n",
       "1        200750.0                   115723.0   8920.0         30027.0   \n",
       "6        962285.0                   575517.0  81867.0        136937.0   \n",
       "12      1499900.0                   911514.0  98508.0        212374.0   \n",
       "\n",
       "    Not enough genes   bp prophage  \n",
       "5              120.0  2.071120e+05  \n",
       "3              445.0  7.922670e+05  \n",
       "8              348.0  1.163162e+06  \n",
       "13             319.0  1.739537e+06  \n",
       "14            1067.0  1.177506e+06  \n",
       "10             836.0  2.240749e+06  \n",
       "4             1373.0  3.809000e+06  \n",
       "11            2751.0  6.938966e+06  \n",
       "9             2600.0  1.033916e+07  \n",
       "7             8737.0  7.250677e+07  \n",
       "2            15327.0  6.845617e+07  \n",
       "0            39858.0  1.369885e+08  \n",
       "1            76776.0  2.372777e+08  \n",
       "6           356713.0  2.064300e+09  \n",
       "12          600632.0  2.751603e+09  \n",
       "\n",
       "[15 rows x 79 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ochre_phyla = ochre_phage.groupby('phylum').agg('sum').reset_index()\n",
    "ochre_phyla.sort_values('Kept')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1ca6764",
   "metadata": {},
   "source": [
    "# Acid Mines\n",
    "\n",
    "This is the email from Sarah Harmer:\n",
    "\n",
    ">My lab has the following microbes, it would be great to know about their phage:\n",
    ">\n",
    ">*Acidithiobacillus ferrooxidans*<br />\n",
    ">*Leptospirillum ferrooxidans*<br />\n",
    ">*Acidianus brierleyi* (note: Archaea that is not included in this analysis)\n",
    ">\n",
    ">These microbes will be of interest too:<br />\n",
    ">*Thermoplasmales, Sulfobacillus, Acidimicrobium*, and *Acidiphilium* groups\n",
    ">\n",
    ">Anerobic sulfate reducing bacteria are definitely of interest.\n",
    "\n",
    "See also this table of interesting microbes:\n",
    "![I bet some of these have phages!](Australia/amd_microbes.png \"Do you want to bet some of these have phages?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13639787",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
