{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "portuguese-exclusion",
   "metadata": {},
   "source": [
    "# Merge the prophage predictions with RAST metadata\n",
    "\n",
    "Combine RAST metadata and prophage predictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "frequent-lancaster",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "# for parsing collection dates\n",
    "from dateutil.parser import parse, ParserError\n",
    "import pytz\n",
    "\n",
    "import subprocess\n",
    "import gzip\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "prospective-republican",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this is a neat trick for getting markdown in our output\n",
    "# see https://stackoverflow.com/questions/23271575/printing-bold-colored-etc-text-in-ipython-qtconsole\n",
    "# for the inspiration\n",
    "from IPython.display import Markdown, display\n",
    "def printmd(string, color=\"black\"):\n",
    "    colorstr = \"<span style='color:{}'>{}</span>\".format(color, string)\n",
    "    display(Markdown(colorstr))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cleared-shuttle",
   "metadata": {},
   "source": [
    "# Read the phage data. Check the version!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "overall-venice",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Please note that this was run with git commit 361ddcb that has 553,082 genomes parsed and 20,946,107 total prophages\n"
     ]
    }
   ],
   "source": [
    "#phagesdf = pd.read_csv(\"../small_data/phages_per_genome.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "phagesdf = pd.read_csv(\"../data/phages_per_genome.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "githash = subprocess.check_output([\"git\", \"describe\", \"--always\"]).strip().decode()\n",
    "print(f\"Please note that this was run with git commit {githash} that has {phagesdf.shape[0]:,} genomes parsed and {phagesdf['Total Predicted Prophages'].sum():,} total prophages\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "concrete-credits",
   "metadata": {},
   "source": [
    "### Date conversion\n",
    "\n",
    "This routine is used to convert the dates into something useable. Here we choose a point in time (i.e. Feb 24, 1977) and set all dates to be relative to that. We use years as our unit, but could also use seconds. ([Hint](https://pubmed.ncbi.nlm.nih.gov/870828/))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "pending-insured",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert the collection date to a number. We take todays date, and then convert relative to that\n",
    "# it allows dates pre-epoch\n",
    "\n",
    "yr = re.compile(\"^[~]*(\\\\d{4})['s]*$\") # eg ~1970's\n",
    "yrrange = re.compile(\"^(\\\\d{4})\\\\s*.\\\\s*(\\\\d{4})$\") # eg 2001-2002\n",
    "myd = re.compile('^\\\\d{1,4}/\\\\d{1,2}/\\\\d{1,4}$') # eg 2/3/21\n",
    "moyr = re.compile('^\\\\w{3}-\\\\d{2}$') # eg Dec-09 should be 01-Dec-09 else parsed as 9th Dec this year\n",
    "damoyrrange = re.compile('^(\\\\d+\\\\s+\\\\w+\\\\s+\\\\d{4})\\\\s*.{1,3}?\\\\s*(\\\\d+\\\\s+\\\\w+\\\\s+\\\\d{4})$') \n",
    "modacyrrange = re.compile('^(\\\\w+\\\\s+\\\\d+,*\\\\s+\\\\d{4})\\\\s*.{1,3}?\\\\s*(\\\\w+\\\\s+\\\\d+,*\\\\s+\\\\d{4})$')\n",
    "moyrrange = re.compile('^(\\\\w+\\\\s+\\\\d{4})\\\\s*.{1,3}?\\\\s*(\\\\w+\\\\s+\\\\d{4})$')\n",
    "year42 = re.compile('^(\\\\d{4})-\\\\d{2}$')\n",
    "lem = re.compile(\"^late\\\\s*|^early\\\\s*|^mid\\\\s*|^prior to\\\\s*|^before\\\\s*|^pre-\", re.IGNORECASE)\n",
    "splitseen = set()\n",
    "\n",
    "try:\n",
    "    # adate = parse(\"24/02/1977\")\n",
    "    adate = parse(\"1/1/0001\")\n",
    "except ParserError as e:\n",
    "    sys.stderr.write(\"Error parsing adate: {e}\\n\")\n",
    "    sys.exit(1)\n",
    "    \n",
    "adate = adate.replace(tzinfo=pytz.UTC)\n",
    "\n",
    "def try_parsing(x):\n",
    "    \"\"\"\n",
    "    Attempt to parse a date, and catch an error.\n",
    "    \n",
    "    If we fail, we return None, otherwise we return the years since now()\n",
    "    \"\"\"\n",
    "    try:\n",
    "        dt = parse(x)\n",
    "    except:\n",
    "        return None\n",
    "    \n",
    "    dt = dt.replace(tzinfo=pytz.UTC)\n",
    "    \n",
    "    if dt < adate:\n",
    "        tdelt = adate - dt\n",
    "        seconds = -1 * ((tdelt.days * 86400) + tdelt.seconds)\n",
    "    else:\n",
    "        tdelt = dt - adate\n",
    "        seconds = (tdelt.days * 86400) + tdelt.seconds\n",
    "    # convert seconds to years\n",
    "    # then we add one because our epoch is now Jan 1, 0001\n",
    "    return (seconds/31557600)+1\n",
    "    \n",
    "\n",
    "def convert_date(x, verbose=False):\n",
    "    \"\"\"\n",
    "    Convert the date to years and fractions.\n",
    "    \n",
    "    We try several times, and clean it up as we go along.    \n",
    "    \n",
    "    :param x: the date string to convert\n",
    "    :param verbose: more output information\n",
    "    \"\"\"\n",
    "    if pd.isna(x):\n",
    "        return np.nan\n",
    "    \n",
    "    # we need to fix this before trying to parse\n",
    "    m = moyr.match(x)\n",
    "    if m:\n",
    "        x = '01-' + x\n",
    "\n",
    "    # can we parse this date? If so, lets do it and return the value\n",
    "    attempt = try_parsing(x)\n",
    "    if attempt:\n",
    "        if verbose:\n",
    "            sys.stderr.write(\"Parsed at step 1\\n\")\n",
    "        return attempt\n",
    "    orix = x\n",
    "    \n",
    "    if x.lower() in ['restricted access', 'none', 'not collected', 'not applicable', \n",
    "                     'not available: not collected', 'unspecified']:\n",
    "        return np.nan\n",
    "   \n",
    "    # a few one off cases that are just easier to fix\n",
    "    if 'May 2015-Nov 2015' == x:\n",
    "        x = 'May 2015'\n",
    "    \n",
    "    if '1954-65' == x:\n",
    "        x = '01 January 1954'\n",
    "        \n",
    "    if 'Jul-00' == x:\n",
    "        x = 'Jul-2000'\n",
    "\n",
    "    if '2015_9' == x:\n",
    "        x = 'Sep-2015'\n",
    "    \n",
    "    if '31-Mac-2013' == x:\n",
    "        x = '31-May-2013'\n",
    "    \n",
    "    if '2010-0916' == x:\n",
    "        x = '16 Sep 2010'\n",
    "     \n",
    "    x = lem.sub('', x)\n",
    "    \n",
    "        \n",
    "    if '_' in x:\n",
    "        x = x.replace('_', '-')\n",
    "        \n",
    "    x = x.replace(' or earlier', '')\n",
    "    x = x.replace('collected in the ', '')\n",
    "        \n",
    "    # some regular expressions of variants of day month year - day month year ranges. We choose 1\n",
    "    m = yrrange.match(x)\n",
    "    if m:\n",
    "        x = '01 January ' + m.groups()[1]\n",
    "\n",
    "    m = yr.match(x)\n",
    "    if m:\n",
    "        x = '01 January ' + m.groups()[0]\n",
    "\n",
    "    m = year42.match(x)\n",
    "    if m:\n",
    "        x = '01 January ' + m.groups()[0]\n",
    "            \n",
    "    m = modacyrrange.match(x)\n",
    "    if m:\n",
    "        x = m.groups()[1]\n",
    "        \n",
    "    m = damoyrrange.match(x)\n",
    "    if m:\n",
    "        x = m.groups()[1]\n",
    "    \n",
    "    m = moyrrange.match(x)\n",
    "    if m:\n",
    "        x = '01 ' + m.groups()[1]\n",
    "        \n",
    "    \n",
    "        \n",
    "    # can we parse this date? If so, lets do it and return the value\n",
    "    attempt = try_parsing(x)\n",
    "    if attempt:\n",
    "        if verbose:\n",
    "            sys.stderr.write(f\"Parsed at step 2. Now {orix} is {x}\\n\")\n",
    "        return attempt\n",
    "          \n",
    "    if '/' in x:\n",
    "        if x not in splitseen:\n",
    "            # sys.stderr.write(f\"Splitting {x}\\n\")\n",
    "            splitseen.add(x)\n",
    "        p = x.split('/')\n",
    "        x = p[1]\n",
    "   \n",
    "    # can we parse this date? If so, lets do it and return the value\n",
    "    attempt = try_parsing(x)\n",
    "    if attempt:\n",
    "        if verbose:\n",
    "            sys.stderr.write(f\"Parsed at step 3. Now {orix} is {x}\\n\")\n",
    "        return attempt\n",
    "   \n",
    "    if x.endswith('-00'):\n",
    "        x = x.replace('-00', '-2000')\n",
    "\n",
    "    # can we parse this date? If so, lets do it and return the value\n",
    "    attempt = try_parsing(x)\n",
    "    if attempt:\n",
    "        if verbose:\n",
    "            sys.stderr.write(f\"Parsed at step 4. Now {orix} is {x}\\n\")\n",
    "        return attempt\n",
    "\n",
    "    \n",
    "    sys.stderr.write(f\"can't parse |{x}| from |{orix}|\\n\")\n",
    "            \n",
    "    return np.nan"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "electrical-color",
   "metadata": {},
   "source": [
    "### Read the data file\n",
    "\n",
    "The metadata is from PATRIC. The phage data is from us.\n",
    "\n",
    "**NOTE:** Some of the PATRIC data refers to specific chromosomes/fragments in the GenBank file (e.g. plasmid, chromosome), but that may not equate to our predictions, because we have used the whole GenBank file. The PATRIC metadata is redundant for many fields, and so we just keep the first entry for each NCBI Assembly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "controlling-fiber",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\workspace\\edwa0468\\githubs\\phispyanalysis\\venv\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3165: DtypeWarning: Columns (2,6,7,8,9,10,11,12,14,19,21,34,40,41,42,43,44,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,65) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "can't parse |1999 and 2000| from |1999 and 2000|\n",
      "can't parse |0-Apr-2008| from |0-Apr-2008|\n",
      "can't parse |CGMCC No.5392| from |CGMCC No.5392|\n",
      "can't parse |CGMCC No.5392| from |CGMCC No.5392|\n"
     ]
    }
   ],
   "source": [
    "# the full data set. Don't try this at home!\n",
    "# metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "metadf = pd.read_csv(\"../data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "metadf['isolation_date'] = metadf.collection_date.apply(convert_date)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "outside-porter",
   "metadata": {},
   "source": [
    "## Filter for only the first of each genome assembly\n",
    "\n",
    "    The metadata contains multiple entries for a genome assembly if it is submitted more than once, so here we just filter for the first instance. We might think about something smarter,  but this seems to work"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "greatest-norfolk",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadf = metadf.groupby('assembly_accession').first().reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecological-creator",
   "metadata": {},
   "source": [
    "## Clean up the data\n",
    "\n",
    "These are specific things in the data that we need to clean/replace. Add things here as you find them!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "dramatic-diana",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadf['isolation_country'] = metadf['isolation_country'].replace('USA', 'United States')\n",
    "metadf['isolation_country'] = metadf['isolation_country'].replace('Ecully', 'France')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "proved-application",
   "metadata": {},
   "source": [
    "#### Drop Geographic Location\n",
    "\n",
    "This is just a proxy for `isolation_country` and we don't want it duplicated."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "excellent-surfing",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isolation_country</th>\n",
       "      <th>geographic_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Canada</td>\n",
       "      <td>Canada: Quebec</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>United States</td>\n",
       "      <td>New York State</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>United States</td>\n",
       "      <td>West Palm Beach, Florida</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Denmark</td>\n",
       "      <td>Denmark: Aarhus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Norway</td>\n",
       "      <td>Kongsvinger, Norway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320148</th>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>United Kingdom:London, United Kingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320149</th>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>United Kingdom:Scotland, UK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320163</th>\n",
       "      <td>United States</td>\n",
       "      <td>USA:Muskegon County, MI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320164</th>\n",
       "      <td>Slovenia</td>\n",
       "      <td>Slovenia: Pragersko</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320167</th>\n",
       "      <td>United States</td>\n",
       "      <td>USA:Seattle, Washington</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144217 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       isolation_country                    geographic_location\n",
       "1                 Canada                         Canada: Quebec\n",
       "5          United States                         New York State\n",
       "7          United States               West Palm Beach, Florida\n",
       "10               Denmark                        Denmark: Aarhus\n",
       "22                Norway                    Kongsvinger, Norway\n",
       "...                  ...                                    ...\n",
       "320148    United Kingdom  United Kingdom:London, United Kingdom\n",
       "320149    United Kingdom            United Kingdom:Scotland, UK\n",
       "320163     United States                USA:Muskegon County, MI\n",
       "320164          Slovenia                    Slovenia: Pragersko\n",
       "320167     United States                USA:Seattle, Washington\n",
       "\n",
       "[144217 rows x 2 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadf['geographic_location'] = metadf['geographic_location'].replace('USA', 'United States')\n",
    "metadf[\n",
    "    (metadf['geographic_location'].notnull()) &\n",
    "    (metadf['isolation_country'].notnull()) & \n",
    "    (metadf['geographic_location'] != metadf['isolation_country'])\n",
    "][['isolation_country', 'geographic_location']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "adjustable-overhead",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadf = metadf.drop('geographic_location', axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "delayed-application",
   "metadata": {},
   "source": [
    "## Read our categories\n",
    "\n",
    "This is the data from Laura"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "alleged-mathematics",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>Category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>food</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>soil</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>human blood</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>soil</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000006155.2</td>\n",
       "      <td>human other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54640</th>\n",
       "      <td>GCA_902860175.1</td>\n",
       "      <td>built environment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54641</th>\n",
       "      <td>GCA_902860185.1</td>\n",
       "      <td>human other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54642</th>\n",
       "      <td>GCA_902860195.1</td>\n",
       "      <td>human other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54643</th>\n",
       "      <td>GCA_902860235.1</td>\n",
       "      <td>plant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54644</th>\n",
       "      <td>assembly_accession</td>\n",
       "      <td>Category</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54645 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession           Category\n",
       "0         GCA_000003645.1               food\n",
       "1         GCA_000003925.1               soil\n",
       "2         GCA_000003955.1        human blood\n",
       "3         GCA_000005825.2               soil\n",
       "4         GCA_000006155.2        human other\n",
       "...                   ...                ...\n",
       "54640     GCA_902860175.1  built environment\n",
       "54641     GCA_902860185.1        human other\n",
       "54642     GCA_902860195.1        human other\n",
       "54643     GCA_902860235.1              plant\n",
       "54644  assembly_accession           Category\n",
       "\n",
       "[54645 rows x 2 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "catdf = pd.read_csv(\"../data/categories.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "if 'gbff' in catdf:\n",
    "    catdf = catdf.drop('gbff', axis=1)\n",
    "catdf = catdf.groupby('assembly_accession').first().reset_index()\n",
    "catdf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "global-mouth",
   "metadata": {},
   "source": [
    "### Make sure we have the right column\n",
    "\n",
    "PATRIC calls the column `assembly_accession` while GTDB calls it `ncbi_genbank_assembly_accession`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "solar-intermediate",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         GCA_000003135.1\n",
       "1         GCA_000003215.1\n",
       "2         GCA_000003645.1\n",
       "3         GCA_000003925.1\n",
       "4         GCA_000003955.1\n",
       "               ...       \n",
       "320171    GCF_900167595.1\n",
       "320172    GCF_900167605.1\n",
       "320173    GCF_900167615.1\n",
       "320174    GCF_900167625.1\n",
       "320175    GCF_900167635.1\n",
       "Name: assembly_accession, Length: 320176, dtype: object"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acccol = 'assembly_accession'\n",
    "metadf[acccol]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "accessory-guarantee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0           GCA_000003645.1\n",
       "1           GCA_000003925.1\n",
       "2           GCA_000003955.1\n",
       "3           GCA_000005825.2\n",
       "4           GCA_000006155.2\n",
       "                ...        \n",
       "54640       GCA_902860175.1\n",
       "54641       GCA_902860185.1\n",
       "54642       GCA_902860195.1\n",
       "54643       GCA_902860235.1\n",
       "54644    assembly_accession\n",
       "Name: assembly_accession, Length: 54645, dtype: object"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "catdf[acccol]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "lightweight-kennedy",
   "metadata": {},
   "source": [
    "### Abstract out the accession and name into new columns\n",
    "\n",
    "This will allow us to merge the data with the PATRIC and GTDB data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "cooperative-sleeve",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>Name</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Phage Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>ASM313v1</td>\n",
       "      <td>GCA_000003135.1_ASM313v1_genomic.gbff.gz</td>\n",
       "      <td>2396359</td>\n",
       "      <td>114</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>48916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>ASM364v1</td>\n",
       "      <td>GCA_000003645.1_ASM364v1_genomic.gbff.gz</td>\n",
       "      <td>5269725</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>40297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>ASM392v1</td>\n",
       "      <td>GCA_000003925.1_ASM392v1_genomic.gbff.gz</td>\n",
       "      <td>5561906</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>268081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>ASM395v1</td>\n",
       "      <td>GCA_000003955.1_ASM395v1_genomic.gbff.gz</td>\n",
       "      <td>5790501</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>166286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>GCA_000005825.2_ASM582v2_genomic.gbff.gz</td>\n",
       "      <td>4249248</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>93416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553077</th>\n",
       "      <td>GCA_902860175.1</td>\n",
       "      <td>LMG_5997</td>\n",
       "      <td>GCA_902860175.1_LMG_5997_genomic.gbff.gz</td>\n",
       "      <td>7197255</td>\n",
       "      <td>38</td>\n",
       "      <td>21</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>17</td>\n",
       "      <td>69051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553078</th>\n",
       "      <td>GCA_902860185.1</td>\n",
       "      <td>LMG_6103</td>\n",
       "      <td>GCA_902860185.1_LMG_6103_genomic.gbff.gz</td>\n",
       "      <td>6497464</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553079</th>\n",
       "      <td>GCA_902860195.1</td>\n",
       "      <td>LMG_7053</td>\n",
       "      <td>GCA_902860195.1_LMG_7053_genomic.gbff.gz</td>\n",
       "      <td>6702936</td>\n",
       "      <td>200</td>\n",
       "      <td>148</td>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>12819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553080</th>\n",
       "      <td>GCA_902860205.1</td>\n",
       "      <td>LMG_6001</td>\n",
       "      <td>GCA_902860205.1_LMG_6001_genomic.gbff.gz</td>\n",
       "      <td>6320373</td>\n",
       "      <td>36</td>\n",
       "      <td>19</td>\n",
       "      <td>35</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>12</td>\n",
       "      <td>41572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553081</th>\n",
       "      <td>GCA_902860235.1</td>\n",
       "      <td>LMG_9964</td>\n",
       "      <td>GCA_902860235.1_LMG_9964_genomic.gbff.gz</td>\n",
       "      <td>7602199</td>\n",
       "      <td>42</td>\n",
       "      <td>26</td>\n",
       "      <td>37</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>71710</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>553082 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession      Name                                    Contig  \\\n",
       "0         GCA_000003135.1  ASM313v1  GCA_000003135.1_ASM313v1_genomic.gbff.gz   \n",
       "1         GCA_000003645.1  ASM364v1  GCA_000003645.1_ASM364v1_genomic.gbff.gz   \n",
       "2         GCA_000003925.1  ASM392v1  GCA_000003925.1_ASM392v1_genomic.gbff.gz   \n",
       "3         GCA_000003955.1  ASM395v1  GCA_000003955.1_ASM395v1_genomic.gbff.gz   \n",
       "4         GCA_000005825.2  ASM582v2  GCA_000005825.2_ASM582v2_genomic.gbff.gz   \n",
       "...                   ...       ...                                       ...   \n",
       "553077    GCA_902860175.1  LMG_5997  GCA_902860175.1_LMG_5997_genomic.gbff.gz   \n",
       "553078    GCA_902860185.1  LMG_6103  GCA_902860185.1_LMG_6103_genomic.gbff.gz   \n",
       "553079    GCA_902860195.1  LMG_7053  GCA_902860195.1_LMG_7053_genomic.gbff.gz   \n",
       "553080    GCA_902860205.1  LMG_6001  GCA_902860205.1_LMG_6001_genomic.gbff.gz   \n",
       "553081    GCA_902860235.1  LMG_9964  GCA_902860235.1_LMG_9964_genomic.gbff.gz   \n",
       "\n",
       "        Genome length  Contigs  Phage Contigs  Total Predicted Prophages  \\\n",
       "0             2396359      114             10                         16   \n",
       "1             5269725        1              1                         31   \n",
       "2             5561906        1              1                         38   \n",
       "3             5790501        1              1                         46   \n",
       "4             4249248        3              3                         33   \n",
       "...               ...      ...            ...                        ...   \n",
       "553077        7197255       38             21                         33   \n",
       "553078        6497464       13              8                         22   \n",
       "553079        6702936      200            148                         33   \n",
       "553080        6320373       36             19                         35   \n",
       "553081        7602199       42             26                         37   \n",
       "\n",
       "        Kept  No phage genes  Not enough genes  bp prophage  \n",
       "0          2               1                13        48916  \n",
       "1          1              10                20        40297  \n",
       "2          6              13                19       268081  \n",
       "3          6              11                29       166286  \n",
       "4          3               9                21        93416  \n",
       "...      ...             ...               ...          ...  \n",
       "553077     2              14                17        69051  \n",
       "553078     0              10                12            0  \n",
       "553079     1              11                21        12819  \n",
       "553080     2              21                12        41572  \n",
       "553081     2              20                15        71710  \n",
       "\n",
       "[553082 rows x 11 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_acc_name(x):\n",
    "    regexp = re.compile('(\\w+\\.\\d+)_([\\w\\.\\-]+)_genomic.gbff.gz')\n",
    "    m = regexp.match(x)\n",
    "    if not m:\n",
    "        sys.stderr.write(f\"WARNING: Regexp did not match {x}\\n\")\n",
    "        return (None, None)\n",
    "    return list(m.groups())\n",
    "\n",
    "phagesdf = pd.concat([pd.DataFrame.from_records(phagesdf['Contig'].apply(get_acc_name), columns=[acccol, 'Name']), phagesdf], axis=1)\n",
    "phagesdf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "divine-identification",
   "metadata": {},
   "source": [
    "### Merge the dataframes\n",
    "\n",
    "First, select some columns we want to keep from PATRIC, and then merge the data frames. We write this to a tsv file for Laura to append isolation information to."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "bridal-launch",
   "metadata": {},
   "outputs": [],
   "source": [
    "interesting_cols = [acccol, 'isolation_site', 'isolation_country', 'latitude', 'longitude', 'altitude', 'depth',\n",
    "                    'other_environmental', 'host_name', 'host_gender', 'host_age', 'host_health', \n",
    "                    'body_sample_site', 'body_sample_subsite', 'other_clinical', 'gram_stain', 'cell_shape',\n",
    "                    'motility', 'sporulation', 'temperature_range', 'optimal_temperature', 'salinity',\n",
    "                    'oxygen_requirement', 'habitat', 'disease', 'isolation_date']\n",
    "\n",
    "few_interesting_cols = [acccol,  'isolation_country',  'gram_stain',  \n",
    "                    'motility', 'sporulation', 'temperature_range', 'habitat', 'disease', 'isolation_date']\n",
    "\n",
    "\n",
    "tempdf = metadf[interesting_cols]\n",
    "# tempdf = metadf[few_interesting_cols]\n",
    "temp1 = pd.merge(tempdf, catdf, how='left', left_on=acccol, right_on=acccol)\n",
    "# phagemeta = pd.merge(tempdf, phagesdf, how='inner', left_on=acccol, right_on=acccol)\n",
    "# phagemeta.to_csv(os.path.join('results', 'example_isolations.tsv'), sep='\\t')\n",
    "\n",
    "phagemeta = pd.merge(temp1, phagesdf, how='right', left_on=acccol, right_on=acccol)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "stylish-peripheral",
   "metadata": {},
   "source": [
    "# Define the columns\n",
    "\n",
    "These are for the columns we are going to use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "enabling-refrigerator",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isolation_site</th>\n",
       "      <th>isolation_country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>altitude</th>\n",
       "      <th>depth</th>\n",
       "      <th>other_environmental</th>\n",
       "      <th>host_name</th>\n",
       "      <th>host_gender</th>\n",
       "      <th>host_age</th>\n",
       "      <th>...</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>isolation_date</th>\n",
       "      <th>Category</th>\n",
       "      <th>Name</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Genome length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Human, Homo sapiens</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>Anaerobic</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM313v1</td>\n",
       "      <td>GCA_000003135.1_ASM313v1_genomic.gbff.gz</td>\n",
       "      <td>2396359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>NaN</td>\n",
       "      <td>food</td>\n",
       "      <td>ASM364v1</td>\n",
       "      <td>GCA_000003645.1_ASM364v1_genomic.gbff.gz</td>\n",
       "      <td>5269725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>soil</td>\n",
       "      <td>ASM392v1</td>\n",
       "      <td>GCA_000003925.1_ASM392v1_genomic.gbff.gz</td>\n",
       "      <td>5561906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>Iceland</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Human, Homo sapiens</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human blood</td>\n",
       "      <td>ASM395v1</td>\n",
       "      <td>GCA_000003955.1_ASM395v1_genomic.gbff.gz</td>\n",
       "      <td>5790501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>United States</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>None</td>\n",
       "      <td>Facultative</td>\n",
       "      <td>Terrestrial</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>soil</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>GCA_000005825.2_ASM582v2_genomic.gbff.gz</td>\n",
       "      <td>4249248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553077</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1999.956879</td>\n",
       "      <td>built environment</td>\n",
       "      <td>LMG_5997</td>\n",
       "      <td>GCA_902860175.1_LMG_5997_genomic.gbff.gz</td>\n",
       "      <td>7197255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553078</th>\n",
       "      <td>None</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1999.956879</td>\n",
       "      <td>human other</td>\n",
       "      <td>LMG_6103</td>\n",
       "      <td>GCA_902860185.1_LMG_6103_genomic.gbff.gz</td>\n",
       "      <td>6497464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553079</th>\n",
       "      <td>None</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1999.956879</td>\n",
       "      <td>human other</td>\n",
       "      <td>LMG_7053</td>\n",
       "      <td>GCA_902860195.1_LMG_7053_genomic.gbff.gz</td>\n",
       "      <td>6702936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553080</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1999.956879</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LMG_6001</td>\n",
       "      <td>GCA_902860205.1_LMG_6001_genomic.gbff.gz</td>\n",
       "      <td>6320373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553081</th>\n",
       "      <td>None</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plant</td>\n",
       "      <td>LMG_9964</td>\n",
       "      <td>GCA_902860235.1_LMG_9964_genomic.gbff.gz</td>\n",
       "      <td>7602199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>553082 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       isolation_site isolation_country latitude longitude altitude depth  \\\n",
       "0                None              None     None      None     None  None   \n",
       "1                None              None     None      None     None  None   \n",
       "2                None              None     None      None     None  None   \n",
       "3                None           Iceland     None      None     None  None   \n",
       "4                None     United States     None      None     None  None   \n",
       "...               ...               ...      ...       ...      ...   ...   \n",
       "553077           None              None     None      None     None  None   \n",
       "553078           None    United Kingdom     None      None     None  None   \n",
       "553079           None    United Kingdom     None      None     None  None   \n",
       "553080           None              None     None      None     None  None   \n",
       "553081           None            Brazil     None      None     None  None   \n",
       "\n",
       "       other_environmental            host_name host_gender host_age  ...  \\\n",
       "0                     None  Human, Homo sapiens        None     None  ...   \n",
       "1                     None                 None        None     None  ...   \n",
       "2                     None                 None        None     None  ...   \n",
       "3                     None  Human, Homo sapiens        None     None  ...   \n",
       "4                     None                 None        None     None  ...   \n",
       "...                    ...                  ...         ...      ...  ...   \n",
       "553077                None                 None        None     None  ...   \n",
       "553078                None                 None        None     None  ...   \n",
       "553079                None                 None        None     None  ...   \n",
       "553080                None                 None        None     None  ...   \n",
       "553081                None                 None        None     None  ...   \n",
       "\n",
       "       optimal_temperature salinity oxygen_requirement          habitat  \\\n",
       "0                        -     None          Anaerobic  Host-associated   \n",
       "1                        -     None            Aerobic         Multiple   \n",
       "2                        -     None               None             None   \n",
       "3                        -     None            Aerobic         Multiple   \n",
       "4                        -     None        Facultative      Terrestrial   \n",
       "...                    ...      ...                ...              ...   \n",
       "553077                None     None               None             None   \n",
       "553078                None     None               None             None   \n",
       "553079                None     None               None             None   \n",
       "553080                None     None               None             None   \n",
       "553081                None     None               None             None   \n",
       "\n",
       "               disease isolation_date           Category      Name  \\\n",
       "0                 None            NaN                NaN  ASM313v1   \n",
       "1       Food poisoning            NaN               food  ASM364v1   \n",
       "2                 None            NaN               soil  ASM392v1   \n",
       "3       Food poisoning            NaN        human blood  ASM395v1   \n",
       "4                 None            NaN               soil  ASM582v2   \n",
       "...                ...            ...                ...       ...   \n",
       "553077            None    1999.956879  built environment  LMG_5997   \n",
       "553078            None    1999.956879        human other  LMG_6103   \n",
       "553079            None    1999.956879        human other  LMG_7053   \n",
       "553080            None    1999.956879                NaN  LMG_6001   \n",
       "553081            None            NaN              plant  LMG_9964   \n",
       "\n",
       "                                          Contig Genome length  \n",
       "0       GCA_000003135.1_ASM313v1_genomic.gbff.gz       2396359  \n",
       "1       GCA_000003645.1_ASM364v1_genomic.gbff.gz       5269725  \n",
       "2       GCA_000003925.1_ASM392v1_genomic.gbff.gz       5561906  \n",
       "3       GCA_000003955.1_ASM395v1_genomic.gbff.gz       5790501  \n",
       "4       GCA_000005825.2_ASM582v2_genomic.gbff.gz       4249248  \n",
       "...                                          ...           ...  \n",
       "553077  GCA_902860175.1_LMG_5997_genomic.gbff.gz       7197255  \n",
       "553078  GCA_902860185.1_LMG_6103_genomic.gbff.gz       6497464  \n",
       "553079  GCA_902860195.1_LMG_7053_genomic.gbff.gz       6702936  \n",
       "553080  GCA_902860205.1_LMG_6001_genomic.gbff.gz       6320373  \n",
       "553081  GCA_902860235.1_LMG_9964_genomic.gbff.gz       7602199  \n",
       "\n",
       "[553082 rows x 29 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "begdata = 1 # ignore the first column which is the assembly name\n",
    "# enddata = 9 # this is if we are using a smaller dataset\n",
    "enddata = 30 # this is all the metadata\n",
    "phagemeta.iloc[:,begdata:enddata]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "following-candy",
   "metadata": {},
   "source": [
    "## Sanity check\n",
    "\n",
    "This is just to make sure that we are all working on the same data. Here, we make a temporary data frame that has the number of contigs and the number of basepairs in the genome, and use those to compare to see if we think we are working on the same thing.\n",
    "\n",
    "This is tricky because PATRIC has duplicates in the data, thus `GCA_000005825.2` appears three times in the PATRIC metadata, twice for the plasmids and once for the assembly.\n",
    "\n",
    "We can maybe merge on two fields:\n",
    "\n",
    "```\n",
    "\n",
    "comparison = pd.merge(metadf[[acccol, 'contigs', 'genome_length']],\n",
    "                      phagesdf[[acccol, 'Name', 'Contigs', 'Genome length']],\n",
    "                      how='right', left_on=[acccol, 'genome_length'], right_on=[acccol, 'Genome length'])\n",
    "```\n",
    "\n",
    "and then look for entries in phagesdf that are missing from the merged (or just use the entries in the merged throughout)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "useful-fundamental",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "444,503 rows are not the same\n",
      "108,579 rows are the same\n"
     ]
    }
   ],
   "source": [
    "def are_same(x):\n",
    "    # currently fixing contig counts, so when that is done delete the next line and uncomment the one after\n",
    "    return x['genome_length'] == x['Genome length']\n",
    "    # return x['genome_length'] == x['Genome length'] and x['contigs'] == x['Contigs']\n",
    "\n",
    "\n",
    "comparison = pd.merge(metadf[[acccol, 'contigs', 'genome_length']],\n",
    "                      phagesdf[[acccol, 'Name', 'Contigs', 'Genome length']],\n",
    "                      how='right', left_on=acccol, right_on=acccol)\n",
    "\n",
    "comparison['same'] = comparison.apply(are_same, axis=1)\n",
    "# look for rows that are not the same\n",
    "print(f\"{comparison[comparison['same']==False].shape[0]:,} rows are not the same\")\n",
    "print(f\"{comparison[comparison['same']==True].shape[0]:,} rows are the same\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "smaller-thursday",
   "metadata": {},
   "source": [
    "## How many NaNs are there?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "foreign-train",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "isolation_site\t128\n",
      "isolation_country\t96876\n",
      "latitude\t2910\n",
      "longitude\t2909\n",
      "altitude\t1033\n",
      "depth\t8094\n",
      "other_environmental\t9841\n",
      "host_name\t59086\n",
      "host_gender\t4069\n",
      "host_age\t3610\n",
      "host_health\t11651\n",
      "body_sample_site\t2032\n",
      "body_sample_subsite\t0\n",
      "other_clinical\t11223\n",
      "gram_stain\t8097\n",
      "cell_shape\t7380\n",
      "motility\t5267\n",
      "sporulation\t3873\n",
      "temperature_range\t4015\n",
      "optimal_temperature\t10737\n",
      "salinity\t705\n",
      "oxygen_requirement\t6440\n",
      "habitat\t4344\n",
      "disease\t2096\n",
      "isolation_date\t90756\n",
      "Category\t54644\n",
      "Name\t553082\n",
      "Contig\t553082\n",
      "Genome length\t553082\n"
     ]
    }
   ],
   "source": [
    "for c in phagemeta.iloc[:,begdata:enddata].columns:\n",
    "    print(f\"{c}\\t{phagemeta[phagemeta[c].notnull()].shape[0]}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "congressional-amount",
   "metadata": {},
   "source": [
    "## Encode the data\n",
    "\n",
    "This converts every column into categories so we can (hopefully) use it in the RF models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "intense-dictionary",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isolation_site</th>\n",
       "      <th>isolation_country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>altitude</th>\n",
       "      <th>depth</th>\n",
       "      <th>other_environmental</th>\n",
       "      <th>host_name</th>\n",
       "      <th>host_gender</th>\n",
       "      <th>host_age</th>\n",
       "      <th>...</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>isolation_date</th>\n",
       "      <th>Category</th>\n",
       "      <th>Name</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Genome length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1044</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>-1</td>\n",
       "      <td>105501</td>\n",
       "      <td>0</td>\n",
       "      <td>95431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357518</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>-1</td>\n",
       "      <td>485889</td>\n",
       "      <td>357518</td>\n",
       "      <td>226581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357517</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>-1</td>\n",
       "      <td>485885</td>\n",
       "      <td>357517</td>\n",
       "      <td>188813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357516</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>-1</td>\n",
       "      <td>485887</td>\n",
       "      <td>357516</td>\n",
       "      <td>190524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357515</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>-1</td>\n",
       "      <td>485883</td>\n",
       "      <td>357515</td>\n",
       "      <td>334948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503279</th>\n",
       "      <td>-1</td>\n",
       "      <td>224</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>104</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>2020.063655</td>\n",
       "      <td>46</td>\n",
       "      <td>55559</td>\n",
       "      <td>503279</td>\n",
       "      <td>456799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503278</th>\n",
       "      <td>-1</td>\n",
       "      <td>224</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>104</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>2020.063655</td>\n",
       "      <td>46</td>\n",
       "      <td>55558</td>\n",
       "      <td>503278</td>\n",
       "      <td>466952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503277</th>\n",
       "      <td>-1</td>\n",
       "      <td>224</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>104</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>2020.063655</td>\n",
       "      <td>46</td>\n",
       "      <td>55557</td>\n",
       "      <td>503277</td>\n",
       "      <td>456794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503275</th>\n",
       "      <td>-1</td>\n",
       "      <td>224</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>104</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>2020.063655</td>\n",
       "      <td>46</td>\n",
       "      <td>55554</td>\n",
       "      <td>503275</td>\n",
       "      <td>461313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505307</th>\n",
       "      <td>-1</td>\n",
       "      <td>140</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>2020.238877</td>\n",
       "      <td>-1</td>\n",
       "      <td>56312</td>\n",
       "      <td>505307</td>\n",
       "      <td>164651</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>553082 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        isolation_site  isolation_country  latitude  longitude  altitude  \\\n",
       "0                   -1                 -1        -1         -1        -1   \n",
       "357518              -1                 -1        -1         -1        -1   \n",
       "357517              -1                 -1        -1         -1        -1   \n",
       "357516              -1                 -1        -1         -1        -1   \n",
       "357515              -1                 -1        -1         -1        -1   \n",
       "...                ...                ...       ...        ...       ...   \n",
       "503279              -1                224        -1         -1        -1   \n",
       "503278              -1                224        -1         -1        -1   \n",
       "503277              -1                224        -1         -1        -1   \n",
       "503275              -1                224        -1         -1        -1   \n",
       "505307              -1                140        -1         -1        -1   \n",
       "\n",
       "        depth  other_environmental  host_name  host_gender  host_age  ...  \\\n",
       "0          -1                   -1       1044           -1        -1  ...   \n",
       "357518     -1                   -1         -1           -1        -1  ...   \n",
       "357517     -1                   -1         -1           -1        -1  ...   \n",
       "357516     -1                   -1         -1           -1        -1  ...   \n",
       "357515     -1                   -1         -1           -1        -1  ...   \n",
       "...       ...                  ...        ...          ...       ...  ...   \n",
       "503279    104                   -1         -1           -1        -1  ...   \n",
       "503278    104                   -1         -1           -1        -1  ...   \n",
       "503277    104                   -1         -1           -1        -1  ...   \n",
       "503275    104                   -1         -1           -1        -1  ...   \n",
       "505307     -1                   -1         -1           -1        -1  ...   \n",
       "\n",
       "        optimal_temperature  salinity  oxygen_requirement  habitat  disease  \\\n",
       "0                         1        -1                   3       14       -1   \n",
       "357518                   -1        -1                  -1       -1       -1   \n",
       "357517                   -1        -1                  -1       -1       -1   \n",
       "357516                   -1        -1                  -1       -1       -1   \n",
       "357515                   -1        -1                  -1       -1       -1   \n",
       "...                     ...       ...                 ...      ...      ...   \n",
       "503279                   -1        -1                  -1       -1       -1   \n",
       "503278                   -1        -1                  -1       -1       -1   \n",
       "503277                   -1        -1                  -1       -1       -1   \n",
       "503275                   -1        -1                  -1       -1       -1   \n",
       "505307                   -1        -1                  -1       -1       -1   \n",
       "\n",
       "        isolation_date  Category    Name  Contig  Genome length  \n",
       "0            -1.000000        -1  105501       0          95431  \n",
       "357518       -1.000000        -1  485889  357518         226581  \n",
       "357517       -1.000000        -1  485885  357517         188813  \n",
       "357516       -1.000000        -1  485887  357516         190524  \n",
       "357515       -1.000000        -1  485883  357515         334948  \n",
       "...                ...       ...     ...     ...            ...  \n",
       "503279     2020.063655        46   55559  503279         456799  \n",
       "503278     2020.063655        46   55558  503278         466952  \n",
       "503277     2020.063655        46   55557  503277         456794  \n",
       "503275     2020.063655        46   55554  503275         461313  \n",
       "505307     2020.238877        -1   56312  505307         164651  \n",
       "\n",
       "[553082 rows x 29 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmenc = pd.DataFrame()\n",
    "for c in phagemeta.iloc[:,begdata:enddata].columns:\n",
    "    pmenc[c] = phagemeta[c].astype('category').cat.codes\n",
    "pmenc['isolation_date'] = phagemeta['isolation_date'].fillna(-1)\n",
    "pmenc.sort_values('isolation_date')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "inner-child",
   "metadata": {},
   "source": [
    "## Random Forest\n",
    "\n",
    "Can we identify which features best predict the number of phages predicted. Note we should compare total/kept/etc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "gorgeous-colors",
   "metadata": {},
   "source": [
    "First, an RF with all the data. You might want to skip this!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "straight-nightlife",
   "metadata": {},
   "outputs": [],
   "source": [
    "clf = RandomForestClassifier(random_state=42, n_estimators=1000, bootstrap=True, n_jobs=-1, oob_score=True)\n",
    "rf = clf.fit(pmenc, phagemeta.Kept.values.ravel())\n",
    "print(rf)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "reasonable-telephone",
   "metadata": {},
   "source": [
    "## Important features\n",
    "\n",
    "What are the most important features for predicting the number of prophages in a genome?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "determined-department",
   "metadata": {},
   "outputs": [],
   "source": [
    "fi = pd.DataFrame(zip(pmenc.columns, rf.feature_importances_), columns=['feature', 'importance'])\n",
    "impdf = fi.sort_values('importance', ascending=False)\n",
    "# impdf['Number Not Null'] = phagemeta[phagemeta[impdf['feature']].notnull()].shape[0]\n",
    "impdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "weird-queen",
   "metadata": {},
   "outputs": [],
   "source": [
    "def number_not_null(x):\n",
    "    return phagemeta[phagemeta[x.feature].notnull()].shape[0]\n",
    "\n",
    "fi = pd.DataFrame(zip(pmenc.columns, rf.feature_importances_), columns=['feature', 'importance'])\n",
    "impdf = fi.sort_values('importance', ascending=False)\n",
    "impdf[\"Number of observations\"] = impdf.apply(number_not_null, axis=1)\n",
    "impdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "incorporated-pathology",
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(11, 8))\n",
    "ax = sns.scatterplot(x='feature', y='importance', data=impdf, ax=ax)\n",
    "plt.xticks(rotation=45,  horizontalalignment='right')\n",
    "fig.savefig(f\"images/rf_importance.svg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "pharmaceutical-davis",
   "metadata": {},
   "outputs": [],
   "source": [
    "printmd(\"**Random Forest output**\", \"blue\")\n",
    "printmd(f\"There were {rf.n_features_} _features_ (columns in the dataframe), and {rf.n_classes_} _classes_ (unique categories)\")\n",
    "printmd(f\"The OOB error was {rf.oob_score_}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "sustained-gregory",
   "metadata": {},
   "source": [
    "# Exploring the data\n",
    "\n",
    "Lets take a look at some of these important things."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "advanced-plain",
   "metadata": {},
   "source": [
    "## Isolation date of the phage\n",
    "\n",
    "Note that above we do a lot of work to create the `isolation_date` field, which is now the year of isolation as a decimal so that it plots out nicely! \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "african-witch",
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(11, 8))\n",
    "ax = sns.scatterplot(x='isolation_date', y='Kept', data=phagemeta[phagemeta['isolation_date'] != -1], ax=ax, marker='+', color='k')\n",
    "ax.set_ylabel(\"Number of phages predicted\")\n",
    "ax.set_xlabel(\"Isolation date\")\n",
    "#ax.axhline(y=0, color='k')\n",
    "ax.spines['bottom'].set_position('zero')\n",
    "ax.spines['top'].set_color('none')\n",
    "ax.spines['right'].set_color('none')\n",
    "\n",
    "\n",
    "fig.savefig(f\"images/phages_vs_ya.svg\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "noted-anthony",
   "metadata": {},
   "source": [
    "## Isolation Country\n",
    "\n",
    "Here we look at isolation country. Note that earlier we fixed a few countries (e.g. USA -> United States), etc.\n",
    "\n",
    "Note that we have kept true zero counts: these are genomes for which we have no prophages isolated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "commercial-chester",
   "metadata": {},
   "outputs": [],
   "source": [
    "phagemeta[(phagemeta['isolation_country'] == 'United States') & (phagemeta['Kept'] == 0)][['assembly_accession', 'Kept']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "chemical-defensive",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = phagemeta[phagemeta['isolation_country'].notnull()].sort_values('Kept', ascending=False)\n",
    "fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(11, 8))\n",
    "ax = sns.violinplot(x='isolation_country', y='Kept', data=tmp, ax=ax)\n",
    "ax.set_ylabel(\"Number of phages predicted\")\n",
    "ax.set_xlabel(\"Isolation country\")\n",
    "#ax.axhline(y=0, color='k')\n",
    "ax.spines['bottom'].set_position('zero')\n",
    "ax.spines['top'].set_color('none')\n",
    "ax.spines['right'].set_color('none')\n",
    "plt.xticks(rotation=45,  horizontalalignment='right')\n",
    "ax.set_title('All countries sort of sorted')\n",
    "fig.savefig(f\"images/phages_vs_country_all.svg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "technical-tonight",
   "metadata": {},
   "outputs": [],
   "source": [
    "topnc = 50\n",
    "tmp = phagemeta[phagemeta['isolation_country'].isin(phagemeta.groupby('isolation_country').count().sort_values('assembly_accession', ascending=False).head(topnc).index)]\n",
    "fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(11, 8))\n",
    "ax = sns.violinplot(x='isolation_country', y='Kept', data=tmp, ax=ax)\n",
    "ax.set_ylabel(\"Number of phages predicted\")\n",
    "ax.set_xlabel(\"Isolation country\")\n",
    "ax.spines['bottom'].set_position('zero')\n",
    "ax.spines['top'].set_color('none')\n",
    "ax.spines['right'].set_color('none')\n",
    "ax.set_title(f'Top {topnc} countries with the most genome isolates')\n",
    "plt.xticks(rotation=45,  horizontalalignment='right')\n",
    "\n",
    "fig.savefig(f\"images/phages_vs_country_top.svg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "interesting-sector",
   "metadata": {},
   "outputs": [],
   "source": [
    "topnc = 100\n",
    "tmp = phagemeta[phagemeta['isolation_country'].notnull()].sort_values('Kept', ascending=False)\n",
    "fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(11, 8))\n",
    "sortedorder = phagemeta.groupby('isolation_country')[\"Kept\"].median().sort_values(ascending=False).index[0:topnc]\n",
    "ax = sns.violinplot(x='isolation_country', y='Kept', data=tmp, ax=ax, order=sortedorder)\n",
    "ax.set_ylabel(\"Number of phages predicted\")\n",
    "ax.set_xlabel(\"Isolation country\")\n",
    "#ax.axhline(y=0, color='k')\n",
    "ax.spines['bottom'].set_position('zero')\n",
    "ax.spines['top'].set_color('none')\n",
    "ax.spines['right'].set_color('none')\n",
    "ax.set_title(f'Top {topnc} genomes by median number of prophages')\n",
    "plt.xticks(rotation=45,  horizontalalignment='right')\n",
    "\n",
    "fig.savefig(f\"images/phages_vs_country_100_median.svg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "convinced-porter",
   "metadata": {},
   "outputs": [],
   "source": [
    "phagemeta.groupby('isolation_country')[\"Kept\"].median().sort_values(ascending=False).index[0:50]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "finnish-benchmark",
   "metadata": {},
   "outputs": [],
   "source": [
    "topnc = 50\n",
    "genomen = 5 \n",
    "tmpc = pd.DataFrame(phagemeta.groupby('isolation_country').count()['assembly_accession'] > genomen)\n",
    "tmpc[tmpc['assembly_accession']].index\n",
    "tmp = phagemeta[phagemeta['isolation_country'].isin(tmpc[tmpc['assembly_accession']].index)]\n",
    "\n",
    "fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(11, 8))\n",
    "sortedorder = tmp.groupby('isolation_country')[\"Kept\"].median().sort_values(ascending=False).index[0:topnc]\n",
    "ax = sns.violinplot(x='isolation_country', y='Kept', data=tmp, ax=ax, order=sortedorder)\n",
    "ax.set_ylabel(\"Number of phages predicted\")\n",
    "ax.set_xlabel(\"Isolation country\")\n",
    "#ax.axhline(y=0, color='k')\n",
    "ax.spines['bottom'].set_position('zero')\n",
    "ax.spines['top'].set_color('none')\n",
    "ax.spines['right'].set_color('none')\n",
    "ax.set_title(f'Top {topnc} countries with more than {genomen} genomes\\nSorted by median prophage abundance')\n",
    "plt.xticks(rotation=45,  horizontalalignment='right')\n",
    "\n",
    "fig.savefig(f\"images/phages_vs_country_{topnc}_genomes_{genomen}_phages_median.svg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "raised-location",
   "metadata": {},
   "outputs": [],
   "source": [
    "topnc = 50\n",
    "genomen = 5 \n",
    "tmpc = pd.DataFrame(phagemeta.groupby('isolation_country').count()['assembly_accession'] > genomen)\n",
    "tmpc[tmpc['assembly_accession']].index\n",
    "tmp = phagemeta[phagemeta['isolation_country'].isin(tmpc[tmpc['assembly_accession']].index)]\n",
    "\n",
    "fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(11, 8))\n",
    "sortedorder = tmp.groupby('isolation_country')[\"Kept\"].sum().sort_values(ascending=False).index[0:topnc]\n",
    "ax = sns.violinplot(x='isolation_country', y='Kept', data=tmp, ax=ax, order=sortedorder)\n",
    "ax.set_ylabel(\"Number of phages predicted\")\n",
    "ax.set_xlabel(\"Isolation country\")\n",
    "#ax.axhline(y=0, color='k')\n",
    "ax.spines['bottom'].set_position('zero')\n",
    "ax.spines['top'].set_color('none')\n",
    "ax.spines['right'].set_color('none')\n",
    "ax.set_title(f'Top {topnc} countries with more than {genomen} genomes\\nSorted by total prophage abundance')\n",
    "plt.xticks(rotation=45,  horizontalalignment='right')\n",
    "\n",
    "fig.savefig(f\"images/phages_vs_country_{topnc}_genomes_{genomen}_phages_sum.svg\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "broadband-shanghai",
   "metadata": {},
   "source": [
    "## Host Name\n",
    "\n",
    "This is generally where the bacteria was isolated from"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "designing-karma",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.unique(phagemeta[phagemeta['host_name'].notnull()]['host_name'])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
