{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge the prophage predictions with RAST metadata\n",
    "\n",
    "Combine RAST metadata and prophage predictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.ticker as mticker\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway\n",
    "from sklearn.linear_model import LinearRegression\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "import subprocess\n",
    "import gzip\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# this is a neat trick for getting markdown in our output\n",
    "# see https://stackoverflow.com/questions/23271575/printing-bold-colored-etc-text-in-ipython-qtconsole\n",
    "# for the inspiration\n",
    "from IPython.display import Markdown, display\n",
    "def printmd(string, color=\"black\"):\n",
    "    colorstr = \"<span style='color:{}'>{}</span>\".format(color, string)\n",
    "    display(Markdown(colorstr))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read the data file\n",
    "\n",
    "The metadata is from PATRIC. The phage data is from us.\n",
    "\n",
    "**NOTE:** Some of the PATRIC data refers to specific chromosomes/fragments in the GenBank file (e.g. plasmid, chromosome), but that may not equate to our predictions, because we have used the whole GenBank file. The PATRIC metadata is redundant for many fields, and so we just keep the first entry for each NCBI Assembly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/redwards/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (2,6,7,8,9,10,11,12,14,19,21,34,40,41,42,43,44,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,65) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>mlst</th>\n",
       "      <th>...</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>469009.4</td>\n",
       "      <td>\"'Brassica napus' phytoplasma strain TW1\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>469009</td>\n",
       "      <td>WGS</td>\n",
       "      <td>TW1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequence of a strain of bacteria that c...</td>\n",
       "      <td>sample_type:metagenomic assembly;collected_by:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1309411.5</td>\n",
       "      <td>\"'Deinococcus soli' Cha et al. 2014 strain N5\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1309411</td>\n",
       "      <td>Complete</td>\n",
       "      <td>N5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing of a Gamma-Radiation-Resista...</td>\n",
       "      <td>sample_type:bacterial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1123738.3</td>\n",
       "      <td>\"'Echinacea purpurea' witches'-broom phytoplas...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1123738</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NCHU2014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'Echinacea purpurea' witches'-broom phytoplasm...</td>\n",
       "      <td>lab_host:Catharanthus roseus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>551115.6</td>\n",
       "      <td>\"'Nostoc azollae' 0708\"</td>\n",
       "      <td>'Nostoc azollae' 0708</td>\n",
       "      <td>551115</td>\n",
       "      <td>Complete</td>\n",
       "      <td>708</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nostoc azollae 0708. Nostoc azollae 0708, also...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1856298.3</td>\n",
       "      <td>\"'Osedax' symbiont bacterium Rs2_46_30_T18 str...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1856298</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Rs2_46_30_T18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In this study, we simulate the Deepwater Horiz...</td>\n",
       "      <td>sample_type:metagenomic assembly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433517</th>\n",
       "      <td>1131286.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>1131286</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-137-J06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433518</th>\n",
       "      <td>1131287.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>1131287</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-C20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433519</th>\n",
       "      <td>1131288.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>1131288</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-E04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433520</th>\n",
       "      <td>1131289.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>1131289</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-B04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433521</th>\n",
       "      <td>1131290.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>1131290</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-O16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433522 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        genome_id                                        genome_name  \\\n",
       "0        469009.4          \"'Brassica napus' phytoplasma strain TW1\"   \n",
       "1       1309411.5     \"'Deinococcus soli' Cha et al. 2014 strain N5\"   \n",
       "2       1123738.3  \"'Echinacea purpurea' witches'-broom phytoplas...   \n",
       "3        551115.6                            \"'Nostoc azollae' 0708\"   \n",
       "4       1856298.3  \"'Osedax' symbiont bacterium Rs2_46_30_T18 str...   \n",
       "...           ...                                                ...   \n",
       "433517  1131286.3               zeta proteobacterium SCGC AB-137-J06   \n",
       "433518  1131287.3               zeta proteobacterium SCGC AB-602-C20   \n",
       "433519  1131288.3               zeta proteobacterium SCGC AB-602-E04   \n",
       "433520  1131289.3               zeta proteobacterium SCGC AB-604-B04   \n",
       "433521  1131290.3               zeta proteobacterium SCGC AB-604-O16   \n",
       "\n",
       "                               organism_name  taxon_id genome_status  \\\n",
       "0                                        NaN    469009           WGS   \n",
       "1                                        NaN   1309411      Complete   \n",
       "2                                        NaN   1123738           WGS   \n",
       "3                      'Nostoc azollae' 0708    551115      Complete   \n",
       "4                                        NaN   1856298           WGS   \n",
       "...                                      ...       ...           ...   \n",
       "433517  zeta proteobacterium SCGC AB-137-J06   1131286           WGS   \n",
       "433518  zeta proteobacterium SCGC AB-602-C20   1131287           WGS   \n",
       "433519  zeta proteobacterium SCGC AB-602-E04   1131288           WGS   \n",
       "433520  zeta proteobacterium SCGC AB-604-B04   1131289           WGS   \n",
       "433521  zeta proteobacterium SCGC AB-604-O16   1131290           WGS   \n",
       "\n",
       "                 strain serovar biovar pathovar mlst  ... motility  \\\n",
       "0                   TW1     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1                    N5     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "2              NCHU2014     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "3                   708     NaN    NaN      NaN  NaN  ...      Yes   \n",
       "4         Rs2_46_30_T18     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "...                 ...     ...    ...      ...  ...  ...      ...   \n",
       "433517  SCGC AB-137-J06     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433518  SCGC AB-602-C20     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433519  SCGC AB-602-E04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433520  SCGC AB-604-B04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "433521  SCGC AB-604-O16     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "\n",
       "       sporulation temperature_range optimal_temperature salinity  \\\n",
       "0              NaN               NaN                 NaN      NaN   \n",
       "1              NaN               NaN                 NaN      NaN   \n",
       "2              NaN               NaN                   C      NaN   \n",
       "3              NaN        Mesophilic                   -      NaN   \n",
       "4              NaN               NaN                 NaN      NaN   \n",
       "...            ...               ...                 ...      ...   \n",
       "433517         NaN               NaN                 NaN      NaN   \n",
       "433518         NaN               NaN                 NaN      NaN   \n",
       "433519         NaN               NaN                 NaN      NaN   \n",
       "433520         NaN               NaN                 NaN      NaN   \n",
       "433521         NaN               NaN                 NaN      NaN   \n",
       "\n",
       "       oxygen_requirement   habitat disease  \\\n",
       "0                     NaN       NaN     NaN   \n",
       "1                     NaN       NaN     NaN   \n",
       "2                     NaN       NaN     NaN   \n",
       "3                 Aerobic  Multiple     NaN   \n",
       "4                     NaN       NaN     NaN   \n",
       "...                   ...       ...     ...   \n",
       "433517                NaN       NaN     NaN   \n",
       "433518                NaN       NaN     NaN   \n",
       "433519                NaN       NaN     NaN   \n",
       "433520                NaN       NaN     NaN   \n",
       "433521                NaN       NaN     NaN   \n",
       "\n",
       "                                                 comments  \\\n",
       "0       Genome sequence of a strain of bacteria that c...   \n",
       "1       Genome sequencing of a Gamma-Radiation-Resista...   \n",
       "2       'Echinacea purpurea' witches'-broom phytoplasm...   \n",
       "3       Nostoc azollae 0708. Nostoc azollae 0708, also...   \n",
       "4       In this study, we simulate the Deepwater Horiz...   \n",
       "...                                                   ...   \n",
       "433517  Single cell genome sequencing of biomineralizi...   \n",
       "433518  Single cell genome sequencing of biomineralizi...   \n",
       "433519  Single cell genome sequencing of biomineralizi...   \n",
       "433520  Single cell genome sequencing of biomineralizi...   \n",
       "433521  Single cell genome sequencing of biomineralizi...   \n",
       "\n",
       "                                      additional_metadata  \n",
       "0       sample_type:metagenomic assembly;collected_by:...  \n",
       "1                                   sample_type:bacterial  \n",
       "2                            lab_host:Catharanthus roseus  \n",
       "3                                                     NaN  \n",
       "4                        sample_type:metagenomic assembly  \n",
       "...                                                   ...  \n",
       "433517                                                NaN  \n",
       "433518                                                NaN  \n",
       "433519                                                NaN  \n",
       "433520                                                NaN  \n",
       "433521                                                NaN  \n",
       "\n",
       "[433522 rows x 66 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# the full data set. Don't try this at home!\n",
    "# metadf = pd.read_csv(\"../small_data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "metadf = pd.read_csv(\"../data/patric_genome_metadata.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "metadf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make sure we have the right column\n",
    "\n",
    "PATRIC calls the column `assembly_accession` while GTDB calls it `ncbi_genbank_assembly_accession`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         GCA_003181115.1\n",
       "1         GCF_001007995.1\n",
       "2         GCF_001307505.1\n",
       "3         GCA_000196515.1\n",
       "4         GCA_002163025.1\n",
       "               ...       \n",
       "433517    GCA_000379245.1\n",
       "433518    GCA_000379345.1\n",
       "433519    GCA_000379265.1\n",
       "433520    GCA_000379205.1\n",
       "433521    GCA_000372125.1\n",
       "Name: assembly_accession, Length: 433522, dtype: object"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acccol = 'assembly_accession'\n",
    "metadf[acccol]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Abstract out the accession and name into new columns\n",
    "\n",
    "This will allow us to merge the data with the PATRIC and GTDB data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>Name</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>ASM313v1</td>\n",
       "      <td>GCA_000003135.1_ASM313v1_genomic.gbff.gz</td>\n",
       "      <td>2396359</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>48916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>ASM364v1</td>\n",
       "      <td>GCA_000003645.1_ASM364v1_genomic.gbff.gz</td>\n",
       "      <td>5269725</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>40297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>ASM392v1</td>\n",
       "      <td>GCA_000003925.1_ASM392v1_genomic.gbff.gz</td>\n",
       "      <td>5561906</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>268081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>ASM395v1</td>\n",
       "      <td>GCA_000003955.1_ASM395v1_genomic.gbff.gz</td>\n",
       "      <td>5790501</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>166286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>GCA_000005825.2_ASM582v2_genomic.gbff.gz</td>\n",
       "      <td>4249248</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>93416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399577</th>\n",
       "      <td>GCA_902860175.1</td>\n",
       "      <td>LMG_5997</td>\n",
       "      <td>GCA_902860175.1_LMG_5997_genomic.gbff.gz</td>\n",
       "      <td>7197255</td>\n",
       "      <td>21</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>17</td>\n",
       "      <td>69051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399578</th>\n",
       "      <td>GCA_902860185.1</td>\n",
       "      <td>LMG_6103</td>\n",
       "      <td>GCA_902860185.1_LMG_6103_genomic.gbff.gz</td>\n",
       "      <td>6497464</td>\n",
       "      <td>8</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399579</th>\n",
       "      <td>GCA_902860195.1</td>\n",
       "      <td>LMG_7053</td>\n",
       "      <td>GCA_902860195.1_LMG_7053_genomic.gbff.gz</td>\n",
       "      <td>6702936</td>\n",
       "      <td>148</td>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>12819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399580</th>\n",
       "      <td>GCA_902860205.1</td>\n",
       "      <td>LMG_6001</td>\n",
       "      <td>GCA_902860205.1_LMG_6001_genomic.gbff.gz</td>\n",
       "      <td>6320373</td>\n",
       "      <td>19</td>\n",
       "      <td>35</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>12</td>\n",
       "      <td>41572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399581</th>\n",
       "      <td>GCA_902860235.1</td>\n",
       "      <td>LMG_9964</td>\n",
       "      <td>GCA_902860235.1_LMG_9964_genomic.gbff.gz</td>\n",
       "      <td>7602199</td>\n",
       "      <td>26</td>\n",
       "      <td>37</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>71710</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>399582 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession      Name                                    Contig  \\\n",
       "0         GCA_000003135.1  ASM313v1  GCA_000003135.1_ASM313v1_genomic.gbff.gz   \n",
       "1         GCA_000003645.1  ASM364v1  GCA_000003645.1_ASM364v1_genomic.gbff.gz   \n",
       "2         GCA_000003925.1  ASM392v1  GCA_000003925.1_ASM392v1_genomic.gbff.gz   \n",
       "3         GCA_000003955.1  ASM395v1  GCA_000003955.1_ASM395v1_genomic.gbff.gz   \n",
       "4         GCA_000005825.2  ASM582v2  GCA_000005825.2_ASM582v2_genomic.gbff.gz   \n",
       "...                   ...       ...                                       ...   \n",
       "399577    GCA_902860175.1  LMG_5997  GCA_902860175.1_LMG_5997_genomic.gbff.gz   \n",
       "399578    GCA_902860185.1  LMG_6103  GCA_902860185.1_LMG_6103_genomic.gbff.gz   \n",
       "399579    GCA_902860195.1  LMG_7053  GCA_902860195.1_LMG_7053_genomic.gbff.gz   \n",
       "399580    GCA_902860205.1  LMG_6001  GCA_902860205.1_LMG_6001_genomic.gbff.gz   \n",
       "399581    GCA_902860235.1  LMG_9964  GCA_902860235.1_LMG_9964_genomic.gbff.gz   \n",
       "\n",
       "        Genome length  Contigs  Total Predicted Prophages  Kept  \\\n",
       "0             2396359       10                         16     2   \n",
       "1             5269725        1                         31     1   \n",
       "2             5561906        1                         38     6   \n",
       "3             5790501        1                         46     6   \n",
       "4             4249248        3                         33     3   \n",
       "...               ...      ...                        ...   ...   \n",
       "399577        7197255       21                         33     2   \n",
       "399578        6497464        8                         22     0   \n",
       "399579        6702936      148                         33     1   \n",
       "399580        6320373       19                         35     2   \n",
       "399581        7602199       26                         37     2   \n",
       "\n",
       "        No phage genes  Not enough genes  bp prophage  \n",
       "0                    1                13        48916  \n",
       "1                   10                20        40297  \n",
       "2                   13                19       268081  \n",
       "3                   11                29       166286  \n",
       "4                    9                21        93416  \n",
       "...                ...               ...          ...  \n",
       "399577              14                17        69051  \n",
       "399578              10                12            0  \n",
       "399579              11                21        12819  \n",
       "399580              21                12        41572  \n",
       "399581              20                15        71710  \n",
       "\n",
       "[399582 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_acc_name(x):\n",
    "    regexp = re.compile('(\\w+\\.\\d+)_([\\w\\.\\-]+)_genomic.gbff.gz')\n",
    "    m = regexp.match(x)\n",
    "    if not m:\n",
    "        sys.stderr.write(f\"WARNING: Regexp did not match {x}\\n\")\n",
    "        return (None, None)\n",
    "    return list(m.groups())\n",
    "\n",
    "#phagesdf = pd.read_csv(\"../small_data/phages_per_genome.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "phagesdf = pd.read_csv(\"../data/phages_per_genome.tsv.gz\", compression='gzip', header=0, delimiter=\"\\t\")\n",
    "phagesdf = pd.concat([pd.DataFrame.from_records(phagesdf['Contig'].apply(get_acc_name), columns=[acccol, 'Name']), phagesdf], axis=1)\n",
    "phagesdf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge the dataframes\n",
    "\n",
    "First, select some columns we want to keep from PATRIC, and then merge the data frames. We write this to a tsv file for Laura to append isolation information to."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "interesting_cols = [acccol, 'isolation_site', 'isolation_source', 'isolation_comments', 'collection_date',\n",
    "                    'isolation_country', 'geographic_location', 'latitude', 'longitude', 'altitude', 'depth',\n",
    "                    'other_environmental', 'host_name', 'host_gender', 'host_age', 'host_health', \n",
    "                    'body_sample_site', 'body_sample_subsite', 'other_clinical', 'gram_stain', 'cell_shape',\n",
    "                    'motility', 'sporulation', 'temperature_range', 'optimal_temperature', 'salinity',\n",
    "                    'oxygen_requirement', 'habitat', 'disease']\n",
    "\n",
    "tempdf = metadf[interesting_cols]\n",
    "phagemeta = pd.merge(tempdf, phagesdf, how='right', left_on=acccol, right_on=acccol)\n",
    "phagemeta.to_csv(os.path.join('results', 'example_isolations.tsv'), sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>isolation_site</th>\n",
       "      <th>isolation_source</th>\n",
       "      <th>isolation_comments</th>\n",
       "      <th>collection_date</th>\n",
       "      <th>isolation_country</th>\n",
       "      <th>geographic_location</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>altitude</th>\n",
       "      <th>...</th>\n",
       "      <th>disease</th>\n",
       "      <th>Name</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Genome length</th>\n",
       "      <th>Contigs</th>\n",
       "      <th>Total Predicted Prophages</th>\n",
       "      <th>Kept</th>\n",
       "      <th>No phage genes</th>\n",
       "      <th>Not enough genes</th>\n",
       "      <th>bp prophage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM313v1</td>\n",
       "      <td>GCA_000003135.1_ASM313v1_genomic.gbff.gz</td>\n",
       "      <td>2396359</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>48916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cream cheese</td>\n",
       "      <td>isolated from cream cheese</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>ASM364v1</td>\n",
       "      <td>GCA_000003645.1_ASM364v1_genomic.gbff.gz</td>\n",
       "      <td>5269725</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>40297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>soil</td>\n",
       "      <td>isolated from soil</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM392v1</td>\n",
       "      <td>GCA_000003925.1_ASM392v1_genomic.gbff.gz</td>\n",
       "      <td>5561906</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>268081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human blood</td>\n",
       "      <td>isolated from human blood in Iceland</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Iceland</td>\n",
       "      <td>Iceland</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Food poisoning</td>\n",
       "      <td>ASM395v1</td>\n",
       "      <td>GCA_000003955.1_ASM395v1_genomic.gbff.gz</td>\n",
       "      <td>5790501</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>166286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>soil</td>\n",
       "      <td>isolated from soil in New York State by alkali...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>United States</td>\n",
       "      <td>New York State</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM582v2</td>\n",
       "      <td>GCA_000005825.2_ASM582v2_genomic.gbff.gz</td>\n",
       "      <td>4249248</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>93416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403548</th>\n",
       "      <td>GCA_902860175.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Laboratory sink</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LMG_5997</td>\n",
       "      <td>GCA_902860175.1_LMG_5997_genomic.gbff.gz</td>\n",
       "      <td>7197255</td>\n",
       "      <td>21</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>17</td>\n",
       "      <td>69051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403549</th>\n",
       "      <td>GCA_902860185.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LMG_6103</td>\n",
       "      <td>GCA_902860185.1_LMG_6103_genomic.gbff.gz</td>\n",
       "      <td>6497464</td>\n",
       "      <td>8</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403550</th>\n",
       "      <td>GCA_902860195.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LMG_7053</td>\n",
       "      <td>GCA_902860195.1_LMG_7053_genomic.gbff.gz</td>\n",
       "      <td>6702936</td>\n",
       "      <td>148</td>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>12819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403551</th>\n",
       "      <td>GCA_902860205.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LMG_6001</td>\n",
       "      <td>GCA_902860205.1_LMG_6001_genomic.gbff.gz</td>\n",
       "      <td>6320373</td>\n",
       "      <td>19</td>\n",
       "      <td>35</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>12</td>\n",
       "      <td>41572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403552</th>\n",
       "      <td>GCA_902860235.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Acacia decurrens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LMG_9964</td>\n",
       "      <td>GCA_902860235.1_LMG_9964_genomic.gbff.gz</td>\n",
       "      <td>7602199</td>\n",
       "      <td>26</td>\n",
       "      <td>37</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>71710</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>403553 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession isolation_site  isolation_source  \\\n",
       "0         GCA_000003135.1            NaN               NaN   \n",
       "1         GCA_000003645.1            NaN      cream cheese   \n",
       "2         GCA_000003925.1            NaN              soil   \n",
       "3         GCA_000003955.1            NaN       human blood   \n",
       "4         GCA_000005825.2            NaN              soil   \n",
       "...                   ...            ...               ...   \n",
       "403548    GCA_902860175.1            NaN   Laboratory sink   \n",
       "403549    GCA_902860185.1            NaN             Human   \n",
       "403550    GCA_902860195.1            NaN             Human   \n",
       "403551    GCA_902860205.1            NaN               NaN   \n",
       "403552    GCA_902860235.1            NaN  Acacia decurrens   \n",
       "\n",
       "                                       isolation_comments collection_date  \\\n",
       "0                                                     NaN             NaN   \n",
       "1                              isolated from cream cheese             NaN   \n",
       "2                                      isolated from soil             NaN   \n",
       "3                    isolated from human blood in Iceland             NaN   \n",
       "4       isolated from soil in New York State by alkali...             NaN   \n",
       "...                                                   ...             ...   \n",
       "403548                                                NaN            0000   \n",
       "403549                                                NaN            0000   \n",
       "403550                                                NaN            0000   \n",
       "403551                                                NaN            0000   \n",
       "403552                                                NaN             NaN   \n",
       "\n",
       "       isolation_country geographic_location latitude longitude altitude  ...  \\\n",
       "0                    NaN                 NaN      NaN       NaN      NaN  ...   \n",
       "1                    NaN                 NaN      NaN       NaN      NaN  ...   \n",
       "2                    NaN                 NaN      NaN       NaN      NaN  ...   \n",
       "3                Iceland             Iceland      NaN       NaN      NaN  ...   \n",
       "4          United States      New York State      NaN       NaN      NaN  ...   \n",
       "...                  ...                 ...      ...       ...      ...  ...   \n",
       "403548               NaN                 NaN      NaN       NaN      NaN  ...   \n",
       "403549    United Kingdom      United Kingdom      NaN       NaN      NaN  ...   \n",
       "403550    United Kingdom      United Kingdom      NaN       NaN      NaN  ...   \n",
       "403551               NaN                 NaN      NaN       NaN      NaN  ...   \n",
       "403552            Brazil              Brazil      NaN       NaN      NaN  ...   \n",
       "\n",
       "               disease      Name                                    Contig  \\\n",
       "0                  NaN  ASM313v1  GCA_000003135.1_ASM313v1_genomic.gbff.gz   \n",
       "1       Food poisoning  ASM364v1  GCA_000003645.1_ASM364v1_genomic.gbff.gz   \n",
       "2                  NaN  ASM392v1  GCA_000003925.1_ASM392v1_genomic.gbff.gz   \n",
       "3       Food poisoning  ASM395v1  GCA_000003955.1_ASM395v1_genomic.gbff.gz   \n",
       "4                  NaN  ASM582v2  GCA_000005825.2_ASM582v2_genomic.gbff.gz   \n",
       "...                ...       ...                                       ...   \n",
       "403548             NaN  LMG_5997  GCA_902860175.1_LMG_5997_genomic.gbff.gz   \n",
       "403549             NaN  LMG_6103  GCA_902860185.1_LMG_6103_genomic.gbff.gz   \n",
       "403550             NaN  LMG_7053  GCA_902860195.1_LMG_7053_genomic.gbff.gz   \n",
       "403551             NaN  LMG_6001  GCA_902860205.1_LMG_6001_genomic.gbff.gz   \n",
       "403552             NaN  LMG_9964  GCA_902860235.1_LMG_9964_genomic.gbff.gz   \n",
       "\n",
       "       Genome length Contigs Total Predicted Prophages Kept No phage genes  \\\n",
       "0            2396359      10                        16    2              1   \n",
       "1            5269725       1                        31    1             10   \n",
       "2            5561906       1                        38    6             13   \n",
       "3            5790501       1                        46    6             11   \n",
       "4            4249248       3                        33    3              9   \n",
       "...              ...     ...                       ...  ...            ...   \n",
       "403548       7197255      21                        33    2             14   \n",
       "403549       6497464       8                        22    0             10   \n",
       "403550       6702936     148                        33    1             11   \n",
       "403551       6320373      19                        35    2             21   \n",
       "403552       7602199      26                        37    2             20   \n",
       "\n",
       "       Not enough genes bp prophage  \n",
       "0                    13       48916  \n",
       "1                    20       40297  \n",
       "2                    19      268081  \n",
       "3                    29      166286  \n",
       "4                    21       93416  \n",
       "...                 ...         ...  \n",
       "403548               17       69051  \n",
       "403549               12           0  \n",
       "403550               21       12819  \n",
       "403551               12       41572  \n",
       "403552               15       71710  \n",
       "\n",
       "[403553 rows x 38 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagemeta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isolation_source</th>\n",
       "      <th>isolation_comments</th>\n",
       "      <th>collection_date</th>\n",
       "      <th>isolation_country</th>\n",
       "      <th>geographic_location</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>altitude</th>\n",
       "      <th>depth</th>\n",
       "      <th>other_environmental</th>\n",
       "      <th>...</th>\n",
       "      <th>gram_stain</th>\n",
       "      <th>cell_shape</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>Rod</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Anaerobic</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cream cheese</td>\n",
       "      <td>isolated from cream cheese</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>Rod</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>soil</td>\n",
       "      <td>isolated from soil</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>Rod</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>human blood</td>\n",
       "      <td>isolated from human blood in Iceland</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Iceland</td>\n",
       "      <td>Iceland</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>Rod</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Food poisoning</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>soil</td>\n",
       "      <td>isolated from soil in New York State by alkali...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>United States</td>\n",
       "      <td>New York State</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Facultative</td>\n",
       "      <td>Terrestrial</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403548</th>\n",
       "      <td>Laboratory sink</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403549</th>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403550</th>\n",
       "      <td>Human</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403551</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403552</th>\n",
       "      <td>Acacia decurrens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>403553 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        isolation_source                                 isolation_comments  \\\n",
       "0                    NaN                                                NaN   \n",
       "1           cream cheese                         isolated from cream cheese   \n",
       "2                   soil                                 isolated from soil   \n",
       "3            human blood               isolated from human blood in Iceland   \n",
       "4                   soil  isolated from soil in New York State by alkali...   \n",
       "...                  ...                                                ...   \n",
       "403548   Laboratory sink                                                NaN   \n",
       "403549             Human                                                NaN   \n",
       "403550             Human                                                NaN   \n",
       "403551               NaN                                                NaN   \n",
       "403552  Acacia decurrens                                                NaN   \n",
       "\n",
       "       collection_date isolation_country geographic_location latitude  \\\n",
       "0                  NaN               NaN                 NaN      NaN   \n",
       "1                  NaN               NaN                 NaN      NaN   \n",
       "2                  NaN               NaN                 NaN      NaN   \n",
       "3                  NaN           Iceland             Iceland      NaN   \n",
       "4                  NaN     United States      New York State      NaN   \n",
       "...                ...               ...                 ...      ...   \n",
       "403548            0000               NaN                 NaN      NaN   \n",
       "403549            0000    United Kingdom      United Kingdom      NaN   \n",
       "403550            0000    United Kingdom      United Kingdom      NaN   \n",
       "403551            0000               NaN                 NaN      NaN   \n",
       "403552             NaN            Brazil              Brazil      NaN   \n",
       "\n",
       "       longitude altitude depth other_environmental  ... gram_stain  \\\n",
       "0            NaN      NaN   NaN                 NaN  ...          +   \n",
       "1            NaN      NaN   NaN                 NaN  ...          +   \n",
       "2            NaN      NaN   NaN                 NaN  ...          +   \n",
       "3            NaN      NaN   NaN                 NaN  ...          +   \n",
       "4            NaN      NaN   NaN                 NaN  ...        NaN   \n",
       "...          ...      ...   ...                 ...  ...        ...   \n",
       "403548       NaN      NaN   NaN                 NaN  ...        NaN   \n",
       "403549       NaN      NaN   NaN                 NaN  ...        NaN   \n",
       "403550       NaN      NaN   NaN                 NaN  ...        NaN   \n",
       "403551       NaN      NaN   NaN                 NaN  ...        NaN   \n",
       "403552       NaN      NaN   NaN                 NaN  ...        NaN   \n",
       "\n",
       "       cell_shape motility sporulation temperature_range optimal_temperature  \\\n",
       "0             Rod       No          No        Mesophilic                   -   \n",
       "1             Rod      Yes         Yes        Mesophilic                   -   \n",
       "2             Rod       No         Yes               NaN                   -   \n",
       "3             Rod      Yes         Yes        Mesophilic                   -   \n",
       "4         Bacilli      Yes         NaN        Mesophilic                 NaN   \n",
       "...           ...      ...         ...               ...                 ...   \n",
       "403548        NaN      NaN         NaN               NaN                 NaN   \n",
       "403549        NaN      NaN         NaN               NaN                 NaN   \n",
       "403550        NaN      NaN         NaN               NaN                 NaN   \n",
       "403551        NaN      NaN         NaN               NaN                 NaN   \n",
       "403552        NaN      NaN         NaN               NaN                 NaN   \n",
       "\n",
       "       salinity oxygen_requirement          habitat         disease  \n",
       "0           NaN          Anaerobic  Host-associated             NaN  \n",
       "1           NaN            Aerobic         Multiple  Food poisoning  \n",
       "2           NaN                NaN              NaN             NaN  \n",
       "3           NaN            Aerobic         Multiple  Food poisoning  \n",
       "4           NaN        Facultative      Terrestrial             NaN  \n",
       "...         ...                ...              ...             ...  \n",
       "403548      NaN                NaN              NaN             NaN  \n",
       "403549      NaN                NaN              NaN             NaN  \n",
       "403550      NaN                NaN              NaN             NaN  \n",
       "403551      NaN                NaN              NaN             NaN  \n",
       "403552      NaN                NaN              NaN             NaN  \n",
       "\n",
       "[403553 rows x 27 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagemeta.iloc[:,2:29]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0           -1\n",
       "1         5261\n",
       "2         8897\n",
       "3         6481\n",
       "4         8897\n",
       "          ... \n",
       "403548    2025\n",
       "403549    1762\n",
       "403550    1762\n",
       "403551      -1\n",
       "403552     115\n",
       "Length: 403553, dtype: int16"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagemeta['isolation_source'].astype('category').cat.codes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Kept</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403548</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403549</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403550</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403551</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403552</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>403553 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Kept\n",
       "0          2\n",
       "1          1\n",
       "2          6\n",
       "3          6\n",
       "4          3\n",
       "...      ...\n",
       "403548     2\n",
       "403549     0\n",
       "403550     1\n",
       "403551     2\n",
       "403552     2\n",
       "\n",
       "[403553 rows x 1 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phagemeta.iloc[:,34:35]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Encode the data\n",
    "\n",
    "This converts every column into categories so we can (hopefully) use it in the RF models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isolation_source</th>\n",
       "      <th>isolation_comments</th>\n",
       "      <th>collection_date</th>\n",
       "      <th>isolation_country</th>\n",
       "      <th>geographic_location</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>altitude</th>\n",
       "      <th>depth</th>\n",
       "      <th>other_environmental</th>\n",
       "      <th>...</th>\n",
       "      <th>gram_stain</th>\n",
       "      <th>cell_shape</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5261</td>\n",
       "      <td>1990</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8897</td>\n",
       "      <td>2657</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6481</td>\n",
       "      <td>2208</td>\n",
       "      <td>-1</td>\n",
       "      <td>93</td>\n",
       "      <td>2666</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8897</td>\n",
       "      <td>2688</td>\n",
       "      <td>-1</td>\n",
       "      <td>228</td>\n",
       "      <td>4088</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>-1</td>\n",
       "      <td>8</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>5</td>\n",
       "      <td>37</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403548</th>\n",
       "      <td>2025</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403549</th>\n",
       "      <td>1762</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>227</td>\n",
       "      <td>7171</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403550</th>\n",
       "      <td>1762</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>227</td>\n",
       "      <td>7171</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403551</th>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403552</th>\n",
       "      <td>115</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>33</td>\n",
       "      <td>469</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>403553 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        isolation_source  isolation_comments  collection_date  \\\n",
       "0                     -1                  -1               -1   \n",
       "1                   5261                1990               -1   \n",
       "2                   8897                2657               -1   \n",
       "3                   6481                2208               -1   \n",
       "4                   8897                2688               -1   \n",
       "...                  ...                 ...              ...   \n",
       "403548              2025                  -1                0   \n",
       "403549              1762                  -1                0   \n",
       "403550              1762                  -1                0   \n",
       "403551                -1                  -1                0   \n",
       "403552               115                  -1               -1   \n",
       "\n",
       "        isolation_country  geographic_location  latitude  longitude  altitude  \\\n",
       "0                      -1                   -1        -1         -1        -1   \n",
       "1                      -1                   -1        -1         -1        -1   \n",
       "2                      -1                   -1        -1         -1        -1   \n",
       "3                      93                 2666        -1         -1        -1   \n",
       "4                     228                 4088        -1         -1        -1   \n",
       "...                   ...                  ...       ...        ...       ...   \n",
       "403548                 -1                   -1        -1         -1        -1   \n",
       "403549                227                 7171        -1         -1        -1   \n",
       "403550                227                 7171        -1         -1        -1   \n",
       "403551                 -1                   -1        -1         -1        -1   \n",
       "403552                 33                  469        -1         -1        -1   \n",
       "\n",
       "        depth  other_environmental  ...  gram_stain  cell_shape  motility  \\\n",
       "0          -1                   -1  ...           0          56         5   \n",
       "1          -1                   -1  ...           0          56         9   \n",
       "2          -1                   -1  ...           0          56         5   \n",
       "3          -1                   -1  ...           0          56         9   \n",
       "4          -1                   -1  ...          -1          13         9   \n",
       "...       ...                  ...  ...         ...         ...       ...   \n",
       "403548     -1                   -1  ...          -1          -1        -1   \n",
       "403549     -1                   -1  ...          -1          -1        -1   \n",
       "403550     -1                   -1  ...          -1          -1        -1   \n",
       "403551     -1                   -1  ...          -1          -1        -1   \n",
       "403552     -1                   -1  ...          -1          -1        -1   \n",
       "\n",
       "        sporulation  temperature_range  optimal_temperature  salinity  \\\n",
       "0                 3                  8                    1        -1   \n",
       "1                 8                  8                    1        -1   \n",
       "2                 8                 -1                    1        -1   \n",
       "3                 8                  8                    1        -1   \n",
       "4                -1                  8                   -1        -1   \n",
       "...             ...                ...                  ...       ...   \n",
       "403548           -1                 -1                   -1        -1   \n",
       "403549           -1                 -1                   -1        -1   \n",
       "403550           -1                 -1                   -1        -1   \n",
       "403551           -1                 -1                   -1        -1   \n",
       "403552           -1                 -1                   -1        -1   \n",
       "\n",
       "        oxygen_requirement  habitat  disease  \n",
       "0                        3       14       -1  \n",
       "1                        1       23      159  \n",
       "2                       -1       -1       -1  \n",
       "3                        1       23      159  \n",
       "4                        5       37       -1  \n",
       "...                    ...      ...      ...  \n",
       "403548                  -1       -1       -1  \n",
       "403549                  -1       -1       -1  \n",
       "403550                  -1       -1       -1  \n",
       "403551                  -1       -1       -1  \n",
       "403552                  -1       -1       -1  \n",
       "\n",
       "[403553 rows x 27 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmenc = pd.DataFrame()\n",
    "for c in phagemeta.iloc[:,2:29].columns:\n",
    "    pmenc[c] = phagemeta[c].astype('category').cat.codes\n",
    "pmenc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Random Forest\n",
    "\n",
    "Can we identify which features best predict the number of phages predicted. Note we should compare total/kept/etc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# which data frame are we going to use here.\n",
    "# df = scdnonzero\n",
    "df = phagemeta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clf = RandomForestClassifier(random_state=42, n_estimators=1000, bootstrap=True, n_jobs=-1)\n",
    "rf = clf.fit(pmenc, phagemeta.iloc[:,34:35].values.ravel())\n",
    "print(rf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fi = pd.DataFrame(zip(df.iloc[:,1:].columns, rf.feature_importances_), columns=['genome', 'importance'])\n",
    "fi.sort_values('importance', ascending=False).head(5)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
